// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/apis/RUMApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __await = (this && this.__await) || function (v) { return this instanceof __await ? (this.v = v, this) : new __await(v); }
var __asyncGenerator = (this && this.__asyncGenerator) || function (thisArg, _arguments, generator) {
    if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
    var g = generator.apply(thisArg, _arguments || []), i, q = [];
    return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
    function verb(n) { if (g[n]) i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
    function resume(n, v) { try { step(g[n](v)); } catch (e) { settle(q[0][3], e); } }
    function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
    function fulfill(value) { resume("next", value); }
    function reject(value) { resume("throw", value); }
    function settle(f, v) { if (f(v), q.shift(), q.length) resume(q[0][0], q[0][1]); }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RUMApi = exports.RUMApiResponseProcessor = exports.RUMApiRequestFactory = void 0;
const baseapi_1 = require("../../datadog-api-client-common/baseapi");
const configuration_1 = require("../../datadog-api-client-common/configuration");
const http_1 = require("../../datadog-api-client-common/http/http");
const logger_1 = require("../../../logger");
const ObjectSerializer_1 = require("../models/ObjectSerializer");
const exception_1 = require("../../datadog-api-client-common/exception");
const RUMQueryPageOptions_1 = require("../models/RUMQueryPageOptions");
class RUMApiRequestFactory extends baseapi_1.BaseAPIRequestFactory {
    aggregateRUMEvents(body, _options) {
        return __awaiter(this, void 0, void 0, function* () {
            const _config = _options || this.configuration;
            // verify required parameter 'body' is not null or undefined
            if (body === null || body === undefined) {
                throw new baseapi_1.RequiredError("body", "aggregateRUMEvents");
            }
            // Path Params
            const localVarPath = "/api/v2/rum/analytics/aggregate";
            // Make Request Context
            const requestContext = _config
                .getServer("v2.RUMApi.aggregateRUMEvents")
                .makeRequestContext(localVarPath, http_1.HttpMethod.POST);
            requestContext.setHeaderParam("Accept", "application/json");
            requestContext.setHttpConfig(_config.httpConfig);
            // Body Params
            const contentType = ObjectSerializer_1.ObjectSerializer.getPreferredMediaType([
                "application/json",
            ]);
            requestContext.setHeaderParam("Content-Type", contentType);
            const serializedBody = ObjectSerializer_1.ObjectSerializer.stringify(ObjectSerializer_1.ObjectSerializer.serialize(body, "RUMAggregateRequest", ""), contentType);
            requestContext.setBody(serializedBody);
            // Apply auth methods
            (0, configuration_1.applySecurityAuthentication)(_config, requestContext, [
                "apiKeyAuth",
                "appKeyAuth",
            ]);
            return requestContext;
        });
    }
    createRUMApplication(body, _options) {
        return __awaiter(this, void 0, void 0, function* () {
            const _config = _options || this.configuration;
            // verify required parameter 'body' is not null or undefined
            if (body === null || body === undefined) {
                throw new baseapi_1.RequiredError("body", "createRUMApplication");
            }
            // Path Params
            const localVarPath = "/api/v2/rum/applications";
            // Make Request Context
            const requestContext = _config
                .getServer("v2.RUMApi.createRUMApplication")
                .makeRequestContext(localVarPath, http_1.HttpMethod.POST);
            requestContext.setHeaderParam("Accept", "application/json");
            requestContext.setHttpConfig(_config.httpConfig);
            // Body Params
            const contentType = ObjectSerializer_1.ObjectSerializer.getPreferredMediaType([
                "application/json",
            ]);
            requestContext.setHeaderParam("Content-Type", contentType);
            const serializedBody = ObjectSerializer_1.ObjectSerializer.stringify(ObjectSerializer_1.ObjectSerializer.serialize(body, "RUMApplicationCreateRequest", ""), contentType);
            requestContext.setBody(serializedBody);
            // Apply auth methods
            (0, configuration_1.applySecurityAuthentication)(_config, requestContext, [
                "apiKeyAuth",
                "appKeyAuth",
            ]);
            return requestContext;
        });
    }
    deleteRUMApplication(id, _options) {
        return __awaiter(this, void 0, void 0, function* () {
            const _config = _options || this.configuration;
            // verify required parameter 'id' is not null or undefined
            if (id === null || id === undefined) {
                throw new baseapi_1.RequiredError("id", "deleteRUMApplication");
            }
            // Path Params
            const localVarPath = "/api/v2/rum/applications/{id}".replace("{id}", encodeURIComponent(String(id)));
            // Make Request Context
            const requestContext = _config
                .getServer("v2.RUMApi.deleteRUMApplication")
                .makeRequestContext(localVarPath, http_1.HttpMethod.DELETE);
            requestContext.setHeaderParam("Accept", "*/*");
            requestContext.setHttpConfig(_config.httpConfig);
            // Apply auth methods
            (0, configuration_1.applySecurityAuthentication)(_config, requestContext, [
                "apiKeyAuth",
                "appKeyAuth",
            ]);
            return requestContext;
        });
    }
    deleteSourcemaps(mapkind, dryRun, filterService, filterVersion, filterVariant, filterId, filterBuildId, filterUuid, filterPlatform, filterBuildNumber, filterBundleName, filterArch, filterSymbolSource, filterOrigin, filterOriginVersion, filterFilename, filterDebugId, filterGnuBuildId, filterGoBuildId, filterFileHash, _options) {
        return __awaiter(this, void 0, void 0, function* () {
            const _config = _options || this.configuration;
            logger_1.logger.warn("Using unstable operation 'deleteSourcemaps'");
            if (!_config.unstableOperations["v2.deleteSourcemaps"]) {
                throw new Error("Unstable operation 'deleteSourcemaps' is disabled");
            }
            // verify required parameter 'mapkind' is not null or undefined
            if (mapkind === null || mapkind === undefined) {
                throw new baseapi_1.RequiredError("mapkind", "deleteSourcemaps");
            }
            // verify required parameter 'dryRun' is not null or undefined
            if (dryRun === null || dryRun === undefined) {
                throw new baseapi_1.RequiredError("dryRun", "deleteSourcemaps");
            }
            // Path Params
            const localVarPath = "/api/v2/sourcemaps";
            // Make Request Context
            const requestContext = _config
                .getServer("v2.RUMApi.deleteSourcemaps")
                .makeRequestContext(localVarPath, http_1.HttpMethod.DELETE);
            requestContext.setHeaderParam("Accept", "application/json");
            requestContext.setHttpConfig(_config.httpConfig);
            // Query Params
            if (mapkind !== undefined) {
                requestContext.setQueryParam("mapkind", ObjectSerializer_1.ObjectSerializer.serialize(mapkind, "SourcemapMapKind", ""), "");
            }
            if (dryRun !== undefined) {
                requestContext.setQueryParam("dry_run", ObjectSerializer_1.ObjectSerializer.serialize(dryRun, "boolean", ""), "");
            }
            if (filterService !== undefined) {
                requestContext.setQueryParam("filter[service]", ObjectSerializer_1.ObjectSerializer.serialize(filterService, "Array<string>", ""), "multi");
            }
            if (filterVersion !== undefined) {
                requestContext.setQueryParam("filter[version]", ObjectSerializer_1.ObjectSerializer.serialize(filterVersion, "Array<string>", ""), "multi");
            }
            if (filterVariant !== undefined) {
                requestContext.setQueryParam("filter[variant]", ObjectSerializer_1.ObjectSerializer.serialize(filterVariant, "Array<string>", ""), "multi");
            }
            if (filterId !== undefined) {
                requestContext.setQueryParam("filter[id]", ObjectSerializer_1.ObjectSerializer.serialize(filterId, "Array<string>", ""), "multi");
            }
            if (filterBuildId !== undefined) {
                requestContext.setQueryParam("filter[build_id]", ObjectSerializer_1.ObjectSerializer.serialize(filterBuildId, "Array<string>", ""), "multi");
            }
            if (filterUuid !== undefined) {
                requestContext.setQueryParam("filter[uuid]", ObjectSerializer_1.ObjectSerializer.serialize(filterUuid, "Array<string>", ""), "multi");
            }
            if (filterPlatform !== undefined) {
                requestContext.setQueryParam("filter[platform]", ObjectSerializer_1.ObjectSerializer.serialize(filterPlatform, "Array<string>", ""), "multi");
            }
            if (filterBuildNumber !== undefined) {
                requestContext.setQueryParam("filter[build_number]", ObjectSerializer_1.ObjectSerializer.serialize(filterBuildNumber, "Array<string>", ""), "multi");
            }
            if (filterBundleName !== undefined) {
                requestContext.setQueryParam("filter[bundle_name]", ObjectSerializer_1.ObjectSerializer.serialize(filterBundleName, "Array<string>", ""), "multi");
            }
            if (filterArch !== undefined) {
                requestContext.setQueryParam("filter[arch]", ObjectSerializer_1.ObjectSerializer.serialize(filterArch, "Array<string>", ""), "multi");
            }
            if (filterSymbolSource !== undefined) {
                requestContext.setQueryParam("filter[symbol_source]", ObjectSerializer_1.ObjectSerializer.serialize(filterSymbolSource, "Array<string>", ""), "multi");
            }
            if (filterOrigin !== undefined) {
                requestContext.setQueryParam("filter[origin]", ObjectSerializer_1.ObjectSerializer.serialize(filterOrigin, "Array<string>", ""), "multi");
            }
            if (filterOriginVersion !== undefined) {
                requestContext.setQueryParam("filter[origin_version]", ObjectSerializer_1.ObjectSerializer.serialize(filterOriginVersion, "Array<string>", ""), "multi");
            }
            if (filterFilename !== undefined) {
                requestContext.setQueryParam("filter[filename]", ObjectSerializer_1.ObjectSerializer.serialize(filterFilename, "string", ""), "");
            }
            if (filterDebugId !== undefined) {
                requestContext.setQueryParam("filter[debug_id]", ObjectSerializer_1.ObjectSerializer.serialize(filterDebugId, "string", ""), "");
            }
            if (filterGnuBuildId !== undefined) {
                requestContext.setQueryParam("filter[gnu_build_id]", ObjectSerializer_1.ObjectSerializer.serialize(filterGnuBuildId, "string", ""), "");
            }
            if (filterGoBuildId !== undefined) {
                requestContext.setQueryParam("filter[go_build_id]", ObjectSerializer_1.ObjectSerializer.serialize(filterGoBuildId, "string", ""), "");
            }
            if (filterFileHash !== undefined) {
                requestContext.setQueryParam("filter[file_hash]", ObjectSerializer_1.ObjectSerializer.serialize(filterFileHash, "string", ""), "");
            }
            // Apply auth methods
            (0, configuration_1.applySecurityAuthentication)(_config, requestContext, [
                "apiKeyAuth",
                "appKeyAuth",
            ]);
            return requestContext;
        });
    }
    getRUMApplication(id, _options) {
        return __awaiter(this, void 0, void 0, function* () {
            const _config = _options || this.configuration;
            // verify required parameter 'id' is not null or undefined
            if (id === null || id === undefined) {
                throw new baseapi_1.RequiredError("id", "getRUMApplication");
            }
            // Path Params
            const localVarPath = "/api/v2/rum/applications/{id}".replace("{id}", encodeURIComponent(String(id)));
            // Make Request Context
            const requestContext = _config
                .getServer("v2.RUMApi.getRUMApplication")
                .makeRequestContext(localVarPath, http_1.HttpMethod.GET);
            requestContext.setHeaderParam("Accept", "application/json");
            requestContext.setHttpConfig(_config.httpConfig);
            // Apply auth methods
            (0, configuration_1.applySecurityAuthentication)(_config, requestContext, [
                "apiKeyAuth",
                "appKeyAuth",
            ]);
            return requestContext;
        });
    }
    getRUMApplications(_options) {
        return __awaiter(this, void 0, void 0, function* () {
            const _config = _options || this.configuration;
            // Path Params
            const localVarPath = "/api/v2/rum/applications";
            // Make Request Context
            const requestContext = _config
                .getServer("v2.RUMApi.getRUMApplications")
                .makeRequestContext(localVarPath, http_1.HttpMethod.GET);
            requestContext.setHeaderParam("Accept", "application/json");
            requestContext.setHttpConfig(_config.httpConfig);
            // Apply auth methods
            (0, configuration_1.applySecurityAuthentication)(_config, requestContext, [
                "apiKeyAuth",
                "appKeyAuth",
            ]);
            return requestContext;
        });
    }
    getServiceRepositoryInfo(body, _options) {
        return __awaiter(this, void 0, void 0, function* () {
            const _config = _options || this.configuration;
            logger_1.logger.warn("Using unstable operation 'getServiceRepositoryInfo'");
            if (!_config.unstableOperations["v2.getServiceRepositoryInfo"]) {
                throw new Error("Unstable operation 'getServiceRepositoryInfo' is disabled");
            }
            // verify required parameter 'body' is not null or undefined
            if (body === null || body === undefined) {
                throw new baseapi_1.RequiredError("body", "getServiceRepositoryInfo");
            }
            // Path Params
            const localVarPath = "/api/v2/sourcemaps/service_repository_info";
            // Make Request Context
            const requestContext = _config
                .getServer("v2.RUMApi.getServiceRepositoryInfo")
                .makeRequestContext(localVarPath, http_1.HttpMethod.POST);
            requestContext.setHeaderParam("Accept", "application/json");
            requestContext.setHttpConfig(_config.httpConfig);
            // Body Params
            const contentType = ObjectSerializer_1.ObjectSerializer.getPreferredMediaType([
                "application/json",
            ]);
            requestContext.setHeaderParam("Content-Type", contentType);
            const serializedBody = ObjectSerializer_1.ObjectSerializer.stringify(ObjectSerializer_1.ObjectSerializer.serialize(body, "ServiceRepositoryInfoRequest", ""), contentType);
            requestContext.setBody(serializedBody);
            // Apply auth methods
            (0, configuration_1.applySecurityAuthentication)(_config, requestContext, [
                "apiKeyAuth",
                "appKeyAuth",
            ]);
            return requestContext;
        });
    }
    getSourcemaps(filename, service, version, _options) {
        return __awaiter(this, void 0, void 0, function* () {
            const _config = _options || this.configuration;
            logger_1.logger.warn("Using unstable operation 'getSourcemaps'");
            if (!_config.unstableOperations["v2.getSourcemaps"]) {
                throw new Error("Unstable operation 'getSourcemaps' is disabled");
            }
            // verify required parameter 'filename' is not null or undefined
            if (filename === null || filename === undefined) {
                throw new baseapi_1.RequiredError("filename", "getSourcemaps");
            }
            // verify required parameter 'service' is not null or undefined
            if (service === null || service === undefined) {
                throw new baseapi_1.RequiredError("service", "getSourcemaps");
            }
            // verify required parameter 'version' is not null or undefined
            if (version === null || version === undefined) {
                throw new baseapi_1.RequiredError("version", "getSourcemaps");
            }
            // Path Params
            const localVarPath = "/api/v2/sourcemaps";
            // Make Request Context
            const requestContext = _config
                .getServer("v2.RUMApi.getSourcemaps")
                .makeRequestContext(localVarPath, http_1.HttpMethod.GET);
            requestContext.setHeaderParam("Accept", "application/json");
            requestContext.setHttpConfig(_config.httpConfig);
            // Query Params
            if (filename !== undefined) {
                requestContext.setQueryParam("filename", ObjectSerializer_1.ObjectSerializer.serialize(filename, "string", ""), "");
            }
            if (service !== undefined) {
                requestContext.setQueryParam("service", ObjectSerializer_1.ObjectSerializer.serialize(service, "string", ""), "");
            }
            if (version !== undefined) {
                requestContext.setQueryParam("version", ObjectSerializer_1.ObjectSerializer.serialize(version, "string", ""), "");
            }
            // Apply auth methods
            (0, configuration_1.applySecurityAuthentication)(_config, requestContext, [
                "apiKeyAuth",
                "appKeyAuth",
            ]);
            return requestContext;
        });
    }
    listRUMEvents(filterQuery, filterFrom, filterTo, sort, pageCursor, pageLimit, _options) {
        return __awaiter(this, void 0, void 0, function* () {
            const _config = _options || this.configuration;
            // Path Params
            const localVarPath = "/api/v2/rum/events";
            // Make Request Context
            const requestContext = _config
                .getServer("v2.RUMApi.listRUMEvents")
                .makeRequestContext(localVarPath, http_1.HttpMethod.GET);
            requestContext.setHeaderParam("Accept", "application/json");
            requestContext.setHttpConfig(_config.httpConfig);
            // Query Params
            if (filterQuery !== undefined) {
                requestContext.setQueryParam("filter[query]", ObjectSerializer_1.ObjectSerializer.serialize(filterQuery, "string", ""), "");
            }
            if (filterFrom !== undefined) {
                requestContext.setQueryParam("filter[from]", ObjectSerializer_1.ObjectSerializer.serialize(filterFrom, "Date", "date-time"), "");
            }
            if (filterTo !== undefined) {
                requestContext.setQueryParam("filter[to]", ObjectSerializer_1.ObjectSerializer.serialize(filterTo, "Date", "date-time"), "");
            }
            if (sort !== undefined) {
                requestContext.setQueryParam("sort", ObjectSerializer_1.ObjectSerializer.serialize(sort, "RUMSort", ""), "");
            }
            if (pageCursor !== undefined) {
                requestContext.setQueryParam("page[cursor]", ObjectSerializer_1.ObjectSerializer.serialize(pageCursor, "string", ""), "");
            }
            if (pageLimit !== undefined) {
                requestContext.setQueryParam("page[limit]", ObjectSerializer_1.ObjectSerializer.serialize(pageLimit, "number", "int32"), "");
            }
            // Apply auth methods
            (0, configuration_1.applySecurityAuthentication)(_config, requestContext, [
                "apiKeyAuth",
                "appKeyAuth",
            ]);
            return requestContext;
        });
    }
    listSourcemaps(mapkind, pageSize, pageNumber, filterService, filterVersion, filterVariant, filterId, filterBuildId, filterUuid, filterPlatform, filterBuildNumber, filterBundleName, filterArch, filterSymbolSource, filterOrigin, filterOriginVersion, filterFilename, filterDebugId, filterGnuBuildId, filterGoBuildId, filterFileHash, _options) {
        return __awaiter(this, void 0, void 0, function* () {
            const _config = _options || this.configuration;
            logger_1.logger.warn("Using unstable operation 'listSourcemaps'");
            if (!_config.unstableOperations["v2.listSourcemaps"]) {
                throw new Error("Unstable operation 'listSourcemaps' is disabled");
            }
            // Path Params
            const localVarPath = "/api/v2/sourcemaps/list";
            // Make Request Context
            const requestContext = _config
                .getServer("v2.RUMApi.listSourcemaps")
                .makeRequestContext(localVarPath, http_1.HttpMethod.GET);
            requestContext.setHeaderParam("Accept", "application/json");
            requestContext.setHttpConfig(_config.httpConfig);
            // Query Params
            if (mapkind !== undefined) {
                requestContext.setQueryParam("mapkind", ObjectSerializer_1.ObjectSerializer.serialize(mapkind, "SourcemapMapKind", ""), "");
            }
            if (pageSize !== undefined) {
                requestContext.setQueryParam("page[size]", ObjectSerializer_1.ObjectSerializer.serialize(pageSize, "number", "int64"), "");
            }
            if (pageNumber !== undefined) {
                requestContext.setQueryParam("page[number]", ObjectSerializer_1.ObjectSerializer.serialize(pageNumber, "number", "int64"), "");
            }
            if (filterService !== undefined) {
                requestContext.setQueryParam("filter[service]", ObjectSerializer_1.ObjectSerializer.serialize(filterService, "Array<string>", ""), "multi");
            }
            if (filterVersion !== undefined) {
                requestContext.setQueryParam("filter[version]", ObjectSerializer_1.ObjectSerializer.serialize(filterVersion, "Array<string>", ""), "multi");
            }
            if (filterVariant !== undefined) {
                requestContext.setQueryParam("filter[variant]", ObjectSerializer_1.ObjectSerializer.serialize(filterVariant, "Array<string>", ""), "multi");
            }
            if (filterId !== undefined) {
                requestContext.setQueryParam("filter[id]", ObjectSerializer_1.ObjectSerializer.serialize(filterId, "Array<string>", ""), "multi");
            }
            if (filterBuildId !== undefined) {
                requestContext.setQueryParam("filter[build_id]", ObjectSerializer_1.ObjectSerializer.serialize(filterBuildId, "Array<string>", ""), "multi");
            }
            if (filterUuid !== undefined) {
                requestContext.setQueryParam("filter[uuid]", ObjectSerializer_1.ObjectSerializer.serialize(filterUuid, "Array<string>", ""), "multi");
            }
            if (filterPlatform !== undefined) {
                requestContext.setQueryParam("filter[platform]", ObjectSerializer_1.ObjectSerializer.serialize(filterPlatform, "Array<string>", ""), "multi");
            }
            if (filterBuildNumber !== undefined) {
                requestContext.setQueryParam("filter[build_number]", ObjectSerializer_1.ObjectSerializer.serialize(filterBuildNumber, "Array<string>", ""), "multi");
            }
            if (filterBundleName !== undefined) {
                requestContext.setQueryParam("filter[bundle_name]", ObjectSerializer_1.ObjectSerializer.serialize(filterBundleName, "Array<string>", ""), "multi");
            }
            if (filterArch !== undefined) {
                requestContext.setQueryParam("filter[arch]", ObjectSerializer_1.ObjectSerializer.serialize(filterArch, "Array<string>", ""), "multi");
            }
            if (filterSymbolSource !== undefined) {
                requestContext.setQueryParam("filter[symbol_source]", ObjectSerializer_1.ObjectSerializer.serialize(filterSymbolSource, "Array<string>", ""), "multi");
            }
            if (filterOrigin !== undefined) {
                requestContext.setQueryParam("filter[origin]", ObjectSerializer_1.ObjectSerializer.serialize(filterOrigin, "Array<string>", ""), "multi");
            }
            if (filterOriginVersion !== undefined) {
                requestContext.setQueryParam("filter[origin_version]", ObjectSerializer_1.ObjectSerializer.serialize(filterOriginVersion, "Array<string>", ""), "multi");
            }
            if (filterFilename !== undefined) {
                requestContext.setQueryParam("filter[filename]", ObjectSerializer_1.ObjectSerializer.serialize(filterFilename, "string", ""), "");
            }
            if (filterDebugId !== undefined) {
                requestContext.setQueryParam("filter[debug_id]", ObjectSerializer_1.ObjectSerializer.serialize(filterDebugId, "string", ""), "");
            }
            if (filterGnuBuildId !== undefined) {
                requestContext.setQueryParam("filter[gnu_build_id]", ObjectSerializer_1.ObjectSerializer.serialize(filterGnuBuildId, "string", ""), "");
            }
            if (filterGoBuildId !== undefined) {
                requestContext.setQueryParam("filter[go_build_id]", ObjectSerializer_1.ObjectSerializer.serialize(filterGoBuildId, "string", ""), "");
            }
            if (filterFileHash !== undefined) {
                requestContext.setQueryParam("filter[file_hash]", ObjectSerializer_1.ObjectSerializer.serialize(filterFileHash, "string", ""), "");
            }
            // Apply auth methods
            (0, configuration_1.applySecurityAuthentication)(_config, requestContext, [
                "apiKeyAuth",
                "appKeyAuth",
            ]);
            return requestContext;
        });
    }
    restoreSourcemaps(mapkind, dryRun, filterService, filterVersion, filterVariant, filterId, filterBuildId, filterUuid, filterPlatform, filterBuildNumber, filterBundleName, filterArch, filterSymbolSource, filterOrigin, filterOriginVersion, filterFilename, filterDebugId, filterGnuBuildId, filterGoBuildId, filterFileHash, _options) {
        return __awaiter(this, void 0, void 0, function* () {
            const _config = _options || this.configuration;
            logger_1.logger.warn("Using unstable operation 'restoreSourcemaps'");
            if (!_config.unstableOperations["v2.restoreSourcemaps"]) {
                throw new Error("Unstable operation 'restoreSourcemaps' is disabled");
            }
            // verify required parameter 'mapkind' is not null or undefined
            if (mapkind === null || mapkind === undefined) {
                throw new baseapi_1.RequiredError("mapkind", "restoreSourcemaps");
            }
            // verify required parameter 'dryRun' is not null or undefined
            if (dryRun === null || dryRun === undefined) {
                throw new baseapi_1.RequiredError("dryRun", "restoreSourcemaps");
            }
            // Path Params
            const localVarPath = "/api/v2/sourcemaps/restore";
            // Make Request Context
            const requestContext = _config
                .getServer("v2.RUMApi.restoreSourcemaps")
                .makeRequestContext(localVarPath, http_1.HttpMethod.PATCH);
            requestContext.setHeaderParam("Accept", "application/json");
            requestContext.setHttpConfig(_config.httpConfig);
            // Query Params
            if (mapkind !== undefined) {
                requestContext.setQueryParam("mapkind", ObjectSerializer_1.ObjectSerializer.serialize(mapkind, "SourcemapMapKind", ""), "");
            }
            if (dryRun !== undefined) {
                requestContext.setQueryParam("dry_run", ObjectSerializer_1.ObjectSerializer.serialize(dryRun, "boolean", ""), "");
            }
            if (filterService !== undefined) {
                requestContext.setQueryParam("filter[service]", ObjectSerializer_1.ObjectSerializer.serialize(filterService, "Array<string>", ""), "multi");
            }
            if (filterVersion !== undefined) {
                requestContext.setQueryParam("filter[version]", ObjectSerializer_1.ObjectSerializer.serialize(filterVersion, "Array<string>", ""), "multi");
            }
            if (filterVariant !== undefined) {
                requestContext.setQueryParam("filter[variant]", ObjectSerializer_1.ObjectSerializer.serialize(filterVariant, "Array<string>", ""), "multi");
            }
            if (filterId !== undefined) {
                requestContext.setQueryParam("filter[id]", ObjectSerializer_1.ObjectSerializer.serialize(filterId, "Array<string>", ""), "multi");
            }
            if (filterBuildId !== undefined) {
                requestContext.setQueryParam("filter[build_id]", ObjectSerializer_1.ObjectSerializer.serialize(filterBuildId, "Array<string>", ""), "multi");
            }
            if (filterUuid !== undefined) {
                requestContext.setQueryParam("filter[uuid]", ObjectSerializer_1.ObjectSerializer.serialize(filterUuid, "Array<string>", ""), "multi");
            }
            if (filterPlatform !== undefined) {
                requestContext.setQueryParam("filter[platform]", ObjectSerializer_1.ObjectSerializer.serialize(filterPlatform, "Array<string>", ""), "multi");
            }
            if (filterBuildNumber !== undefined) {
                requestContext.setQueryParam("filter[build_number]", ObjectSerializer_1.ObjectSerializer.serialize(filterBuildNumber, "Array<string>", ""), "multi");
            }
            if (filterBundleName !== undefined) {
                requestContext.setQueryParam("filter[bundle_name]", ObjectSerializer_1.ObjectSerializer.serialize(filterBundleName, "Array<string>", ""), "multi");
            }
            if (filterArch !== undefined) {
                requestContext.setQueryParam("filter[arch]", ObjectSerializer_1.ObjectSerializer.serialize(filterArch, "Array<string>", ""), "multi");
            }
            if (filterSymbolSource !== undefined) {
                requestContext.setQueryParam("filter[symbol_source]", ObjectSerializer_1.ObjectSerializer.serialize(filterSymbolSource, "Array<string>", ""), "multi");
            }
            if (filterOrigin !== undefined) {
                requestContext.setQueryParam("filter[origin]", ObjectSerializer_1.ObjectSerializer.serialize(filterOrigin, "Array<string>", ""), "multi");
            }
            if (filterOriginVersion !== undefined) {
                requestContext.setQueryParam("filter[origin_version]", ObjectSerializer_1.ObjectSerializer.serialize(filterOriginVersion, "Array<string>", ""), "multi");
            }
            if (filterFilename !== undefined) {
                requestContext.setQueryParam("filter[filename]", ObjectSerializer_1.ObjectSerializer.serialize(filterFilename, "string", ""), "");
            }
            if (filterDebugId !== undefined) {
                requestContext.setQueryParam("filter[debug_id]", ObjectSerializer_1.ObjectSerializer.serialize(filterDebugId, "string", ""), "");
            }
            if (filterGnuBuildId !== undefined) {
                requestContext.setQueryParam("filter[gnu_build_id]", ObjectSerializer_1.ObjectSerializer.serialize(filterGnuBuildId, "string", ""), "");
            }
            if (filterGoBuildId !== undefined) {
                requestContext.setQueryParam("filter[go_build_id]", ObjectSerializer_1.ObjectSerializer.serialize(filterGoBuildId, "string", ""), "");
            }
            if (filterFileHash !== undefined) {
                requestContext.setQueryParam("filter[file_hash]", ObjectSerializer_1.ObjectSerializer.serialize(filterFileHash, "string", ""), "");
            }
            // Apply auth methods
            (0, configuration_1.applySecurityAuthentication)(_config, requestContext, [
                "apiKeyAuth",
                "appKeyAuth",
            ]);
            return requestContext;
        });
    }
    searchRUMEvents(body, _options) {
        return __awaiter(this, void 0, void 0, function* () {
            const _config = _options || this.configuration;
            // verify required parameter 'body' is not null or undefined
            if (body === null || body === undefined) {
                throw new baseapi_1.RequiredError("body", "searchRUMEvents");
            }
            // Path Params
            const localVarPath = "/api/v2/rum/events/search";
            // Make Request Context
            const requestContext = _config
                .getServer("v2.RUMApi.searchRUMEvents")
                .makeRequestContext(localVarPath, http_1.HttpMethod.POST);
            requestContext.setHeaderParam("Accept", "application/json");
            requestContext.setHttpConfig(_config.httpConfig);
            // Body Params
            const contentType = ObjectSerializer_1.ObjectSerializer.getPreferredMediaType([
                "application/json",
            ]);
            requestContext.setHeaderParam("Content-Type", contentType);
            const serializedBody = ObjectSerializer_1.ObjectSerializer.stringify(ObjectSerializer_1.ObjectSerializer.serialize(body, "RUMSearchEventsRequest", ""), contentType);
            requestContext.setBody(serializedBody);
            // Apply auth methods
            (0, configuration_1.applySecurityAuthentication)(_config, requestContext, [
                "apiKeyAuth",
                "appKeyAuth",
            ]);
            return requestContext;
        });
    }
    updateRUMApplication(id, body, _options) {
        return __awaiter(this, void 0, void 0, function* () {
            const _config = _options || this.configuration;
            // verify required parameter 'id' is not null or undefined
            if (id === null || id === undefined) {
                throw new baseapi_1.RequiredError("id", "updateRUMApplication");
            }
            // verify required parameter 'body' is not null or undefined
            if (body === null || body === undefined) {
                throw new baseapi_1.RequiredError("body", "updateRUMApplication");
            }
            // Path Params
            const localVarPath = "/api/v2/rum/applications/{id}".replace("{id}", encodeURIComponent(String(id)));
            // Make Request Context
            const requestContext = _config
                .getServer("v2.RUMApi.updateRUMApplication")
                .makeRequestContext(localVarPath, http_1.HttpMethod.PATCH);
            requestContext.setHeaderParam("Accept", "application/json");
            requestContext.setHttpConfig(_config.httpConfig);
            // Body Params
            const contentType = ObjectSerializer_1.ObjectSerializer.getPreferredMediaType([
                "application/json",
            ]);
            requestContext.setHeaderParam("Content-Type", contentType);
            const serializedBody = ObjectSerializer_1.ObjectSerializer.stringify(ObjectSerializer_1.ObjectSerializer.serialize(body, "RUMApplicationUpdateRequest", ""), contentType);
            requestContext.setBody(serializedBody);
            // Apply auth methods
            (0, configuration_1.applySecurityAuthentication)(_config, requestContext, [
                "apiKeyAuth",
                "appKeyAuth",
            ]);
            return requestContext;
        });
    }
}
exports.RUMApiRequestFactory = RUMApiRequestFactory;
class RUMApiResponseProcessor {
    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to aggregateRUMEvents
     * @throws ApiException if the response code was not in [200, 299]
     */
    aggregateRUMEvents(response) {
        return __awaiter(this, void 0, void 0, function* () {
            const contentType = ObjectSerializer_1.ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
            if (response.httpStatusCode === 200) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "RUMAnalyticsAggregateResponse");
                return body;
            }
            if (response.httpStatusCode === 400 ||
                response.httpStatusCode === 403 ||
                response.httpStatusCode === 429) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "APIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            // Work around for missing responses in specification, e.g. for petstore.yaml
            if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "RUMAnalyticsAggregateResponse", "");
                return body;
            }
            const body = (yield response.body.text()) || "";
            throw new exception_1.ApiException(response.httpStatusCode, 'Unknown API Status Code!\nBody: "' + body + '"');
        });
    }
    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to createRUMApplication
     * @throws ApiException if the response code was not in [200, 299]
     */
    createRUMApplication(response) {
        return __awaiter(this, void 0, void 0, function* () {
            const contentType = ObjectSerializer_1.ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
            if (response.httpStatusCode === 200) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "RUMApplicationResponse");
                return body;
            }
            if (response.httpStatusCode === 400 || response.httpStatusCode === 429) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "APIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            // Work around for missing responses in specification, e.g. for petstore.yaml
            if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "RUMApplicationResponse", "");
                return body;
            }
            const body = (yield response.body.text()) || "";
            throw new exception_1.ApiException(response.httpStatusCode, 'Unknown API Status Code!\nBody: "' + body + '"');
        });
    }
    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to deleteRUMApplication
     * @throws ApiException if the response code was not in [200, 299]
     */
    deleteRUMApplication(response) {
        return __awaiter(this, void 0, void 0, function* () {
            const contentType = ObjectSerializer_1.ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
            if (response.httpStatusCode === 204) {
                return;
            }
            if (response.httpStatusCode === 404 || response.httpStatusCode === 429) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "APIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            // Work around for missing responses in specification, e.g. for petstore.yaml
            if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
                return;
            }
            const body = (yield response.body.text()) || "";
            throw new exception_1.ApiException(response.httpStatusCode, 'Unknown API Status Code!\nBody: "' + body + '"');
        });
    }
    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to deleteSourcemaps
     * @throws ApiException if the response code was not in [200, 299]
     */
    deleteSourcemaps(response) {
        return __awaiter(this, void 0, void 0, function* () {
            const contentType = ObjectSerializer_1.ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
            if (response.httpStatusCode === 200) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "SourcemapsResponse");
                return body;
            }
            if (response.httpStatusCode === 400 ||
                response.httpStatusCode === 404 ||
                response.httpStatusCode === 500) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "JSONAPIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            if (response.httpStatusCode === 429) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "APIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            // Work around for missing responses in specification, e.g. for petstore.yaml
            if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "SourcemapsResponse", "");
                return body;
            }
            const body = (yield response.body.text()) || "";
            throw new exception_1.ApiException(response.httpStatusCode, 'Unknown API Status Code!\nBody: "' + body + '"');
        });
    }
    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to getRUMApplication
     * @throws ApiException if the response code was not in [200, 299]
     */
    getRUMApplication(response) {
        return __awaiter(this, void 0, void 0, function* () {
            const contentType = ObjectSerializer_1.ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
            if (response.httpStatusCode === 200) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "RUMApplicationResponse");
                return body;
            }
            if (response.httpStatusCode === 404 || response.httpStatusCode === 429) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "APIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            // Work around for missing responses in specification, e.g. for petstore.yaml
            if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "RUMApplicationResponse", "");
                return body;
            }
            const body = (yield response.body.text()) || "";
            throw new exception_1.ApiException(response.httpStatusCode, 'Unknown API Status Code!\nBody: "' + body + '"');
        });
    }
    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to getRUMApplications
     * @throws ApiException if the response code was not in [200, 299]
     */
    getRUMApplications(response) {
        return __awaiter(this, void 0, void 0, function* () {
            const contentType = ObjectSerializer_1.ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
            if (response.httpStatusCode === 200) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "RUMApplicationsResponse");
                return body;
            }
            if (response.httpStatusCode === 404 || response.httpStatusCode === 429) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "APIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            // Work around for missing responses in specification, e.g. for petstore.yaml
            if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "RUMApplicationsResponse", "");
                return body;
            }
            const body = (yield response.body.text()) || "";
            throw new exception_1.ApiException(response.httpStatusCode, 'Unknown API Status Code!\nBody: "' + body + '"');
        });
    }
    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to getServiceRepositoryInfo
     * @throws ApiException if the response code was not in [200, 299]
     */
    getServiceRepositoryInfo(response) {
        return __awaiter(this, void 0, void 0, function* () {
            const contentType = ObjectSerializer_1.ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
            if (response.httpStatusCode === 200) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "ServiceRepositoryInfoResponse");
                return body;
            }
            if (response.httpStatusCode === 401 || response.httpStatusCode === 500) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "JSONAPIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            if (response.httpStatusCode === 429) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "APIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            // Work around for missing responses in specification, e.g. for petstore.yaml
            if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "ServiceRepositoryInfoResponse", "");
                return body;
            }
            const body = (yield response.body.text()) || "";
            throw new exception_1.ApiException(response.httpStatusCode, 'Unknown API Status Code!\nBody: "' + body + '"');
        });
    }
    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to getSourcemaps
     * @throws ApiException if the response code was not in [200, 299]
     */
    getSourcemaps(response) {
        return __awaiter(this, void 0, void 0, function* () {
            const contentType = ObjectSerializer_1.ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
            if (response.httpStatusCode === 200) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "SourcemapFileResponse");
                return body;
            }
            if (response.httpStatusCode === 400 ||
                response.httpStatusCode === 404 ||
                response.httpStatusCode === 500) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "JSONAPIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            if (response.httpStatusCode === 429) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "APIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            // Work around for missing responses in specification, e.g. for petstore.yaml
            if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "SourcemapFileResponse", "");
                return body;
            }
            const body = (yield response.body.text()) || "";
            throw new exception_1.ApiException(response.httpStatusCode, 'Unknown API Status Code!\nBody: "' + body + '"');
        });
    }
    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to listRUMEvents
     * @throws ApiException if the response code was not in [200, 299]
     */
    listRUMEvents(response) {
        return __awaiter(this, void 0, void 0, function* () {
            const contentType = ObjectSerializer_1.ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
            if (response.httpStatusCode === 200) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "RUMEventsResponse");
                return body;
            }
            if (response.httpStatusCode === 400 ||
                response.httpStatusCode === 403 ||
                response.httpStatusCode === 429) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "APIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            // Work around for missing responses in specification, e.g. for petstore.yaml
            if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "RUMEventsResponse", "");
                return body;
            }
            const body = (yield response.body.text()) || "";
            throw new exception_1.ApiException(response.httpStatusCode, 'Unknown API Status Code!\nBody: "' + body + '"');
        });
    }
    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to listSourcemaps
     * @throws ApiException if the response code was not in [200, 299]
     */
    listSourcemaps(response) {
        return __awaiter(this, void 0, void 0, function* () {
            const contentType = ObjectSerializer_1.ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
            if (response.httpStatusCode === 200) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "ListSourcemapsResponse");
                return body;
            }
            if (response.httpStatusCode === 400 ||
                response.httpStatusCode === 413 ||
                response.httpStatusCode === 500) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "JSONAPIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            if (response.httpStatusCode === 429) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "APIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            // Work around for missing responses in specification, e.g. for petstore.yaml
            if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "ListSourcemapsResponse", "");
                return body;
            }
            const body = (yield response.body.text()) || "";
            throw new exception_1.ApiException(response.httpStatusCode, 'Unknown API Status Code!\nBody: "' + body + '"');
        });
    }
    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to restoreSourcemaps
     * @throws ApiException if the response code was not in [200, 299]
     */
    restoreSourcemaps(response) {
        return __awaiter(this, void 0, void 0, function* () {
            const contentType = ObjectSerializer_1.ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
            if (response.httpStatusCode === 200) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "SourcemapsResponse");
                return body;
            }
            if (response.httpStatusCode === 400 ||
                response.httpStatusCode === 404 ||
                response.httpStatusCode === 500) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "JSONAPIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            if (response.httpStatusCode === 429) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "APIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            // Work around for missing responses in specification, e.g. for petstore.yaml
            if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "SourcemapsResponse", "");
                return body;
            }
            const body = (yield response.body.text()) || "";
            throw new exception_1.ApiException(response.httpStatusCode, 'Unknown API Status Code!\nBody: "' + body + '"');
        });
    }
    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to searchRUMEvents
     * @throws ApiException if the response code was not in [200, 299]
     */
    searchRUMEvents(response) {
        return __awaiter(this, void 0, void 0, function* () {
            const contentType = ObjectSerializer_1.ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
            if (response.httpStatusCode === 200) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "RUMEventsResponse");
                return body;
            }
            if (response.httpStatusCode === 400 ||
                response.httpStatusCode === 403 ||
                response.httpStatusCode === 429) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "APIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            // Work around for missing responses in specification, e.g. for petstore.yaml
            if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "RUMEventsResponse", "");
                return body;
            }
            const body = (yield response.body.text()) || "";
            throw new exception_1.ApiException(response.httpStatusCode, 'Unknown API Status Code!\nBody: "' + body + '"');
        });
    }
    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to updateRUMApplication
     * @throws ApiException if the response code was not in [200, 299]
     */
    updateRUMApplication(response) {
        return __awaiter(this, void 0, void 0, function* () {
            const contentType = ObjectSerializer_1.ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
            if (response.httpStatusCode === 200) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "RUMApplicationResponse");
                return body;
            }
            if (response.httpStatusCode === 400 ||
                response.httpStatusCode === 404 ||
                response.httpStatusCode === 422 ||
                response.httpStatusCode === 429) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "APIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            // Work around for missing responses in specification, e.g. for petstore.yaml
            if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "RUMApplicationResponse", "");
                return body;
            }
            const body = (yield response.body.text()) || "";
            throw new exception_1.ApiException(response.httpStatusCode, 'Unknown API Status Code!\nBody: "' + body + '"');
        });
    }
}
exports.RUMApiResponseProcessor = RUMApiResponseProcessor;
class RUMApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.configuration = configuration;
        this.requestFactory =
            requestFactory || new RUMApiRequestFactory(configuration);
        this.responseProcessor = responseProcessor || new RUMApiResponseProcessor();
    }
    /**
     * The API endpoint to aggregate RUM events into buckets of computed metrics and timeseries.
     * @param param The request object
     */
    aggregateRUMEvents(param, options) {
        const requestContextPromise = this.requestFactory.aggregateRUMEvents(param.body, options);
        return requestContextPromise.then((requestContext) => {
            return this.configuration.httpApi
                .send(requestContext)
                .then((responseContext) => {
                return this.responseProcessor.aggregateRUMEvents(responseContext);
            });
        });
    }
    /**
     * Create a new RUM application in your organization.
     * @param param The request object
     */
    createRUMApplication(param, options) {
        const requestContextPromise = this.requestFactory.createRUMApplication(param.body, options);
        return requestContextPromise.then((requestContext) => {
            return this.configuration.httpApi
                .send(requestContext)
                .then((responseContext) => {
                return this.responseProcessor.createRUMApplication(responseContext);
            });
        });
    }
    /**
     * Delete an existing RUM application in your organization.
     * @param param The request object
     */
    deleteRUMApplication(param, options) {
        const requestContextPromise = this.requestFactory.deleteRUMApplication(param.id, options);
        return requestContextPromise.then((requestContext) => {
            return this.configuration.httpApi
                .send(requestContext)
                .then((responseContext) => {
                return this.responseProcessor.deleteRUMApplication(responseContext);
            });
        });
    }
    /**
     * Deletes source maps matching the specified filter criteria. Supports
     * dry-run mode to preview which source maps would be deleted without
     * performing the actual deletion.
     * @param param The request object
     */
    deleteSourcemaps(param, options) {
        const requestContextPromise = this.requestFactory.deleteSourcemaps(param.mapkind, param.dryRun, param.filterService, param.filterVersion, param.filterVariant, param.filterId, param.filterBuildId, param.filterUuid, param.filterPlatform, param.filterBuildNumber, param.filterBundleName, param.filterArch, param.filterSymbolSource, param.filterOrigin, param.filterOriginVersion, param.filterFilename, param.filterDebugId, param.filterGnuBuildId, param.filterGoBuildId, param.filterFileHash, options);
        return requestContextPromise.then((requestContext) => {
            return this.configuration.httpApi
                .send(requestContext)
                .then((responseContext) => {
                return this.responseProcessor.deleteSourcemaps(responseContext);
            });
        });
    }
    /**
     * Get the RUM application with given ID in your organization.
     * @param param The request object
     */
    getRUMApplication(param, options) {
        const requestContextPromise = this.requestFactory.getRUMApplication(param.id, options);
        return requestContextPromise.then((requestContext) => {
            return this.configuration.httpApi
                .send(requestContext)
                .then((responseContext) => {
                return this.responseProcessor.getRUMApplication(responseContext);
            });
        });
    }
    /**
     * List all the RUM applications in your organization.
     * @param param The request object
     */
    getRUMApplications(options) {
        const requestContextPromise = this.requestFactory.getRUMApplications(options);
        return requestContextPromise.then((requestContext) => {
            return this.configuration.httpApi
                .send(requestContext)
                .then((responseContext) => {
                return this.responseProcessor.getRUMApplications(responseContext);
            });
        });
    }
    /**
     * Returns the repository URL and commit SHA associated with a given service and version.
     * @param param The request object
     */
    getServiceRepositoryInfo(param, options) {
        const requestContextPromise = this.requestFactory.getServiceRepositoryInfo(param.body, options);
        return requestContextPromise.then((requestContext) => {
            return this.configuration.httpApi
                .send(requestContext)
                .then((responseContext) => {
                return this.responseProcessor.getServiceRepositoryInfo(responseContext);
            });
        });
    }
    /**
     * Retrieves the content of a specific JavaScript source map file by its
     * filename, service name, and version.
     * @param param The request object
     */
    getSourcemaps(param, options) {
        const requestContextPromise = this.requestFactory.getSourcemaps(param.filename, param.service, param.version, options);
        return requestContextPromise.then((requestContext) => {
            return this.configuration.httpApi
                .send(requestContext)
                .then((responseContext) => {
                return this.responseProcessor.getSourcemaps(responseContext);
            });
        });
    }
    /**
     * List endpoint returns events that match a RUM search query.
     * [Results are paginated][1].
     *
     * Use this endpoint to see your latest RUM events.
     *
     * [1]: https://docs.datadoghq.com/logs/guide/collect-multiple-logs-with-pagination
     * @param param The request object
     */
    listRUMEvents(param = {}, options) {
        const requestContextPromise = this.requestFactory.listRUMEvents(param.filterQuery, param.filterFrom, param.filterTo, param.sort, param.pageCursor, param.pageLimit, options);
        return requestContextPromise.then((requestContext) => {
            return this.configuration.httpApi
                .send(requestContext)
                .then((responseContext) => {
                return this.responseProcessor.listRUMEvents(responseContext);
            });
        });
    }
    /**
     * Provide a paginated version of listRUMEvents returning a generator with all the items.
     */
    listRUMEventsWithPagination(param = {}, options) {
        return __asyncGenerator(this, arguments, function* listRUMEventsWithPagination_1() {
            let pageSize = 10;
            if (param.pageLimit !== undefined) {
                pageSize = param.pageLimit;
            }
            param.pageLimit = pageSize;
            while (true) {
                const requestContext = yield __await(this.requestFactory.listRUMEvents(param.filterQuery, param.filterFrom, param.filterTo, param.sort, param.pageCursor, param.pageLimit, options));
                const responseContext = yield __await(this.configuration.httpApi.send(requestContext));
                const response = yield __await(this.responseProcessor.listRUMEvents(responseContext));
                const responseData = response.data;
                if (responseData === undefined) {
                    break;
                }
                const results = responseData;
                for (const item of results) {
                    yield yield __await(item);
                }
                if (results.length === 0) {
                    break;
                }
                const cursorMeta = response.meta;
                if (cursorMeta === undefined) {
                    break;
                }
                const cursorMetaPage = cursorMeta.page;
                if (cursorMetaPage === undefined) {
                    break;
                }
                const cursorMetaPageAfter = cursorMetaPage.after;
                if (cursorMetaPageAfter === undefined) {
                    break;
                }
                param.pageCursor = cursorMetaPageAfter;
            }
        });
    }
    /**
     * Retrieves a paginated list of source maps matching the specified filter criteria.
     * @param param The request object
     */
    listSourcemaps(param = {}, options) {
        const requestContextPromise = this.requestFactory.listSourcemaps(param.mapkind, param.pageSize, param.pageNumber, param.filterService, param.filterVersion, param.filterVariant, param.filterId, param.filterBuildId, param.filterUuid, param.filterPlatform, param.filterBuildNumber, param.filterBundleName, param.filterArch, param.filterSymbolSource, param.filterOrigin, param.filterOriginVersion, param.filterFilename, param.filterDebugId, param.filterGnuBuildId, param.filterGoBuildId, param.filterFileHash, options);
        return requestContextPromise.then((requestContext) => {
            return this.configuration.httpApi
                .send(requestContext)
                .then((responseContext) => {
                return this.responseProcessor.listSourcemaps(responseContext);
            });
        });
    }
    /**
     * Restores previously deleted source maps matching the specified filter
     * criteria. Supports dry-run mode to preview which source maps would be
     * restored without performing the actual restoration.
     * @param param The request object
     */
    restoreSourcemaps(param, options) {
        const requestContextPromise = this.requestFactory.restoreSourcemaps(param.mapkind, param.dryRun, param.filterService, param.filterVersion, param.filterVariant, param.filterId, param.filterBuildId, param.filterUuid, param.filterPlatform, param.filterBuildNumber, param.filterBundleName, param.filterArch, param.filterSymbolSource, param.filterOrigin, param.filterOriginVersion, param.filterFilename, param.filterDebugId, param.filterGnuBuildId, param.filterGoBuildId, param.filterFileHash, options);
        return requestContextPromise.then((requestContext) => {
            return this.configuration.httpApi
                .send(requestContext)
                .then((responseContext) => {
                return this.responseProcessor.restoreSourcemaps(responseContext);
            });
        });
    }
    /**
     * List endpoint returns RUM events that match a RUM search query.
     * [Results are paginated][1].
     *
     * Use this endpoint to build complex RUM events filtering and search.
     *
     * [1]: https://docs.datadoghq.com/logs/guide/collect-multiple-logs-with-pagination
     * @param param The request object
     */
    searchRUMEvents(param, options) {
        const requestContextPromise = this.requestFactory.searchRUMEvents(param.body, options);
        return requestContextPromise.then((requestContext) => {
            return this.configuration.httpApi
                .send(requestContext)
                .then((responseContext) => {
                return this.responseProcessor.searchRUMEvents(responseContext);
            });
        });
    }
    /**
     * Provide a paginated version of searchRUMEvents returning a generator with all the items.
     */
    searchRUMEventsWithPagination(param, options) {
        return __asyncGenerator(this, arguments, function* searchRUMEventsWithPagination_1() {
            let pageSize = 10;
            if (param.body.page === undefined) {
                param.body.page = new RUMQueryPageOptions_1.RUMQueryPageOptions();
            }
            if (param.body.page.limit === undefined) {
                param.body.page.limit = pageSize;
            }
            else {
                pageSize = param.body.page.limit;
            }
            while (true) {
                const requestContext = yield __await(this.requestFactory.searchRUMEvents(param.body, options));
                const responseContext = yield __await(this.configuration.httpApi.send(requestContext));
                const response = yield __await(this.responseProcessor.searchRUMEvents(responseContext));
                const responseData = response.data;
                if (responseData === undefined) {
                    break;
                }
                const results = responseData;
                for (const item of results) {
                    yield yield __await(item);
                }
                if (results.length === 0) {
                    break;
                }
                const cursorMeta = response.meta;
                if (cursorMeta === undefined) {
                    break;
                }
                const cursorMetaPage = cursorMeta.page;
                if (cursorMetaPage === undefined) {
                    break;
                }
                const cursorMetaPageAfter = cursorMetaPage.after;
                if (cursorMetaPageAfter === undefined) {
                    break;
                }
                param.body.page.cursor = cursorMetaPageAfter;
            }
        });
    }
    /**
     * Update the RUM application with given ID in your organization.
     * @param param The request object
     */
    updateRUMApplication(param, options) {
        const requestContextPromise = this.requestFactory.updateRUMApplication(param.id, param.body, options);
        return requestContextPromise.then((requestContext) => {
            return this.configuration.httpApi
                .send(requestContext)
                .then((responseContext) => {
                return this.responseProcessor.updateRUMApplication(responseContext);
            });
        });
    }
}
exports.RUMApi = RUMApi;
