// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/apis/RumRateLimitApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RumRateLimitApi = exports.RumRateLimitApiResponseProcessor = exports.RumRateLimitApiRequestFactory = void 0;
const baseapi_1 = require("../../datadog-api-client-common/baseapi");
const configuration_1 = require("../../datadog-api-client-common/configuration");
const http_1 = require("../../datadog-api-client-common/http/http");
const logger_1 = require("../../../logger");
const ObjectSerializer_1 = require("../models/ObjectSerializer");
const exception_1 = require("../../datadog-api-client-common/exception");
class RumRateLimitApiRequestFactory extends baseapi_1.BaseAPIRequestFactory {
    deleteRumRateLimitConfig(scopeType, scopeId, _options) {
        return __awaiter(this, void 0, void 0, function* () {
            const _config = _options || this.configuration;
            logger_1.logger.warn("Using unstable operation 'deleteRumRateLimitConfig'");
            if (!_config.unstableOperations["v2.deleteRumRateLimitConfig"]) {
                throw new Error("Unstable operation 'deleteRumRateLimitConfig' is disabled");
            }
            // verify required parameter 'scopeType' is not null or undefined
            if (scopeType === null || scopeType === undefined) {
                throw new baseapi_1.RequiredError("scopeType", "deleteRumRateLimitConfig");
            }
            // verify required parameter 'scopeId' is not null or undefined
            if (scopeId === null || scopeId === undefined) {
                throw new baseapi_1.RequiredError("scopeId", "deleteRumRateLimitConfig");
            }
            // Path Params
            const localVarPath = "/api/v2/rum/config/rate-limit/{scope_type}/{scope_id}"
                .replace("{scope_type}", encodeURIComponent(String(scopeType)))
                .replace("{scope_id}", encodeURIComponent(String(scopeId)));
            // Make Request Context
            const requestContext = _config
                .getServer("v2.RumRateLimitApi.deleteRumRateLimitConfig")
                .makeRequestContext(localVarPath, http_1.HttpMethod.DELETE);
            requestContext.setHeaderParam("Accept", "*/*");
            requestContext.setHttpConfig(_config.httpConfig);
            // Apply auth methods
            (0, configuration_1.applySecurityAuthentication)(_config, requestContext, [
                "apiKeyAuth",
                "appKeyAuth",
            ]);
            return requestContext;
        });
    }
    getRumRateLimitConfig(scopeType, scopeId, _options) {
        return __awaiter(this, void 0, void 0, function* () {
            const _config = _options || this.configuration;
            logger_1.logger.warn("Using unstable operation 'getRumRateLimitConfig'");
            if (!_config.unstableOperations["v2.getRumRateLimitConfig"]) {
                throw new Error("Unstable operation 'getRumRateLimitConfig' is disabled");
            }
            // verify required parameter 'scopeType' is not null or undefined
            if (scopeType === null || scopeType === undefined) {
                throw new baseapi_1.RequiredError("scopeType", "getRumRateLimitConfig");
            }
            // verify required parameter 'scopeId' is not null or undefined
            if (scopeId === null || scopeId === undefined) {
                throw new baseapi_1.RequiredError("scopeId", "getRumRateLimitConfig");
            }
            // Path Params
            const localVarPath = "/api/v2/rum/config/rate-limit/{scope_type}/{scope_id}"
                .replace("{scope_type}", encodeURIComponent(String(scopeType)))
                .replace("{scope_id}", encodeURIComponent(String(scopeId)));
            // Make Request Context
            const requestContext = _config
                .getServer("v2.RumRateLimitApi.getRumRateLimitConfig")
                .makeRequestContext(localVarPath, http_1.HttpMethod.GET);
            requestContext.setHeaderParam("Accept", "application/json");
            requestContext.setHttpConfig(_config.httpConfig);
            // Apply auth methods
            (0, configuration_1.applySecurityAuthentication)(_config, requestContext, [
                "apiKeyAuth",
                "appKeyAuth",
            ]);
            return requestContext;
        });
    }
    updateRumRateLimitConfig(scopeType, scopeId, body, _options) {
        return __awaiter(this, void 0, void 0, function* () {
            const _config = _options || this.configuration;
            logger_1.logger.warn("Using unstable operation 'updateRumRateLimitConfig'");
            if (!_config.unstableOperations["v2.updateRumRateLimitConfig"]) {
                throw new Error("Unstable operation 'updateRumRateLimitConfig' is disabled");
            }
            // verify required parameter 'scopeType' is not null or undefined
            if (scopeType === null || scopeType === undefined) {
                throw new baseapi_1.RequiredError("scopeType", "updateRumRateLimitConfig");
            }
            // verify required parameter 'scopeId' is not null or undefined
            if (scopeId === null || scopeId === undefined) {
                throw new baseapi_1.RequiredError("scopeId", "updateRumRateLimitConfig");
            }
            // verify required parameter 'body' is not null or undefined
            if (body === null || body === undefined) {
                throw new baseapi_1.RequiredError("body", "updateRumRateLimitConfig");
            }
            // Path Params
            const localVarPath = "/api/v2/rum/config/rate-limit/{scope_type}/{scope_id}"
                .replace("{scope_type}", encodeURIComponent(String(scopeType)))
                .replace("{scope_id}", encodeURIComponent(String(scopeId)));
            // Make Request Context
            const requestContext = _config
                .getServer("v2.RumRateLimitApi.updateRumRateLimitConfig")
                .makeRequestContext(localVarPath, http_1.HttpMethod.PUT);
            requestContext.setHeaderParam("Accept", "application/json");
            requestContext.setHttpConfig(_config.httpConfig);
            // Body Params
            const contentType = ObjectSerializer_1.ObjectSerializer.getPreferredMediaType([
                "application/json",
            ]);
            requestContext.setHeaderParam("Content-Type", contentType);
            const serializedBody = ObjectSerializer_1.ObjectSerializer.stringify(ObjectSerializer_1.ObjectSerializer.serialize(body, "RumRateLimitConfigUpdateRequest", ""), contentType);
            requestContext.setBody(serializedBody);
            // Apply auth methods
            (0, configuration_1.applySecurityAuthentication)(_config, requestContext, [
                "apiKeyAuth",
                "appKeyAuth",
            ]);
            return requestContext;
        });
    }
}
exports.RumRateLimitApiRequestFactory = RumRateLimitApiRequestFactory;
class RumRateLimitApiResponseProcessor {
    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to deleteRumRateLimitConfig
     * @throws ApiException if the response code was not in [200, 299]
     */
    deleteRumRateLimitConfig(response) {
        return __awaiter(this, void 0, void 0, function* () {
            const contentType = ObjectSerializer_1.ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
            if (response.httpStatusCode === 204) {
                return;
            }
            if (response.httpStatusCode === 400 ||
                response.httpStatusCode === 403 ||
                response.httpStatusCode === 404 ||
                response.httpStatusCode === 429) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "APIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            // Work around for missing responses in specification, e.g. for petstore.yaml
            if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
                return;
            }
            const body = (yield response.body.text()) || "";
            throw new exception_1.ApiException(response.httpStatusCode, 'Unknown API Status Code!\nBody: "' + body + '"');
        });
    }
    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to getRumRateLimitConfig
     * @throws ApiException if the response code was not in [200, 299]
     */
    getRumRateLimitConfig(response) {
        return __awaiter(this, void 0, void 0, function* () {
            const contentType = ObjectSerializer_1.ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
            if (response.httpStatusCode === 200) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "RumRateLimitConfigResponse");
                return body;
            }
            if (response.httpStatusCode === 400 ||
                response.httpStatusCode === 403 ||
                response.httpStatusCode === 404 ||
                response.httpStatusCode === 429) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "APIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            // Work around for missing responses in specification, e.g. for petstore.yaml
            if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "RumRateLimitConfigResponse", "");
                return body;
            }
            const body = (yield response.body.text()) || "";
            throw new exception_1.ApiException(response.httpStatusCode, 'Unknown API Status Code!\nBody: "' + body + '"');
        });
    }
    /**
     * Unwraps the actual response sent by the server from the response context and deserializes the response content
     * to the expected objects
     *
     * @params response Response returned by the server for a request to updateRumRateLimitConfig
     * @throws ApiException if the response code was not in [200, 299]
     */
    updateRumRateLimitConfig(response) {
        return __awaiter(this, void 0, void 0, function* () {
            const contentType = ObjectSerializer_1.ObjectSerializer.normalizeMediaType(response.headers["content-type"]);
            if (response.httpStatusCode === 200) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "RumRateLimitConfigResponse");
                return body;
            }
            if (response.httpStatusCode === 400 ||
                response.httpStatusCode === 403 ||
                response.httpStatusCode === 404 ||
                response.httpStatusCode === 429) {
                const bodyText = ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType);
                let body;
                try {
                    body = ObjectSerializer_1.ObjectSerializer.deserialize(bodyText, "APIErrorResponse");
                }
                catch (error) {
                    logger_1.logger.debug(`Got error deserializing error: ${error}`);
                    throw new exception_1.ApiException(response.httpStatusCode, bodyText);
                }
                throw new exception_1.ApiException(response.httpStatusCode, body);
            }
            // Work around for missing responses in specification, e.g. for petstore.yaml
            if (response.httpStatusCode >= 200 && response.httpStatusCode <= 299) {
                const body = ObjectSerializer_1.ObjectSerializer.deserialize(ObjectSerializer_1.ObjectSerializer.parse(yield response.body.text(), contentType), "RumRateLimitConfigResponse", "");
                return body;
            }
            const body = (yield response.body.text()) || "";
            throw new exception_1.ApiException(response.httpStatusCode, 'Unknown API Status Code!\nBody: "' + body + '"');
        });
    }
}
exports.RumRateLimitApiResponseProcessor = RumRateLimitApiResponseProcessor;
class RumRateLimitApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.configuration = configuration;
        this.requestFactory =
            requestFactory || new RumRateLimitApiRequestFactory(configuration);
        this.responseProcessor =
            responseProcessor || new RumRateLimitApiResponseProcessor();
    }
    /**
     * Delete the RUM rate limit configuration for a given scope.
     * @param param The request object
     */
    deleteRumRateLimitConfig(param, options) {
        const requestContextPromise = this.requestFactory.deleteRumRateLimitConfig(param.scopeType, param.scopeId, options);
        return requestContextPromise.then((requestContext) => {
            return this.configuration.httpApi
                .send(requestContext)
                .then((responseContext) => {
                return this.responseProcessor.deleteRumRateLimitConfig(responseContext);
            });
        });
    }
    /**
     * Get the RUM rate limit configuration for a given scope.
     * @param param The request object
     */
    getRumRateLimitConfig(param, options) {
        const requestContextPromise = this.requestFactory.getRumRateLimitConfig(param.scopeType, param.scopeId, options);
        return requestContextPromise.then((requestContext) => {
            return this.configuration.httpApi
                .send(requestContext)
                .then((responseContext) => {
                return this.responseProcessor.getRumRateLimitConfig(responseContext);
            });
        });
    }
    /**
     * Create or update the RUM rate limit configuration for a given scope.
     * Returns the rate limit configuration object when the request is successful.
     * @param param The request object
     */
    updateRumRateLimitConfig(param, options) {
        const requestContextPromise = this.requestFactory.updateRumRateLimitConfig(param.scopeType, param.scopeId, param.body, options);
        return requestContextPromise.then((requestContext) => {
            return this.configuration.httpApi
                .send(requestContext)
                .then((responseContext) => {
                return this.responseProcessor.updateRumRateLimitConfig(responseContext);
            });
        });
    }
}
exports.RumRateLimitApi = RumRateLimitApi;
