// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/HostMapWidgetInfrastructureRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HostMapWidgetInfrastructureRequest = void 0;
/**
 * Infrastructure-backed request for the host map widget. Supports entity-based
 * visualization with metric query enrichments, tag-based filtering, flexible grouping,
 * and hierarchical views.
 */
class HostMapWidgetInfrastructureRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return HostMapWidgetInfrastructureRequest.attributeTypeMap;
    }
}
exports.HostMapWidgetInfrastructureRequest = HostMapWidgetInfrastructureRequest;
/**
 * @ignore
 */
HostMapWidgetInfrastructureRequest.attributeTypeMap = {
    child: {
        baseName: "child",
        type: "HostMapWidgetInfrastructureRequestLeaf",
    },
    conditionalFormats: {
        baseName: "conditional_formats",
        type: "Array<WidgetConditionalFormat>",
    },
    enrichments: {
        baseName: "enrichments",
        type: "Array<HostMapWidgetScalarRequest>",
        required: true,
    },
    filter: {
        baseName: "filter",
        type: "string",
    },
    groupBy: {
        baseName: "group_by",
        type: "Array<HostMapWidgetGroupBy>",
    },
    noGroupHosts: {
        baseName: "no_group_hosts",
        type: "boolean",
    },
    noMetricHosts: {
        baseName: "no_metric_hosts",
        type: "boolean",
    },
    nodeType: {
        baseName: "node_type",
        type: "HostMapWidgetNodeType",
        required: true,
    },
    requestType: {
        baseName: "request_type",
        type: "HostMapWidgetInfrastructureRequestRequestType",
        required: true,
    },
    style: {
        baseName: "style",
        type: "HostMapWidgetInfrastructureStyle",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
