// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/HostMapWidgetScalarRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HostMapWidgetScalarRequest = void 0;
/**
 * Scalar formula request for the infrastructure host map widget. Each formula specifies
 * which visual dimension it drives.
 */
class HostMapWidgetScalarRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return HostMapWidgetScalarRequest.attributeTypeMap;
    }
}
exports.HostMapWidgetScalarRequest = HostMapWidgetScalarRequest;
/**
 * @ignore
 */
HostMapWidgetScalarRequest.attributeTypeMap = {
    formulas: {
        baseName: "formulas",
        type: "Array<HostMapWidgetFormula>",
        required: true,
    },
    queries: {
        baseName: "queries",
        type: "Array<FormulaAndFunctionQueryDefinition>",
        required: true,
    },
    responseFormat: {
        baseName: "response_format",
        type: "HostMapWidgetScalarRequestResponseFormat",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
