// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/HostMapWidgetInfrastructureRequestLeaf.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HostMapWidgetInfrastructureRequestLeaf = void 0;
/**
 * Infrastructure-backed host map child request (leaf node, no further nesting supported).
 */
class HostMapWidgetInfrastructureRequestLeaf {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return HostMapWidgetInfrastructureRequestLeaf.attributeTypeMap;
    }
}
exports.HostMapWidgetInfrastructureRequestLeaf = HostMapWidgetInfrastructureRequestLeaf;
/**
 * @ignore
 */
HostMapWidgetInfrastructureRequestLeaf.attributeTypeMap = {
    conditionalFormats: {
        baseName: "conditional_formats",
        type: "Array<WidgetConditionalFormat>",
    },
    enrichments: {
        baseName: "enrichments",
        type: "Array<HostMapWidgetScalarRequest>",
        required: true,
    },
    filter: {
        baseName: "filter",
        type: "string",
    },
    groupBy: {
        baseName: "group_by",
        type: "Array<HostMapWidgetGroupBy>",
    },
    noGroupHosts: {
        baseName: "no_group_hosts",
        type: "boolean",
    },
    noMetricHosts: {
        baseName: "no_metric_hosts",
        type: "boolean",
    },
    nodeType: {
        baseName: "node_type",
        type: "HostMapWidgetNodeType",
        required: true,
    },
    requestType: {
        baseName: "request_type",
        type: "HostMapWidgetInfrastructureRequestRequestType",
        required: true,
    },
    style: {
        baseName: "style",
        type: "HostMapWidgetInfrastructureStyle",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
