// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/HostMapWidgetDefinitionRequests.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HostMapWidgetDefinitionRequests = void 0;
/**
 * Query definition for the host map widget. Supports two mutually exclusive formats distinguished by the presence of `request_type`: the legacy metric-based format (`fill`/`size`) and the infrastructure-backed format (`request_type`, `node_type`, `enrichments`).
 */
class HostMapWidgetDefinitionRequests {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return HostMapWidgetDefinitionRequests.attributeTypeMap;
    }
}
exports.HostMapWidgetDefinitionRequests = HostMapWidgetDefinitionRequests;
/**
 * @ignore
 */
HostMapWidgetDefinitionRequests.attributeTypeMap = {
    child: {
        baseName: "child",
        type: "HostMapWidgetInfrastructureRequest",
    },
    conditionalFormats: {
        baseName: "conditional_formats",
        type: "Array<WidgetConditionalFormat>",
    },
    enrichments: {
        baseName: "enrichments",
        type: "Array<HostMapWidgetScalarRequest>",
    },
    fill: {
        baseName: "fill",
        type: "HostMapRequest",
    },
    filter: {
        baseName: "filter",
        type: "string",
    },
    groupBy: {
        baseName: "group_by",
        type: "Array<HostMapWidgetGroupBy>",
    },
    noGroupHosts: {
        baseName: "no_group_hosts",
        type: "boolean",
    },
    noMetricHosts: {
        baseName: "no_metric_hosts",
        type: "boolean",
    },
    nodeType: {
        baseName: "node_type",
        type: "HostMapWidgetNodeType",
    },
    requestType: {
        baseName: "request_type",
        type: "HostMapWidgetInfrastructureRequestRequestType",
    },
    size: {
        baseName: "size",
        type: "HostMapRequest",
    },
    style: {
        baseName: "style",
        type: "HostMapWidgetInfrastructureStyle",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
