// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/HostMapWidgetFormula.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HostMapWidgetFormula = void 0;
/**
 * Formula for the infrastructure host map widget that specifies both the expression
 * and the visual dimension it populates.
 */
class HostMapWidgetFormula {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return HostMapWidgetFormula.attributeTypeMap;
    }
}
exports.HostMapWidgetFormula = HostMapWidgetFormula;
/**
 * @ignore
 */
HostMapWidgetFormula.attributeTypeMap = {
    alias: {
        baseName: "alias",
        type: "string",
    },
    dimension: {
        baseName: "dimension",
        type: "HostMapWidgetDimension",
        required: true,
    },
    formula: {
        baseName: "formula",
        type: "string",
        required: true,
    },
    numberFormat: {
        baseName: "number_format",
        type: "WidgetNumberFormat",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
