// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/HostMapWidgetGroupBy.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.HostMapWidgetGroupBy = void 0;
/**
 * Defines a grouping dimension for the infrastructure host map.
 */
class HostMapWidgetGroupBy {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return HostMapWidgetGroupBy.attributeTypeMap;
    }
}
exports.HostMapWidgetGroupBy = HostMapWidgetGroupBy;
/**
 * @ignore
 */
HostMapWidgetGroupBy.attributeTypeMap = {
    column: {
        baseName: "column",
        type: "string",
        required: true,
    },
    key: {
        baseName: "key",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
