// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/ListStreamQuery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListStreamQuery = void 0;
/**
 * Updated list stream widget.
 */
class ListStreamQuery {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ListStreamQuery.attributeTypeMap;
    }
}
exports.ListStreamQuery = ListStreamQuery;
/**
 * @ignore
 */
ListStreamQuery.attributeTypeMap = {
    assigneeUuids: {
        baseName: "assignee_uuids",
        type: "Array<string>",
    },
    clusteringPatternFieldPath: {
        baseName: "clustering_pattern_field_path",
        type: "string",
    },
    compute: {
        baseName: "compute",
        type: "Array<ListStreamComputeItems>",
    },
    dataSource: {
        baseName: "data_source",
        type: "ListStreamSource",
        required: true,
    },
    eventSize: {
        baseName: "event_size",
        type: "WidgetEventSize",
    },
    groupBy: {
        baseName: "group_by",
        type: "Array<ListStreamGroupByItems>",
    },
    indexes: {
        baseName: "indexes",
        type: "Array<string>",
    },
    persona: {
        baseName: "persona",
        type: "ListStreamIssuePersona",
    },
    queryString: {
        baseName: "query_string",
        type: "string",
        required: true,
    },
    sort: {
        baseName: "sort",
        type: "WidgetFieldSort",
    },
    states: {
        baseName: "states",
        type: "Array<ListStreamIssueState>",
    },
    storage: {
        baseName: "storage",
        type: "string",
    },
    suspectedCauses: {
        baseName: "suspected_causes",
        type: "Array<string>",
    },
    teamHandles: {
        baseName: "team_handles",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
