// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/LogsArrayMapAttributeRemapper.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogsArrayMapAttributeRemapper = void 0;
/**
 * An attribute remapper sub-processor for use inside an array-map processor.
 * Unlike the top-level attribute remapper, `is_enabled`, `source_type`, and
 * `target_type` are not supported.
 */
class LogsArrayMapAttributeRemapper {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LogsArrayMapAttributeRemapper.attributeTypeMap;
    }
}
exports.LogsArrayMapAttributeRemapper = LogsArrayMapAttributeRemapper;
/**
 * @ignore
 */
LogsArrayMapAttributeRemapper.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
    },
    overrideOnConflict: {
        baseName: "override_on_conflict",
        type: "boolean",
    },
    preserveSource: {
        baseName: "preserve_source",
        type: "boolean",
    },
    sources: {
        baseName: "sources",
        type: "Array<string>",
        required: true,
    },
    target: {
        baseName: "target",
        type: "string",
        required: true,
    },
    targetFormat: {
        baseName: "target_format",
        type: "TargetFormatType",
    },
    type: {
        baseName: "type",
        type: "LogsAttributeRemapperType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
