// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/LogsArrayMapArithmeticSubProcessor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogsArrayMapArithmeticSubProcessor = void 0;
/**
 * An arithmetic sub-processor for use inside an array-map processor.
 * Unlike the top-level arithmetic processor, `is_enabled` is not supported.
 */
class LogsArrayMapArithmeticSubProcessor {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LogsArrayMapArithmeticSubProcessor.attributeTypeMap;
    }
}
exports.LogsArrayMapArithmeticSubProcessor = LogsArrayMapArithmeticSubProcessor;
/**
 * @ignore
 */
LogsArrayMapArithmeticSubProcessor.attributeTypeMap = {
    expression: {
        baseName: "expression",
        type: "string",
        required: true,
    },
    isReplaceMissing: {
        baseName: "is_replace_missing",
        type: "boolean",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    target: {
        baseName: "target",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LogsArithmeticProcessorType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
