// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/LogsExcludeAttributeProcessor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogsExcludeAttributeProcessor = void 0;
/**
 * Use this processor to remove an attribute from a log during processing.
 * The processor strips the specified attribute from the log event, which is useful
 * when the attribute contains sensitive data or is no longer needed downstream.
 */
class LogsExcludeAttributeProcessor {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LogsExcludeAttributeProcessor.attributeTypeMap;
    }
}
exports.LogsExcludeAttributeProcessor = LogsExcludeAttributeProcessor;
/**
 * @ignore
 */
LogsExcludeAttributeProcessor.attributeTypeMap = {
    attributeToExclude: {
        baseName: "attribute_to_exclude",
        type: "string",
        required: true,
    },
    isEnabled: {
        baseName: "is_enabled",
        type: "boolean",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "LogsExcludeAttributeProcessorType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
