// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/LogsArrayMapStringBuilderSubProcessor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogsArrayMapStringBuilderSubProcessor = void 0;
/**
 * A string builder sub-processor for use inside an array-map processor.
 * Unlike the top-level string builder processor, `is_enabled` is not supported.
 */
class LogsArrayMapStringBuilderSubProcessor {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LogsArrayMapStringBuilderSubProcessor.attributeTypeMap;
    }
}
exports.LogsArrayMapStringBuilderSubProcessor = LogsArrayMapStringBuilderSubProcessor;
/**
 * @ignore
 */
LogsArrayMapStringBuilderSubProcessor.attributeTypeMap = {
    isReplaceMissing: {
        baseName: "is_replace_missing",
        type: "boolean",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    target: {
        baseName: "target",
        type: "string",
        required: true,
    },
    template: {
        baseName: "template",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LogsStringBuilderProcessorType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
