// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/LogsArrayMapProcessor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogsArrayMapProcessor = void 0;
/**
 * The array-map processor transforms each element of a source array by applying
 * sub-processors in order and collecting the results into a target array.
 * Results can be written to a new array, to the source array (in-place), or to
 * an existing target array. Sub-processors can read from `$sourceElem.<field>`
 * (object element field), bare `$sourceElem` (primitive element), or any parent
 * log attribute path. Sub-processors write to `$targetElem.<field>` (object
 * output field) or bare `$targetElem` (primitive output).
 */
class LogsArrayMapProcessor {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LogsArrayMapProcessor.attributeTypeMap;
    }
}
exports.LogsArrayMapProcessor = LogsArrayMapProcessor;
/**
 * @ignore
 */
LogsArrayMapProcessor.attributeTypeMap = {
    isEnabled: {
        baseName: "is_enabled",
        type: "boolean",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    preserveSource: {
        baseName: "preserve_source",
        type: "boolean",
    },
    processors: {
        baseName: "processors",
        type: "Array<LogsArrayMapSubProcessor>",
        required: true,
    },
    source: {
        baseName: "source",
        type: "string",
        required: true,
    },
    target: {
        baseName: "target",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LogsArrayMapProcessorType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
