// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/LogsArrayMapCategorySubProcessor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogsArrayMapCategorySubProcessor = void 0;
/**
 * A category sub-processor for use inside an array-map processor.
 * Unlike the top-level category processor, `is_enabled` is not supported.
 */
class LogsArrayMapCategorySubProcessor {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LogsArrayMapCategorySubProcessor.attributeTypeMap;
    }
}
exports.LogsArrayMapCategorySubProcessor = LogsArrayMapCategorySubProcessor;
/**
 * @ignore
 */
LogsArrayMapCategorySubProcessor.attributeTypeMap = {
    categories: {
        baseName: "categories",
        type: "Array<LogsCategoryProcessorCategory>",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
    },
    target: {
        baseName: "target",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LogsCategoryProcessorType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
