// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/MonitorFormulaAndFunctionDataJobsQueryDefinition.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitorFormulaAndFunctionDataJobsQueryDefinition = void 0;
/**
 * A formula and functions data jobs query.
 */
class MonitorFormulaAndFunctionDataJobsQueryDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MonitorFormulaAndFunctionDataJobsQueryDefinition.attributeTypeMap;
    }
}
exports.MonitorFormulaAndFunctionDataJobsQueryDefinition = MonitorFormulaAndFunctionDataJobsQueryDefinition;
/**
 * @ignore
 */
MonitorFormulaAndFunctionDataJobsQueryDefinition.attributeTypeMap = {
    jobType: {
        baseName: "job_type",
        type: "string",
        required: true,
    },
    jobsQuery: {
        baseName: "jobs_query",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    queryDialect: {
        baseName: "query_dialect",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
