// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SyntheticsBasicAuthJWT.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsBasicAuthJWT = void 0;
/**
 * Object to handle JWT authentication when performing the test.
 */
class SyntheticsBasicAuthJWT {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsBasicAuthJWT.attributeTypeMap;
    }
}
exports.SyntheticsBasicAuthJWT = SyntheticsBasicAuthJWT;
/**
 * @ignore
 */
SyntheticsBasicAuthJWT.attributeTypeMap = {
    addClaims: {
        baseName: "addClaims",
        type: "SyntheticsBasicAuthJWTAddClaims",
    },
    algorithm: {
        baseName: "algorithm",
        type: "SyntheticsBasicAuthJWTAlgorithm",
        required: true,
    },
    expiresIn: {
        baseName: "expiresIn",
        type: "number",
        format: "int64",
    },
    header: {
        baseName: "header",
        type: "string",
    },
    payload: {
        baseName: "payload",
        type: "string",
        required: true,
    },
    secret: {
        baseName: "secret",
        type: "string",
        required: true,
    },
    tokenPrefix: {
        baseName: "tokenPrefix",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "SyntheticsBasicAuthJWTType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
