// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SyntheticsBasicAuthJWTAddClaims.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsBasicAuthJWTAddClaims = void 0;
/**
 * Standard JWT claims to automatically inject.
 */
class SyntheticsBasicAuthJWTAddClaims {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsBasicAuthJWTAddClaims.attributeTypeMap;
    }
}
exports.SyntheticsBasicAuthJWTAddClaims = SyntheticsBasicAuthJWTAddClaims;
/**
 * @ignore
 */
SyntheticsBasicAuthJWTAddClaims.attributeTypeMap = {
    exp: {
        baseName: "exp",
        type: "boolean",
    },
    iat: {
        baseName: "iat",
        type: "boolean",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
