// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/models/SyntheticsAssertionMCPServerCapabilitiesTarget.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SyntheticsAssertionMCPServerCapabilitiesTarget = void 0;
/**
 * An assertion that checks that an MCP server advertises the expected capabilities.
 */
class SyntheticsAssertionMCPServerCapabilitiesTarget {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SyntheticsAssertionMCPServerCapabilitiesTarget.attributeTypeMap;
    }
}
exports.SyntheticsAssertionMCPServerCapabilitiesTarget = SyntheticsAssertionMCPServerCapabilitiesTarget;
/**
 * @ignore
 */
SyntheticsAssertionMCPServerCapabilitiesTarget.attributeTypeMap = {
    operator: {
        baseName: "operator",
        type: "SyntheticsAssertionOperator",
        required: true,
    },
    target: {
        baseName: "target",
        type: "Array<SyntheticsMCPServerCapability>",
        required: true,
    },
    type: {
        baseName: "type",
        type: "SyntheticsAssertionMCPServerCapabilitiesType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
