// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AccessTokenListItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccessTokenListItem = void 0;
/**
 * An access token entry returned by the personal access tokens list endpoint. May represent either a personal or a service access token.
 */
class AccessTokenListItem {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AccessTokenListItem.attributeTypeMap;
    }
}
exports.AccessTokenListItem = AccessTokenListItem;
/**
 * @ignore
 */
AccessTokenListItem.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "PersonalAccessTokenAttributes",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    relationships: {
        baseName: "relationships",
        type: "AccessTokenListItemRelationships",
    },
    type: {
        baseName: "type",
        type: "AccessTokensType",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
