// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AggregatedHighFrozenFrameRate.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedHighFrozenFrameRate = void 0;
/**
 * Aggregated high frozen frame rate detection at view level.
 */
class AggregatedHighFrozenFrameRate {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AggregatedHighFrozenFrameRate.attributeTypeMap;
    }
}
exports.AggregatedHighFrozenFrameRate = AggregatedHighFrozenFrameRate;
/**
 * @ignore
 */
AggregatedHighFrozenFrameRate.attributeTypeMap = {
    avgFrozenFrameRate: {
        baseName: "avg_frozen_frame_rate",
        type: "number",
        required: true,
        format: "double",
    },
    avgSegmentDuration: {
        baseName: "avg_segment_duration",
        type: "number",
        required: true,
        format: "int64",
    },
    avgTotalFrozenDuration: {
        baseName: "avg_total_frozen_duration",
        type: "number",
        required: true,
        format: "int64",
    },
    fingerprint: {
        baseName: "fingerprint",
        type: "string",
        required: true,
    },
    impactScore: {
        baseName: "impact_score",
        type: "number",
        required: true,
        format: "double",
    },
    viewOccurrences: {
        baseName: "view_occurrences",
        type: "number",
        required: true,
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
