// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AggregatedResourceTimingBreakdown.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedResourceTimingBreakdown = void 0;
/**
 * Average timing breakdown per network phase for a resource.
 */
class AggregatedResourceTimingBreakdown {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AggregatedResourceTimingBreakdown.attributeTypeMap;
    }
}
exports.AggregatedResourceTimingBreakdown = AggregatedResourceTimingBreakdown;
/**
 * @ignore
 */
AggregatedResourceTimingBreakdown.attributeTypeMap = {
    avgConnectMs: {
        baseName: "avg_connect_ms",
        type: "number",
        required: true,
        format: "double",
    },
    avgDnsMs: {
        baseName: "avg_dns_ms",
        type: "number",
        required: true,
        format: "double",
    },
    avgDownloadMs: {
        baseName: "avg_download_ms",
        type: "number",
        required: true,
        format: "double",
    },
    avgFirstByteMs: {
        baseName: "avg_first_byte_ms",
        type: "number",
        required: true,
        format: "double",
    },
    avgRedirectMs: {
        baseName: "avg_redirect_ms",
        type: "number",
        required: true,
        format: "double",
    },
    avgSslMs: {
        baseName: "avg_ssl_ms",
        type: "number",
        required: true,
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
