// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AggregatedLongTasksResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedLongTasksResponseData = void 0;
/**
 * Data envelope for an aggregated long tasks response.
 */
class AggregatedLongTasksResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AggregatedLongTasksResponseData.attributeTypeMap;
    }
}
exports.AggregatedLongTasksResponseData = AggregatedLongTasksResponseData;
/**
 * @ignore
 */
AggregatedLongTasksResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "AggregatedLongTasksResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "AggregatedLongTasksRequestType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
