// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AggregatedLongTasksResponseAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedLongTasksResponseAttributes = void 0;
/**
 * Attributes of an aggregated long tasks response.
 */
class AggregatedLongTasksResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AggregatedLongTasksResponseAttributes.attributeTypeMap;
    }
}
exports.AggregatedLongTasksResponseAttributes = AggregatedLongTasksResponseAttributes;
/**
 * @ignore
 */
AggregatedLongTasksResponseAttributes.attributeTypeMap = {
    applicationId: {
        baseName: "application_id",
        type: "string",
        required: true,
    },
    criteria: {
        baseName: "criteria",
        type: "AggregatedWaterfallPerformanceCriteria",
    },
    from: {
        baseName: "from",
        type: "number",
        required: true,
        format: "int64",
    },
    longTasksByInvokerType: {
        baseName: "long_tasks_by_invoker_type",
        type: "Array<AggregatedLongTasksByInvokerType>",
        required: true,
    },
    sampledViewIds: {
        baseName: "sampled_view_ids",
        type: "Array<string>",
        required: true,
    },
    to: {
        baseName: "to",
        type: "number",
        required: true,
        format: "int64",
    },
    viewCount: {
        baseName: "view_count",
        type: "number",
        required: true,
        format: "int32",
    },
    viewName: {
        baseName: "view_name",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
