// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AggregatedResource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedResource = void 0;
/**
 * Aggregated performance statistics for a single network resource across sampled view instances.
 */
class AggregatedResource {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AggregatedResource.attributeTypeMap;
    }
}
exports.AggregatedResource = AggregatedResource;
/**
 * @ignore
 */
AggregatedResource.attributeTypeMap = {
    avgDurationMs: {
        baseName: "avg_duration_ms",
        type: "number",
        required: true,
        format: "double",
    },
    avgStartTimeMs: {
        baseName: "avg_start_time_ms",
        type: "number",
        required: true,
        format: "double",
    },
    cacheHitRatePct: {
        baseName: "cache_hit_rate_pct",
        type: "number",
        required: true,
        format: "double",
    },
    cachedCount: {
        baseName: "cached_count",
        type: "number",
        required: true,
        format: "int32",
    },
    downloadedCount: {
        baseName: "downloaded_count",
        type: "number",
        required: true,
        format: "int32",
    },
    globalP75DurationMs: {
        baseName: "global_p75_duration_ms",
        type: "number",
        format: "double",
    },
    globalViewNameCount: {
        baseName: "global_view_name_count",
        type: "number",
        format: "int32",
    },
    globalViewNamePct: {
        baseName: "global_view_name_pct",
        type: "number",
        format: "double",
    },
    httpMethod: {
        baseName: "http_method",
        type: "string",
        required: true,
    },
    loadFrequencyPct: {
        baseName: "load_frequency_pct",
        type: "number",
        required: true,
        format: "double",
    },
    maxDurationMs: {
        baseName: "max_duration_ms",
        type: "number",
        required: true,
        format: "double",
    },
    medianDurationMs: {
        baseName: "median_duration_ms",
        type: "number",
        required: true,
        format: "double",
    },
    minDurationMs: {
        baseName: "min_duration_ms",
        type: "number",
        required: true,
        format: "double",
    },
    p75DurationMs: {
        baseName: "p75_duration_ms",
        type: "number",
        required: true,
        format: "double",
    },
    p95DurationMs: {
        baseName: "p95_duration_ms",
        type: "number",
        required: true,
        format: "double",
    },
    resourceType: {
        baseName: "resource_type",
        type: "string",
        required: true,
    },
    resourceUrlPathGroup: {
        baseName: "resource_url_path_group",
        type: "string",
        required: true,
    },
    timingBreakdown: {
        baseName: "timing_breakdown",
        type: "AggregatedResourceTimingBreakdown",
        required: true,
    },
    totalRequests: {
        baseName: "total_requests",
        type: "number",
        required: true,
        format: "int32",
    },
    viewsWithResource: {
        baseName: "views_with_resource",
        type: "number",
        required: true,
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
