// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AggregatedWaterfallResponseAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedWaterfallResponseAttributes = void 0;
/**
 * Attributes of an aggregated waterfall response.
 */
class AggregatedWaterfallResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AggregatedWaterfallResponseAttributes.attributeTypeMap;
    }
}
exports.AggregatedWaterfallResponseAttributes = AggregatedWaterfallResponseAttributes;
/**
 * @ignore
 */
AggregatedWaterfallResponseAttributes.attributeTypeMap = {
    applicationId: {
        baseName: "application_id",
        type: "string",
        required: true,
    },
    criteria: {
        baseName: "criteria",
        type: "AggregatedWaterfallPerformanceCriteria",
    },
    from: {
        baseName: "from",
        type: "number",
        required: true,
        format: "int64",
    },
    resources: {
        baseName: "resources",
        type: "Array<AggregatedResource>",
        required: true,
    },
    sampledViewIds: {
        baseName: "sampled_view_ids",
        type: "Array<string>",
        required: true,
    },
    to: {
        baseName: "to",
        type: "number",
        required: true,
        format: "int64",
    },
    totalCacheHitRatePct: {
        baseName: "total_cache_hit_rate_pct",
        type: "number",
        required: true,
        format: "double",
    },
    viewCount: {
        baseName: "view_count",
        type: "number",
        required: true,
        format: "int32",
    },
    viewName: {
        baseName: "view_name",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
