// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AggregatedLongTasksRequestAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedLongTasksRequestAttributes = void 0;
/**
 * Attributes for an aggregated long tasks query.
 */
class AggregatedLongTasksRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AggregatedLongTasksRequestAttributes.attributeTypeMap;
    }
}
exports.AggregatedLongTasksRequestAttributes = AggregatedLongTasksRequestAttributes;
/**
 * @ignore
 */
AggregatedLongTasksRequestAttributes.attributeTypeMap = {
    applicationId: {
        baseName: "application_id",
        type: "string",
        required: true,
    },
    criteria: {
        baseName: "criteria",
        type: "AggregatedWaterfallPerformanceCriteria",
    },
    filter: {
        baseName: "filter",
        type: "string",
    },
    from: {
        baseName: "from",
        type: "number",
        required: true,
        format: "int64",
    },
    sampleSize: {
        baseName: "sample_size",
        type: "number",
        required: true,
        format: "int32",
    },
    to: {
        baseName: "to",
        type: "number",
        required: true,
        format: "int64",
    },
    viewName: {
        baseName: "view_name",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
