// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AggregatedWaterfallRequestAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedWaterfallRequestAttributes = void 0;
/**
 * Attributes for an aggregated waterfall query.
 */
class AggregatedWaterfallRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AggregatedWaterfallRequestAttributes.attributeTypeMap;
    }
}
exports.AggregatedWaterfallRequestAttributes = AggregatedWaterfallRequestAttributes;
/**
 * @ignore
 */
AggregatedWaterfallRequestAttributes.attributeTypeMap = {
    applicationId: {
        baseName: "application_id",
        type: "string",
        required: true,
    },
    criteria: {
        baseName: "criteria",
        type: "AggregatedWaterfallPerformanceCriteria",
    },
    filter: {
        baseName: "filter",
        type: "string",
    },
    from: {
        baseName: "from",
        type: "number",
        required: true,
        format: "int64",
    },
    includeGlobalAppearance: {
        baseName: "include_global_appearance",
        type: "boolean",
    },
    sampleSize: {
        baseName: "sample_size",
        type: "number",
        required: true,
        format: "int32",
    },
    to: {
        baseName: "to",
        type: "number",
        required: true,
        format: "int64",
    },
    viewName: {
        baseName: "view_name",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
