// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AggregatedSlowInteractionLongTask.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedSlowInteractionLongTask = void 0;
/**
 * Aggregated slow interaction with long task detection grouped by action and selector.
 */
class AggregatedSlowInteractionLongTask {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AggregatedSlowInteractionLongTask.attributeTypeMap;
    }
}
exports.AggregatedSlowInteractionLongTask = AggregatedSlowInteractionLongTask;
/**
 * @ignore
 */
AggregatedSlowInteractionLongTask.attributeTypeMap = {
    actionType: {
        baseName: "action_type",
        type: "string",
        required: true,
    },
    avgBlockingDuration: {
        baseName: "avg_blocking_duration",
        type: "number",
        required: true,
        format: "int64",
    },
    avgDuration: {
        baseName: "avg_duration",
        type: "number",
        required: true,
        format: "int64",
    },
    fingerprint: {
        baseName: "fingerprint",
        type: "string",
        required: true,
    },
    impactScore: {
        baseName: "impact_score",
        type: "number",
        required: true,
        format: "double",
    },
    instanceCount: {
        baseName: "instance_count",
        type: "number",
        required: true,
        format: "int32",
    },
    selector: {
        baseName: "selector",
        type: "string",
        required: true,
    },
    selectorNormalized: {
        baseName: "selector_normalized",
        type: "string",
        required: true,
    },
    viewOccurrences: {
        baseName: "view_occurrences",
        type: "number",
        required: true,
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
