// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AggregatedWaterfallPerformanceCriteria.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedWaterfallPerformanceCriteria = void 0;
/**
 * Performance criteria to filter view instances by a metric threshold.
 */
class AggregatedWaterfallPerformanceCriteria {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AggregatedWaterfallPerformanceCriteria.attributeTypeMap;
    }
}
exports.AggregatedWaterfallPerformanceCriteria = AggregatedWaterfallPerformanceCriteria;
/**
 * @ignore
 */
AggregatedWaterfallPerformanceCriteria.attributeTypeMap = {
    max: {
        baseName: "max",
        type: "number",
        format: "double",
    },
    metric: {
        baseName: "metric",
        type: "AggregatedWaterfallPerformanceCriteriaMetric",
        required: true,
    },
    min: {
        baseName: "min",
        type: "number",
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
