// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AggregatedHighScriptEval.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedHighScriptEval = void 0;
/**
 * Aggregated high script evaluation detection grouped by source.
 */
class AggregatedHighScriptEval {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AggregatedHighScriptEval.attributeTypeMap;
    }
}
exports.AggregatedHighScriptEval = AggregatedHighScriptEval;
/**
 * @ignore
 */
AggregatedHighScriptEval.attributeTypeMap = {
    avgDuration: {
        baseName: "avg_duration",
        type: "number",
        required: true,
        format: "int64",
    },
    avgForcedStyleLayout: {
        baseName: "avg_forced_style_layout",
        type: "number",
        required: true,
        format: "int64",
    },
    fingerprint: {
        baseName: "fingerprint",
        type: "string",
        required: true,
    },
    impactScore: {
        baseName: "impact_score",
        type: "number",
        required: true,
        format: "double",
    },
    instanceCount: {
        baseName: "instance_count",
        type: "number",
        required: true,
        format: "int32",
    },
    invokerType: {
        baseName: "invoker_type",
        type: "string",
        required: true,
    },
    sourceCategory: {
        baseName: "source_category",
        type: "string",
        required: true,
    },
    sourceFunctionName: {
        baseName: "source_function_name",
        type: "string",
        required: true,
    },
    sourceUrl: {
        baseName: "source_url",
        type: "string",
        required: true,
    },
    viewOccurrences: {
        baseName: "view_occurrences",
        type: "number",
        required: true,
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
