// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AggregatedSlowFCPHighBytes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedSlowFCPHighBytes = void 0;
/**
 * Aggregated slow first contentful paint with high byte count detection.
 */
class AggregatedSlowFCPHighBytes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AggregatedSlowFCPHighBytes.attributeTypeMap;
    }
}
exports.AggregatedSlowFCPHighBytes = AggregatedSlowFCPHighBytes;
/**
 * @ignore
 */
AggregatedSlowFCPHighBytes.attributeTypeMap = {
    avgBytesBeforeFcpBytes: {
        baseName: "avg_bytes_before_fcp_bytes",
        type: "number",
        required: true,
        format: "int64",
    },
    avgFirstContentfulPaintMs: {
        baseName: "avg_first_contentful_paint_ms",
        type: "number",
        required: true,
        format: "int64",
    },
    avgResourceCountBeforeFcp: {
        baseName: "avg_resource_count_before_fcp",
        type: "number",
        required: true,
        format: "int64",
    },
    fingerprint: {
        baseName: "fingerprint",
        type: "string",
        required: true,
    },
    impactScore: {
        baseName: "impact_score",
        type: "number",
        required: true,
        format: "double",
    },
    platform: {
        baseName: "platform",
        type: "string",
        required: true,
    },
    viewOccurrences: {
        baseName: "view_occurrences",
        type: "number",
        required: true,
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
