// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AggregatedSignalsProblemsResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedSignalsProblemsResponseData = void 0;
/**
 * Data envelope for an aggregated signals and problems response.
 */
class AggregatedSignalsProblemsResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AggregatedSignalsProblemsResponseData.attributeTypeMap;
    }
}
exports.AggregatedSignalsProblemsResponseData = AggregatedSignalsProblemsResponseData;
/**
 * @ignore
 */
AggregatedSignalsProblemsResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "AggregatedSignalsProblemsResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "AggregatedSignalsProblemsRequestType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
