// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AggregatedUncompressedResource.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedUncompressedResource = void 0;
/**
 * Aggregated uncompressed resource detection grouped by URL path.
 */
class AggregatedUncompressedResource {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AggregatedUncompressedResource.attributeTypeMap;
    }
}
exports.AggregatedUncompressedResource = AggregatedUncompressedResource;
/**
 * @ignore
 */
AggregatedUncompressedResource.attributeTypeMap = {
    avgBodySize: {
        baseName: "avg_body_size",
        type: "number",
        required: true,
        format: "int64",
    },
    avgDuration: {
        baseName: "avg_duration",
        type: "number",
        required: true,
        format: "int64",
    },
    fingerprint: {
        baseName: "fingerprint",
        type: "string",
        required: true,
    },
    impactScore: {
        baseName: "impact_score",
        type: "number",
        required: true,
        format: "double",
    },
    instanceCount: {
        baseName: "instance_count",
        type: "number",
        required: true,
        format: "int32",
    },
    providerType: {
        baseName: "provider_type",
        type: "string",
        required: true,
    },
    renderBlocking: {
        baseName: "render_blocking",
        type: "string",
        required: true,
    },
    resourceType: {
        baseName: "resource_type",
        type: "string",
        required: true,
    },
    urlPathGroup: {
        baseName: "url_path_group",
        type: "string",
        required: true,
    },
    viewOccurrences: {
        baseName: "view_occurrences",
        type: "number",
        required: true,
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
