// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AggregatedWaterfallResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedWaterfallResponseData = void 0;
/**
 * Data envelope for an aggregated waterfall response.
 */
class AggregatedWaterfallResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AggregatedWaterfallResponseData.attributeTypeMap;
    }
}
exports.AggregatedWaterfallResponseData = AggregatedWaterfallResponseData;
/**
 * @ignore
 */
AggregatedWaterfallResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "AggregatedWaterfallResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "AggregatedWaterfallRequestType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
