// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AggregatedMobileScrollFriction.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedMobileScrollFriction = void 0;
/**
 * Aggregated mobile scroll friction detection at view level.
 */
class AggregatedMobileScrollFriction {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AggregatedMobileScrollFriction.attributeTypeMap;
    }
}
exports.AggregatedMobileScrollFriction = AggregatedMobileScrollFriction;
/**
 * @ignore
 */
AggregatedMobileScrollFriction.attributeTypeMap = {
    avgScrollFrozenFrameCount: {
        baseName: "avg_scroll_frozen_frame_count",
        type: "number",
        required: true,
        format: "int32",
    },
    fingerprint: {
        baseName: "fingerprint",
        type: "string",
        required: true,
    },
    impactScore: {
        baseName: "impact_score",
        type: "number",
        required: true,
        format: "double",
    },
    viewOccurrences: {
        baseName: "view_occurrences",
        type: "number",
        required: true,
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
