// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AggregatedLowCacheHitRate.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedLowCacheHitRate = void 0;
/**
 * Aggregated low cache hit rate detection at view level.
 */
class AggregatedLowCacheHitRate {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AggregatedLowCacheHitRate.attributeTypeMap;
    }
}
exports.AggregatedLowCacheHitRate = AggregatedLowCacheHitRate;
/**
 * @ignore
 */
AggregatedLowCacheHitRate.attributeTypeMap = {
    avgCacheHitRate: {
        baseName: "avg_cache_hit_rate",
        type: "number",
        required: true,
        format: "double",
    },
    avgResourceDownloadSizeBytes: {
        baseName: "avg_resource_download_size_bytes",
        type: "number",
        required: true,
        format: "int64",
    },
    fingerprint: {
        baseName: "fingerprint",
        type: "string",
        required: true,
    },
    impactScore: {
        baseName: "impact_score",
        type: "number",
        required: true,
        format: "double",
    },
    viewOccurrences: {
        baseName: "view_occurrences",
        type: "number",
        required: true,
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
