// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AggregatedSignalsProblemsResponseAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedSignalsProblemsResponseAttributes = void 0;
/**
 * Attributes of an aggregated signals and problems response.
 */
class AggregatedSignalsProblemsResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AggregatedSignalsProblemsResponseAttributes.attributeTypeMap;
    }
}
exports.AggregatedSignalsProblemsResponseAttributes = AggregatedSignalsProblemsResponseAttributes;
/**
 * @ignore
 */
AggregatedSignalsProblemsResponseAttributes.attributeTypeMap = {
    applicationId: {
        baseName: "application_id",
        type: "string",
        required: true,
    },
    criteria: {
        baseName: "criteria",
        type: "AggregatedWaterfallPerformanceCriteria",
    },
    from: {
        baseName: "from",
        type: "number",
        required: true,
        format: "int64",
    },
    problemDetections: {
        baseName: "problem_detections",
        type: "SignalsProblemsDetections",
        required: true,
    },
    sampleMetadata: {
        baseName: "sample_metadata",
        type: "SignalsProblemsSampleMetadata",
        required: true,
    },
    to: {
        baseName: "to",
        type: "number",
        required: true,
        format: "int64",
    },
    viewName: {
        baseName: "view_name",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
