// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AggregatedLongTasksByInvokerType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AggregatedLongTasksByInvokerType = void 0;
/**
 * Aggregated long task statistics for a single invoker type.
 */
class AggregatedLongTasksByInvokerType {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AggregatedLongTasksByInvokerType.attributeTypeMap;
    }
}
exports.AggregatedLongTasksByInvokerType = AggregatedLongTasksByInvokerType;
/**
 * @ignore
 */
AggregatedLongTasksByInvokerType.attributeTypeMap = {
    criteriaViewOccurrences: {
        baseName: "criteria_view_occurrences",
        type: "number",
        format: "int32",
    },
    impactScore: {
        baseName: "impact_score",
        type: "number",
        format: "double",
    },
    invokerType: {
        baseName: "invoker_type",
        type: "string",
        required: true,
    },
    statsPerView: {
        baseName: "stats_per_view",
        type: "LongTaskStatsPerView",
        required: true,
    },
    topInvokers: {
        baseName: "top_invokers",
        type: "Array<TopLongTaskInvoker>",
        required: true,
    },
    viewOccurrences: {
        baseName: "view_occurrences",
        type: "number",
        required: true,
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
