// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AiCustomRuleRevisionResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiCustomRuleRevisionResponseData = void 0;
/**
 * Response data for an AI custom rule revision.
 */
class AiCustomRuleRevisionResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AiCustomRuleRevisionResponseData.attributeTypeMap;
    }
}
exports.AiCustomRuleRevisionResponseData = AiCustomRuleRevisionResponseData;
/**
 * @ignore
 */
AiCustomRuleRevisionResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "AiCustomRuleRevisionResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "AiCustomRuleRevisionDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
