// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AiCustomRuleItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiCustomRuleItem = void 0;
/**
 * An AI custom rule embedded within a ruleset response.
 */
class AiCustomRuleItem {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AiCustomRuleItem.attributeTypeMap;
    }
}
exports.AiCustomRuleItem = AiCustomRuleItem;
/**
 * @ignore
 */
AiCustomRuleItem.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    createdBy: {
        baseName: "created_by",
        type: "string",
        required: true,
    },
    lastRevision: {
        baseName: "last_revision",
        type: "AiCustomRuleRevisionResponseAttributes",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
