// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AiCustomRuleRevisionRequestAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiCustomRuleRevisionRequestAttributes = void 0;
/**
 * Attributes for creating an AI custom rule revision.
 */
class AiCustomRuleRevisionRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AiCustomRuleRevisionRequestAttributes.attributeTypeMap;
    }
}
exports.AiCustomRuleRevisionRequestAttributes = AiCustomRuleRevisionRequestAttributes;
/**
 * @ignore
 */
AiCustomRuleRevisionRequestAttributes.attributeTypeMap = {
    category: {
        baseName: "category",
        type: "CustomRuleRevisionAttributesCategory",
        required: true,
    },
    content: {
        baseName: "content",
        type: "string",
        required: true,
    },
    cwe: {
        baseName: "cwe",
        type: "string",
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    directories: {
        baseName: "directories",
        type: "Array<string>",
        required: true,
    },
    executionMode: {
        baseName: "execution_mode",
        type: "AiCustomRuleRevisionExecutionMode",
        required: true,
    },
    globs: {
        baseName: "globs",
        type: "Array<string>",
        required: true,
    },
    isPublished: {
        baseName: "is_published",
        type: "boolean",
        required: true,
    },
    isTesting: {
        baseName: "is_testing",
        type: "boolean",
        required: true,
    },
    severity: {
        baseName: "severity",
        type: "CustomRuleRevisionAttributesSeverity",
        required: true,
    },
    shortDescription: {
        baseName: "short_description",
        type: "string",
        required: true,
    },
    versionId: {
        baseName: "version_id",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
