// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AiCustomRulesetResponseAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiCustomRulesetResponseAttributes = void 0;
/**
 * Response attributes of an AI custom ruleset.
 */
class AiCustomRulesetResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AiCustomRulesetResponseAttributes.attributeTypeMap;
    }
}
exports.AiCustomRulesetResponseAttributes = AiCustomRulesetResponseAttributes;
/**
 * @ignore
 */
AiCustomRulesetResponseAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    createdBy: {
        baseName: "created_by",
        type: "string",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    rules: {
        baseName: "rules",
        type: "Array<AiCustomRuleItem>",
        required: true,
    },
    shortDescription: {
        baseName: "short_description",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
