// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AiCustomRuleRevisionResponseAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiCustomRuleRevisionResponseAttributes = void 0;
/**
 * Response attributes of an AI custom rule revision.
 */
class AiCustomRuleRevisionResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AiCustomRuleRevisionResponseAttributes.attributeTypeMap;
    }
}
exports.AiCustomRuleRevisionResponseAttributes = AiCustomRuleRevisionResponseAttributes;
/**
 * @ignore
 */
AiCustomRuleRevisionResponseAttributes.attributeTypeMap = {
    category: {
        baseName: "category",
        type: "CustomRuleRevisionAttributesCategory",
        required: true,
    },
    checksum: {
        baseName: "checksum",
        type: "string",
        required: true,
    },
    content: {
        baseName: "content",
        type: "string",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    createdBy: {
        baseName: "created_by",
        type: "string",
        required: true,
    },
    cwe: {
        baseName: "cwe",
        type: "string",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    directories: {
        baseName: "directories",
        type: "Array<string>",
        required: true,
    },
    executionMode: {
        baseName: "execution_mode",
        type: "AiCustomRuleRevisionExecutionMode",
        required: true,
    },
    globs: {
        baseName: "globs",
        type: "Array<string>",
        required: true,
    },
    isDefault: {
        baseName: "is_default",
        type: "boolean",
        required: true,
    },
    isPublished: {
        baseName: "is_published",
        type: "boolean",
        required: true,
    },
    isTesting: {
        baseName: "is_testing",
        type: "boolean",
        required: true,
    },
    severity: {
        baseName: "severity",
        type: "CustomRuleRevisionAttributesSeverity",
        required: true,
    },
    shortDescription: {
        baseName: "short_description",
        type: "string",
        required: true,
    },
    versionId: {
        baseName: "version_id",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
