// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AiCustomRulesetUpdateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiCustomRulesetUpdateAttributes = void 0;
/**
 * Attributes for updating an AI custom ruleset.
 */
class AiCustomRulesetUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AiCustomRulesetUpdateAttributes.attributeTypeMap;
    }
}
exports.AiCustomRulesetUpdateAttributes = AiCustomRulesetUpdateAttributes;
/**
 * @ignore
 */
AiCustomRulesetUpdateAttributes.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    shortDescription: {
        baseName: "short_description",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
