// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AiMemoryViolationResultRequestAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiMemoryViolationResultRequestAttributes = void 0;
/**
 * Attributes for creating an AI memory violation result.
 */
class AiMemoryViolationResultRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AiMemoryViolationResultRequestAttributes.attributeTypeMap;
    }
}
exports.AiMemoryViolationResultRequestAttributes = AiMemoryViolationResultRequestAttributes;
/**
 * @ignore
 */
AiMemoryViolationResultRequestAttributes.attributeTypeMap = {
    line: {
        baseName: "line",
        type: "number",
        required: true,
        format: "int64",
    },
    message: {
        baseName: "message",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    repositoryId: {
        baseName: "repository_id",
        type: "string",
        required: true,
    },
    rule: {
        baseName: "rule",
        type: "string",
        required: true,
    },
    sha: {
        baseName: "sha",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "AiMemoryViolationType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
