// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AiMemoryViolationResultResponseAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiMemoryViolationResultResponseAttributes = void 0;
/**
 * Response attributes of an AI memory violation result.
 */
class AiMemoryViolationResultResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AiMemoryViolationResultResponseAttributes.attributeTypeMap;
    }
}
exports.AiMemoryViolationResultResponseAttributes = AiMemoryViolationResultResponseAttributes;
/**
 * @ignore
 */
AiMemoryViolationResultResponseAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    createdBy: {
        baseName: "created_by",
        type: "string",
        required: true,
    },
    line: {
        baseName: "line",
        type: "number",
        required: true,
        format: "int64",
    },
    message: {
        baseName: "message",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    repositoryId: {
        baseName: "repository_id",
        type: "string",
        required: true,
    },
    rule: {
        baseName: "rule",
        type: "string",
        required: true,
    },
    sha: {
        baseName: "sha",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "AiMemoryViolationType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
