// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AiMemoryViolationResultResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiMemoryViolationResultResponseData = void 0;
/**
 * Response data for an AI memory violation result.
 */
class AiMemoryViolationResultResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AiMemoryViolationResultResponseData.attributeTypeMap;
    }
}
exports.AiMemoryViolationResultResponseData = AiMemoryViolationResultResponseData;
/**
 * @ignore
 */
AiMemoryViolationResultResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "AiMemoryViolationResultResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "AiMemoryViolationResultDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
