// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AiPromptResponseAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiPromptResponseAttributes = void 0;
/**
 * Response attributes of an AI prompt.
 */
class AiPromptResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AiPromptResponseAttributes.attributeTypeMap;
    }
}
exports.AiPromptResponseAttributes = AiPromptResponseAttributes;
/**
 * @ignore
 */
AiPromptResponseAttributes.attributeTypeMap = {
    category: {
        baseName: "category",
        type: "CustomRuleRevisionAttributesCategory",
        required: true,
    },
    checksum: {
        baseName: "checksum",
        type: "string",
        required: true,
    },
    content: {
        baseName: "content",
        type: "string",
        required: true,
    },
    cwe: {
        baseName: "cwe",
        type: "string",
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    directories: {
        baseName: "directories",
        type: "Array<string>",
        required: true,
    },
    executionMode: {
        baseName: "execution_mode",
        type: "AiCustomRuleRevisionExecutionMode",
        required: true,
    },
    fileSearchKeywords: {
        baseName: "file_search_keywords",
        type: "Array<string>",
        required: true,
    },
    globs: {
        baseName: "globs",
        type: "Array<string>",
        required: true,
    },
    isDefault: {
        baseName: "is_default",
        type: "boolean",
        required: true,
    },
    isTesting: {
        baseName: "is_testing",
        type: "boolean",
        required: true,
    },
    language: {
        baseName: "language",
        type: "Language",
    },
    resultKeywordsExclude: {
        baseName: "result_keywords_exclude",
        type: "Array<string>",
        required: true,
    },
    ruleVersion: {
        baseName: "rule_version",
        type: "string",
        required: true,
    },
    severity: {
        baseName: "severity",
        type: "CustomRuleRevisionAttributesSeverity",
        required: true,
    },
    shortDescription: {
        baseName: "short_description",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
