// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AnalysisRequestDataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnalysisRequestDataAttributes = void 0;
/**
 * The attributes of the analysis request, containing the source code and rules to apply.
 */
class AnalysisRequestDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AnalysisRequestDataAttributes.attributeTypeMap;
    }
}
exports.AnalysisRequestDataAttributes = AnalysisRequestDataAttributes;
/**
 * @ignore
 */
AnalysisRequestDataAttributes.attributeTypeMap = {
    code: {
        baseName: "code",
        type: "string",
        required: true,
    },
    fileEncoding: {
        baseName: "file_encoding",
        type: "string",
        required: true,
    },
    filename: {
        baseName: "filename",
        type: "string",
        required: true,
    },
    language: {
        baseName: "language",
        type: "string",
        required: true,
    },
    rules: {
        baseName: "rules",
        type: "Array<AnalysisRequestRule>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
