// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AnalysisRuleResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnalysisRuleResponse = void 0;
/**
 * The result of applying a single static analysis rule to the analyzed source code.
 */
class AnalysisRuleResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AnalysisRuleResponse.attributeTypeMap;
    }
}
exports.AnalysisRuleResponse = AnalysisRuleResponse;
/**
 * @ignore
 */
AnalysisRuleResponse.attributeTypeMap = {
    errors: {
        baseName: "errors",
        type: "Array<string>",
        required: true,
    },
    executionError: {
        baseName: "execution_error",
        type: "string",
        required: true,
    },
    executionTimeMs: {
        baseName: "execution_time_ms",
        type: "number",
        required: true,
        format: "int64",
    },
    identifier: {
        baseName: "identifier",
        type: "string",
        required: true,
    },
    output: {
        baseName: "output",
        type: "string",
        required: true,
    },
    violations: {
        baseName: "violations",
        type: "Array<AnalysisViolation>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
