// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AnalysisRequestRule.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnalysisRequestRule = void 0;
/**
 * A static analysis rule to apply during code analysis.
 */
class AnalysisRequestRule {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AnalysisRequestRule.attributeTypeMap;
    }
}
exports.AnalysisRequestRule = AnalysisRequestRule;
/**
 * @ignore
 */
AnalysisRequestRule.attributeTypeMap = {
    category: {
        baseName: "category",
        type: "string",
        required: true,
    },
    checksum: {
        baseName: "checksum",
        type: "string",
        required: true,
    },
    code: {
        baseName: "code",
        type: "string",
        required: true,
    },
    entityChecked: {
        baseName: "entity_checked",
        type: "string",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    language: {
        baseName: "language",
        type: "string",
        required: true,
    },
    regex: {
        baseName: "regex",
        type: "string",
    },
    severity: {
        baseName: "severity",
        type: "string",
        required: true,
    },
    treeSitterQuery: {
        baseName: "tree_sitter_query",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
