// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AnalysisViolation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnalysisViolation = void 0;
/**
 * A rule violation found in the analyzed source code.
 */
class AnalysisViolation {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AnalysisViolation.attributeTypeMap;
    }
}
exports.AnalysisViolation = AnalysisViolation;
/**
 * @ignore
 */
AnalysisViolation.attributeTypeMap = {
    category: {
        baseName: "category",
        type: "string",
        required: true,
    },
    end: {
        baseName: "end",
        type: "AnalysisPosition",
        required: true,
    },
    fixes: {
        baseName: "fixes",
        type: "Array<AnalysisFix>",
        required: true,
    },
    message: {
        baseName: "message",
        type: "string",
        required: true,
    },
    severity: {
        baseName: "severity",
        type: "string",
        required: true,
    },
    start: {
        baseName: "start",
        type: "AnalysisPosition",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
