// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AnnotationAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnnotationAttributes = void 0;
/**
 * Attributes of an annotation returned in a response.
 */
class AnnotationAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AnnotationAttributes.attributeTypeMap;
    }
}
exports.AnnotationAttributes = AnnotationAttributes;
/**
 * @ignore
 */
AnnotationAttributes.attributeTypeMap = {
    authorId: {
        baseName: "author_id",
        type: "string",
        required: true,
    },
    color: {
        baseName: "color",
        type: "AnnotationColor",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "number",
        required: true,
        format: "int64",
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    endTime: {
        baseName: "end_time",
        type: "number",
        required: true,
        format: "int64",
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "number",
        required: true,
        format: "int64",
    },
    pageId: {
        baseName: "page_id",
        type: "string",
        required: true,
    },
    startTime: {
        baseName: "start_time",
        type: "number",
        required: true,
        format: "int64",
    },
    type: {
        baseName: "type",
        type: "AnnotationKind",
        required: true,
    },
    widgetIds: {
        baseName: "widget_ids",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
