// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AnnotationInPage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnnotationInPage = void 0;
/**
 * A flat annotation object as it appears within a page annotations response.
 */
class AnnotationInPage {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AnnotationInPage.attributeTypeMap;
    }
}
exports.AnnotationInPage = AnnotationInPage;
/**
 * @ignore
 */
AnnotationInPage.attributeTypeMap = {
    authorId: {
        baseName: "author_id",
        type: "string",
        required: true,
    },
    color: {
        baseName: "color",
        type: "AnnotationColor",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "number",
        required: true,
        format: "int64",
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    endTime: {
        baseName: "end_time",
        type: "number",
        required: true,
        format: "int64",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
        format: "uuid",
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "number",
        required: true,
        format: "int64",
    },
    pageId: {
        baseName: "page_id",
        type: "string",
        required: true,
    },
    startTime: {
        baseName: "start_time",
        type: "number",
        required: true,
        format: "int64",
    },
    type: {
        baseName: "type",
        type: "AnnotationKind",
        required: true,
    },
    widgetIds: {
        baseName: "widget_ids",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
