// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AnnotationCreateAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnnotationCreateAttributes = void 0;
/**
 * Attributes for creating or updating an annotation.
 */
class AnnotationCreateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AnnotationCreateAttributes.attributeTypeMap;
    }
}
exports.AnnotationCreateAttributes = AnnotationCreateAttributes;
/**
 * @ignore
 */
AnnotationCreateAttributes.attributeTypeMap = {
    color: {
        baseName: "color",
        type: "AnnotationColor",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    endTime: {
        baseName: "end_time",
        type: "number",
        format: "int64",
    },
    pageId: {
        baseName: "page_id",
        type: "string",
        required: true,
    },
    startTime: {
        baseName: "start_time",
        type: "number",
        required: true,
        format: "int64",
    },
    type: {
        baseName: "type",
        type: "AnnotationKind",
        required: true,
    },
    widgetIds: {
        baseName: "widget_ids",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
