// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/APMTraceSpan.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.APMTraceSpan = void 0;
/**
 * A single APM span returned as part of a trace.
 */
class APMTraceSpan {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return APMTraceSpan.attributeTypeMap;
    }
}
exports.APMTraceSpan = APMTraceSpan;
/**
 * @ignore
 */
APMTraceSpan.attributeTypeMap = {
    duration: {
        baseName: "duration",
        type: "number",
        required: true,
        format: "int64",
    },
    endTime: {
        baseName: "endTime",
        type: "number",
        required: true,
        format: "int64",
    },
    error: {
        baseName: "error",
        type: "APMSpanErrorFlag",
        required: true,
    },
    meta: {
        baseName: "meta",
        type: "{ [key: string]: string; }",
        required: true,
    },
    metrics: {
        baseName: "metrics",
        type: "{ [key: string]: number; }",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    parentId: {
        baseName: "parentID",
        type: "number",
        required: true,
        format: "int64",
    },
    resource: {
        baseName: "resource",
        type: "string",
        required: true,
    },
    resourceHash: {
        baseName: "resourceHash",
        type: "string",
    },
    restricted: {
        baseName: "restricted",
        type: "boolean",
    },
    selfTime: {
        baseName: "self_time",
        type: "number",
        format: "double",
    },
    service: {
        baseName: "service",
        type: "string",
        required: true,
    },
    spanId: {
        baseName: "spanID",
        type: "number",
        required: true,
        format: "int64",
    },
    startTime: {
        baseName: "startTime",
        type: "number",
        required: true,
        format: "int64",
    },
    traceId: {
        baseName: "traceID",
        type: "number",
        required: true,
        format: "int64",
    },
    traceIdFull: {
        baseName: "traceIDFull",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
