// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ApplicationSecurityServiceResource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationSecurityServiceResource = void 0;
/**
 * A JSON:API resource describing a service and its Application Security details.
 */
class ApplicationSecurityServiceResource {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ApplicationSecurityServiceResource.attributeTypeMap;
    }
}
exports.ApplicationSecurityServiceResource = ApplicationSecurityServiceResource;
/**
 * @ignore
 */
ApplicationSecurityServiceResource.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "ApplicationSecurityServiceAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "ApplicationSecurityServiceType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
