// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ApplicationSecurityPolicyRulesetOverride.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationSecurityPolicyRulesetOverride = void 0;
/**
 * Deprecated: Override WAF ruleset parameters. Use `protectionPresets` instead.
 */
class ApplicationSecurityPolicyRulesetOverride {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ApplicationSecurityPolicyRulesetOverride.attributeTypeMap;
    }
}
exports.ApplicationSecurityPolicyRulesetOverride = ApplicationSecurityPolicyRulesetOverride;
/**
 * @ignore
 */
ApplicationSecurityPolicyRulesetOverride.attributeTypeMap = {
    blocking: {
        baseName: "blocking",
        type: "boolean",
        required: true,
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
