// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ApplicationSecurityServicesMetadata.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationSecurityServicesMetadata = void 0;
/**
 * Metadata returned alongside the list of services.
 */
class ApplicationSecurityServicesMetadata {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ApplicationSecurityServicesMetadata.attributeTypeMap;
    }
}
exports.ApplicationSecurityServicesMetadata = ApplicationSecurityServicesMetadata;
/**
 * @ignore
 */
ApplicationSecurityServicesMetadata.attributeTypeMap = {
    numServicesWithAppsec: {
        baseName: "num_services_with_appsec",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
