// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ApplicationSecurityServiceAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationSecurityServiceAttributes = void 0;
/**
 * Application Security details describing a service in a given environment.
 */
class ApplicationSecurityServiceAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ApplicationSecurityServiceAttributes.attributeTypeMap;
    }
}
exports.ApplicationSecurityServiceAttributes = ApplicationSecurityServiceAttributes;
/**
 * @ignore
 */
ApplicationSecurityServiceAttributes.attributeTypeMap = {
    agentVersions: {
        baseName: "agent_versions",
        type: "Array<string>",
        required: true,
    },
    appType: {
        baseName: "app_type",
        type: "string",
        required: true,
    },
    asmThreatCompatible: {
        baseName: "asm_threat_compatible",
        type: "boolean",
        required: true,
    },
    backendWafEventCount: {
        baseName: "backend_waf_event_count",
        type: "number",
        required: true,
        format: "int64",
    },
    businessLogic: {
        baseName: "business_logic",
        type: "Array<string>",
        required: true,
    },
    color: {
        baseName: "color",
        type: "string",
        required: true,
    },
    env: {
        baseName: "env",
        type: "string",
        required: true,
    },
    eventCount: {
        baseName: "event_count",
        type: "number",
        required: true,
        format: "int64",
    },
    eventTrend: {
        baseName: "event_trend",
        type: "Array<number>",
        required: true,
    },
    hasAppsecEnabled: {
        baseName: "has_appsec_enabled",
        type: "boolean",
        required: true,
    },
    hits: {
        baseName: "hits",
        type: "number",
        required: true,
        format: "int64",
    },
    iastProductActivation: {
        baseName: "iast_product_activation",
        type: "boolean",
        required: true,
    },
    iastProductCompatibility: {
        baseName: "iast_product_compatibility",
        type: "string",
        required: true,
    },
    iastProductCompatibilityReasons: {
        baseName: "iast_product_compatibility_reasons",
        type: "Array<string>",
        required: true,
    },
    languages: {
        baseName: "languages",
        type: "Array<string>",
        required: true,
    },
    lastIngestedSpans: {
        baseName: "last_ingested_spans",
        type: "number",
        required: true,
        format: "int64",
    },
    rcCapabilities: {
        baseName: "rc_capabilities",
        type: "Array<string>",
        required: true,
    },
    recommendedBusinessLogic: {
        baseName: "recommended_business_logic",
        type: "Array<string>",
        required: true,
    },
    riskProductActivation: {
        baseName: "risk_product_activation",
        type: "boolean",
        required: true,
    },
    riskProductCompatibility: {
        baseName: "risk_product_compatibility",
        type: "string",
        required: true,
    },
    riskProductCompatibilityReasons: {
        baseName: "risk_product_compatibility_reasons",
        type: "Array<string>",
        required: true,
    },
    rulesVersion: {
        baseName: "rules_version",
        type: "Array<string>",
        required: true,
    },
    service: {
        baseName: "service",
        type: "string",
        required: true,
    },
    signalCount: {
        baseName: "signal_count",
        type: "number",
        required: true,
        format: "int64",
    },
    signalTrend: {
        baseName: "signal_trend",
        type: "Array<number>",
        required: true,
    },
    source: {
        baseName: "source",
        type: "Array<string>",
        required: true,
    },
    teams: {
        baseName: "teams",
        type: "Array<string>",
        required: true,
    },
    tracerVersions: {
        baseName: "tracer_versions",
        type: "Array<string>",
        required: true,
    },
    vmActivation: {
        baseName: "vm-activation",
        type: "string",
        required: true,
    },
    vulnCriticalCount: {
        baseName: "vuln_critical_count",
        type: "number",
        required: true,
        format: "int64",
    },
    vulnHighCount: {
        baseName: "vuln_high_count",
        type: "number",
        required: true,
        format: "int64",
    },
    withoutFilterServices: {
        baseName: "without_filter_services",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
