// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ApplicationSecurityPolicyCreateAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationSecurityPolicyCreateAttributes = void 0;
/**
 * Create a new WAF policy.
 */
class ApplicationSecurityPolicyCreateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ApplicationSecurityPolicyCreateAttributes.attributeTypeMap;
    }
}
exports.ApplicationSecurityPolicyCreateAttributes = ApplicationSecurityPolicyCreateAttributes;
/**
 * @ignore
 */
ApplicationSecurityPolicyCreateAttributes.attributeTypeMap = {
    basedOn: {
        baseName: "basedOn",
        type: "string",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    isDefault: {
        baseName: "isDefault",
        type: "boolean",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    protectionPresets: {
        baseName: "protectionPresets",
        type: "Array<string>",
    },
    rules: {
        baseName: "rules",
        type: "Array<ApplicationSecurityPolicyRuleOverride>",
    },
    rulesets: {
        baseName: "rulesets",
        type: "Array<ApplicationSecurityPolicyRulesetOverride>",
    },
    scope: {
        baseName: "scope",
        type: "Array<ApplicationSecurityPolicyScope>",
    },
    version: {
        baseName: "version",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
