// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AssignmentResult.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssignmentResult = void 0;
/**
 * Per-finding outcome of an assign or unassign operation.
 */
class AssignmentResult {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AssignmentResult.attributeTypeMap;
    }
}
exports.AssignmentResult = AssignmentResult;
/**
 * @ignore
 */
AssignmentResult.attributeTypeMap = {
    detail: {
        baseName: "detail",
        type: "string",
        required: true,
    },
    findingId: {
        baseName: "finding_id",
        type: "string",
        required: true,
    },
    status: {
        baseName: "status",
        type: "number",
        required: true,
        format: "int32",
    },
    title: {
        baseName: "title",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
