// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AttachServiceNowTicketRequestData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AttachServiceNowTicketRequestData = void 0;
/**
 * Data of the ServiceNow ticket to attach security findings to.
 */
class AttachServiceNowTicketRequestData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AttachServiceNowTicketRequestData.attributeTypeMap;
    }
}
exports.AttachServiceNowTicketRequestData = AttachServiceNowTicketRequestData;
/**
 * @ignore
 */
AttachServiceNowTicketRequestData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "AttachServiceNowTicketRequestDataAttributes",
        required: true,
    },
    relationships: {
        baseName: "relationships",
        type: "AttachServiceNowTicketRequestDataRelationships",
        required: true,
    },
    type: {
        baseName: "type",
        type: "ServiceNowTicketsDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
