// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AttachServiceNowTicketRequestDataAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AttachServiceNowTicketRequestDataAttributes = void 0;
/**
 * Attributes of the ServiceNow ticket to attach security findings to.
 */
class AttachServiceNowTicketRequestDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AttachServiceNowTicketRequestDataAttributes.attributeTypeMap;
    }
}
exports.AttachServiceNowTicketRequestDataAttributes = AttachServiceNowTicketRequestDataAttributes;
/**
 * @ignore
 */
AttachServiceNowTicketRequestDataAttributes.attributeTypeMap = {
    servicenowTicketUrl: {
        baseName: "servicenow_ticket_url",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
