// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AutomationRuleTriggerData.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutomationRuleTriggerData = void 0;
/**
 * Additional configuration for the trigger, dependent on the trigger type. For `status_transitioned` triggers, specify `from_status_name` and `to_status_name`. For `attribute_value_changed` triggers, specify `field` and `change_type`.
 */
class AutomationRuleTriggerData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AutomationRuleTriggerData.attributeTypeMap;
    }
}
exports.AutomationRuleTriggerData = AutomationRuleTriggerData;
/**
 * @ignore
 */
AutomationRuleTriggerData.attributeTypeMap = {
    approvalType: {
        baseName: "approval_type",
        type: "string",
    },
    changeType: {
        baseName: "change_type",
        type: "string",
    },
    field: {
        baseName: "field",
        type: "string",
    },
    fromStatusName: {
        baseName: "from_status_name",
        type: "string",
    },
    toStatusName: {
        baseName: "to_status_name",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
