// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AutomationRuleCreateAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutomationRuleCreateAttributes = void 0;
/**
 * Attributes required to create an automation rule.
 */
class AutomationRuleCreateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AutomationRuleCreateAttributes.attributeTypeMap;
    }
}
exports.AutomationRuleCreateAttributes = AutomationRuleCreateAttributes;
/**
 * @ignore
 */
AutomationRuleCreateAttributes.attributeTypeMap = {
    action: {
        baseName: "action",
        type: "AutomationRuleAction",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    state: {
        baseName: "state",
        type: "CaseAutomationRuleState",
    },
    trigger: {
        baseName: "trigger",
        type: "AutomationRuleTrigger",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
