// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AutomationRuleActionData.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutomationRuleActionData = void 0;
/**
 * Configuration for the action to execute, dependent on the action type.
 */
class AutomationRuleActionData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AutomationRuleActionData.attributeTypeMap;
    }
}
exports.AutomationRuleActionData = AutomationRuleActionData;
/**
 * @ignore
 */
AutomationRuleActionData.attributeTypeMap = {
    agentType: {
        baseName: "agent_type",
        type: "string",
    },
    assignedAgentId: {
        baseName: "assigned_agent_id",
        type: "string",
    },
    handle: {
        baseName: "handle",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
