// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AutomationRule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutomationRule = void 0;
/**
 * An automation rule that executes an action (such as running a Datadog workflow or assigning an AI agent) when a specified case event occurs within a project.
 */
class AutomationRule {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AutomationRule.attributeTypeMap;
    }
}
exports.AutomationRule = AutomationRule;
/**
 * @ignore
 */
AutomationRule.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "AutomationRuleAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    relationships: {
        baseName: "relationships",
        type: "AutomationRuleRelationships",
    },
    type: {
        baseName: "type",
        type: "CaseAutomationRuleResourceType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
