// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AutomationRuleAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutomationRuleAttributes = void 0;
/**
 * Core attributes of an automation rule, including its name, trigger condition, action to execute, and current state.
 */
class AutomationRuleAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AutomationRuleAttributes.attributeTypeMap;
    }
}
exports.AutomationRuleAttributes = AutomationRuleAttributes;
/**
 * @ignore
 */
AutomationRuleAttributes.attributeTypeMap = {
    action: {
        baseName: "action",
        type: "AutomationRuleAction",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        format: "date-time",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    state: {
        baseName: "state",
        type: "CaseAutomationRuleState",
        required: true,
    },
    trigger: {
        baseName: "trigger",
        type: "AutomationRuleTrigger",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
