// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AWSCcmConfigValidationResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AWSCcmConfigValidationResponseData = void 0;
/**
 * AWS CCM config validation response data.
 */
class AWSCcmConfigValidationResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AWSCcmConfigValidationResponseData.attributeTypeMap;
    }
}
exports.AWSCcmConfigValidationResponseData = AWSCcmConfigValidationResponseData;
/**
 * @ignore
 */
AWSCcmConfigValidationResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "AWSCcmConfigValidationResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "AWSCcmConfigValidationType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
