// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AWSCcmConfigValidationIssueCode.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.QUERY_STATEMENT_INVALID = exports.OVERWRITE_MODE_INVALID = exports.REFRESH_CADENCE_INVALID = exports.INCLUDE_RESOURCES_DISABLED = exports.FILE_FORMAT_INVALID = exports.TIME_GRANULARITY_INVALID = exports.EXPORT_STATUS_UNHEALTHY = exports.EXPORT_NOT_FOUND = exports.EXPORT_GET_PERMISSION_MISSING = exports.EXPORT_LIST_PERMISSION_MISSING = exports.S3_BUCKET_NOT_ACCESSIBLE = exports.S3_BUCKET_REGION_MISMATCH = exports.S3_GET_PERMISSION_MISSING = exports.S3_LIST_PERMISSION_MISSING = exports.BUCKET_NAME_INVALID_GOVCLOUD = exports.CREDENTIAL_ERROR = exports.ISSUE_CODE_UNSPECIFIED = void 0;
exports.ISSUE_CODE_UNSPECIFIED = "ISSUE_CODE_UNSPECIFIED";
exports.CREDENTIAL_ERROR = "CREDENTIAL_ERROR";
exports.BUCKET_NAME_INVALID_GOVCLOUD = "BUCKET_NAME_INVALID_GOVCLOUD";
exports.S3_LIST_PERMISSION_MISSING = "S3_LIST_PERMISSION_MISSING";
exports.S3_GET_PERMISSION_MISSING = "S3_GET_PERMISSION_MISSING";
exports.S3_BUCKET_REGION_MISMATCH = "S3_BUCKET_REGION_MISMATCH";
exports.S3_BUCKET_NOT_ACCESSIBLE = "S3_BUCKET_NOT_ACCESSIBLE";
exports.EXPORT_LIST_PERMISSION_MISSING = "EXPORT_LIST_PERMISSION_MISSING";
exports.EXPORT_GET_PERMISSION_MISSING = "EXPORT_GET_PERMISSION_MISSING";
exports.EXPORT_NOT_FOUND = "EXPORT_NOT_FOUND";
exports.EXPORT_STATUS_UNHEALTHY = "EXPORT_STATUS_UNHEALTHY";
exports.TIME_GRANULARITY_INVALID = "TIME_GRANULARITY_INVALID";
exports.FILE_FORMAT_INVALID = "FILE_FORMAT_INVALID";
exports.INCLUDE_RESOURCES_DISABLED = "INCLUDE_RESOURCES_DISABLED";
exports.REFRESH_CADENCE_INVALID = "REFRESH_CADENCE_INVALID";
exports.OVERWRITE_MODE_INVALID = "OVERWRITE_MODE_INVALID";
exports.QUERY_STATEMENT_INVALID = "QUERY_STATEMENT_INVALID";
