// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/AWSCcmConfigValidationRequestAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AWSCcmConfigValidationRequestAttributes = void 0;
/**
 * Attributes for an AWS CCM config validation request.
 */
class AWSCcmConfigValidationRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return AWSCcmConfigValidationRequestAttributes.attributeTypeMap;
    }
}
exports.AWSCcmConfigValidationRequestAttributes = AWSCcmConfigValidationRequestAttributes;
/**
 * @ignore
 */
AWSCcmConfigValidationRequestAttributes.attributeTypeMap = {
    accountId: {
        baseName: "account_id",
        type: "string",
        required: true,
    },
    bucketName: {
        baseName: "bucket_name",
        type: "string",
        required: true,
    },
    bucketRegion: {
        baseName: "bucket_region",
        type: "string",
        required: true,
    },
    reportName: {
        baseName: "report_name",
        type: "string",
        required: true,
    },
    reportPrefix: {
        baseName: "report_prefix",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
