// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/BlueprintMetadataAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlueprintMetadataAttributes = void 0;
/**
 * The attributes of a blueprint metadata resource.
 */
class BlueprintMetadataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return BlueprintMetadataAttributes.attributeTypeMap;
    }
}
exports.BlueprintMetadataAttributes = BlueprintMetadataAttributes;
/**
 * @ignore
 */
BlueprintMetadataAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    slug: {
        baseName: "slug",
        type: "string",
        required: true,
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
    tileBackground: {
        baseName: "tile_background",
        type: "string",
    },
    tileIconActionFqn: {
        baseName: "tile_icon_action_fqn",
        type: "string",
    },
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
