// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/BlueprintAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlueprintAttributes = void 0;
/**
 * The attributes of a blueprint resource.
 */
class BlueprintAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return BlueprintAttributes.attributeTypeMap;
    }
}
exports.BlueprintAttributes = BlueprintAttributes;
/**
 * @ignore
 */
BlueprintAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    definition: {
        baseName: "definition",
        type: "AppDefinitionType",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    embeddedDatastoreBlueprints: {
        baseName: "embedded_datastore_blueprints",
        type: "{ [key: string]: any; }",
    },
    embeddedNativeActions: {
        baseName: "embedded_native_actions",
        type: "Array<{ [key: string]: any; }>",
    },
    embeddedWorkflowBlueprints: {
        baseName: "embedded_workflow_blueprints",
        type: "{ [key: string]: any; }",
    },
    integrationId: {
        baseName: "integration_id",
        type: "string",
    },
    mockedOutputs: {
        baseName: "mocked_outputs",
        type: "{ [key: string]: any; }",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    slug: {
        baseName: "slug",
        type: "string",
        required: true,
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
    tileBackground: {
        baseName: "tile_background",
        type: "string",
    },
    tileIconActionFqn: {
        baseName: "tile_icon_action_fqn",
        type: "string",
    },
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
