// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/BudgetAttributesCosts.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BudgetAttributesCosts = void 0;
/**
 * Aggregated cost data for the budget over the requested period.
 */
class BudgetAttributesCosts {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return BudgetAttributesCosts.attributeTypeMap;
    }
}
exports.BudgetAttributesCosts = BudgetAttributesCosts;
/**
 * @ignore
 */
BudgetAttributesCosts.attributeTypeMap = {
    actual: {
        baseName: "actual",
        type: "number",
        format: "double",
    },
    amount: {
        baseName: "amount",
        type: "number",
        format: "double",
    },
    forecast: {
        baseName: "forecast",
        type: "number",
        format: "double",
    },
    ootbForecast: {
        baseName: "ootb_forecast",
        type: "number",
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
