// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CaseLinkAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CaseLinkAttributes = void 0;
/**
 * Attributes describing a directional relationship between two entities (cases, incidents, or pages).
 */
class CaseLinkAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CaseLinkAttributes.attributeTypeMap;
    }
}
exports.CaseLinkAttributes = CaseLinkAttributes;
/**
 * @ignore
 */
CaseLinkAttributes.attributeTypeMap = {
    childEntityId: {
        baseName: "child_entity_id",
        type: "string",
        required: true,
    },
    childEntityType: {
        baseName: "child_entity_type",
        type: "string",
        required: true,
    },
    parentEntityId: {
        baseName: "parent_entity_id",
        type: "string",
        required: true,
    },
    parentEntityType: {
        baseName: "parent_entity_type",
        type: "string",
        required: true,
    },
    relationship: {
        baseName: "relationship",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
