// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CaseViewCreateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CaseViewCreateAttributes = void 0;
/**
 * Attributes required to create a case view.
 */
class CaseViewCreateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CaseViewCreateAttributes.attributeTypeMap;
    }
}
exports.CaseViewCreateAttributes = CaseViewCreateAttributes;
/**
 * @ignore
 */
CaseViewCreateAttributes.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    npRuleId: {
        baseName: "np_rule_id",
        type: "string",
    },
    projectId: {
        baseName: "project_id",
        type: "string",
        required: true,
    },
    query: {
        baseName: "query",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
