// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CaseAggregateGroup.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CaseAggregateGroup = void 0;
/**
 * A single group within the aggregation results, containing the group key and its associated count values.
 */
class CaseAggregateGroup {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CaseAggregateGroup.attributeTypeMap;
    }
}
exports.CaseAggregateGroup = CaseAggregateGroup;
/**
 * @ignore
 */
CaseAggregateGroup.attributeTypeMap = {
    group: {
        baseName: "group",
        type: "string",
        required: true,
    },
    value: {
        baseName: "value",
        type: "Array<number>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
