// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CaseCountGroupValue.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CaseCountGroupValue = void 0;
/**
 * A single value within a count group, representing the number of cases with that specific field value.
 */
class CaseCountGroupValue {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CaseCountGroupValue.attributeTypeMap;
    }
}
exports.CaseCountGroupValue = CaseCountGroupValue;
/**
 * @ignore
 */
CaseCountGroupValue.attributeTypeMap = {
    count: {
        baseName: "count",
        type: "number",
        required: true,
        format: "int64",
    },
    value: {
        baseName: "value",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
