// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CaseInsightType.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PROFILING_DOWNSIZING_EXPERIMENT = exports.SCM_URL = exports.APM_RECOMMENDATION = exports.RESOURCE_POLICY = exports.INSIGHT_SCORECARD_CAMPAIGN = exports.SECURITY_FINDING = exports.WIDGET = exports.WATCHDOG_STORY = exports.EVENT = exports.SENSITIVE_DATA_SCANNER_ISSUE = exports.INCIDENT = exports.CLOUD_COST_RECOMMENDATION = exports.ERROR_TRACKING = exports.EVENT_CORRELATION = exports.MONITOR = exports.SECURITY_SIGNAL = void 0;
exports.SECURITY_SIGNAL = "SECURITY_SIGNAL";
exports.MONITOR = "MONITOR";
exports.EVENT_CORRELATION = "EVENT_CORRELATION";
exports.ERROR_TRACKING = "ERROR_TRACKING";
exports.CLOUD_COST_RECOMMENDATION = "CLOUD_COST_RECOMMENDATION";
exports.INCIDENT = "INCIDENT";
exports.SENSITIVE_DATA_SCANNER_ISSUE = "SENSITIVE_DATA_SCANNER_ISSUE";
exports.EVENT = "EVENT";
exports.WATCHDOG_STORY = "WATCHDOG_STORY";
exports.WIDGET = "WIDGET";
exports.SECURITY_FINDING = "SECURITY_FINDING";
exports.INSIGHT_SCORECARD_CAMPAIGN = "INSIGHT_SCORECARD_CAMPAIGN";
exports.RESOURCE_POLICY = "RESOURCE_POLICY";
exports.APM_RECOMMENDATION = "APM_RECOMMENDATION";
exports.SCM_URL = "SCM_URL";
exports.PROFILING_DOWNSIZING_EXPERIMENT = "PROFILING_DOWNSIZING_EXPERIMENT";
