// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CaseView.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CaseView = void 0;
/**
 * A saved case view that provides a filtered, reusable list of cases matching a specific query. Views act as persistent dashboards for monitoring case subsets.
 */
class CaseView {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CaseView.attributeTypeMap;
    }
}
exports.CaseView = CaseView;
/**
 * @ignore
 */
CaseView.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "CaseViewAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    relationships: {
        baseName: "relationships",
        type: "CaseViewRelationships",
    },
    type: {
        baseName: "type",
        type: "CaseViewResourceType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
