// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CaseBulkUpdateRequestAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CaseBulkUpdateRequestAttributes = void 0;
/**
 * Attributes for the bulk update, specifying which cases to update and the action to apply.
 */
class CaseBulkUpdateRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CaseBulkUpdateRequestAttributes.attributeTypeMap;
    }
}
exports.CaseBulkUpdateRequestAttributes = CaseBulkUpdateRequestAttributes;
/**
 * @ignore
 */
CaseBulkUpdateRequestAttributes.attributeTypeMap = {
    caseIds: {
        baseName: "case_ids",
        type: "Array<string>",
        required: true,
    },
    payload: {
        baseName: "payload",
        type: "{ [key: string]: string; }",
    },
    type: {
        baseName: "type",
        type: "CaseBulkActionType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
