// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CaseAggregateGroupBy.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CaseAggregateGroupBy = void 0;
/**
 * Configuration for grouping aggregated results by one or more case fields.
 */
class CaseAggregateGroupBy {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CaseAggregateGroupBy.attributeTypeMap;
    }
}
exports.CaseAggregateGroupBy = CaseAggregateGroupBy;
/**
 * @ignore
 */
CaseAggregateGroupBy.attributeTypeMap = {
    groups: {
        baseName: "groups",
        type: "Array<string>",
        required: true,
    },
    limit: {
        baseName: "limit",
        type: "number",
        required: true,
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
