// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CaseViewAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CaseViewAttributes = void 0;
/**
 * Attributes of a case view, including the filter query and optional notification rule.
 */
class CaseViewAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CaseViewAttributes.attributeTypeMap;
    }
}
exports.CaseViewAttributes = CaseViewAttributes;
/**
 * @ignore
 */
CaseViewAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        format: "date-time",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    npRuleId: {
        baseName: "np_rule_id",
        type: "string",
    },
    query: {
        baseName: "query",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
