// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CostRecommendationDataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CostRecommendationDataAttributes = void 0;
/**
 * Attributes describing a single cost recommendation.
 */
class CostRecommendationDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CostRecommendationDataAttributes.attributeTypeMap;
    }
}
exports.CostRecommendationDataAttributes = CostRecommendationDataAttributes;
/**
 * @ignore
 */
CostRecommendationDataAttributes.attributeTypeMap = {
    ddResourceKey: {
        baseName: "dd_resource_key",
        type: "string",
    },
    potentialDailySavings: {
        baseName: "potential_daily_savings",
        type: "CostRecommendationDataAttributesPotentialDailySavings",
    },
    recommendationType: {
        baseName: "recommendation_type",
        type: "string",
    },
    resourceId: {
        baseName: "resource_id",
        type: "string",
    },
    resourceType: {
        baseName: "resource_type",
        type: "string",
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
