// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CostTagDescriptionUpsertRequestDataAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CostTagDescriptionUpsertRequestDataAttributes = void 0;
/**
 * Mutable attributes set when creating or updating a Cloud Cost Management tag key description.
 */
class CostTagDescriptionUpsertRequestDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CostTagDescriptionUpsertRequestDataAttributes.attributeTypeMap;
    }
}
exports.CostTagDescriptionUpsertRequestDataAttributes = CostTagDescriptionUpsertRequestDataAttributes;
/**
 * @ignore
 */
CostTagDescriptionUpsertRequestDataAttributes.attributeTypeMap = {
    cloud: {
        baseName: "cloud",
        type: "string",
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
