// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CostTagDescriptionUpsertRequestData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CostTagDescriptionUpsertRequestData = void 0;
/**
 * Resource envelope carrying the tag key description being upserted. The `id` is informational; the authoritative tag key is taken from the URL path.
 */
class CostTagDescriptionUpsertRequestData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CostTagDescriptionUpsertRequestData.attributeTypeMap;
    }
}
exports.CostTagDescriptionUpsertRequestData = CostTagDescriptionUpsertRequestData;
/**
 * @ignore
 */
CostTagDescriptionUpsertRequestData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "CostTagDescriptionUpsertRequestDataAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "CostTagDescriptionType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
