// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CostRecommendationDataAttributesPotentialDailySavings.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CostRecommendationDataAttributesPotentialDailySavings = void 0;
/**
 * Estimated daily savings if the recommendation is applied.
 */
class CostRecommendationDataAttributesPotentialDailySavings {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CostRecommendationDataAttributesPotentialDailySavings.attributeTypeMap;
    }
}
exports.CostRecommendationDataAttributesPotentialDailySavings = CostRecommendationDataAttributesPotentialDailySavings;
/**
 * @ignore
 */
CostRecommendationDataAttributesPotentialDailySavings.attributeTypeMap = {
    amount: {
        baseName: "amount",
        type: "number",
        format: "double",
    },
    currency: {
        baseName: "currency",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
