// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CostTagKeyMetadataAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CostTagKeyMetadataAttributes = void 0;
/**
 * Attributes of a Cloud Cost Management tag key metadata entry.
 */
class CostTagKeyMetadataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CostTagKeyMetadataAttributes.attributeTypeMap;
    }
}
exports.CostTagKeyMetadataAttributes = CostTagKeyMetadataAttributes;
/**
 * @ignore
 */
CostTagKeyMetadataAttributes.attributeTypeMap = {
    cardinalityByAccount: {
        baseName: "cardinality_by_account",
        type: "{ [key: string]: number; }",
        required: true,
    },
    costCovered: {
        baseName: "cost_covered",
        type: "number",
        required: true,
        format: "double",
    },
    date: {
        baseName: "date",
        type: "string",
    },
    metric: {
        baseName: "metric",
        type: "string",
        required: true,
    },
    rowCount: {
        baseName: "row_count",
        type: "number",
        required: true,
        format: "int64",
    },
    tagSources: {
        baseName: "tag_sources",
        type: "Array<string>",
        required: true,
    },
    topValuesByAccount: {
        baseName: "top_values_by_account",
        type: "{ [key: string]: Array<string>; }",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
