// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CostTagKeySourceAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CostTagKeySourceAttributes = void 0;
/**
 * Attributes of a Cloud Cost Management tag source.
 */
class CostTagKeySourceAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CostTagKeySourceAttributes.attributeTypeMap;
    }
}
exports.CostTagKeySourceAttributes = CostTagKeySourceAttributes;
/**
 * @ignore
 */
CostTagKeySourceAttributes.attributeTypeMap = {
    tagKey: {
        baseName: "tag_key",
        type: "string",
        required: true,
    },
    tagSources: {
        baseName: "tag_sources",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
