// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CreateServiceNowTicketRequestData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateServiceNowTicketRequestData = void 0;
/**
 * Data of the ServiceNow ticket to create.
 */
class CreateServiceNowTicketRequestData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CreateServiceNowTicketRequestData.attributeTypeMap;
    }
}
exports.CreateServiceNowTicketRequestData = CreateServiceNowTicketRequestData;
/**
 * @ignore
 */
CreateServiceNowTicketRequestData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "CreateServiceNowTicketRequestDataAttributes",
    },
    relationships: {
        baseName: "relationships",
        type: "CreateServiceNowTicketRequestDataRelationships",
        required: true,
    },
    type: {
        baseName: "type",
        type: "ServiceNowTicketsDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
