// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CreateFormDataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateFormDataAttributes = void 0;
/**
 * The attributes for creating a form.
 */
class CreateFormDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CreateFormDataAttributes.attributeTypeMap;
    }
}
exports.CreateFormDataAttributes = CreateFormDataAttributes;
/**
 * @ignore
 */
CreateFormDataAttributes.attributeTypeMap = {
    anonymous: {
        baseName: "anonymous",
        type: "boolean",
    },
    dataDefinition: {
        baseName: "data_definition",
        type: "FormDataDefinition",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
    },
    idpSurvey: {
        baseName: "idp_survey",
        type: "boolean",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    singleResponse: {
        baseName: "single_response",
        type: "boolean",
    },
    uiDefinition: {
        baseName: "ui_definition",
        type: "FormUiDefinition",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
