// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CreateServiceNowTicketRequestDataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateServiceNowTicketRequestDataAttributes = void 0;
/**
 * Attributes of the ServiceNow ticket to create.
 */
class CreateServiceNowTicketRequestDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CreateServiceNowTicketRequestDataAttributes.attributeTypeMap;
    }
}
exports.CreateServiceNowTicketRequestDataAttributes = CreateServiceNowTicketRequestDataAttributes;
/**
 * @ignore
 */
CreateServiceNowTicketRequestDataAttributes.attributeTypeMap = {
    assigneeId: {
        baseName: "assignee_id",
        type: "string",
    },
    description: {
        baseName: "description",
        type: "string",
    },
    priority: {
        baseName: "priority",
        type: "CasePriority",
    },
    title: {
        baseName: "title",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
