// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CsmAgentlessHostFacetAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CsmAgentlessHostFacetAttributes = void 0;
/**
 * Attributes of an agentless host facet.
 */
class CsmAgentlessHostFacetAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CsmAgentlessHostFacetAttributes.attributeTypeMap;
    }
}
exports.CsmAgentlessHostFacetAttributes = CsmAgentlessHostFacetAttributes;
/**
 * @ignore
 */
CsmAgentlessHostFacetAttributes.attributeTypeMap = {
    bounded: {
        baseName: "bounded",
        type: "boolean",
        required: true,
    },
    bundled: {
        baseName: "bundled",
        type: "boolean",
        required: true,
    },
    bundledAndUsed: {
        baseName: "bundledAndUsed",
        type: "boolean",
        required: true,
    },
    defaultValues: {
        baseName: "defaultValues",
        type: "Array<string>",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    editable: {
        baseName: "editable",
        type: "boolean",
        required: true,
    },
    facetType: {
        baseName: "facetType",
        type: "string",
        required: true,
    },
    groups: {
        baseName: "groups",
        type: "Array<string>",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    path: {
        baseName: "path",
        type: "string",
        required: true,
    },
    source: {
        baseName: "source",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "string",
        required: true,
    },
    values: {
        baseName: "values",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
