// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CsmAgentlessHostAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CsmAgentlessHostAttributes = void 0;
/**
 * Attributes of an agentless host.
 */
class CsmAgentlessHostAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CsmAgentlessHostAttributes.attributeTypeMap;
    }
}
exports.CsmAgentlessHostAttributes = CsmAgentlessHostAttributes;
/**
 * @ignore
 */
CsmAgentlessHostAttributes.attributeTypeMap = {
    accountId: {
        baseName: "account_id",
        type: "string",
        required: true,
    },
    cloudProvider: {
        baseName: "cloud_provider",
        type: "CsmCloudProvider",
        required: true,
    },
    hasPostureManagement: {
        baseName: "has_posture_management",
        type: "boolean",
        required: true,
    },
    hasVulnerabilityScanning: {
        baseName: "has_vulnerability_scanning",
        type: "boolean",
        required: true,
    },
    resourceType: {
        baseName: "resource_type",
        type: "CsmAgentlessHostResourceType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
