// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CsmHostFacetInfoData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CsmHostFacetInfoData = void 0;
/**
 * The data wrapper for a facet info response.
 */
class CsmHostFacetInfoData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CsmHostFacetInfoData.attributeTypeMap;
    }
}
exports.CsmHostFacetInfoData = CsmHostFacetInfoData;
/**
 * @ignore
 */
CsmHostFacetInfoData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "CsmHostFacetInfoAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    meta: {
        baseName: "meta",
        type: "CsmHostFacetInfoMeta",
        required: true,
    },
    type: {
        baseName: "type",
        type: "CsmFacetInfoType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
