// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CsmHostFacetInfoItem.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CsmHostFacetInfoItem = void 0;
/**
 * A single value and its occurrence count for a facet.
 */
class CsmHostFacetInfoItem {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CsmHostFacetInfoItem.attributeTypeMap;
    }
}
exports.CsmHostFacetInfoItem = CsmHostFacetInfoItem;
/**
 * @ignore
 */
CsmHostFacetInfoItem.attributeTypeMap = {
    count: {
        baseName: "count",
        type: "number",
        required: true,
        format: "int64",
    },
    value: {
        baseName: "value",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
