// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CsmUnifiedHostsMeta.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CsmUnifiedHostsMeta = void 0;
/**
 * Pagination metadata for a unified hosts list response.
 */
class CsmUnifiedHostsMeta {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CsmUnifiedHostsMeta.attributeTypeMap;
    }
}
exports.CsmUnifiedHostsMeta = CsmUnifiedHostsMeta;
/**
 * @ignore
 */
CsmUnifiedHostsMeta.attributeTypeMap = {
    pageIndex: {
        baseName: "page_index",
        type: "number",
        required: true,
        format: "int64",
    },
    pageSize: {
        baseName: "page_size",
        type: "number",
        required: true,
        format: "int64",
    },
    totalFiltered: {
        baseName: "total_filtered",
        type: "number",
        required: true,
        format: "int64",
    },
    totalPages: {
        baseName: "total_pages",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
