// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CsmUnifiedHostAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CsmUnifiedHostAttributes = void 0;
/**
 * Attributes of a unified host, combining data from agent and agentless sources.
 */
class CsmUnifiedHostAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CsmUnifiedHostAttributes.attributeTypeMap;
    }
}
exports.CsmUnifiedHostAttributes = CsmUnifiedHostAttributes;
/**
 * @ignore
 */
CsmUnifiedHostAttributes.attributeTypeMap = {
    accountId: {
        baseName: "account_id",
        type: "string",
    },
    agentCsmVmContainersEnabled: {
        baseName: "agent_csm_vm_containers_enabled",
        type: "boolean",
    },
    agentCsmVmHostsEnabled: {
        baseName: "agent_csm_vm_hosts_enabled",
        type: "boolean",
    },
    agentCwsEnabled: {
        baseName: "agent_cws_enabled",
        type: "boolean",
    },
    agentPostureManagement: {
        baseName: "agent_posture_management",
        type: "boolean",
    },
    agentVersion: {
        baseName: "agent_version",
        type: "string",
    },
    agentlessPostureManagement: {
        baseName: "agentless_posture_management",
        type: "boolean",
    },
    agentlessVulnerabilityScanning: {
        baseName: "agentless_vulnerability_scanning",
        type: "boolean",
    },
    cloudProvider: {
        baseName: "cloud_provider",
        type: "CsmCloudProvider",
    },
    clusterName: {
        baseName: "cluster_name",
        type: "string",
    },
    datadogAgentKey: {
        baseName: "datadog_agent_key",
        type: "string",
    },
    env: {
        baseName: "env",
        type: "Array<string>",
    },
    hostId: {
        baseName: "host_id",
        type: "number",
        format: "int64",
    },
    installMethodTool: {
        baseName: "install_method_tool",
        type: "string",
    },
    os: {
        baseName: "os",
        type: "string",
    },
    resourceType: {
        baseName: "resource_type",
        type: "CsmAgentlessHostResourceType",
    },
    source: {
        baseName: "source",
        type: "CsmUnifiedHostSource",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
