// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CustomForecastEntryTagFilter.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomForecastEntryTagFilter = void 0;
/**
 * A tag filter that scopes a custom forecast entry to specific resource tags.
 */
class CustomForecastEntryTagFilter {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CustomForecastEntryTagFilter.attributeTypeMap;
    }
}
exports.CustomForecastEntryTagFilter = CustomForecastEntryTagFilter;
/**
 * @ignore
 */
CustomForecastEntryTagFilter.attributeTypeMap = {
    tagKey: {
        baseName: "tag_key",
        type: "string",
        required: true,
    },
    tagValue: {
        baseName: "tag_value",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
