// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CustomForecastResponseDataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomForecastResponseDataAttributes = void 0;
/**
 * Attributes of a custom forecast.
 */
class CustomForecastResponseDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CustomForecastResponseDataAttributes.attributeTypeMap;
    }
}
exports.CustomForecastResponseDataAttributes = CustomForecastResponseDataAttributes;
/**
 * @ignore
 */
CustomForecastResponseDataAttributes.attributeTypeMap = {
    budgetUid: {
        baseName: "budget_uid",
        type: "string",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "number",
        required: true,
        format: "int64",
    },
    createdBy: {
        baseName: "created_by",
        type: "string",
        required: true,
    },
    entries: {
        baseName: "entries",
        type: "Array<CustomForecastEntry>",
        required: true,
    },
    updatedAt: {
        baseName: "updated_at",
        type: "number",
        required: true,
        format: "int64",
    },
    updatedBy: {
        baseName: "updated_by",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
