// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CustomAttributeConfigUpdateAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomAttributeConfigUpdateAttributes = void 0;
/**
 * Attributes that can be updated on a custom attribute configuration. All fields are optional; only provided fields are changed.
 */
class CustomAttributeConfigUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CustomAttributeConfigUpdateAttributes.attributeTypeMap;
    }
}
exports.CustomAttributeConfigUpdateAttributes = CustomAttributeConfigUpdateAttributes;
/**
 * @ignore
 */
CustomAttributeConfigUpdateAttributes.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    displayName: {
        baseName: "display_name",
        type: "string",
    },
    mapFrom: {
        baseName: "map_from",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "CustomAttributeType",
    },
    typeData: {
        baseName: "type_data",
        type: "CustomAttributeTypeData",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
