// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CustomAttributeConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomAttributeConfig = void 0;
/**
 * A custom attribute configuration that defines an organization-specific metadata field on cases. Custom attributes are scoped to a case type and can hold text, URLs, numbers, or predefined select options.
 */
class CustomAttributeConfig {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CustomAttributeConfig.attributeTypeMap;
    }
}
exports.CustomAttributeConfig = CustomAttributeConfig;
/**
 * @ignore
 */
CustomAttributeConfig.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "CustomAttributeConfigResourceAttributes",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "CustomAttributeConfigResourceType",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
