// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CustomerOrgDisableRequestAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerOrgDisableRequestAttributes = void 0;
/**
 * Optional attributes for a customer org disable request. When supplied, `org_uuid`
 * must match the authenticated organization or the request is rejected.
 */
class CustomerOrgDisableRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CustomerOrgDisableRequestAttributes.attributeTypeMap;
    }
}
exports.CustomerOrgDisableRequestAttributes = CustomerOrgDisableRequestAttributes;
/**
 * @ignore
 */
CustomerOrgDisableRequestAttributes.attributeTypeMap = {
    orgUuid: {
        baseName: "org_uuid",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
