// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CustomerOrgDisableResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerOrgDisableResponseData = void 0;
/**
 * Data object returned after disabling the customer organization.
 */
class CustomerOrgDisableResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CustomerOrgDisableResponseData.attributeTypeMap;
    }
}
exports.CustomerOrgDisableResponseData = CustomerOrgDisableResponseData;
/**
 * @ignore
 */
CustomerOrgDisableResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "CustomerOrgDisableResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "CustomerOrgDisableResponseType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
