// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CycloneDXBom.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CycloneDXBom = void 0;
/**
 * A CycloneDX 1.5 Bill of Materials (BOM) document containing vulnerability data.
 */
class CycloneDXBom {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CycloneDXBom.attributeTypeMap;
    }
}
exports.CycloneDXBom = CycloneDXBom;
/**
 * @ignore
 */
CycloneDXBom.attributeTypeMap = {
    bomFormat: {
        baseName: "bomFormat",
        type: "string",
        required: true,
    },
    components: {
        baseName: "components",
        type: "Array<CycloneDXComponent>",
        required: true,
    },
    metadata: {
        baseName: "metadata",
        type: "CycloneDXMetadata",
        required: true,
    },
    specVersion: {
        baseName: "specVersion",
        type: "string",
        required: true,
    },
    version: {
        baseName: "version",
        type: "number",
        format: "int64",
    },
    vulnerabilities: {
        baseName: "vulnerabilities",
        type: "Array<CycloneDXVulnerability>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
