// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CycloneDXMetadataComponent.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CycloneDXMetadataComponent = void 0;
/**
 * The asset that was scanned (for example, a host or container image).
 */
class CycloneDXMetadataComponent {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CycloneDXMetadataComponent.attributeTypeMap;
    }
}
exports.CycloneDXMetadataComponent = CycloneDXMetadataComponent;
/**
 * @ignore
 */
CycloneDXMetadataComponent.attributeTypeMap = {
    bomRef: {
        baseName: "bom-ref",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
