// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/CycloneDXComponent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CycloneDXComponent = void 0;
/**
 * A software component identified during scanning.
 */
class CycloneDXComponent {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return CycloneDXComponent.attributeTypeMap;
    }
}
exports.CycloneDXComponent = CycloneDXComponent;
/**
 * @ignore
 */
CycloneDXComponent.attributeTypeMap = {
    bomRef: {
        baseName: "bom-ref",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    purl: {
        baseName: "purl",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "CycloneDXComponentType",
        required: true,
    },
    version: {
        baseName: "version",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
