// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DashboardUsageUser.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DashboardUsageUser = void 0;
/**
 * A user referenced from a dashboard usage record (author or viewer).
 */
class DashboardUsageUser {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DashboardUsageUser.attributeTypeMap;
    }
}
exports.DashboardUsageUser = DashboardUsageUser;
/**
 * @ignore
 */
DashboardUsageUser.attributeTypeMap = {
    handle: {
        baseName: "handle",
        type: "string",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    isDisabled: {
        baseName: "is_disabled",
        type: "boolean",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
