// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DashboardUsageAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DashboardUsageAttributes = void 0;
/**
 * Usage statistics for a dashboard. The `viewer` field and all view-count fields (`total_views`, `viewed_at`, `total_views_by_type`) are populated only when Real User Monitoring (RUM) is active for the org.
 */
class DashboardUsageAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DashboardUsageAttributes.attributeTypeMap;
    }
}
exports.DashboardUsageAttributes = DashboardUsageAttributes;
/**
 * @ignore
 */
DashboardUsageAttributes.attributeTypeMap = {
    author: {
        baseName: "author",
        type: "DashboardUsageUser",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        format: "date-time",
    },
    dashboardQualityScore: {
        baseName: "dashboard_quality_score",
        type: "number",
        format: "double",
    },
    editedAt: {
        baseName: "edited_at",
        type: "Date",
        format: "date-time",
    },
    orgId: {
        baseName: "org_id",
        type: "number",
        required: true,
        format: "int64",
    },
    teams: {
        baseName: "teams",
        type: "Array<string>",
    },
    title: {
        baseName: "title",
        type: "string",
    },
    totalViews: {
        baseName: "total_views",
        type: "number",
        format: "int64",
    },
    totalViewsByType: {
        baseName: "total_views_by_type",
        type: "{ [key: string]: number; }",
    },
    viewedAt: {
        baseName: "viewed_at",
        type: "Date",
        format: "date-time",
    },
    viewer: {
        baseName: "viewer",
        type: "DashboardUsageUser",
    },
    widgetCount: {
        baseName: "widget_count",
        type: "number",
        format: "int64",
    },
    widgetCountByType: {
        baseName: "widget_count_by_type",
        type: "{ [key: string]: number; }",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
