// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DefaultRulesetsPerLanguageDataAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DefaultRulesetsPerLanguageDataAttributes = void 0;
/**
 * The attributes of the default rulesets per language response, containing the list of default ruleset names.
 */
class DefaultRulesetsPerLanguageDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DefaultRulesetsPerLanguageDataAttributes.attributeTypeMap;
    }
}
exports.DefaultRulesetsPerLanguageDataAttributes = DefaultRulesetsPerLanguageDataAttributes;
/**
 * @ignore
 */
DefaultRulesetsPerLanguageDataAttributes.attributeTypeMap = {
    rulesets: {
        baseName: "rulesets",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
