// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DefaultRulesetsPerLanguageData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DefaultRulesetsPerLanguageData = void 0;
/**
 * The primary data object in the default rulesets per language response.
 */
class DefaultRulesetsPerLanguageData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DefaultRulesetsPerLanguageData.attributeTypeMap;
    }
}
exports.DefaultRulesetsPerLanguageData = DefaultRulesetsPerLanguageData;
/**
 * @ignore
 */
DefaultRulesetsPerLanguageData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "DefaultRulesetsPerLanguageDataAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "DefaultRulesetsPerLanguageDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
