// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DegradationDataAttributesUpdatesItems.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DegradationDataAttributesUpdatesItems = void 0;
/**
 * A status update recorded during a degradation.
 */
class DegradationDataAttributesUpdatesItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DegradationDataAttributesUpdatesItems.attributeTypeMap;
    }
}
exports.DegradationDataAttributesUpdatesItems = DegradationDataAttributesUpdatesItems;
/**
 * @ignore
 */
DegradationDataAttributesUpdatesItems.attributeTypeMap = {
    componentsAffected: {
        baseName: "components_affected",
        type: "Array<DegradationDataAttributesUpdatesItemsComponentsAffectedItems>",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        format: "date-time",
    },
    deletedAt: {
        baseName: "deleted_at",
        type: "string",
    },
    deletedByUserUuid: {
        baseName: "deleted_by_user_uuid",
        type: "string",
    },
    description: {
        baseName: "description",
        type: "string",
    },
    id: {
        baseName: "id",
        type: "string",
        format: "uuid",
    },
    lastModifiedByUserUuid: {
        baseName: "last_modified_by_user_uuid",
        type: "string",
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        format: "date-time",
    },
    startedAt: {
        baseName: "started_at",
        type: "Date",
        format: "date-time",
    },
    status: {
        baseName: "status",
        type: "CreateDegradationRequestDataAttributesStatus",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
