// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DegradationUpdateDataRelationships.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DegradationUpdateDataRelationships = void 0;
/**
 * Relationships of a degradation update resource.
 */
class DegradationUpdateDataRelationships {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DegradationUpdateDataRelationships.attributeTypeMap;
    }
}
exports.DegradationUpdateDataRelationships = DegradationUpdateDataRelationships;
/**
 * @ignore
 */
DegradationUpdateDataRelationships.attributeTypeMap = {
    createdByUser: {
        baseName: "created_by_user",
        type: "DegradationUpdateDataRelationshipsUser",
    },
    degradation: {
        baseName: "degradation",
        type: "DegradationUpdateDataRelationshipsDegradation",
    },
    deletedByUser: {
        baseName: "deleted_by_user",
        type: "DegradationUpdateDataRelationshipsUser",
    },
    lastModifiedByUser: {
        baseName: "last_modified_by_user",
        type: "DegradationUpdateDataRelationshipsUser",
    },
    statusPage: {
        baseName: "status_page",
        type: "DegradationUpdateDataRelationshipsStatusPage",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
