// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DegradationUpdateDataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DegradationUpdateDataAttributes = void 0;
/**
 * Attributes of a degradation update resource.
 */
class DegradationUpdateDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DegradationUpdateDataAttributes.attributeTypeMap;
    }
}
exports.DegradationUpdateDataAttributes = DegradationUpdateDataAttributes;
/**
 * @ignore
 */
DegradationUpdateDataAttributes.attributeTypeMap = {
    componentsAffected: {
        baseName: "components_affected",
        type: "Array<DegradationUpdateDataAttributesComponentsAffectedItems>",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        format: "date-time",
    },
    deletedAt: {
        baseName: "deleted_at",
        type: "Date",
        format: "date-time",
    },
    description: {
        baseName: "description",
        type: "string",
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        format: "date-time",
    },
    startedAt: {
        baseName: "started_at",
        type: "Date",
        format: "date-time",
    },
    status: {
        baseName: "status",
        type: "CreateDegradationRequestDataAttributesStatus",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
