// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DeploymentGatesEvaluationConfiguration.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentGatesEvaluationConfiguration = void 0;
/**
 * Inline rule definitions for a deployment gate evaluation. When provided, rules are evaluated
 * directly from this configuration instead of using the preconfigured gate rules.
 * At least one rule is required.
 */
class DeploymentGatesEvaluationConfiguration {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DeploymentGatesEvaluationConfiguration.attributeTypeMap;
    }
}
exports.DeploymentGatesEvaluationConfiguration = DeploymentGatesEvaluationConfiguration;
/**
 * @ignore
 */
DeploymentGatesEvaluationConfiguration.attributeTypeMap = {
    dryRun: {
        baseName: "dry_run",
        type: "boolean",
    },
    rules: {
        baseName: "rules",
        type: "Array<DeploymentGatesEvaluationRule>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
