// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DeploymentGatesEvaluationRequestAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentGatesEvaluationRequestAttributes = void 0;
/**
 * Attributes for a deployment gate evaluation request.
 * When `configuration` is provided, rules are evaluated inline from that configuration.
 * When omitted, rules are resolved from the preconfigured gate for the given service and environment.
 */
class DeploymentGatesEvaluationRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DeploymentGatesEvaluationRequestAttributes.attributeTypeMap;
    }
}
exports.DeploymentGatesEvaluationRequestAttributes = DeploymentGatesEvaluationRequestAttributes;
/**
 * @ignore
 */
DeploymentGatesEvaluationRequestAttributes.attributeTypeMap = {
    configuration: {
        baseName: "configuration",
        type: "DeploymentGatesEvaluationConfiguration",
    },
    env: {
        baseName: "env",
        type: "string",
        required: true,
    },
    identifier: {
        baseName: "identifier",
        type: "string",
    },
    primaryTag: {
        baseName: "primary_tag",
        type: "string",
    },
    service: {
        baseName: "service",
        type: "string",
        required: true,
    },
    version: {
        baseName: "version",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
