// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DeploymentGatesFDDRule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentGatesFDDRule = void 0;
/**
 * A faulty deployment detection rule to evaluate as part of a deployment gate evaluation.
 */
class DeploymentGatesFDDRule {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DeploymentGatesFDDRule.attributeTypeMap;
    }
}
exports.DeploymentGatesFDDRule = DeploymentGatesFDDRule;
/**
 * @ignore
 */
DeploymentGatesFDDRule.attributeTypeMap = {
    dryRun: {
        baseName: "dry_run",
        type: "boolean",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    options: {
        baseName: "options",
        type: "DeploymentGatesFDDRuleOptions",
    },
    type: {
        baseName: "type",
        type: "DeploymentGatesFDDRuleType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
