// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DeploymentGatesMonitorRule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentGatesMonitorRule = void 0;
/**
 * A monitor rule to evaluate as part of a deployment gate evaluation.
 */
class DeploymentGatesMonitorRule {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DeploymentGatesMonitorRule.attributeTypeMap;
    }
}
exports.DeploymentGatesMonitorRule = DeploymentGatesMonitorRule;
/**
 * @ignore
 */
DeploymentGatesMonitorRule.attributeTypeMap = {
    dryRun: {
        baseName: "dry_run",
        type: "boolean",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    options: {
        baseName: "options",
        type: "DeploymentGatesMonitorRuleOptions",
    },
    type: {
        baseName: "type",
        type: "DeploymentGatesMonitorRuleType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
