// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DueDateRuleAttributesResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DueDateRuleAttributesResponse = void 0;
/**
 * Attributes of a due date rule returned by the API.
 */
class DueDateRuleAttributesResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DueDateRuleAttributesResponse.attributeTypeMap;
    }
}
exports.DueDateRuleAttributesResponse = DueDateRuleAttributesResponse;
/**
 * @ignore
 */
DueDateRuleAttributesResponse.attributeTypeMap = {
    action: {
        baseName: "action",
        type: "DueDateRuleAction",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "number",
        required: true,
        format: "int64",
    },
    createdBy: {
        baseName: "created_by",
        type: "AutomationRuleCreatedBy",
        required: true,
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
        required: true,
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "number",
        required: true,
        format: "int64",
    },
    modifiedBy: {
        baseName: "modified_by",
        type: "AutomationRuleModifiedBy",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    rule: {
        baseName: "rule",
        type: "AutomationRuleScope",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
