// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/DueDateRuleAttributesCreate.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DueDateRuleAttributesCreate = void 0;
/**
 * Attributes for creating or updating a due date rule.
 */
class DueDateRuleAttributesCreate {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return DueDateRuleAttributesCreate.attributeTypeMap;
    }
}
exports.DueDateRuleAttributesCreate = DueDateRuleAttributesCreate;
/**
 * @ignore
 */
DueDateRuleAttributesCreate.attributeTypeMap = {
    action: {
        baseName: "action",
        type: "DueDateRuleAction",
        required: true,
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    rule: {
        baseName: "rule",
        type: "AutomationRuleScope",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
