// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ELFSourcemapAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ELFSourcemapAttributes = void 0;
/**
 * Attributes of an ELF symbol file.
 */
class ELFSourcemapAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ELFSourcemapAttributes.attributeTypeMap;
    }
}
exports.ELFSourcemapAttributes = ELFSourcemapAttributes;
/**
 * @ignore
 */
ELFSourcemapAttributes.attributeTypeMap = {
    arch: {
        baseName: "arch",
        type: "string",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    fileHash: {
        baseName: "file_hash",
        type: "string",
    },
    fileName: {
        baseName: "file_name",
        type: "string",
    },
    gnuBuildId: {
        baseName: "gnu_build_id",
        type: "string",
    },
    goBuildId: {
        baseName: "go_build_id",
        type: "string",
    },
    mapkind: {
        baseName: "mapkind",
        type: "string",
        required: true,
    },
    origin: {
        baseName: "origin",
        type: "string",
    },
    originVersion: {
        baseName: "origin_version",
        type: "string",
    },
    size: {
        baseName: "size",
        type: "number",
        required: true,
        format: "int64",
    },
    symbolSource: {
        baseName: "symbol_source",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
