// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/EntityIntegrationConfigAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntityIntegrationConfigAttributes = void 0;
/**
 * The organization ID, integration identifier, and integration-specific configuration payload for an entity integration configuration.
 */
class EntityIntegrationConfigAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return EntityIntegrationConfigAttributes.attributeTypeMap;
    }
}
exports.EntityIntegrationConfigAttributes = EntityIntegrationConfigAttributes;
/**
 * @ignore
 */
EntityIntegrationConfigAttributes.attributeTypeMap = {
    config: {
        baseName: "config",
        type: "{ [key: string]: any; }",
        required: true,
    },
    integrationId: {
        baseName: "integration_id",
        type: "string",
        required: true,
    },
    orgId: {
        baseName: "org_id",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
