// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/EntityContextRevision.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntityContextRevision = void 0;
/**
 * A single historical revision of an entity, including the time range during which the revision was observed.
 */
class EntityContextRevision {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return EntityContextRevision.attributeTypeMap;
    }
}
exports.EntityContextRevision = EntityContextRevision;
/**
 * @ignore
 */
EntityContextRevision.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "{ [key: string]: any; }",
        required: true,
    },
    firstSeenAt: {
        baseName: "first_seen_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    lastSeenAt: {
        baseName: "last_seen_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
