// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/EntityIntegrationConfigRequestAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntityIntegrationConfigRequestAttributes = void 0;
/**
 * Attributes used to create or update an entity integration configuration.
 */
class EntityIntegrationConfigRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return EntityIntegrationConfigRequestAttributes.attributeTypeMap;
    }
}
exports.EntityIntegrationConfigRequestAttributes = EntityIntegrationConfigRequestAttributes;
/**
 * @ignore
 */
EntityIntegrationConfigRequestAttributes.attributeTypeMap = {
    config: {
        baseName: "config",
        type: "{ [key: string]: any; }",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
