// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FindingServiceNowTicketResult.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FindingServiceNowTicketResult = void 0;
/**
 * Result of the ServiceNow ticket creation or attachment.
 */
class FindingServiceNowTicketResult {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FindingServiceNowTicketResult.attributeTypeMap;
    }
}
exports.FindingServiceNowTicketResult = FindingServiceNowTicketResult;
/**
 * @ignore
 */
FindingServiceNowTicketResult.attributeTypeMap = {
    instanceName: {
        baseName: "instance_name",
        type: "string",
    },
    sysId: {
        baseName: "sys_id",
        type: "string",
    },
    sysTargetLink: {
        baseName: "sys_target_link",
        type: "string",
    },
    sysTargetSysId: {
        baseName: "sys_target_sys_id",
        type: "string",
    },
    tableName: {
        baseName: "table_name",
        type: "string",
    },
    url: {
        baseName: "url",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
