// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FormDatastoreConfigAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormDatastoreConfigAttributes = void 0;
/**
 * The datastore configuration for a form.
 */
class FormDatastoreConfigAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FormDatastoreConfigAttributes.attributeTypeMap;
    }
}
exports.FormDatastoreConfigAttributes = FormDatastoreConfigAttributes;
/**
 * @ignore
 */
FormDatastoreConfigAttributes.attributeTypeMap = {
    datastoreId: {
        baseName: "datastore_id",
        type: "string",
        required: true,
        format: "uuid",
    },
    primaryColumnName: {
        baseName: "primary_column_name",
        type: "string",
        required: true,
    },
    primaryKeyGenerationStrategy: {
        baseName: "primary_key_generation_strategy",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
