// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FormVersionAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormVersionAttributes = void 0;
/**
 * The attributes of a form version.
 */
class FormVersionAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FormVersionAttributes.attributeTypeMap;
    }
}
exports.FormVersionAttributes = FormVersionAttributes;
/**
 * @ignore
 */
FormVersionAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    dataDefinition: {
        baseName: "data_definition",
        type: "FormDataDefinition",
        required: true,
    },
    definitionSignature: {
        baseName: "definition_signature",
        type: "string",
        required: true,
    },
    etag: {
        baseName: "etag",
        type: "string",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    state: {
        baseName: "state",
        type: "FormVersionState",
        required: true,
    },
    uiDefinition: {
        baseName: "ui_definition",
        type: "FormUiDefinition",
        required: true,
    },
    userId: {
        baseName: "user_id",
        type: "number",
        required: true,
        format: "int64",
    },
    userUuid: {
        baseName: "user_uuid",
        type: "string",
        required: true,
        format: "uuid",
    },
    version: {
        baseName: "version",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
