// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FormDataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormDataAttributes = void 0;
/**
 * The attributes of a form.
 */
class FormDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FormDataAttributes.attributeTypeMap;
    }
}
exports.FormDataAttributes = FormDataAttributes;
/**
 * @ignore
 */
FormDataAttributes.attributeTypeMap = {
    active: {
        baseName: "active",
        type: "boolean",
        required: true,
    },
    anonymous: {
        baseName: "anonymous",
        type: "boolean",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    datastoreConfig: {
        baseName: "datastore_config",
        type: "FormDatastoreConfigAttributes",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    endDate: {
        baseName: "end_date",
        type: "Date",
        format: "date-time",
    },
    hasSubmitted: {
        baseName: "has_submitted",
        type: "boolean",
    },
    idpSurvey: {
        baseName: "idp_survey",
        type: "boolean",
        required: true,
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    orgId: {
        baseName: "org_id",
        type: "number",
        required: true,
        format: "int64",
    },
    publication: {
        baseName: "publication",
        type: "FormPublicationAttributes",
    },
    selfService: {
        baseName: "self_service",
        type: "boolean",
        required: true,
    },
    singleResponse: {
        baseName: "single_response",
        type: "boolean",
        required: true,
    },
    userId: {
        baseName: "user_id",
        type: "number",
        required: true,
        format: "int64",
    },
    userUuid: {
        baseName: "user_uuid",
        type: "string",
        required: true,
        format: "uuid",
    },
    version: {
        baseName: "version",
        type: "FormVersionAttributes",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
