// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FormPublicationAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormPublicationAttributes = void 0;
/**
 * The attributes of a form publication.
 */
class FormPublicationAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FormPublicationAttributes.attributeTypeMap;
    }
}
exports.FormPublicationAttributes = FormPublicationAttributes;
/**
 * @ignore
 */
FormPublicationAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    formId: {
        baseName: "form_id",
        type: "string",
        required: true,
        format: "uuid",
    },
    formVersion: {
        baseName: "form_version",
        type: "number",
        required: true,
        format: "int64",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    orgId: {
        baseName: "org_id",
        type: "number",
        required: true,
        format: "int64",
    },
    publishSeq: {
        baseName: "publish_seq",
        type: "number",
        required: true,
        format: "int64",
    },
    userId: {
        baseName: "user_id",
        type: "number",
        required: true,
        format: "int64",
    },
    userUuid: {
        baseName: "user_uuid",
        type: "string",
        required: true,
        format: "uuid",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
