// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/FormDataDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormDataDefinition = void 0;
/**
 * A JSON Schema definition that describes the form's data fields.
 */
class FormDataDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return FormDataDefinition.attributeTypeMap;
    }
}
exports.FormDataDefinition = FormDataDefinition;
/**
 * @ignore
 */
FormDataDefinition.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    properties: {
        baseName: "properties",
        type: "{ [key: string]: any; }",
    },
    required: {
        baseName: "required",
        type: "Array<string>",
    },
    title: {
        baseName: "title",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "FormDataDefinitionType",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
