// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GeneratedCostTagDescription.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GeneratedCostTagDescription = void 0;
/**
 * AI-generated Cloud Cost Management tag key description returned by the generate endpoint. The result is returned to the client but is not persisted by this endpoint.
 */
class GeneratedCostTagDescription {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GeneratedCostTagDescription.attributeTypeMap;
    }
}
exports.GeneratedCostTagDescription = GeneratedCostTagDescription;
/**
 * @ignore
 */
GeneratedCostTagDescription.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "GeneratedCostTagDescriptionAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "GeneratedCostTagDescriptionType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
