// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GeneratedCostTagDescriptionAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GeneratedCostTagDescriptionAttributes = void 0;
/**
 * Attributes of an AI-generated Cloud Cost Management tag key description.
 */
class GeneratedCostTagDescriptionAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GeneratedCostTagDescriptionAttributes.attributeTypeMap;
    }
}
exports.GeneratedCostTagDescriptionAttributes = GeneratedCostTagDescriptionAttributes;
/**
 * @ignore
 */
GeneratedCostTagDescriptionAttributes.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
