// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GetAstRequestDataAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetAstRequestDataAttributes = void 0;
/**
 * The attributes of the get-AST request, containing the source code to parse.
 */
class GetAstRequestDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GetAstRequestDataAttributes.attributeTypeMap;
    }
}
exports.GetAstRequestDataAttributes = GetAstRequestDataAttributes;
/**
 * @ignore
 */
GetAstRequestDataAttributes.attributeTypeMap = {
    code: {
        baseName: "code",
        type: "string",
        required: true,
    },
    fileEncoding: {
        baseName: "file_encoding",
        type: "string",
        required: true,
    },
    language: {
        baseName: "language",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
