// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GetAstResponseDataAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetAstResponseDataAttributes = void 0;
/**
 * The attributes of the get-AST response, containing the parsed abstract syntax tree.
 */
class GetAstResponseDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GetAstResponseDataAttributes.attributeTypeMap;
    }
}
exports.GetAstResponseDataAttributes = GetAstResponseDataAttributes;
/**
 * @ignore
 */
GetAstResponseDataAttributes.attributeTypeMap = {
    ast: {
        baseName: "ast",
        type: "{ [key: string]: any; }",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
