// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GetDataObservabilityMonitorRunStatusResponseAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetDataObservabilityMonitorRunStatusResponseAttributes = void 0;
/**
 * The attributes of a data observability monitor run status response.
 */
class GetDataObservabilityMonitorRunStatusResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GetDataObservabilityMonitorRunStatusResponseAttributes.attributeTypeMap;
    }
}
exports.GetDataObservabilityMonitorRunStatusResponseAttributes = GetDataObservabilityMonitorRunStatusResponseAttributes;
/**
 * @ignore
 */
GetDataObservabilityMonitorRunStatusResponseAttributes.attributeTypeMap = {
    errorMessage: {
        baseName: "error_message",
        type: "string",
    },
    status: {
        baseName: "status",
        type: "DataObservabilityMonitorRunStatus",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
