// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GetDataObservabilityMonitorRunStatusResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetDataObservabilityMonitorRunStatusResponseData = void 0;
/**
 * The data object for a data observability monitor run status response.
 */
class GetDataObservabilityMonitorRunStatusResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GetDataObservabilityMonitorRunStatusResponseData.attributeTypeMap;
    }
}
exports.GetDataObservabilityMonitorRunStatusResponseData = GetDataObservabilityMonitorRunStatusResponseData;
/**
 * @ignore
 */
GetDataObservabilityMonitorRunStatusResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "GetDataObservabilityMonitorRunStatusResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "DataObservabilityMonitorRunType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
