// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GlobalOrgUser.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GlobalOrgUser = void 0;
/**
 * User information for a global organization association.
 */
class GlobalOrgUser {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GlobalOrgUser.attributeTypeMap;
    }
}
exports.GlobalOrgUser = GlobalOrgUser;
/**
 * @ignore
 */
GlobalOrgUser.attributeTypeMap = {
    handle: {
        baseName: "handle",
        type: "string",
        required: true,
    },
    uuid: {
        baseName: "uuid",
        type: "string",
        required: true,
        format: "uuid",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
