// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GlobalOrg.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GlobalOrg = void 0;
/**
 * Organization information for a global organization association.
 */
class GlobalOrg {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GlobalOrg.attributeTypeMap;
    }
}
exports.GlobalOrg = GlobalOrg;
/**
 * @ignore
 */
GlobalOrg.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    publicId: {
        baseName: "public_id",
        type: "string",
    },
    subdomain: {
        baseName: "subdomain",
        type: "string",
    },
    uuid: {
        baseName: "uuid",
        type: "string",
        required: true,
        format: "uuid",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
