// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GoogleChatTargetAudienceCreateRequestAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GoogleChatTargetAudienceCreateRequestAttributes = void 0;
/**
 * Attributes for creating a Google Chat target audience.
 */
class GoogleChatTargetAudienceCreateRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GoogleChatTargetAudienceCreateRequestAttributes.attributeTypeMap;
    }
}
exports.GoogleChatTargetAudienceCreateRequestAttributes = GoogleChatTargetAudienceCreateRequestAttributes;
/**
 * @ignore
 */
GoogleChatTargetAudienceCreateRequestAttributes.attributeTypeMap = {
    audienceId: {
        baseName: "audience_id",
        type: "string",
        required: true,
    },
    audienceName: {
        baseName: "audience_name",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
