// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GoogleChatOrganizationData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GoogleChatOrganizationData = void 0;
/**
 * Google Chat organization data from a response.
 */
class GoogleChatOrganizationData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GoogleChatOrganizationData.attributeTypeMap;
    }
}
exports.GoogleChatOrganizationData = GoogleChatOrganizationData;
/**
 * @ignore
 */
GoogleChatOrganizationData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "GoogleChatOrganizationAttributes",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    relationships: {
        baseName: "relationships",
        type: "GoogleChatOrganizationRelationships",
    },
    type: {
        baseName: "type",
        type: "GoogleChatOrganizationType",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
