// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GovernanceControlAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GovernanceControlAttributes = void 0;
/**
 * The attributes of a governance control.
 */
class GovernanceControlAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GovernanceControlAttributes.attributeTypeMap;
    }
}
exports.GovernanceControlAttributes = GovernanceControlAttributes;
/**
 * @ignore
 */
GovernanceControlAttributes.attributeTypeMap = {
    activeDetectionsCount: {
        baseName: "active_detections_count",
        type: "number",
        required: true,
        format: "int64",
    },
    category: {
        baseName: "category",
        type: "string",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    createdBy: {
        baseName: "created_by",
        type: "string",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    detectionFrequency: {
        baseName: "detection_frequency",
        type: "string",
        required: true,
    },
    detectionParameters: {
        baseName: "detection_parameters",
        type: "{ [key: string]: any; }",
        required: true,
    },
    detectionType: {
        baseName: "detection_type",
        type: "string",
        required: true,
    },
    featureFlags: {
        baseName: "feature_flags",
        type: "Array<string>",
        required: true,
    },
    insights: {
        baseName: "insights",
        type: "Array<string>",
        required: true,
    },
    lastDetectionAt: {
        baseName: "last_detection_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    mitigatedDetectionsCount: {
        baseName: "mitigated_detections_count",
        type: "number",
        required: true,
        format: "int64",
    },
    mitigationParameters: {
        baseName: "mitigation_parameters",
        type: "{ [key: string]: any; }",
        required: true,
    },
    mitigationType: {
        baseName: "mitigation_type",
        type: "string",
        required: true,
    },
    mitigations: {
        baseName: "mitigations",
        type: "Array<GovernanceControlMitigationDefinition>",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    nextSteps: {
        baseName: "next_steps",
        type: "string",
        required: true,
    },
    notificationFrequency: {
        baseName: "notification_frequency",
        type: "string",
        required: true,
    },
    notificationParameters: {
        baseName: "notification_parameters",
        type: "{ [key: string]: any; }",
        required: true,
    },
    notificationType: {
        baseName: "notification_type",
        type: "string",
        required: true,
    },
    priority: {
        baseName: "priority",
        type: "string",
        required: true,
    },
    product: {
        baseName: "product",
        type: "string",
        required: true,
    },
    releaseStatus: {
        baseName: "release_status",
        type: "string",
        required: true,
    },
    resourceType: {
        baseName: "resource_type",
        type: "string",
        required: true,
    },
    resourceTypeDisplayName: {
        baseName: "resource_type_display_name",
        type: "string",
        required: true,
    },
    supportedDetectionParameters: {
        baseName: "supported_detection_parameters",
        type: "Array<GovernanceControlParameterDefinition>",
        required: true,
    },
    supportedNotificationParameters: {
        baseName: "supported_notification_parameters",
        type: "Array<GovernanceControlParameterDefinition>",
        required: true,
    },
    task: {
        baseName: "task",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "string",
        required: true,
    },
    usageConcern: {
        baseName: "usage_concern",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
