// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GovernanceControlUpdateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GovernanceControlUpdateAttributes = void 0;
/**
 * The attributes of a governance control that can be updated. Only the attributes present in the request are modified.
 */
class GovernanceControlUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GovernanceControlUpdateAttributes.attributeTypeMap;
    }
}
exports.GovernanceControlUpdateAttributes = GovernanceControlUpdateAttributes;
/**
 * @ignore
 */
GovernanceControlUpdateAttributes.attributeTypeMap = {
    detectionFrequency: {
        baseName: "detection_frequency",
        type: "string",
    },
    detectionParameters: {
        baseName: "detection_parameters",
        type: "{ [key: string]: any; }",
    },
    mitigationParameters: {
        baseName: "mitigation_parameters",
        type: "{ [key: string]: any; }",
    },
    mitigationType: {
        baseName: "mitigation_type",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    notificationFrequency: {
        baseName: "notification_frequency",
        type: "string",
    },
    notificationParameters: {
        baseName: "notification_parameters",
        type: "{ [key: string]: any; }",
    },
    notificationType: {
        baseName: "notification_type",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
