// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GovernanceControlSupportedValue.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GovernanceControlSupportedValue = void 0;
/**
 * A supported value for an enumerated parameter.
 */
class GovernanceControlSupportedValue {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GovernanceControlSupportedValue.attributeTypeMap;
    }
}
exports.GovernanceControlSupportedValue = GovernanceControlSupportedValue;
/**
 * @ignore
 */
GovernanceControlSupportedValue.attributeTypeMap = {
    label: {
        baseName: "label",
        type: "string",
        required: true,
    },
    value: {
        baseName: "value",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
