// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GovernanceBestPracticeDefinition.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GovernanceBestPracticeDefinition = void 0;
/**
 * The best practice associated with an insight. Populated with the first active best practice
 * matched to the insight; `null` when no best practice is attached.
 */
class GovernanceBestPracticeDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GovernanceBestPracticeDefinition.attributeTypeMap;
    }
}
exports.GovernanceBestPracticeDefinition = GovernanceBestPracticeDefinition;
/**
 * @ignore
 */
GovernanceBestPracticeDefinition.attributeTypeMap = {
    category: {
        baseName: "category",
        type: "string",
        required: true,
    },
    deepLink: {
        baseName: "deep_link",
        type: "string",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    detectionType: {
        baseName: "detection_type",
        type: "string",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    impact: {
        baseName: "impact",
        type: "string",
        required: true,
    },
    impactHint: {
        baseName: "impact_hint",
        type: "number",
        required: true,
        format: "int64",
    },
    permissions: {
        baseName: "permissions",
        type: "Array<string>",
        required: true,
    },
    status: {
        baseName: "status",
        type: "string",
        required: true,
    },
    summary: {
        baseName: "summary",
        type: "string",
        required: true,
    },
    title: {
        baseName: "title",
        type: "string",
        required: true,
    },
    triggerCondition: {
        baseName: "trigger_condition",
        type: "string",
        required: true,
    },
    triggerType: {
        baseName: "trigger_type",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
