// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GovernanceInsightMetricQuery.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GovernanceInsightMetricQuery = void 0;
/**
 * A metric query used to compute an insight value.
 */
class GovernanceInsightMetricQuery {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GovernanceInsightMetricQuery.attributeTypeMap;
    }
}
exports.GovernanceInsightMetricQuery = GovernanceInsightMetricQuery;
/**
 * @ignore
 */
GovernanceInsightMetricQuery.attributeTypeMap = {
    query: {
        baseName: "query",
        type: "string",
        required: true,
    },
    reducer: {
        baseName: "reducer",
        type: "string",
        required: true,
    },
    source: {
        baseName: "source",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
