// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GovernanceInsightQueryConfig.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GovernanceInsightQueryConfig = void 0;
/**
 * Query execution context that allows the frontend to execute insight queries directly.
 */
class GovernanceInsightQueryConfig {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GovernanceInsightQueryConfig.attributeTypeMap;
    }
}
exports.GovernanceInsightQueryConfig = GovernanceInsightQueryConfig;
/**
 * @ignore
 */
GovernanceInsightQueryConfig.attributeTypeMap = {
    chartType: {
        baseName: "chart_type",
        type: "string",
    },
    comparisonShift: {
        baseName: "comparison_shift",
        type: "string",
        required: true,
    },
    defaultValue: {
        baseName: "default_value",
        type: "number",
        format: "int64",
    },
    directionality: {
        baseName: "directionality",
        type: "string",
    },
    effectiveTimeWindowDays: {
        baseName: "effective_time_window_days",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
