// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GovernanceInsightEventCompute.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GovernanceInsightEventCompute = void 0;
/**
 * The aggregation applied to an event query.
 */
class GovernanceInsightEventCompute {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GovernanceInsightEventCompute.attributeTypeMap;
    }
}
exports.GovernanceInsightEventCompute = GovernanceInsightEventCompute;
/**
 * @ignore
 */
GovernanceInsightEventCompute.attributeTypeMap = {
    aggregation: {
        baseName: "aggregation",
        type: "string",
        required: true,
    },
    interval: {
        baseName: "interval",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
