// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GovernanceControlParameterDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GovernanceControlParameterDefinition = void 0;
/**
 * The definition of a configurable parameter on a control or mitigation.
 */
class GovernanceControlParameterDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GovernanceControlParameterDefinition.attributeTypeMap;
    }
}
exports.GovernanceControlParameterDefinition = GovernanceControlParameterDefinition;
/**
 * @ignore
 */
GovernanceControlParameterDefinition.attributeTypeMap = {
    defaultValue: {
        baseName: "default_value",
        type: "any",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    displayName: {
        baseName: "display_name",
        type: "string",
        required: true,
    },
    hidden: {
        baseName: "hidden",
        type: "boolean",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    required: {
        baseName: "required",
        type: "boolean",
        required: true,
    },
    supportedValues: {
        baseName: "supported_values",
        type: "Array<GovernanceControlSupportedValue>",
        required: true,
    },
    type: {
        baseName: "type",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
