// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GovernanceControlMitigationDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GovernanceControlMitigationDefinition = void 0;
/**
 * The definition of a mitigation available for a control.
 */
class GovernanceControlMitigationDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GovernanceControlMitigationDefinition.attributeTypeMap;
    }
}
exports.GovernanceControlMitigationDefinition = GovernanceControlMitigationDefinition;
/**
 * @ignore
 */
GovernanceControlMitigationDefinition.attributeTypeMap = {
    actionVerb: {
        baseName: "action_verb",
        type: "string",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    executionModes: {
        baseName: "execution_modes",
        type: "Array<string>",
    },
    featureFlags: {
        baseName: "feature_flags",
        type: "Array<string>",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    manualMitigationWarning: {
        baseName: "manual_mitigation_warning",
        type: "string",
        required: true,
    },
    permissions: {
        baseName: "permissions",
        type: "Array<string>",
        required: true,
    },
    requiresAi: {
        baseName: "requires_ai",
        type: "boolean",
        required: true,
    },
    supportedParameters: {
        baseName: "supported_parameters",
        type: "Array<GovernanceControlParameterDefinition>",
        required: true,
    },
    title: {
        baseName: "title",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
