// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GovernanceInsightAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GovernanceInsightAttributes = void 0;
/**
 * The attributes of a governance insight.
 */
class GovernanceInsightAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GovernanceInsightAttributes.attributeTypeMap;
    }
}
exports.GovernanceInsightAttributes = GovernanceInsightAttributes;
/**
 * @ignore
 */
GovernanceInsightAttributes.attributeTypeMap = {
    auditQuery: {
        baseName: "audit_query",
        type: "GovernanceInsightAuditQuery",
        required: true,
    },
    bestPractice: {
        baseName: "best_practice",
        type: "GovernanceBestPracticeDefinition",
        required: true,
    },
    deepLink: {
        baseName: "deep_link",
        type: "string",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    displayName: {
        baseName: "display_name",
        type: "string",
        required: true,
    },
    eventQuery: {
        baseName: "event_query",
        type: "GovernanceInsightEventQuery",
        required: true,
    },
    metricQuery: {
        baseName: "metric_query",
        type: "GovernanceInsightMetricQuery",
        required: true,
    },
    oldValue: {
        baseName: "old_value",
        type: "number",
        required: true,
        format: "double",
    },
    percentageQuery: {
        baseName: "percentage_query",
        type: "GovernanceInsightPercentageQuery",
        required: true,
    },
    product: {
        baseName: "product",
        type: "string",
        required: true,
    },
    queryConfig: {
        baseName: "query_config",
        type: "GovernanceInsightQueryConfig",
    },
    sortOrder: {
        baseName: "sort_order",
        type: "number",
        format: "int64",
    },
    state: {
        baseName: "state",
        type: "string",
        required: true,
    },
    subProduct: {
        baseName: "sub_product",
        type: "string",
        required: true,
    },
    timeRange: {
        baseName: "time_range",
        type: "string",
        required: true,
    },
    unitName: {
        baseName: "unit_name",
        type: "string",
        required: true,
    },
    usageQuery: {
        baseName: "usage_query",
        type: "GovernanceInsightUsageQuery",
        required: true,
    },
    value: {
        baseName: "value",
        type: "number",
        required: true,
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
