// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GovernanceInsightAuditCompute.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GovernanceInsightAuditCompute = void 0;
/**
 * The aggregation applied to an audit log query.
 */
class GovernanceInsightAuditCompute {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GovernanceInsightAuditCompute.attributeTypeMap;
    }
}
exports.GovernanceInsightAuditCompute = GovernanceInsightAuditCompute;
/**
 * @ignore
 */
GovernanceInsightAuditCompute.attributeTypeMap = {
    aggregation: {
        baseName: "aggregation",
        type: "string",
        required: true,
    },
    interval: {
        baseName: "interval",
        type: "number",
        required: true,
        format: "int64",
    },
    metric: {
        baseName: "metric",
        type: "string",
        required: true,
    },
    rollup: {
        baseName: "rollup",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
