// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/GovernanceInsightAuditQuery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GovernanceInsightAuditQuery = void 0;
/**
 * An audit log query used to compute an insight value.
 */
class GovernanceInsightAuditQuery {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return GovernanceInsightAuditQuery.attributeTypeMap;
    }
}
exports.GovernanceInsightAuditQuery = GovernanceInsightAuditQuery;
/**
 * @ignore
 */
GovernanceInsightAuditQuery.attributeTypeMap = {
    compute: {
        baseName: "compute",
        type: "GovernanceInsightAuditCompute",
        required: true,
    },
    indexes: {
        baseName: "indexes",
        type: "Array<string>",
        required: true,
    },
    query: {
        baseName: "query",
        type: "string",
        required: true,
    },
    source: {
        baseName: "source",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
