// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/HistoricalJobQuery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HistoricalJobQuery = void 0;
/**
 * Query for selecting logs analyzed by the historical job.
 */
class HistoricalJobQuery {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return HistoricalJobQuery.attributeTypeMap;
    }
}
exports.HistoricalJobQuery = HistoricalJobQuery;
/**
 * @ignore
 */
HistoricalJobQuery.attributeTypeMap = {
    additionalFilters: {
        baseName: "additionalFilters",
        type: "string",
    },
    aggregation: {
        baseName: "aggregation",
        type: "SecurityMonitoringRuleQueryAggregation",
    },
    correlatedByFields: {
        baseName: "correlatedByFields",
        type: "Array<string>",
    },
    correlatedQueryIndex: {
        baseName: "correlatedQueryIndex",
        type: "number",
        format: "int64",
    },
    customQueryExtension: {
        baseName: "customQueryExtension",
        type: "string",
    },
    dataSource: {
        baseName: "dataSource",
        type: "SecurityMonitoringStandardDataSource",
    },
    datasetIds: {
        baseName: "datasetIds",
        type: "Array<string>",
    },
    distinctFields: {
        baseName: "distinctFields",
        type: "Array<string>",
    },
    groupByFields: {
        baseName: "groupByFields",
        type: "Array<string>",
    },
    hasOptionalGroupByFields: {
        baseName: "hasOptionalGroupByFields",
        type: "boolean",
    },
    index: {
        baseName: "index",
        type: "string",
    },
    indexes: {
        baseName: "indexes",
        type: "Array<string>",
    },
    metrics: {
        baseName: "metrics",
        type: "Array<string>",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    query: {
        baseName: "query",
        type: "string",
    },
    queryLanguage: {
        baseName: "queryLanguage",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
