// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IL2CPPSourcemapAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IL2CPPSourcemapAttributes = void 0;
/**
 * Attributes of an IL2CPP mapping file.
 */
class IL2CPPSourcemapAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IL2CPPSourcemapAttributes.attributeTypeMap;
    }
}
exports.IL2CPPSourcemapAttributes = IL2CPPSourcemapAttributes;
/**
 * @ignore
 */
IL2CPPSourcemapAttributes.attributeTypeMap = {
    buildId: {
        baseName: "build_id",
        type: "string",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    mapkind: {
        baseName: "mapkind",
        type: "string",
        required: true,
    },
    size: {
        baseName: "size",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
