// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IoCTriageWriteResponseAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IoCTriageWriteResponseAttributes = void 0;
/**
 * Attributes of a created or updated triage state.
 */
class IoCTriageWriteResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IoCTriageWriteResponseAttributes.attributeTypeMap;
    }
}
exports.IoCTriageWriteResponseAttributes = IoCTriageWriteResponseAttributes;
/**
 * @ignore
 */
IoCTriageWriteResponseAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        format: "date-time",
    },
    indicator: {
        baseName: "indicator",
        type: "string",
    },
    triageState: {
        baseName: "triage_state",
        type: "IoCTriageState",
    },
    triagedAt: {
        baseName: "triaged_at",
        type: "Date",
        format: "date-time",
    },
    triagedBy: {
        baseName: "triaged_by",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
