// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IOSSourcemapAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IOSSourcemapAttributes = void 0;
/**
 * Attributes of an iOS dSYM source map.
 */
class IOSSourcemapAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IOSSourcemapAttributes.attributeTypeMap;
    }
}
exports.IOSSourcemapAttributes = IOSSourcemapAttributes;
/**
 * @ignore
 */
IOSSourcemapAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    mapkind: {
        baseName: "mapkind",
        type: "string",
        required: true,
    },
    size: {
        baseName: "size",
        type: "number",
        required: true,
        format: "int64",
    },
    uuids: {
        baseName: "uuids",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
