// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/IssueCaseLinearIssueResult.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IssueCaseLinearIssueResult = void 0;
/**
 * Contains the identifiers and URL for a successfully created Linear issue.
 */
class IssueCaseLinearIssueResult {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return IssueCaseLinearIssueResult.attributeTypeMap;
    }
}
exports.IssueCaseLinearIssueResult = IssueCaseLinearIssueResult;
/**
 * @ignore
 */
IssueCaseLinearIssueResult.attributeTypeMap = {
    accountId: {
        baseName: "account_id",
        type: "string",
    },
    issueId: {
        baseName: "issue_id",
        type: "string",
    },
    issueKey: {
        baseName: "issue_key",
        type: "string",
    },
    issueUrl: {
        baseName: "issue_url",
        type: "string",
    },
    teamId: {
        baseName: "team_id",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
