// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/JSSourcemapAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.JSSourcemapAttributes = void 0;
/**
 * Attributes of a JavaScript source map.
 */
class JSSourcemapAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return JSSourcemapAttributes.attributeTypeMap;
    }
}
exports.JSSourcemapAttributes = JSSourcemapAttributes;
/**
 * @ignore
 */
JSSourcemapAttributes.attributeTypeMap = {
    absolutePath: {
        baseName: "absolute_path",
        type: "string",
    },
    blobStorageSourcemapPath: {
        baseName: "blob_storage_sourcemap_path",
        type: "string",
    },
    buildId: {
        baseName: "build_id",
        type: "string",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    domain: {
        baseName: "domain",
        type: "string",
    },
    fileName: {
        baseName: "file_name",
        type: "string",
    },
    mapkind: {
        baseName: "mapkind",
        type: "string",
        required: true,
    },
    service: {
        baseName: "service",
        type: "string",
    },
    size: {
        baseName: "size",
        type: "number",
        required: true,
        format: "int64",
    },
    variant: {
        baseName: "variant",
        type: "string",
    },
    version: {
        baseName: "version",
        type: "string",
    },
    versionCode: {
        baseName: "version_code",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
