// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/JVMSourcemapAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.JVMSourcemapAttributes = void 0;
/**
 * Attributes of a JVM mapping file.
 */
class JVMSourcemapAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return JVMSourcemapAttributes.attributeTypeMap;
    }
}
exports.JVMSourcemapAttributes = JVMSourcemapAttributes;
/**
 * @ignore
 */
JVMSourcemapAttributes.attributeTypeMap = {
    buildId: {
        baseName: "build_id",
        type: "string",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    mapkind: {
        baseName: "mapkind",
        type: "string",
        required: true,
    },
    service: {
        baseName: "service",
        type: "string",
    },
    size: {
        baseName: "size",
        type: "number",
        required: true,
        format: "int64",
    },
    variant: {
        baseName: "variant",
        type: "string",
    },
    version: {
        baseName: "version",
        type: "string",
    },
    versionCode: {
        baseName: "version_code",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
