// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LicensesListResponseDataAttributesLicensesItems.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LicensesListResponseDataAttributesLicensesItems = void 0;
/**
 * An SPDX license entry returned by the licenses list endpoint.
 */
class LicensesListResponseDataAttributesLicensesItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LicensesListResponseDataAttributesLicensesItems.attributeTypeMap;
    }
}
exports.LicensesListResponseDataAttributesLicensesItems = LicensesListResponseDataAttributesLicensesItems;
/**
 * @ignore
 */
LicensesListResponseDataAttributesLicensesItems.attributeTypeMap = {
    displayName: {
        baseName: "display_name",
        type: "string",
        required: true,
    },
    identifier: {
        baseName: "identifier",
        type: "string",
        required: true,
    },
    shortName: {
        baseName: "short_name",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
