// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ListRowsResponseLinks.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListRowsResponseLinks = void 0;
/**
 * Pagination links for the list rows response.
 */
class ListRowsResponseLinks {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ListRowsResponseLinks.attributeTypeMap;
    }
}
exports.ListRowsResponseLinks = ListRowsResponseLinks;
/**
 * @ignore
 */
ListRowsResponseLinks.attributeTypeMap = {
    first: {
        baseName: "first",
        type: "string",
        required: true,
    },
    next: {
        baseName: "next",
        type: "string",
    },
    self: {
        baseName: "self",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
