// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ListWorkflowsResponseMetaPage.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListWorkflowsResponseMetaPage = void 0;
/**
 * Pagination metadata for a List Workflows response.
 */
class ListWorkflowsResponseMetaPage {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ListWorkflowsResponseMetaPage.attributeTypeMap;
    }
}
exports.ListWorkflowsResponseMetaPage = ListWorkflowsResponseMetaPage;
/**
 * @ignore
 */
ListWorkflowsResponseMetaPage.attributeTypeMap = {
    totalCount: {
        baseName: "totalCount",
        type: "number",
        format: "int64",
    },
    totalFilteredCount: {
        baseName: "totalFilteredCount",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
