// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentationAnalyticsValue.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentationAnalyticsValue = void 0;
/**
 * A single analytics result bucket.
 */
class LLMObsExperimentationAnalyticsValue {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentationAnalyticsValue.attributeTypeMap;
    }
}
exports.LLMObsExperimentationAnalyticsValue = LLMObsExperimentationAnalyticsValue;
/**
 * @ignore
 */
LLMObsExperimentationAnalyticsValue.attributeTypeMap = {
    by: {
        baseName: "by",
        type: "{ [key: string]: any; }",
    },
    metrics: {
        baseName: "metrics",
        type: "{ [key: string]: any; }",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
