// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDatasetDraftStateUser.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDatasetDraftStateUser = void 0;
/**
 * User information associated with a dataset draft state.
 */
class LLMObsDatasetDraftStateUser {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDatasetDraftStateUser.attributeTypeMap;
    }
}
exports.LLMObsDatasetDraftStateUser = LLMObsDatasetDraftStateUser;
/**
 * @ignore
 */
LLMObsDatasetDraftStateUser.attributeTypeMap = {
    email: {
        baseName: "email",
        type: "string",
    },
    handle: {
        baseName: "handle",
        type: "string",
    },
    icon: {
        baseName: "icon",
        type: "string",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
