// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentationSearchDataResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentationSearchDataResponse = void 0;
/**
 * JSON:API data object for an experimentation search response.
 */
class LLMObsExperimentationSearchDataResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentationSearchDataResponse.attributeTypeMap;
    }
}
exports.LLMObsExperimentationSearchDataResponse = LLMObsExperimentationSearchDataResponse;
/**
 * @ignore
 */
LLMObsExperimentationSearchDataResponse.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsExperimentationSearchResults",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsExperimentationType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
