// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsSpanToolDefinition.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsSpanToolDefinition = void 0;
/**
 * A tool definition available to an LLM span.
 */
class LLMObsSpanToolDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsSpanToolDefinition.attributeTypeMap;
    }
}
exports.LLMObsSpanToolDefinition = LLMObsSpanToolDefinition;
/**
 * @ignore
 */
LLMObsSpanToolDefinition.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    schema: {
        baseName: "schema",
        type: "{ [key: string]: any; }",
    },
    version: {
        baseName: "version",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
