// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsAnnotatedInteractionsByTraceDataAttributesResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsAnnotatedInteractionsByTraceDataAttributesResponse = void 0;
/**
 * Attributes of the cross-queue annotated interactions response.
 */
class LLMObsAnnotatedInteractionsByTraceDataAttributesResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsAnnotatedInteractionsByTraceDataAttributesResponse.attributeTypeMap;
    }
}
exports.LLMObsAnnotatedInteractionsByTraceDataAttributesResponse = LLMObsAnnotatedInteractionsByTraceDataAttributesResponse;
/**
 * @ignore
 */
LLMObsAnnotatedInteractionsByTraceDataAttributesResponse.attributeTypeMap = {
    annotatedInteractions: {
        baseName: "annotated_interactions",
        type: "Array<LLMObsAnnotatedInteractionByTraceItem>",
        required: true,
    },
    totalCount: {
        baseName: "total_count",
        type: "number",
        required: true,
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
