// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentUpdateDataAttributesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentUpdateDataAttributesRequest = void 0;
/**
 * Attributes for updating an LLM Observability experiment.
 */
class LLMObsExperimentUpdateDataAttributesRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentUpdateDataAttributesRequest.attributeTypeMap;
    }
}
exports.LLMObsExperimentUpdateDataAttributesRequest = LLMObsExperimentUpdateDataAttributesRequest;
/**
 * @ignore
 */
LLMObsExperimentUpdateDataAttributesRequest.attributeTypeMap = {
    datasetId: {
        baseName: "dataset_id",
        type: "string",
    },
    description: {
        baseName: "description",
        type: "string",
    },
    error: {
        baseName: "error",
        type: "string",
    },
    metadata: {
        baseName: "metadata",
        type: "{ [key: string]: any; }",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    status: {
        baseName: "status",
        type: "LLMObsExperimentStatus",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
