// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDisplayBlockAnnotatedInteractionItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDisplayBlockAnnotatedInteractionItem = void 0;
/**
 * A display_block interaction with its associated annotations.
 */
class LLMObsDisplayBlockAnnotatedInteractionItem {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDisplayBlockAnnotatedInteractionItem.attributeTypeMap;
    }
}
exports.LLMObsDisplayBlockAnnotatedInteractionItem = LLMObsDisplayBlockAnnotatedInteractionItem;
/**
 * @ignore
 */
LLMObsDisplayBlockAnnotatedInteractionItem.attributeTypeMap = {
    annotations: {
        baseName: "annotations",
        type: "Array<LLMObsAnnotationItem>",
        required: true,
    },
    contentId: {
        baseName: "content_id",
        type: "string",
        required: true,
    },
    displayBlock: {
        baseName: "display_block",
        type: "Array<LLMObsContentBlock>",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsDisplayBlockInteractionType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
