// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDatasetRestoreVersionDataAttributesRequest.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDatasetRestoreVersionDataAttributesRequest = void 0;
/**
 * Attributes for restoring an LLM Observability dataset to a previous version.
 */
class LLMObsDatasetRestoreVersionDataAttributesRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDatasetRestoreVersionDataAttributesRequest.attributeTypeMap;
    }
}
exports.LLMObsDatasetRestoreVersionDataAttributesRequest = LLMObsDatasetRestoreVersionDataAttributesRequest;
/**
 * @ignore
 */
LLMObsDatasetRestoreVersionDataAttributesRequest.attributeTypeMap = {
    datasetVersion: {
        baseName: "dataset_version",
        type: "number",
        required: true,
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
