// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsConfigSnapshot.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsConfigSnapshot = void 0;
/**
 * Snapshot of the configuration used for a patterns run.
 */
class LLMObsPatternsConfigSnapshot {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsConfigSnapshot.attributeTypeMap;
    }
}
exports.LLMObsPatternsConfigSnapshot = LLMObsPatternsConfigSnapshot;
/**
 * @ignore
 */
LLMObsPatternsConfigSnapshot.attributeTypeMap = {
    accountId: {
        baseName: "account_id",
        type: "string",
    },
    evpQuery: {
        baseName: "evp_query",
        type: "string",
    },
    hierarchyDepth: {
        baseName: "hierarchy_depth",
        type: "number",
        format: "int32",
    },
    integrationProvider: {
        baseName: "integration_provider",
        type: "string",
    },
    modelName: {
        baseName: "model_name",
        type: "string",
    },
    numRecords: {
        baseName: "num_records",
        type: "number",
        format: "int32",
    },
    samplingRatio: {
        baseName: "sampling_ratio",
        type: "number",
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
