// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDatasetBatchUpdateDataAttributesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDatasetBatchUpdateDataAttributesRequest = void 0;
/**
 * Attributes for batch-updating records in an LLM Observability dataset.
 */
class LLMObsDatasetBatchUpdateDataAttributesRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDatasetBatchUpdateDataAttributesRequest.attributeTypeMap;
    }
}
exports.LLMObsDatasetBatchUpdateDataAttributesRequest = LLMObsDatasetBatchUpdateDataAttributesRequest;
/**
 * @ignore
 */
LLMObsDatasetBatchUpdateDataAttributesRequest.attributeTypeMap = {
    createNewVersion: {
        baseName: "create_new_version",
        type: "boolean",
    },
    deleteRecords: {
        baseName: "delete_records",
        type: "Array<string>",
    },
    insertRecords: {
        baseName: "insert_records",
        type: "Array<LLMObsDatasetBatchUpdateInsertRecord>",
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
    updateRecords: {
        baseName: "update_records",
        type: "Array<LLMObsDatasetBatchUpdateUpdateRecord>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
