// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsClusteredPointRef.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsClusteredPointRef = void 0;
/**
 * A clustered point attached inline to a topic. The metric fields are populated
 * only when the request includes `include_metrics=true`.
 */
class LLMObsPatternsClusteredPointRef {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsClusteredPointRef.attributeTypeMap;
    }
}
exports.LLMObsPatternsClusteredPointRef = LLMObsPatternsClusteredPointRef;
/**
 * @ignore
 */
LLMObsPatternsClusteredPointRef.attributeTypeMap = {
    duration: {
        baseName: "duration",
        type: "number",
        format: "double",
    },
    estimatedTotalCost: {
        baseName: "estimated_total_cost",
        type: "number",
        format: "double",
    },
    evaluation: {
        baseName: "evaluation",
        type: "{ [key: string]: any; }",
    },
    inputTokens: {
        baseName: "input_tokens",
        type: "number",
        format: "double",
    },
    outputTokens: {
        baseName: "output_tokens",
        type: "number",
        format: "double",
    },
    spanId: {
        baseName: "span_id",
        type: "string",
        required: true,
    },
    status: {
        baseName: "status",
        type: "string",
    },
    totalTokens: {
        baseName: "total_tokens",
        type: "number",
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
