// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsTopicsResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsTopicsResponseData = void 0;
/**
 * Data object of an LLM Observability patterns topics response.
 */
class LLMObsPatternsTopicsResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsTopicsResponseData.attributeTypeMap;
    }
}
exports.LLMObsPatternsTopicsResponseData = LLMObsPatternsTopicsResponseData;
/**
 * @ignore
 */
LLMObsPatternsTopicsResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsPatternsTopicsResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsPatternsTopicsType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
