// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDeleteAnnotationsDataResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDeleteAnnotationsDataResponse = void 0;
/**
 * Data object for the annotation deletion response.
 */
class LLMObsDeleteAnnotationsDataResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDeleteAnnotationsDataResponse.attributeTypeMap;
    }
}
exports.LLMObsDeleteAnnotationsDataResponse = LLMObsDeleteAnnotationsDataResponse;
/**
 * @ignore
 */
LLMObsDeleteAnnotationsDataResponse.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsDeleteAnnotationsDataAttributesResponse",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsAnnotationsType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
