// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentationAnalyticsDataResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentationAnalyticsDataResponse = void 0;
/**
 * JSON:API data object for an analytics response.
 */
class LLMObsExperimentationAnalyticsDataResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentationAnalyticsDataResponse.attributeTypeMap;
    }
}
exports.LLMObsExperimentationAnalyticsDataResponse = LLMObsExperimentationAnalyticsDataResponse;
/**
 * @ignore
 */
LLMObsExperimentationAnalyticsDataResponse.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsExperimentationAnalyticsDataAttributesResponse",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsExperimentationType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
