// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDatasetRecordTagOperations.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDatasetRecordTagOperations = void 0;
/**
 * Explicit tag operations for updating records. Operations are applied in order, Remove then Add then Set. `set` is the final override; if specified, the result of `remove` and `add` is discarded.
 */
class LLMObsDatasetRecordTagOperations {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDatasetRecordTagOperations.attributeTypeMap;
    }
}
exports.LLMObsDatasetRecordTagOperations = LLMObsDatasetRecordTagOperations;
/**
 * @ignore
 */
LLMObsDatasetRecordTagOperations.attributeTypeMap = {
    add: {
        baseName: "add",
        type: "Array<string>",
    },
    remove: {
        baseName: "remove",
        type: "Array<string>",
    },
    set: {
        baseName: "set",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
