// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsIntegrationInferenceResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsIntegrationInferenceResponse = void 0;
/**
 * The result of an LLM inference request, including input parameters and the model response.
 */
class LLMObsIntegrationInferenceResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsIntegrationInferenceResponse.attributeTypeMap;
    }
}
exports.LLMObsIntegrationInferenceResponse = LLMObsIntegrationInferenceResponse;
/**
 * @ignore
 */
LLMObsIntegrationInferenceResponse.attributeTypeMap = {
    anthropicMetadata: {
        baseName: "anthropic_metadata",
        type: "LLMObsAnthropicMetadata",
    },
    azureOpenaiMetadata: {
        baseName: "azure_openai_metadata",
        type: "LLMObsAzureOpenAIMetadata",
    },
    bedrockMetadata: {
        baseName: "bedrock_metadata",
        type: "LLMObsBedrockMetadata",
    },
    errorResponse: {
        baseName: "error_response",
        type: "LLMObsInferenceErrorResponse",
    },
    frequencyPenalty: {
        baseName: "frequency_penalty",
        type: "number",
        format: "double",
    },
    jsonSchema: {
        baseName: "json_schema",
        type: "string",
    },
    maxCompletionTokens: {
        baseName: "max_completion_tokens",
        type: "number",
        format: "int64",
    },
    maxTokens: {
        baseName: "max_tokens",
        type: "number",
        format: "int64",
    },
    messages: {
        baseName: "messages",
        type: "Array<LLMObsInferenceMessage>",
        required: true,
    },
    modelId: {
        baseName: "model_id",
        type: "string",
        required: true,
    },
    openaiMetadata: {
        baseName: "openai_metadata",
        type: "LLMObsOpenAIMetadata",
    },
    presencePenalty: {
        baseName: "presence_penalty",
        type: "number",
        format: "double",
    },
    response: {
        baseName: "response",
        type: "LLMObsInferenceRunResult",
        required: true,
    },
    temperature: {
        baseName: "temperature",
        type: "number",
        format: "double",
    },
    tools: {
        baseName: "tools",
        type: "Array<LLMObsInferenceTool>",
    },
    topK: {
        baseName: "top_k",
        type: "number",
        format: "int64",
    },
    topP: {
        baseName: "top_p",
        type: "number",
        format: "double",
    },
    vertexAiMetadata: {
        baseName: "vertex_ai_metadata",
        type: "LLMObsVertexAIMetadata",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
