// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsClusteredPointsResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsClusteredPointsResponseData = void 0;
/**
 * Data object of an LLM Observability patterns clustered points response.
 */
class LLMObsPatternsClusteredPointsResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsClusteredPointsResponseData.attributeTypeMap;
    }
}
exports.LLMObsPatternsClusteredPointsResponseData = LLMObsPatternsClusteredPointsResponseData;
/**
 * @ignore
 */
LLMObsPatternsClusteredPointsResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsPatternsClusteredPointsResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsPatternsClusteredPointsType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
