// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsClusteredPoint.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsClusteredPoint = void 0;
/**
 * A single data point grouped into a topic.
 */
class LLMObsPatternsClusteredPoint {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsClusteredPoint.attributeTypeMap;
    }
}
exports.LLMObsPatternsClusteredPoint = LLMObsPatternsClusteredPoint;
/**
 * @ignore
 */
LLMObsPatternsClusteredPoint.attributeTypeMap = {
    eventId: {
        baseName: "event_id",
        type: "string",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    input: {
        baseName: "input",
        type: "string",
        required: true,
    },
    isIncluded: {
        baseName: "is_included",
        type: "boolean",
        required: true,
    },
    isSuggested: {
        baseName: "is_suggested",
        type: "boolean",
        required: true,
    },
    sessionId: {
        baseName: "session_id",
        type: "string",
        required: true,
    },
    spanId: {
        baseName: "span_id",
        type: "string",
        required: true,
    },
    topicId: {
        baseName: "topic_id",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
