// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentationContentPreview.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentationContentPreview = void 0;
/**
 * Options to control content preview truncation.
 */
class LLMObsExperimentationContentPreview {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentationContentPreview.attributeTypeMap;
    }
}
exports.LLMObsExperimentationContentPreview = LLMObsExperimentationContentPreview;
/**
 * @ignore
 */
LLMObsExperimentationContentPreview.attributeTypeMap = {
    limit: {
        baseName: "limit",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
