// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDatasetRestoreVersionDataRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDatasetRestoreVersionDataRequest = void 0;
/**
 * Data object for restoring an LLM Observability dataset to a previous version.
 */
class LLMObsDatasetRestoreVersionDataRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDatasetRestoreVersionDataRequest.attributeTypeMap;
    }
}
exports.LLMObsDatasetRestoreVersionDataRequest = LLMObsDatasetRestoreVersionDataRequest;
/**
 * @ignore
 */
LLMObsDatasetRestoreVersionDataRequest.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsDatasetRestoreVersionDataAttributesRequest",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsDatasetType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
