// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsRunStatusResponseAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsRunStatusResponseAttributes = void 0;
/**
 * Attributes of an LLM Observability patterns run status.
 */
class LLMObsPatternsRunStatusResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsRunStatusResponseAttributes.attributeTypeMap;
    }
}
exports.LLMObsPatternsRunStatusResponseAttributes = LLMObsPatternsRunStatusResponseAttributes;
/**
 * @ignore
 */
LLMObsPatternsRunStatusResponseAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    progress: {
        baseName: "progress",
        type: "Array<LLMObsPatternsActivityProgress>",
        required: true,
    },
    status: {
        baseName: "status",
        type: "string",
        required: true,
    },
    step: {
        baseName: "step",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
