// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsSpanToolCall.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsSpanToolCall = void 0;
/**
 * A tool call made during a span.
 */
class LLMObsSpanToolCall {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsSpanToolCall.attributeTypeMap;
    }
}
exports.LLMObsSpanToolCall = LLMObsSpanToolCall;
/**
 * @ignore
 */
LLMObsSpanToolCall.attributeTypeMap = {
    arguments: {
        baseName: "arguments",
        type: "{ [key: string]: any; }",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    toolId: {
        baseName: "tool_id",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
