// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsActivityProgress.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsActivityProgress = void 0;
/**
 * Progress information for a single step of a patterns run.
 */
class LLMObsPatternsActivityProgress {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsActivityProgress.attributeTypeMap;
    }
}
exports.LLMObsPatternsActivityProgress = LLMObsPatternsActivityProgress;
/**
 * @ignore
 */
LLMObsPatternsActivityProgress.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    startedAt: {
        baseName: "started_at",
        type: "Date",
        format: "date-time",
    },
    status: {
        baseName: "status",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
