// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentationSimpleSearchDataRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentationSimpleSearchDataRequest = void 0;
/**
 * Data object for an experimentation simple search request.
 */
class LLMObsExperimentationSimpleSearchDataRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentationSimpleSearchDataRequest.attributeTypeMap;
    }
}
exports.LLMObsExperimentationSimpleSearchDataRequest = LLMObsExperimentationSimpleSearchDataRequest;
/**
 * @ignore
 */
LLMObsExperimentationSimpleSearchDataRequest.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsExperimentationSimpleSearchDataAttributesRequest",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsExperimentationType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
