// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDatasetBatchUpdateInsertRecord.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDatasetBatchUpdateInsertRecord = void 0;
/**
 * A record to insert as part of a batch update on an LLM Observability dataset.
 */
class LLMObsDatasetBatchUpdateInsertRecord {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDatasetBatchUpdateInsertRecord.attributeTypeMap;
    }
}
exports.LLMObsDatasetBatchUpdateInsertRecord = LLMObsDatasetBatchUpdateInsertRecord;
/**
 * @ignore
 */
LLMObsDatasetBatchUpdateInsertRecord.attributeTypeMap = {
    expectedOutput: {
        baseName: "expected_output",
        type: "AnyValue",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    input: {
        baseName: "input",
        type: "AnyValue",
        required: true,
    },
    metadata: {
        baseName: "metadata",
        type: "{ [key: string]: any; }",
    },
    tagOperations: {
        baseName: "tag_operations",
        type: "LLMObsDatasetRecordTagOperations",
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
