// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsSpanAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsSpanAttributes = void 0;
/**
 * Attributes of an LLM Observability span.
 */
class LLMObsSpanAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsSpanAttributes.attributeTypeMap;
    }
}
exports.LLMObsSpanAttributes = LLMObsSpanAttributes;
/**
 * @ignore
 */
LLMObsSpanAttributes.attributeTypeMap = {
    duration: {
        baseName: "duration",
        type: "number",
        required: true,
        format: "double",
    },
    evaluation: {
        baseName: "evaluation",
        type: "{ [key: string]: LLMObsSpanEvaluationMetric; }",
    },
    input: {
        baseName: "input",
        type: "LLMObsSpanIO",
    },
    intent: {
        baseName: "intent",
        type: "string",
    },
    metadata: {
        baseName: "metadata",
        type: "{ [key: string]: any; }",
    },
    metrics: {
        baseName: "metrics",
        type: "{ [key: string]: number; }",
    },
    mlApp: {
        baseName: "ml_app",
        type: "string",
        required: true,
    },
    modelName: {
        baseName: "model_name",
        type: "string",
    },
    modelProvider: {
        baseName: "model_provider",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    output: {
        baseName: "output",
        type: "LLMObsSpanIO",
    },
    parentId: {
        baseName: "parent_id",
        type: "string",
    },
    spanId: {
        baseName: "span_id",
        type: "string",
        required: true,
    },
    spanKind: {
        baseName: "span_kind",
        type: "string",
        required: true,
    },
    startNs: {
        baseName: "start_ns",
        type: "number",
        required: true,
        format: "int64",
    },
    status: {
        baseName: "status",
        type: "string",
        required: true,
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
    toolDefinitions: {
        baseName: "tool_definitions",
        type: "Array<LLMObsSpanToolDefinition>",
    },
    traceId: {
        baseName: "trace_id",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
