// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsVertexAIMetadata.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsVertexAIMetadata = void 0;
/**
 * Vertex AI-specific metadata for an integration account or inference request.
 */
class LLMObsVertexAIMetadata {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsVertexAIMetadata.attributeTypeMap;
    }
}
exports.LLMObsVertexAIMetadata = LLMObsVertexAIMetadata;
/**
 * @ignore
 */
LLMObsVertexAIMetadata.attributeTypeMap = {
    location: {
        baseName: "location",
        type: "string",
    },
    project: {
        baseName: "project",
        type: "string",
    },
    projectIds: {
        baseName: "project_ids",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
