// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsSpanEvaluationMetric.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsSpanEvaluationMetric = void 0;
/**
 * An evaluation metric associated with an LLM Observability span.
 */
class LLMObsSpanEvaluationMetric {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsSpanEvaluationMetric.attributeTypeMap;
    }
}
exports.LLMObsSpanEvaluationMetric = LLMObsSpanEvaluationMetric;
/**
 * @ignore
 */
LLMObsSpanEvaluationMetric.attributeTypeMap = {
    assessment: {
        baseName: "assessment",
        type: "string",
    },
    evalMetricType: {
        baseName: "eval_metric_type",
        type: "string",
    },
    reasoning: {
        baseName: "reasoning",
        type: "string",
    },
    status: {
        baseName: "status",
        type: "string",
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
    value: {
        baseName: "value",
        type: "any",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
