// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDatasetVersionDataAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDatasetVersionDataAttributes = void 0;
/**
 * Attributes of an LLM Observability dataset version.
 */
class LLMObsDatasetVersionDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDatasetVersionDataAttributes.attributeTypeMap;
    }
}
exports.LLMObsDatasetVersionDataAttributes = LLMObsDatasetVersionDataAttributes;
/**
 * @ignore
 */
LLMObsDatasetVersionDataAttributes.attributeTypeMap = {
    datasetId: {
        baseName: "dataset_id",
        type: "string",
        required: true,
    },
    lastUsed: {
        baseName: "last_used",
        type: "Date",
        required: true,
        format: "date-time",
    },
    versionNumber: {
        baseName: "version_number",
        type: "number",
        required: true,
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
