// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsAnnotatedInteractionByTraceItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsAnnotatedInteractionByTraceItem = void 0;
/**
 * An annotated interaction returned by the cross-queue lookup, including the source queue metadata.
 */
class LLMObsAnnotatedInteractionByTraceItem {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsAnnotatedInteractionByTraceItem.attributeTypeMap;
    }
}
exports.LLMObsAnnotatedInteractionByTraceItem = LLMObsAnnotatedInteractionByTraceItem;
/**
 * @ignore
 */
LLMObsAnnotatedInteractionByTraceItem.attributeTypeMap = {
    annotations: {
        baseName: "annotations",
        type: "Array<LLMObsAnnotationItem>",
        required: true,
    },
    contentId: {
        baseName: "content_id",
        type: "string",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    displayBlock: {
        baseName: "display_block",
        type: "Array<LLMObsContentBlock>",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    queueId: {
        baseName: "queue_id",
        type: "string",
        required: true,
    },
    queueName: {
        baseName: "queue_name",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsAnyInteractionType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
