// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsTopicsWithClusteredPointsResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsTopicsWithClusteredPointsResponseData = void 0;
/**
 * Data object of an LLM Observability patterns topics-with-clustered-points response.
 */
class LLMObsPatternsTopicsWithClusteredPointsResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsTopicsWithClusteredPointsResponseData.attributeTypeMap;
    }
}
exports.LLMObsPatternsTopicsWithClusteredPointsResponseData = LLMObsPatternsTopicsWithClusteredPointsResponseData;
/**
 * @ignore
 */
LLMObsPatternsTopicsWithClusteredPointsResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsPatternsTopicsWithClusteredPointsResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsPatternsTopicsWithClusteredPointsType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
