// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentSpanWithEvals.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentSpanWithEvals = void 0;
/**
 * An experiment span enriched with its associated evaluation metrics.
 */
class LLMObsExperimentSpanWithEvals {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentSpanWithEvals.attributeTypeMap;
    }
}
exports.LLMObsExperimentSpanWithEvals = LLMObsExperimentSpanWithEvals;
/**
 * @ignore
 */
LLMObsExperimentSpanWithEvals.attributeTypeMap = {
    datasetRecordId: {
        baseName: "dataset_record_id",
        type: "string",
    },
    duration: {
        baseName: "duration",
        type: "number",
        format: "double",
    },
    evalMetrics: {
        baseName: "eval_metrics",
        type: "Array<LLMObsExperimentEvalMetricEvent>",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    meta: {
        baseName: "meta",
        type: "LLMObsExperimentSpanMeta",
    },
    metrics: {
        baseName: "metrics",
        type: "{ [key: string]: number; }",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    parentId: {
        baseName: "parent_id",
        type: "string",
    },
    spanId: {
        baseName: "span_id",
        type: "string",
    },
    startNs: {
        baseName: "start_ns",
        type: "number",
        format: "int64",
    },
    status: {
        baseName: "status",
        type: "LLMObsExperimentSpanStatus",
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
    traceId: {
        baseName: "trace_id",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
