// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsSpanFilter.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsSpanFilter = void 0;
/**
 * Filter criteria for an LLM Observability span search.
 */
class LLMObsSpanFilter {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsSpanFilter.attributeTypeMap;
    }
}
exports.LLMObsSpanFilter = LLMObsSpanFilter;
/**
 * @ignore
 */
LLMObsSpanFilter.attributeTypeMap = {
    from: {
        baseName: "from",
        type: "string",
    },
    mlApp: {
        baseName: "ml_app",
        type: "string",
    },
    query: {
        baseName: "query",
        type: "string",
    },
    spanId: {
        baseName: "span_id",
        type: "string",
    },
    spanKind: {
        baseName: "span_kind",
        type: "string",
    },
    spanName: {
        baseName: "span_name",
        type: "string",
    },
    tags: {
        baseName: "tags",
        type: "{ [key: string]: string; }",
    },
    to: {
        baseName: "to",
        type: "string",
    },
    traceId: {
        baseName: "trace_id",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
