// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentEventsV2DataResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentEventsV2DataResponse = void 0;
/**
 * JSON:API data object for an experiment events response.
 */
class LLMObsExperimentEventsV2DataResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentEventsV2DataResponse.attributeTypeMap;
    }
}
exports.LLMObsExperimentEventsV2DataResponse = LLMObsExperimentEventsV2DataResponse;
/**
 * @ignore
 */
LLMObsExperimentEventsV2DataResponse.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsExperimentEventsV2DataAttributesResponse",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsExperimentEventsType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
