// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentationAnalyticsGroupBy.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentationAnalyticsGroupBy = void 0;
/**
 * A field to group analytics results by.
 */
class LLMObsExperimentationAnalyticsGroupBy {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentationAnalyticsGroupBy.attributeTypeMap;
    }
}
exports.LLMObsExperimentationAnalyticsGroupBy = LLMObsExperimentationAnalyticsGroupBy;
/**
 * @ignore
 */
LLMObsExperimentationAnalyticsGroupBy.attributeTypeMap = {
    field: {
        baseName: "field",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
