// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsInferenceFunction.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsInferenceFunction = void 0;
/**
 * A function definition for a tool available to the model.
 */
class LLMObsInferenceFunction {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsInferenceFunction.attributeTypeMap;
    }
}
exports.LLMObsInferenceFunction = LLMObsInferenceFunction;
/**
 * @ignore
 */
LLMObsInferenceFunction.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    parameters: {
        baseName: "parameters",
        type: "{ [key: string]: any; }",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
