// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentationAnalyticsAggregate.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentationAnalyticsAggregate = void 0;
/**
 * Analytics aggregation parameters.
 */
class LLMObsExperimentationAnalyticsAggregate {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentationAnalyticsAggregate.attributeTypeMap;
    }
}
exports.LLMObsExperimentationAnalyticsAggregate = LLMObsExperimentationAnalyticsAggregate;
/**
 * @ignore
 */
LLMObsExperimentationAnalyticsAggregate.attributeTypeMap = {
    compute: {
        baseName: "compute",
        type: "Array<LLMObsExperimentationAnalyticsCompute>",
        required: true,
    },
    datasetVersion: {
        baseName: "dataset_version",
        type: "number",
        format: "int64",
    },
    groupBy: {
        baseName: "group_by",
        type: "Array<LLMObsExperimentationAnalyticsGroupBy>",
    },
    indexes: {
        baseName: "indexes",
        type: "Array<string>",
        required: true,
    },
    limit: {
        baseName: "limit",
        type: "number",
        format: "int32",
    },
    search: {
        baseName: "search",
        type: "LLMObsExperimentationAnalyticsSearch",
        required: true,
    },
    time: {
        baseName: "time",
        type: "LLMObsExperimentationAnalyticsTimeRange",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
