// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsSpansResponseMeta.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsSpansResponseMeta = void 0;
/**
 * Metadata accompanying the spans response.
 */
class LLMObsSpansResponseMeta {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsSpansResponseMeta.attributeTypeMap;
    }
}
exports.LLMObsSpansResponseMeta = LLMObsSpansResponseMeta;
/**
 * @ignore
 */
LLMObsSpansResponseMeta.attributeTypeMap = {
    elapsed: {
        baseName: "elapsed",
        type: "number",
        required: true,
        format: "int64",
    },
    page: {
        baseName: "page",
        type: "LLMObsSpansResponsePage",
        required: true,
    },
    requestId: {
        baseName: "request_id",
        type: "string",
        required: true,
    },
    status: {
        baseName: "status",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
