// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsTopic.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsTopic = void 0;
/**
 * A topic discovered by an LLM Observability patterns run.
 */
class LLMObsPatternsTopic {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsTopic.attributeTypeMap;
    }
}
exports.LLMObsPatternsTopic = LLMObsPatternsTopic;
/**
 * @ignore
 */
LLMObsPatternsTopic.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    firstSeenAt: {
        baseName: "first_seen_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    hierarchyLevel: {
        baseName: "hierarchy_level",
        type: "number",
        required: true,
        format: "int64",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    isValidated: {
        baseName: "is_validated",
        type: "boolean",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    parentTopicId: {
        baseName: "parent_topic_id",
        type: "string",
        required: true,
    },
    pointCount: {
        baseName: "point_count",
        type: "number",
        required: true,
        format: "int64",
    },
    runId: {
        baseName: "run_id",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
