// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentEventsV2DataAttributesResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentEventsV2DataAttributesResponse = void 0;
/**
 * Attributes of an experiment events response.
 */
class LLMObsExperimentEventsV2DataAttributesResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentEventsV2DataAttributesResponse.attributeTypeMap;
    }
}
exports.LLMObsExperimentEventsV2DataAttributesResponse = LLMObsExperimentEventsV2DataAttributesResponse;
/**
 * @ignore
 */
LLMObsExperimentEventsV2DataAttributesResponse.attributeTypeMap = {
    spans: {
        baseName: "spans",
        type: "Array<LLMObsExperimentSpanWithEvals>",
        required: true,
    },
    summaryMetrics: {
        baseName: "summary_metrics",
        type: "Array<LLMObsExperimentEvalMetricEvent>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
