// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentRunDataResponse.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentRunDataResponse = void 0;
/**
 * Data object for an LLM Observability experiment run.
 */
class LLMObsExperimentRunDataResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentRunDataResponse.attributeTypeMap;
    }
}
exports.LLMObsExperimentRunDataResponse = LLMObsExperimentRunDataResponse;
/**
 * @ignore
 */
LLMObsExperimentRunDataResponse.attributeTypeMap = {
    aggregateData: {
        baseName: "aggregate_data",
        type: "{ [key: string]: any; }",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        format: "date-time",
    },
    experimentId: {
        baseName: "experiment_id",
        type: "string",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    runNumber: {
        baseName: "run_number",
        type: "number",
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
