// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsConfigResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsConfigResponseData = void 0;
/**
 * Data object of an LLM Observability patterns configuration.
 */
class LLMObsPatternsConfigResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsConfigResponseData.attributeTypeMap;
    }
}
exports.LLMObsPatternsConfigResponseData = LLMObsPatternsConfigResponseData;
/**
 * @ignore
 */
LLMObsPatternsConfigResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsPatternsConfigAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsPatternsConfigType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
