// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentationAnalyticsSearch.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentationAnalyticsSearch = void 0;
/**
 * Search query for filtering analytics data.
 */
class LLMObsExperimentationAnalyticsSearch {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentationAnalyticsSearch.attributeTypeMap;
    }
}
exports.LLMObsExperimentationAnalyticsSearch = LLMObsExperimentationAnalyticsSearch;
/**
 * @ignore
 */
LLMObsExperimentationAnalyticsSearch.attributeTypeMap = {
    query: {
        baseName: "query",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
