// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsRunSummary.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsRunSummary = void 0;
/**
 * Summary of an LLM Observability patterns run.
 */
class LLMObsPatternsRunSummary {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsRunSummary.attributeTypeMap;
    }
}
exports.LLMObsPatternsRunSummary = LLMObsPatternsRunSummary;
/**
 * @ignore
 */
LLMObsPatternsRunSummary.attributeTypeMap = {
    completedAt: {
        baseName: "completed_at",
        type: "Date",
        format: "date-time",
    },
    configSnapshot: {
        baseName: "config_snapshot",
        type: "LLMObsPatternsConfigSnapshot",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    status: {
        baseName: "status",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
