// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDeleteAnnotationsDataAttributesRequest.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDeleteAnnotationsDataAttributesRequest = void 0;
/**
 * Attributes for deleting annotations.
 */
class LLMObsDeleteAnnotationsDataAttributesRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDeleteAnnotationsDataAttributesRequest.attributeTypeMap;
    }
}
exports.LLMObsDeleteAnnotationsDataAttributesRequest = LLMObsDeleteAnnotationsDataAttributesRequest;
/**
 * @ignore
 */
LLMObsDeleteAnnotationsDataAttributesRequest.attributeTypeMap = {
    annotationIds: {
        baseName: "annotation_ids",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
