// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDataDeletionResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDataDeletionResponseData = void 0;
/**
 * Data object for an LLM Observability data deletion response.
 */
class LLMObsDataDeletionResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDataDeletionResponseData.attributeTypeMap;
    }
}
exports.LLMObsDataDeletionResponseData = LLMObsDataDeletionResponseData;
/**
 * @ignore
 */
LLMObsDataDeletionResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsDataDeletionResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsDataDeletionResponseType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
