// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentUser.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentUser = void 0;
/**
 * User data for the author of an experiment. Only present when `include[user_data]` is `true`.
 */
class LLMObsExperimentUser {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentUser.attributeTypeMap;
    }
}
exports.LLMObsExperimentUser = LLMObsExperimentUser;
/**
 * @ignore
 */
LLMObsExperimentUser.attributeTypeMap = {
    email: {
        baseName: "email",
        type: "string",
    },
    handle: {
        baseName: "handle",
        type: "string",
    },
    icon: {
        baseName: "icon",
        type: "string",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
