// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentationSearchDataAttributesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentationSearchDataAttributesRequest = void 0;
/**
 * Attributes for an experimentation search request.
 */
class LLMObsExperimentationSearchDataAttributesRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentationSearchDataAttributesRequest.attributeTypeMap;
    }
}
exports.LLMObsExperimentationSearchDataAttributesRequest = LLMObsExperimentationSearchDataAttributesRequest;
/**
 * @ignore
 */
LLMObsExperimentationSearchDataAttributesRequest.attributeTypeMap = {
    contentPreview: {
        baseName: "content_preview",
        type: "LLMObsExperimentationContentPreview",
    },
    filter: {
        baseName: "filter",
        type: "LLMObsExperimentationFilter",
        required: true,
    },
    include: {
        baseName: "include",
        type: "LLMObsExperimentationInclude",
    },
    page: {
        baseName: "page",
        type: "LLMObsExperimentationCursorPage",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
