// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentationAnalyticsCompute.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentationAnalyticsCompute = void 0;
/**
 * A single metric computation definition.
 */
class LLMObsExperimentationAnalyticsCompute {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentationAnalyticsCompute.attributeTypeMap;
    }
}
exports.LLMObsExperimentationAnalyticsCompute = LLMObsExperimentationAnalyticsCompute;
/**
 * @ignore
 */
LLMObsExperimentationAnalyticsCompute.attributeTypeMap = {
    metric: {
        baseName: "metric",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
