// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentEvalMetricEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentEvalMetricEvent = void 0;
/**
 * An evaluation metric event associated with an experiment span.
 */
class LLMObsExperimentEvalMetricEvent {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentEvalMetricEvent.attributeTypeMap;
    }
}
exports.LLMObsExperimentEvalMetricEvent = LLMObsExperimentEvalMetricEvent;
/**
 * @ignore
 */
LLMObsExperimentEvalMetricEvent.attributeTypeMap = {
    assessment: {
        baseName: "assessment",
        type: "LLMObsMetricAssessment",
    },
    booleanValue: {
        baseName: "boolean_value",
        type: "boolean",
    },
    categoricalValue: {
        baseName: "categorical_value",
        type: "string",
    },
    evalSourceType: {
        baseName: "eval_source_type",
        type: "string",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    jsonValue: {
        baseName: "json_value",
        type: "{ [key: string]: any; }",
    },
    label: {
        baseName: "label",
        type: "string",
    },
    metadata: {
        baseName: "metadata",
        type: "{ [key: string]: any; }",
    },
    metricSource: {
        baseName: "metric_source",
        type: "string",
    },
    metricType: {
        baseName: "metric_type",
        type: "LLMObsMetricScoreType",
    },
    reasoning: {
        baseName: "reasoning",
        type: "string",
    },
    scoreValue: {
        baseName: "score_value",
        type: "number",
        format: "double",
    },
    spanId: {
        baseName: "span_id",
        type: "string",
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
    timestampMs: {
        baseName: "timestamp_ms",
        type: "number",
        format: "int64",
    },
    traceId: {
        baseName: "trace_id",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
