// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsInferenceRunResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsInferenceRunResult = void 0;
/**
 * The output of a completed LLM inference call.
 */
class LLMObsInferenceRunResult {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsInferenceRunResult.attributeTypeMap;
    }
}
exports.LLMObsInferenceRunResult = LLMObsInferenceRunResult;
/**
 * @ignore
 */
LLMObsInferenceRunResult.attributeTypeMap = {
    assessment: {
        baseName: "assessment",
        type: "string",
        required: true,
    },
    content: {
        baseName: "content",
        type: "string",
        required: true,
    },
    finishReason: {
        baseName: "finish_reason",
        type: "string",
        required: true,
    },
    inferenceCodes: {
        baseName: "inference_codes",
        type: "Array<LLMObsInferenceCode>",
        required: true,
    },
    inputTokens: {
        baseName: "input_tokens",
        type: "number",
        required: true,
        format: "int64",
    },
    internalReasoning: {
        baseName: "internal_reasoning",
        type: "LLMObsInternalReasoning",
    },
    latency: {
        baseName: "latency",
        type: "number",
        required: true,
        format: "int64",
    },
    outputTokens: {
        baseName: "output_tokens",
        type: "number",
        required: true,
        format: "int64",
    },
    tools: {
        baseName: "tools",
        type: "Array<LLMObsInferenceTool>",
        required: true,
    },
    totalTokens: {
        baseName: "total_tokens",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
