// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsAnnotationLabelValue.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsAnnotationLabelValue = void 0;
/**
 * A single label value entry in an annotation.
 * The `value` type must match the label schema type:
 * - `score`: a number within the schema `min`/`max` range (integer if `is_integer` is `true`).
 * - `categorical`: a string that is one of the schema `values`.
 * - `boolean`: `true` or `false`.
 * - `text`: any non-empty string.
 */
class LLMObsAnnotationLabelValue {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsAnnotationLabelValue.attributeTypeMap;
    }
}
exports.LLMObsAnnotationLabelValue = LLMObsAnnotationLabelValue;
/**
 * @ignore
 */
LLMObsAnnotationLabelValue.attributeTypeMap = {
    assessment: {
        baseName: "assessment",
        type: "LLMObsAnnotationAssessment",
    },
    labelSchemaId: {
        baseName: "label_schema_id",
        type: "string",
        required: true,
    },
    reasoning: {
        baseName: "reasoning",
        type: "string",
    },
    value: {
        baseName: "value",
        type: "LLMObsAnnotationLabelValueValue",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
