// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsConfigAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsConfigAttributes = void 0;
/**
 * Attributes of an LLM Observability patterns configuration.
 */
class LLMObsPatternsConfigAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsConfigAttributes.attributeTypeMap;
    }
}
exports.LLMObsPatternsConfigAttributes = LLMObsPatternsConfigAttributes;
/**
 * @ignore
 */
LLMObsPatternsConfigAttributes.attributeTypeMap = {
    accountId: {
        baseName: "account_id",
        type: "string",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    evpQuery: {
        baseName: "evp_query",
        type: "string",
        required: true,
    },
    hierarchyDepth: {
        baseName: "hierarchy_depth",
        type: "number",
        required: true,
        format: "int32",
    },
    integrationProvider: {
        baseName: "integration_provider",
        type: "string",
    },
    modelName: {
        baseName: "model_name",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    numRecords: {
        baseName: "num_records",
        type: "number",
        required: true,
        format: "int32",
    },
    samplingRatio: {
        baseName: "sampling_ratio",
        type: "number",
        required: true,
        format: "double",
    },
    scope: {
        baseName: "scope",
        type: "string",
        required: true,
    },
    template: {
        baseName: "template",
        type: "string",
    },
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
