// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentationFilter.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentationFilter = void 0;
/**
 * Filter criteria for an experimentation search request.
 */
class LLMObsExperimentationFilter {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentationFilter.attributeTypeMap;
    }
}
exports.LLMObsExperimentationFilter = LLMObsExperimentationFilter;
/**
 * @ignore
 */
LLMObsExperimentationFilter.attributeTypeMap = {
    includeDeleted: {
        baseName: "include_deleted",
        type: "boolean",
    },
    isDeleted: {
        baseName: "is_deleted",
        type: "boolean",
    },
    query: {
        baseName: "query",
        type: "string",
    },
    scope: {
        baseName: "scope",
        type: "Array<string>",
        required: true,
    },
    version: {
        baseName: "version",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
