// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDisplayBlockInteractionResponseItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDisplayBlockInteractionResponseItem = void 0;
/**
 * A display_block interaction result.
 */
class LLMObsDisplayBlockInteractionResponseItem {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDisplayBlockInteractionResponseItem.attributeTypeMap;
    }
}
exports.LLMObsDisplayBlockInteractionResponseItem = LLMObsDisplayBlockInteractionResponseItem;
/**
 * @ignore
 */
LLMObsDisplayBlockInteractionResponseItem.attributeTypeMap = {
    alreadyExisted: {
        baseName: "already_existed",
        type: "boolean",
        required: true,
    },
    contentId: {
        baseName: "content_id",
        type: "string",
        required: true,
    },
    displayBlock: {
        baseName: "display_block",
        type: "Array<LLMObsContentBlock>",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsDisplayBlockInteractionType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
