// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsTraceAnnotatedInteractionItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsTraceAnnotatedInteractionItem = void 0;
/**
 * A trace, experiment trace, or session interaction with its associated annotations.
 */
class LLMObsTraceAnnotatedInteractionItem {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsTraceAnnotatedInteractionItem.attributeTypeMap;
    }
}
exports.LLMObsTraceAnnotatedInteractionItem = LLMObsTraceAnnotatedInteractionItem;
/**
 * @ignore
 */
LLMObsTraceAnnotatedInteractionItem.attributeTypeMap = {
    annotations: {
        baseName: "annotations",
        type: "Array<LLMObsAnnotationItem>",
        required: true,
    },
    contentId: {
        baseName: "content_id",
        type: "string",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    type: {
        baseName: "type",
        type: "LLMObsTraceInteractionType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
