// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentationSimpleSearchDataResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentationSimpleSearchDataResponse = void 0;
/**
 * JSON:API data object for a simple search response.
 */
class LLMObsExperimentationSimpleSearchDataResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentationSimpleSearchDataResponse.attributeTypeMap;
    }
}
exports.LLMObsExperimentationSimpleSearchDataResponse = LLMObsExperimentationSimpleSearchDataResponse;
/**
 * @ignore
 */
LLMObsExperimentationSimpleSearchDataResponse.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsExperimentationSearchResults",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsExperimentationType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
