// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsAzureOpenAIMetadata.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsAzureOpenAIMetadata = void 0;
/**
 * Azure OpenAI-specific metadata for an integration account or inference request.
 */
class LLMObsAzureOpenAIMetadata {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsAzureOpenAIMetadata.attributeTypeMap;
    }
}
exports.LLMObsAzureOpenAIMetadata = LLMObsAzureOpenAIMetadata;
/**
 * @ignore
 */
LLMObsAzureOpenAIMetadata.attributeTypeMap = {
    deploymentId: {
        baseName: "deployment_id",
        type: "string",
    },
    modelVersion: {
        baseName: "model_version",
        type: "string",
    },
    resourceName: {
        baseName: "resource_name",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
