// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsSpanSearchOptions.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsSpanSearchOptions = void 0;
/**
 * Additional options for a span search request.
 */
class LLMObsSpanSearchOptions {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsSpanSearchOptions.attributeTypeMap;
    }
}
exports.LLMObsSpanSearchOptions = LLMObsSpanSearchOptions;
/**
 * @ignore
 */
LLMObsSpanSearchOptions.attributeTypeMap = {
    includeAttachments: {
        baseName: "include_attachments",
        type: "boolean",
    },
    timeOffset: {
        baseName: "time_offset",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
