// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsInferenceMessage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsInferenceMessage = void 0;
/**
 * A single message in an LLM inference conversation.
 */
class LLMObsInferenceMessage {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsInferenceMessage.attributeTypeMap;
    }
}
exports.LLMObsInferenceMessage = LLMObsInferenceMessage;
/**
 * @ignore
 */
LLMObsInferenceMessage.attributeTypeMap = {
    content: {
        baseName: "content",
        type: "string",
    },
    contents: {
        baseName: "contents",
        type: "Array<LLMObsInferenceContent>",
    },
    id: {
        baseName: "id",
        type: "string",
    },
    role: {
        baseName: "role",
        type: "string",
    },
    toolCalls: {
        baseName: "tool_calls",
        type: "Array<LLMObsInferenceToolCall>",
    },
    toolResults: {
        baseName: "tool_results",
        type: "Array<LLMObsInferenceToolResult>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
