// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsInferenceErrorResponse.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsInferenceErrorResponse = void 0;
/**
 * Error details returned when an inference provider returns an error.
 */
class LLMObsInferenceErrorResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsInferenceErrorResponse.attributeTypeMap;
    }
}
exports.LLMObsInferenceErrorResponse = LLMObsInferenceErrorResponse;
/**
 * @ignore
 */
LLMObsInferenceErrorResponse.attributeTypeMap = {
    message: {
        baseName: "message",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
