// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentationSearchResults.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentationSearchResults = void 0;
/**
 * The matching experimentation entities grouped by type.
 */
class LLMObsExperimentationSearchResults {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentationSearchResults.attributeTypeMap;
    }
}
exports.LLMObsExperimentationSearchResults = LLMObsExperimentationSearchResults;
/**
 * @ignore
 */
LLMObsExperimentationSearchResults.attributeTypeMap = {
    datasetRecords: {
        baseName: "dataset_records",
        type: "Array<LLMObsDatasetRecordDataResponse>",
    },
    datasets: {
        baseName: "datasets",
        type: "Array<LLMObsDatasetDataResponse>",
    },
    experimentRuns: {
        baseName: "experiment_runs",
        type: "Array<LLMObsExperimentRunDataResponse>",
    },
    experiments: {
        baseName: "experiments",
        type: "Array<LLMObsExperimentDataAttributesResponse>",
    },
    projects: {
        baseName: "projects",
        type: "Array<LLMObsProjectDataResponse>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
