// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsIntegrationAccount.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsIntegrationAccount = void 0;
/**
 * A configured account for an LLM provider integration.
 */
class LLMObsIntegrationAccount {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsIntegrationAccount.attributeTypeMap;
    }
}
exports.LLMObsIntegrationAccount = LLMObsIntegrationAccount;
/**
 * @ignore
 */
LLMObsIntegrationAccount.attributeTypeMap = {
    accountId: {
        baseName: "account_id",
        type: "string",
        required: true,
    },
    accountName: {
        baseName: "account_name",
        type: "string",
        required: true,
    },
    accountRegion: {
        baseName: "account_region",
        type: "string",
    },
    azureOpenaiMetadata: {
        baseName: "azure_openai_metadata",
        type: "LLMObsAzureOpenAIMetadata",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    integration: {
        baseName: "integration",
        type: "string",
        required: true,
    },
    vertexAiMetadata: {
        baseName: "vertex_ai_metadata",
        type: "LLMObsVertexAIMetadata",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
