// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsRunStatusResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsRunStatusResponseData = void 0;
/**
 * Data object of an LLM Observability patterns run status response.
 */
class LLMObsPatternsRunStatusResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsRunStatusResponseData.attributeTypeMap;
    }
}
exports.LLMObsPatternsRunStatusResponseData = LLMObsPatternsRunStatusResponseData;
/**
 * @ignore
 */
LLMObsPatternsRunStatusResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsPatternsRunStatusResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsPatternsRunStatusType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
