// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentDataAttributesResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentDataAttributesResponse = void 0;
/**
 * Attributes of an LLM Observability experiment.
 */
class LLMObsExperimentDataAttributesResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentDataAttributesResponse.attributeTypeMap;
    }
}
exports.LLMObsExperimentDataAttributesResponse = LLMObsExperimentDataAttributesResponse;
/**
 * @ignore
 */
LLMObsExperimentDataAttributesResponse.attributeTypeMap = {
    aggregateData: {
        baseName: "aggregate_data",
        type: "{ [key: string]: any; }",
    },
    author: {
        baseName: "author",
        type: "LLMObsExperimentUser",
    },
    config: {
        baseName: "config",
        type: "{ [key: string]: any; }",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    datasetId: {
        baseName: "dataset_id",
        type: "string",
        required: true,
    },
    datasetName: {
        baseName: "dataset_name",
        type: "string",
    },
    datasetVersion: {
        baseName: "dataset_version",
        type: "number",
        format: "int64",
    },
    deletedAt: {
        baseName: "deleted_at",
        type: "Date",
        format: "date-time",
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    error: {
        baseName: "error",
        type: "string",
    },
    experiment: {
        baseName: "experiment",
        type: "string",
    },
    metadata: {
        baseName: "metadata",
        type: "{ [key: string]: any; }",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    parentExperimentId: {
        baseName: "parent_experiment_id",
        type: "string",
    },
    projectId: {
        baseName: "project_id",
        type: "string",
        required: true,
    },
    runCount: {
        baseName: "run_count",
        type: "number",
        format: "int32",
    },
    status: {
        baseName: "status",
        type: "LLMObsExperimentStatus",
    },
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
