// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDatasetBatchUpdateUpdateRecord.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDatasetBatchUpdateUpdateRecord = void 0;
/**
 * A record update payload as part of a batch update on an LLM Observability dataset.
 */
class LLMObsDatasetBatchUpdateUpdateRecord {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDatasetBatchUpdateUpdateRecord.attributeTypeMap;
    }
}
exports.LLMObsDatasetBatchUpdateUpdateRecord = LLMObsDatasetBatchUpdateUpdateRecord;
/**
 * @ignore
 */
LLMObsDatasetBatchUpdateUpdateRecord.attributeTypeMap = {
    expectedOutput: {
        baseName: "expected_output",
        type: "AnyValue",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    input: {
        baseName: "input",
        type: "AnyValue",
    },
    metadata: {
        baseName: "metadata",
        type: "{ [key: string]: any; }",
    },
    tagOperations: {
        baseName: "tag_operations",
        type: "LLMObsDatasetRecordTagOperations",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
