// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDataDeletionResponseAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDataDeletionResponseAttributes = void 0;
/**
 * Attributes of a submitted LLM Observability data deletion request.
 */
class LLMObsDataDeletionResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDataDeletionResponseAttributes.attributeTypeMap;
    }
}
exports.LLMObsDataDeletionResponseAttributes = LLMObsDataDeletionResponseAttributes;
/**
 * @ignore
 */
LLMObsDataDeletionResponseAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    createdBy: {
        baseName: "created_by",
        type: "string",
        required: true,
    },
    fromTime: {
        baseName: "from_time",
        type: "number",
        required: true,
        format: "int64",
    },
    orgId: {
        baseName: "org_id",
        type: "number",
        required: true,
        format: "int64",
    },
    product: {
        baseName: "product",
        type: "string",
        required: true,
    },
    query: {
        baseName: "query",
        type: "string",
        required: true,
    },
    toTime: {
        baseName: "to_time",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
