// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDisplayBlockInteractionItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDisplayBlockInteractionItem = void 0;
/**
 * An interaction whose rendered content is supplied directly as a list
 * of display blocks. The server generates `content_id` deterministically
 * from the block list.
 */
class LLMObsDisplayBlockInteractionItem {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDisplayBlockInteractionItem.attributeTypeMap;
    }
}
exports.LLMObsDisplayBlockInteractionItem = LLMObsDisplayBlockInteractionItem;
/**
 * @ignore
 */
LLMObsDisplayBlockInteractionItem.attributeTypeMap = {
    displayBlock: {
        baseName: "display_block",
        type: "Array<LLMObsContentBlock>",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsDisplayBlockInteractionType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
