// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDeleteAnnotationError.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDeleteAnnotationError = void 0;
/**
 * A partial error for a single annotation that could not be deleted.
 */
class LLMObsDeleteAnnotationError {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDeleteAnnotationError.attributeTypeMap;
    }
}
exports.LLMObsDeleteAnnotationError = LLMObsDeleteAnnotationError;
/**
 * @ignore
 */
LLMObsDeleteAnnotationError.attributeTypeMap = {
    annotationId: {
        baseName: "annotation_id",
        type: "string",
        required: true,
    },
    error: {
        baseName: "error",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
