// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsSearchSpansRequestAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsSearchSpansRequestAttributes = void 0;
/**
 * Attributes of an LLM Observability spans search request.
 */
class LLMObsSearchSpansRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsSearchSpansRequestAttributes.attributeTypeMap;
    }
}
exports.LLMObsSearchSpansRequestAttributes = LLMObsSearchSpansRequestAttributes;
/**
 * @ignore
 */
LLMObsSearchSpansRequestAttributes.attributeTypeMap = {
    filter: {
        baseName: "filter",
        type: "LLMObsSpanFilter",
    },
    options: {
        baseName: "options",
        type: "LLMObsSpanSearchOptions",
    },
    page: {
        baseName: "page",
        type: "LLMObsSpanPageQuery",
    },
    sort: {
        baseName: "sort",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
