// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsTriggerRequestAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsTriggerRequestAttributes = void 0;
/**
 * Attributes for triggering an LLM Observability patterns run.
 */
class LLMObsPatternsTriggerRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsTriggerRequestAttributes.attributeTypeMap;
    }
}
exports.LLMObsPatternsTriggerRequestAttributes = LLMObsPatternsTriggerRequestAttributes;
/**
 * @ignore
 */
LLMObsPatternsTriggerRequestAttributes.attributeTypeMap = {
    configId: {
        baseName: "config_id",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
