// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsTriggerResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsTriggerResponseData = void 0;
/**
 * Data object of an LLM Observability patterns trigger response.
 */
class LLMObsPatternsTriggerResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsTriggerResponseData.attributeTypeMap;
    }
}
exports.LLMObsPatternsTriggerResponseData = LLMObsPatternsTriggerResponseData;
/**
 * @ignore
 */
LLMObsPatternsTriggerResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsPatternsTriggerResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsPatternsTriggerResponseType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
