// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDatasetBatchUpdateDataRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDatasetBatchUpdateDataRequest = void 0;
/**
 * Data object for batch-updating records in an LLM Observability dataset.
 */
class LLMObsDatasetBatchUpdateDataRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDatasetBatchUpdateDataRequest.attributeTypeMap;
    }
}
exports.LLMObsDatasetBatchUpdateDataRequest = LLMObsDatasetBatchUpdateDataRequest;
/**
 * @ignore
 */
LLMObsDatasetBatchUpdateDataRequest.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsDatasetBatchUpdateDataAttributesRequest",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsDatasetType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
