// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsConfigsResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsConfigsResponseData = void 0;
/**
 * Data object of a list of LLM Observability patterns configurations.
 */
class LLMObsPatternsConfigsResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsConfigsResponseData.attributeTypeMap;
    }
}
exports.LLMObsPatternsConfigsResponseData = LLMObsPatternsConfigsResponseData;
/**
 * @ignore
 */
LLMObsPatternsConfigsResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsPatternsConfigsResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsPatternsConfigsListType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
