// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsAnnotationLabelValueResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsAnnotationLabelValueResponse = void 0;
/**
 * A single label value entry in an annotation response.
 * In addition to the submitted fields, the server populates `type` and
 * `name_when_saved` to mirror the schema state at the time the annotation
 * was created — these help clients display values correctly when the schema
 * has since changed.
 */
class LLMObsAnnotationLabelValueResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsAnnotationLabelValueResponse.attributeTypeMap;
    }
}
exports.LLMObsAnnotationLabelValueResponse = LLMObsAnnotationLabelValueResponse;
/**
 * @ignore
 */
LLMObsAnnotationLabelValueResponse.attributeTypeMap = {
    assessment: {
        baseName: "assessment",
        type: "LLMObsAnnotationAssessment",
    },
    labelSchemaId: {
        baseName: "label_schema_id",
        type: "string",
        required: true,
    },
    nameWhenSaved: {
        baseName: "name_when_saved",
        type: "string",
    },
    reasoning: {
        baseName: "reasoning",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "LLMObsLabelSchemaType",
    },
    value: {
        baseName: "value",
        type: "LLMObsAnnotationLabelValueValue",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
