// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsInferenceCode.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsInferenceCode = void 0;
/**
 * A generated code snippet for running an inference request programmatically.
 */
class LLMObsInferenceCode {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsInferenceCode.attributeTypeMap;
    }
}
exports.LLMObsInferenceCode = LLMObsInferenceCode;
/**
 * @ignore
 */
LLMObsInferenceCode.attributeTypeMap = {
    code: {
        baseName: "code",
        type: "string",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
