// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsTriggerResponseAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsTriggerResponseAttributes = void 0;
/**
 * Attributes of an LLM Observability patterns trigger response.
 */
class LLMObsPatternsTriggerResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsTriggerResponseAttributes.attributeTypeMap;
    }
}
exports.LLMObsPatternsTriggerResponseAttributes = LLMObsPatternsTriggerResponseAttributes;
/**
 * @ignore
 */
LLMObsPatternsTriggerResponseAttributes.attributeTypeMap = {
    configId: {
        baseName: "config_id",
        type: "string",
        required: true,
    },
    runId: {
        baseName: "run_id",
        type: "string",
        required: true,
    },
    status: {
        baseName: "status",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
