// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsContentBlockTimeFrame.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsContentBlockTimeFrame = void 0;
/**
 * Unix-millis time range used by chart blocks.
 */
class LLMObsContentBlockTimeFrame {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsContentBlockTimeFrame.attributeTypeMap;
    }
}
exports.LLMObsContentBlockTimeFrame = LLMObsContentBlockTimeFrame;
/**
 * @ignore
 */
LLMObsContentBlockTimeFrame.attributeTypeMap = {
    end: {
        baseName: "end",
        type: "number",
        required: true,
        format: "int64",
    },
    start: {
        baseName: "start",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
