// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsTraceInteractionResponseItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsTraceInteractionResponseItem = void 0;
/**
 * A trace, experiment trace, or session interaction result.
 */
class LLMObsTraceInteractionResponseItem {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsTraceInteractionResponseItem.attributeTypeMap;
    }
}
exports.LLMObsTraceInteractionResponseItem = LLMObsTraceInteractionResponseItem;
/**
 * @ignore
 */
LLMObsTraceInteractionResponseItem.attributeTypeMap = {
    alreadyExisted: {
        baseName: "already_existed",
        type: "boolean",
        required: true,
    },
    contentId: {
        baseName: "content_id",
        type: "string",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    type: {
        baseName: "type",
        type: "LLMObsTraceInteractionType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
