// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsAnnotatedInteractionsByTraceDataResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsAnnotatedInteractionsByTraceDataResponse = void 0;
/**
 * Data object for the cross-queue annotated interactions response.
 */
class LLMObsAnnotatedInteractionsByTraceDataResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsAnnotatedInteractionsByTraceDataResponse.attributeTypeMap;
    }
}
exports.LLMObsAnnotatedInteractionsByTraceDataResponse = LLMObsAnnotatedInteractionsByTraceDataResponse;
/**
 * @ignore
 */
LLMObsAnnotatedInteractionsByTraceDataResponse.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsAnnotatedInteractionsByTraceDataAttributesResponse",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsAnnotatedInteractionsByTraceType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
