// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsClusteredPointsResponseAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsClusteredPointsResponseAttributes = void 0;
/**
 * Attributes of an LLM Observability patterns clustered points response.
 */
class LLMObsPatternsClusteredPointsResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsClusteredPointsResponseAttributes.attributeTypeMap;
    }
}
exports.LLMObsPatternsClusteredPointsResponseAttributes = LLMObsPatternsClusteredPointsResponseAttributes;
/**
 * @ignore
 */
LLMObsPatternsClusteredPointsResponseAttributes.attributeTypeMap = {
    nextPageToken: {
        baseName: "next_page_token",
        type: "string",
        required: true,
    },
    points: {
        baseName: "points",
        type: "Array<LLMObsPatternsClusteredPoint>",
        required: true,
    },
    topicId: {
        baseName: "topic_id",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
