// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDataDeletionRequestAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDataDeletionRequestAttributes = void 0;
/**
 * Attributes for an LLM Observability data deletion request.
 */
class LLMObsDataDeletionRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDataDeletionRequestAttributes.attributeTypeMap;
    }
}
exports.LLMObsDataDeletionRequestAttributes = LLMObsDataDeletionRequestAttributes;
/**
 * @ignore
 */
LLMObsDataDeletionRequestAttributes.attributeTypeMap = {
    delay: {
        baseName: "delay",
        type: "number",
        format: "int64",
    },
    from: {
        baseName: "from",
        type: "number",
        required: true,
        format: "int64",
    },
    query: {
        baseName: "query",
        type: "{ [key: string]: string; }",
        required: true,
    },
    to: {
        baseName: "to",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
