// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsDatasetCloneDataAttributesRequest.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsDatasetCloneDataAttributesRequest = void 0;
/**
 * Attributes for cloning an LLM Observability dataset.
 */
class LLMObsDatasetCloneDataAttributesRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsDatasetCloneDataAttributesRequest.attributeTypeMap;
    }
}
exports.LLMObsDatasetCloneDataAttributesRequest = LLMObsDatasetCloneDataAttributesRequest;
/**
 * @ignore
 */
LLMObsDatasetCloneDataAttributesRequest.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
