// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsTopicsResponseAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsTopicsResponseAttributes = void 0;
/**
 * Attributes of an LLM Observability patterns topics response.
 */
class LLMObsPatternsTopicsResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsTopicsResponseAttributes.attributeTypeMap;
    }
}
exports.LLMObsPatternsTopicsResponseAttributes = LLMObsPatternsTopicsResponseAttributes;
/**
 * @ignore
 */
LLMObsPatternsTopicsResponseAttributes.attributeTypeMap = {
    completedAt: {
        baseName: "completed_at",
        type: "Date",
        format: "date-time",
    },
    configId: {
        baseName: "config_id",
        type: "string",
        required: true,
    },
    configSnapshot: {
        baseName: "config_snapshot",
        type: "LLMObsPatternsConfigSnapshot",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    previousRunId: {
        baseName: "previous_run_id",
        type: "string",
        required: true,
    },
    runId: {
        baseName: "run_id",
        type: "string",
        required: true,
    },
    topics: {
        baseName: "topics",
        type: "Array<LLMObsPatternsTopic>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
