// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentationAnalyticsDataAttributesResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentationAnalyticsDataAttributesResponse = void 0;
/**
 * Attributes of an analytics response.
 */
class LLMObsExperimentationAnalyticsDataAttributesResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentationAnalyticsDataAttributesResponse.attributeTypeMap;
    }
}
exports.LLMObsExperimentationAnalyticsDataAttributesResponse = LLMObsExperimentationAnalyticsDataAttributesResponse;
/**
 * @ignore
 */
LLMObsExperimentationAnalyticsDataAttributesResponse.attributeTypeMap = {
    hitCount: {
        baseName: "hit_count",
        type: "number",
        required: true,
        format: "int64",
    },
    result: {
        baseName: "result",
        type: "LLMObsExperimentationAnalyticsResult",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
