// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsContentBlock.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsContentBlock = void 0;
/**
 * A single content block rendered inside a `display_block` interaction.
 * `type` discriminates which other fields are meaningful:
 *
 * - `markdown` / `text`: `content` must be a string.
 * - `header`: `content` must be a string; `level`, when set, must be one of `sm`, `md`, `lg`, `xl`.
 * - `json`: `content` must be a well-formed JSON value (object, array, or scalar).
 * - `image`: `url` is required.
 * - `widget`: `tileDef` is required (any well-formed JSON; the frontend owns the renderable schema).
 * - `llmobs_trace`: `traceId` is required; `interactionType`, when set, must be `trace` or `experiment_trace`.
 *
 * `height`, when set, must be positive.
 */
class LLMObsContentBlock {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsContentBlock.attributeTypeMap;
    }
}
exports.LLMObsContentBlock = LLMObsContentBlock;
/**
 * @ignore
 */
LLMObsContentBlock.attributeTypeMap = {
    alt: {
        baseName: "alt",
        type: "string",
    },
    content: {
        baseName: "content",
        type: "any",
    },
    height: {
        baseName: "height",
        type: "number",
        format: "int64",
    },
    interactionType: {
        baseName: "interactionType",
        type: "LLMObsContentBlockLLMObsTraceInteractionType",
    },
    label: {
        baseName: "label",
        type: "string",
    },
    level: {
        baseName: "level",
        type: "LLMObsContentBlockHeaderLevel",
    },
    tileDef: {
        baseName: "tileDef",
        type: "any",
    },
    timeFrame: {
        baseName: "timeFrame",
        type: "LLMObsContentBlockTimeFrame",
    },
    traceId: {
        baseName: "traceId",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "LLMObsContentBlockType",
        required: true,
    },
    url: {
        baseName: "url",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
