// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsPatternsTopicWithClusteredPoints.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsPatternsTopicWithClusteredPoints = void 0;
/**
 * A topic discovered by an LLM Observability patterns run, including the
 * clustered points attached to leaf topics.
 */
class LLMObsPatternsTopicWithClusteredPoints {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsPatternsTopicWithClusteredPoints.attributeTypeMap;
    }
}
exports.LLMObsPatternsTopicWithClusteredPoints = LLMObsPatternsTopicWithClusteredPoints;
/**
 * @ignore
 */
LLMObsPatternsTopicWithClusteredPoints.attributeTypeMap = {
    clusterPoints: {
        baseName: "cluster_points",
        type: "Array<LLMObsPatternsClusteredPointRef>",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    firstSeenAt: {
        baseName: "first_seen_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    hierarchyLevel: {
        baseName: "hierarchy_level",
        type: "number",
        required: true,
        format: "int64",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    isValidated: {
        baseName: "is_validated",
        type: "boolean",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    parentTopicId: {
        baseName: "parent_topic_id",
        type: "string",
        required: true,
    },
    pointCount: {
        baseName: "point_count",
        type: "number",
        required: true,
        format: "int64",
    },
    runId: {
        baseName: "run_id",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
