// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentationAnalyticsTimeRange.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentationAnalyticsTimeRange = void 0;
/**
 * Unix-millisecond time range for filtering analytics data.
 */
class LLMObsExperimentationAnalyticsTimeRange {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentationAnalyticsTimeRange.attributeTypeMap;
    }
}
exports.LLMObsExperimentationAnalyticsTimeRange = LLMObsExperimentationAnalyticsTimeRange;
/**
 * @ignore
 */
LLMObsExperimentationAnalyticsTimeRange.attributeTypeMap = {
    from: {
        baseName: "from",
        type: "number",
        required: true,
        format: "int64",
    },
    to: {
        baseName: "to",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
