// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsIntegrationModel.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsIntegrationModel = void 0;
/**
 * A model available for a given LLM provider integration and account.
 */
class LLMObsIntegrationModel {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsIntegrationModel.attributeTypeMap;
    }
}
exports.LLMObsIntegrationModel = LLMObsIntegrationModel;
/**
 * @ignore
 */
LLMObsIntegrationModel.attributeTypeMap = {
    hasAccess: {
        baseName: "has_access",
        type: "boolean",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    integration: {
        baseName: "integration",
        type: "string",
        required: true,
    },
    integrationDisplayName: {
        baseName: "integration_display_name",
        type: "string",
        required: true,
    },
    jsonSchema: {
        baseName: "json_schema",
        type: "boolean",
        required: true,
    },
    modelDisplayName: {
        baseName: "model_display_name",
        type: "string",
        required: true,
    },
    modelId: {
        baseName: "model_id",
        type: "string",
        required: true,
    },
    provider: {
        baseName: "provider",
        type: "string",
        required: true,
    },
    providerDisplayName: {
        baseName: "provider_display_name",
        type: "string",
        required: true,
    },
    regionPrefixOverrides: {
        baseName: "region_prefix_overrides",
        type: "{ [key: string]: string; }",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
