// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentationSimpleSearchDataAttributesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentationSimpleSearchDataAttributesRequest = void 0;
/**
 * Attributes for an experimentation simple search request.
 */
class LLMObsExperimentationSimpleSearchDataAttributesRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentationSimpleSearchDataAttributesRequest.attributeTypeMap;
    }
}
exports.LLMObsExperimentationSimpleSearchDataAttributesRequest = LLMObsExperimentationSimpleSearchDataAttributesRequest;
/**
 * @ignore
 */
LLMObsExperimentationSimpleSearchDataAttributesRequest.attributeTypeMap = {
    contentPreview: {
        baseName: "content_preview",
        type: "LLMObsExperimentationContentPreview",
    },
    filter: {
        baseName: "filter",
        type: "LLMObsExperimentationFilter",
        required: true,
    },
    include: {
        baseName: "include",
        type: "LLMObsExperimentationInclude",
    },
    page: {
        baseName: "page",
        type: "LLMObsExperimentationNumberPage",
    },
    sort: {
        baseName: "sort",
        type: "Array<LLMObsExperimentationSortField>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
