// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsAnnotationItemResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsAnnotationItemResponse = void 0;
/**
 * A single annotation on an interaction, as returned by the API.
 */
class LLMObsAnnotationItemResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsAnnotationItemResponse.attributeTypeMap;
    }
}
exports.LLMObsAnnotationItemResponse = LLMObsAnnotationItemResponse;
/**
 * @ignore
 */
LLMObsAnnotationItemResponse.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    createdBy: {
        baseName: "created_by",
        type: "string",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    interactionId: {
        baseName: "interaction_id",
        type: "string",
        required: true,
    },
    labelValues: {
        baseName: "label_values",
        type: "Array<LLMObsAnnotationLabelValueResponse>",
        required: true,
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    modifiedBy: {
        baseName: "modified_by",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
