// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LLMObsExperimentSpanDataResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LLMObsExperimentSpanDataResponse = void 0;
/**
 * JSON:API data item wrapping a single experiment span with evaluations.
 */
class LLMObsExperimentSpanDataResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LLMObsExperimentSpanDataResponse.attributeTypeMap;
    }
}
exports.LLMObsExperimentSpanDataResponse = LLMObsExperimentSpanDataResponse;
/**
 * @ignore
 */
LLMObsExperimentSpanDataResponse.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "LLMObsExperimentSpanWithEvals",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "LLMObsExperimentSpanType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
