// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LongTaskStatsPerView.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LongTaskStatsPerView = void 0;
/**
 * Statistical distributions of long task metrics computed per view across sampled views.
 */
class LongTaskStatsPerView {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LongTaskStatsPerView.attributeTypeMap;
    }
}
exports.LongTaskStatsPerView = LongTaskStatsPerView;
/**
 * @ignore
 */
LongTaskStatsPerView.attributeTypeMap = {
    fcpBlockingTimeMs: {
        baseName: "fcp_blocking_time_ms",
        type: "LongTaskMetricStats",
    },
    fcpCount: {
        baseName: "fcp_count",
        type: "LongTaskMetricStats",
    },
    inpOverlapBlockingTimeMs: {
        baseName: "inp_overlap_blocking_time_ms",
        type: "LongTaskMetricStats",
    },
    inpOverlapCount: {
        baseName: "inp_overlap_count",
        type: "LongTaskMetricStats",
    },
    lcpBlockingTimeMs: {
        baseName: "lcp_blocking_time_ms",
        type: "LongTaskMetricStats",
    },
    lcpCount: {
        baseName: "lcp_count",
        type: "LongTaskMetricStats",
    },
    loadingTimeBlockingTimeMs: {
        baseName: "loading_time_blocking_time_ms",
        type: "LongTaskMetricStats",
    },
    loadingTimeCount: {
        baseName: "loading_time_count",
        type: "LongTaskMetricStats",
    },
    totalBlockingTimeMs: {
        baseName: "total_blocking_time_ms",
        type: "LongTaskMetricStats",
    },
    totalCount: {
        baseName: "total_count",
        type: "LongTaskMetricStats",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
