// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/LongTaskMetricStats.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LongTaskMetricStats = void 0;
/**
 * Statistical distribution (average, min, max) of a long task metric across sampled views.
 */
class LongTaskMetricStats {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return LongTaskMetricStats.attributeTypeMap;
    }
}
exports.LongTaskMetricStats = LongTaskMetricStats;
/**
 * @ignore
 */
LongTaskMetricStats.attributeTypeMap = {
    average: {
        baseName: "average",
        type: "number",
        required: true,
        format: "double",
    },
    max: {
        baseName: "max",
        type: "number",
        required: true,
        format: "double",
    },
    min: {
        baseName: "min",
        type: "number",
        required: true,
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
