// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/MaintenanceWindow.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaintenanceWindow = void 0;
/**
 * A maintenance window that defines a scheduled time period during which case-related notifications and automation rules are suppressed. Each maintenance window applies to cases matching a specified query.
 */
class MaintenanceWindow {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MaintenanceWindow.attributeTypeMap;
    }
}
exports.MaintenanceWindow = MaintenanceWindow;
/**
 * @ignore
 */
MaintenanceWindow.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "MaintenanceWindowAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "MaintenanceWindowResourceType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
