// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/MaintenanceWindowAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaintenanceWindowAttributes = void 0;
/**
 * Attributes of a maintenance window, including its schedule and the query that determines which cases are affected.
 */
class MaintenanceWindowAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MaintenanceWindowAttributes.attributeTypeMap;
    }
}
exports.MaintenanceWindowAttributes = MaintenanceWindowAttributes;
/**
 * @ignore
 */
MaintenanceWindowAttributes.attributeTypeMap = {
    createdBy: {
        baseName: "created_by",
        type: "string",
    },
    endAt: {
        baseName: "end_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    query: {
        baseName: "query",
        type: "string",
        required: true,
    },
    startAt: {
        baseName: "start_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    updatedBy: {
        baseName: "updated_by",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
