// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/MaintenanceWindowUpdateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaintenanceWindowUpdateAttributes = void 0;
/**
 * Attributes that can be updated on a maintenance window. All fields are optional; only provided fields are changed.
 */
class MaintenanceWindowUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MaintenanceWindowUpdateAttributes.attributeTypeMap;
    }
}
exports.MaintenanceWindowUpdateAttributes = MaintenanceWindowUpdateAttributes;
/**
 * @ignore
 */
MaintenanceWindowUpdateAttributes.attributeTypeMap = {
    endAt: {
        baseName: "end_at",
        type: "Date",
        format: "date-time",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    query: {
        baseName: "query",
        type: "string",
    },
    startAt: {
        baseName: "start_at",
        type: "Date",
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
