// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/MaintenanceWindowCreateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaintenanceWindowCreateAttributes = void 0;
/**
 * Attributes required to create a maintenance window.
 */
class MaintenanceWindowCreateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MaintenanceWindowCreateAttributes.attributeTypeMap;
    }
}
exports.MaintenanceWindowCreateAttributes = MaintenanceWindowCreateAttributes;
/**
 * @ignore
 */
MaintenanceWindowCreateAttributes.attributeTypeMap = {
    endAt: {
        baseName: "end_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    query: {
        baseName: "query",
        type: "string",
        required: true,
    },
    startAt: {
        baseName: "start_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
