// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/MaxSessionDurationUpdateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaxSessionDurationUpdateAttributes = void 0;
/**
 * Attributes for the maximum session duration update request.
 */
class MaxSessionDurationUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MaxSessionDurationUpdateAttributes.attributeTypeMap;
    }
}
exports.MaxSessionDurationUpdateAttributes = MaxSessionDurationUpdateAttributes;
/**
 * @ignore
 */
MaxSessionDurationUpdateAttributes.attributeTypeMap = {
    maxSessionDuration: {
        baseName: "max_session_duration",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
