// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/McpScanRequestDataAttributesLibrariesItems.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.McpScanRequestDataAttributesLibrariesItems = void 0;
/**
 * A library declaration to include in the dependency scan.
 */
class McpScanRequestDataAttributesLibrariesItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return McpScanRequestDataAttributesLibrariesItems.attributeTypeMap;
    }
}
exports.McpScanRequestDataAttributesLibrariesItems = McpScanRequestDataAttributesLibrariesItems;
/**
 * @ignore
 */
McpScanRequestDataAttributesLibrariesItems.attributeTypeMap = {
    exclusions: {
        baseName: "exclusions",
        type: "Array<string>",
    },
    isDev: {
        baseName: "is_dev",
        type: "boolean",
        required: true,
    },
    isDirect: {
        baseName: "is_direct",
        type: "boolean",
        required: true,
    },
    packageManager: {
        baseName: "package_manager",
        type: "string",
        required: true,
    },
    purl: {
        baseName: "purl",
        type: "string",
        required: true,
    },
    targetFrameworks: {
        baseName: "target_frameworks",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
