// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/McpScanRequestDataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.McpScanRequestDataAttributes = void 0;
/**
 * The attributes of an MCP SCA scan request, describing the libraries to scan and their context.
 */
class McpScanRequestDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return McpScanRequestDataAttributes.attributeTypeMap;
    }
}
exports.McpScanRequestDataAttributes = McpScanRequestDataAttributes;
/**
 * @ignore
 */
McpScanRequestDataAttributes.attributeTypeMap = {
    commitHash: {
        baseName: "commit_hash",
        type: "string",
        required: true,
    },
    libraries: {
        baseName: "libraries",
        type: "Array<McpScanRequestDataAttributesLibrariesItems>",
        required: true,
    },
    resourceName: {
        baseName: "resource_name",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
