// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ModelLabNumericRange.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelLabNumericRange = void 0;
/**
 * The numeric range of values for a facet.
 */
class ModelLabNumericRange {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ModelLabNumericRange.attributeTypeMap;
    }
}
exports.ModelLabNumericRange = ModelLabNumericRange;
/**
 * @ignore
 */
ModelLabNumericRange.attributeTypeMap = {
    max: {
        baseName: "max",
        type: "number",
        required: true,
        format: "double",
    },
    min: {
        baseName: "min",
        type: "number",
        required: true,
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
