// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ModelLabArtifactInfo.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelLabArtifactInfo = void 0;
/**
 * Information about a project-level artifact file.
 */
class ModelLabArtifactInfo {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ModelLabArtifactInfo.attributeTypeMap;
    }
}
exports.ModelLabArtifactInfo = ModelLabArtifactInfo;
/**
 * @ignore
 */
ModelLabArtifactInfo.attributeTypeMap = {
    artifactPath: {
        baseName: "artifact_path",
        type: "string",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    fileSize: {
        baseName: "file_size",
        type: "number",
        format: "int64",
    },
    filename: {
        baseName: "filename",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
