// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ModelLabMetricSummary.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelLabMetricSummary = void 0;
/**
 * Summary statistics for a metric recorded during a Model Lab run.
 */
class ModelLabMetricSummary {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ModelLabMetricSummary.attributeTypeMap;
    }
}
exports.ModelLabMetricSummary = ModelLabMetricSummary;
/**
 * @ignore
 */
ModelLabMetricSummary.attributeTypeMap = {
    count: {
        baseName: "count",
        type: "number",
        required: true,
        format: "int64",
    },
    firstStep: {
        baseName: "first_step",
        type: "number",
        format: "int64",
    },
    key: {
        baseName: "key",
        type: "string",
        required: true,
    },
    lastStep: {
        baseName: "last_step",
        type: "number",
        format: "int64",
    },
    latest: {
        baseName: "latest",
        type: "number",
        format: "double",
    },
    max: {
        baseName: "max",
        type: "number",
        format: "double",
    },
    mean: {
        baseName: "mean",
        type: "number",
        format: "double",
    },
    min: {
        baseName: "min",
        type: "number",
        format: "double",
    },
    stddev: {
        baseName: "stddev",
        type: "number",
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
