// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ModelLabFacetKeysAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelLabFacetKeysAttributes = void 0;
/**
 * Available facet key names for filtering resources.
 */
class ModelLabFacetKeysAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ModelLabFacetKeysAttributes.attributeTypeMap;
    }
}
exports.ModelLabFacetKeysAttributes = ModelLabFacetKeysAttributes;
/**
 * @ignore
 */
ModelLabFacetKeysAttributes.attributeTypeMap = {
    metrics: {
        baseName: "metrics",
        type: "Array<string>",
        required: true,
    },
    parameters: {
        baseName: "parameters",
        type: "Array<string>",
        required: true,
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
