// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ModelLabMetricStatRange.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelLabMetricStatRange = void 0;
/**
 * The range of values for a specific metric statistic.
 */
class ModelLabMetricStatRange {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ModelLabMetricStatRange.attributeTypeMap;
    }
}
exports.ModelLabMetricStatRange = ModelLabMetricStatRange;
/**
 * @ignore
 */
ModelLabMetricStatRange.attributeTypeMap = {
    max: {
        baseName: "max",
        type: "number",
        required: true,
        format: "double",
    },
    min: {
        baseName: "min",
        type: "number",
        required: true,
        format: "double",
    },
    stat: {
        baseName: "stat",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
