// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ModelLabProjectAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelLabProjectAttributes = void 0;
/**
 * Attributes of a Model Lab project.
 */
class ModelLabProjectAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ModelLabProjectAttributes.attributeTypeMap;
    }
}
exports.ModelLabProjectAttributes = ModelLabProjectAttributes;
/**
 * @ignore
 */
ModelLabProjectAttributes.attributeTypeMap = {
    artifactStorageLocation: {
        baseName: "artifact_storage_location",
        type: "string",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    deletedAt: {
        baseName: "deleted_at",
        type: "Date",
        format: "date-time",
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    externalUrl: {
        baseName: "external_url",
        type: "string",
    },
    isStarred: {
        baseName: "is_starred",
        type: "boolean",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    ownerId: {
        baseName: "owner_id",
        type: "string",
    },
    tags: {
        baseName: "tags",
        type: "Array<ModelLabTag>",
        required: true,
    },
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
