// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ModelLabRunAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelLabRunAttributes = void 0;
/**
 * Attributes of a Model Lab run.
 */
class ModelLabRunAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ModelLabRunAttributes.attributeTypeMap;
    }
}
exports.ModelLabRunAttributes = ModelLabRunAttributes;
/**
 * @ignore
 */
ModelLabRunAttributes.attributeTypeMap = {
    completedAt: {
        baseName: "completed_at",
        type: "Date",
        format: "date-time",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    deletedAt: {
        baseName: "deleted_at",
        type: "Date",
        format: "date-time",
    },
    descendantMatch: {
        baseName: "descendant_match",
        type: "boolean",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    duration: {
        baseName: "duration",
        type: "number",
        format: "double",
    },
    externalUrl: {
        baseName: "external_url",
        type: "string",
    },
    hasChildren: {
        baseName: "has_children",
        type: "boolean",
        required: true,
    },
    isPinned: {
        baseName: "is_pinned",
        type: "boolean",
        required: true,
    },
    metricSummaries: {
        baseName: "metric_summaries",
        type: "Array<ModelLabMetricSummary>",
        required: true,
    },
    mlflowArtifactLocation: {
        baseName: "mlflow_artifact_location",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    ownerId: {
        baseName: "owner_id",
        type: "string",
    },
    params: {
        baseName: "params",
        type: "Array<ModelLabRunParam>",
        required: true,
    },
    projectId: {
        baseName: "project_id",
        type: "number",
        required: true,
        format: "int64",
    },
    startedAt: {
        baseName: "started_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    status: {
        baseName: "status",
        type: "ModelLabRunStatus",
        required: true,
    },
    tags: {
        baseName: "tags",
        type: "Array<ModelLabTag>",
        required: true,
    },
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
