// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ModelLabArtifactObjectInfo.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelLabArtifactObjectInfo = void 0;
/**
 * Information about an artifact file or directory within a run.
 */
class ModelLabArtifactObjectInfo {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ModelLabArtifactObjectInfo.attributeTypeMap;
    }
}
exports.ModelLabArtifactObjectInfo = ModelLabArtifactObjectInfo;
/**
 * @ignore
 */
ModelLabArtifactObjectInfo.attributeTypeMap = {
    fileSize: {
        baseName: "file_size",
        type: "number",
        format: "int64",
    },
    isDir: {
        baseName: "is_dir",
        type: "boolean",
        required: true,
    },
    path: {
        baseName: "path",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
