// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ModelLabFacetValuesAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelLabFacetValuesAttributes = void 0;
/**
 * Available values for a specific facet key.
 */
class ModelLabFacetValuesAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ModelLabFacetValuesAttributes.attributeTypeMap;
    }
}
exports.ModelLabFacetValuesAttributes = ModelLabFacetValuesAttributes;
/**
 * @ignore
 */
ModelLabFacetValuesAttributes.attributeTypeMap = {
    facetName: {
        baseName: "facet_name",
        type: "string",
        required: true,
    },
    facetType: {
        baseName: "facet_type",
        type: "string",
        required: true,
    },
    metricStatRanges: {
        baseName: "metric_stat_ranges",
        type: "Array<ModelLabMetricStatRange>",
    },
    numericRange: {
        baseName: "numeric_range",
        type: "ModelLabNumericRange",
    },
    values: {
        baseName: "values",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
