// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/MuteRuleAttributesResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MuteRuleAttributesResponse = void 0;
/**
 * Attributes of a mute rule returned by the API.
 */
class MuteRuleAttributesResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return MuteRuleAttributesResponse.attributeTypeMap;
    }
}
exports.MuteRuleAttributesResponse = MuteRuleAttributesResponse;
/**
 * @ignore
 */
MuteRuleAttributesResponse.attributeTypeMap = {
    action: {
        baseName: "action",
        type: "MuteRuleAction",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "number",
        required: true,
        format: "int64",
    },
    createdBy: {
        baseName: "created_by",
        type: "AutomationRuleCreatedBy",
        required: true,
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
        required: true,
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "number",
        required: true,
        format: "int64",
    },
    modifiedBy: {
        baseName: "modified_by",
        type: "AutomationRuleModifiedBy",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    rule: {
        baseName: "rule",
        type: "AutomationRuleScope",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
