// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/NDKSourcemapAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NDKSourcemapAttributes = void 0;
/**
 * Attributes of an Android NDK symbol file.
 */
class NDKSourcemapAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return NDKSourcemapAttributes.attributeTypeMap;
    }
}
exports.NDKSourcemapAttributes = NDKSourcemapAttributes;
/**
 * @ignore
 */
NDKSourcemapAttributes.attributeTypeMap = {
    arch: {
        baseName: "arch",
        type: "string",
    },
    buildId: {
        baseName: "build_id",
        type: "string",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    fileName: {
        baseName: "file_name",
        type: "string",
    },
    mapkind: {
        baseName: "mapkind",
        type: "string",
        required: true,
    },
    size: {
        baseName: "size",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
