// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/NetworkHealthInsightAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkHealthInsightAttributes = void 0;
/**
 * Detailed attributes of a network health insight.
 */
class NetworkHealthInsightAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return NetworkHealthInsightAttributes.attributeTypeMap;
    }
}
exports.NetworkHealthInsightAttributes = NetworkHealthInsightAttributes;
/**
 * @ignore
 */
NetworkHealthInsightAttributes.attributeTypeMap = {
    accountId: {
        baseName: "account_id",
        type: "string",
    },
    certificateId: {
        baseName: "certificate_id",
        type: "string",
    },
    certificateLifetimePercent: {
        baseName: "certificate_lifetime_percent",
        type: "number",
        format: "double",
    },
    clientRegion: {
        baseName: "client_region",
        type: "string",
    },
    clientService: {
        baseName: "client_service",
        type: "string",
    },
    daysUntilExpiration: {
        baseName: "days_until_expiration",
        type: "number",
        format: "int64",
    },
    dnsQuery: {
        baseName: "dns_query",
        type: "string",
    },
    dnsServer: {
        baseName: "dns_server",
        type: "string",
    },
    domainName: {
        baseName: "domain_name",
        type: "string",
    },
    failureMagnitude: {
        baseName: "failure_magnitude",
        type: "number",
        format: "int64",
    },
    failureRate: {
        baseName: "failure_rate",
        type: "number",
        format: "double",
    },
    failureType: {
        baseName: "failure_type",
        type: "NetworkHealthInsightFailureType",
    },
    loadbalancerId: {
        baseName: "loadbalancer_id",
        type: "string",
    },
    serverRegion: {
        baseName: "server_region",
        type: "string",
    },
    serverService: {
        baseName: "server_service",
        type: "string",
    },
    totalRequests: {
        baseName: "total_requests",
        type: "number",
        format: "int64",
    },
    trafficVolume: {
        baseName: "traffic_volume",
        type: "NetworkHealthInsightTrafficVolume",
    },
    type: {
        baseName: "type",
        type: "NetworkHealthInsightCategory",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
