// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/NetworkHealthInsightTrafficVolume.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkHealthInsightTrafficVolume = void 0;
/**
 * Network traffic volume metrics between the client and server services during the query window.
 */
class NetworkHealthInsightTrafficVolume {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return NetworkHealthInsightTrafficVolume.attributeTypeMap;
    }
}
exports.NetworkHealthInsightTrafficVolume = NetworkHealthInsightTrafficVolume;
/**
 * @ignore
 */
NetworkHealthInsightTrafficVolume.attributeTypeMap = {
    bytesRead: {
        baseName: "bytes_read",
        type: "number",
        format: "int64",
    },
    bytesWritten: {
        baseName: "bytes_written",
        type: "number",
        format: "int64",
    },
    totalTraffic: {
        baseName: "total_traffic",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
