// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OAuthScopesRestrictionResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OAuthScopesRestrictionResponseData = void 0;
/**
 * Data object of an OAuth2 client scopes restriction response.
 */
class OAuthScopesRestrictionResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OAuthScopesRestrictionResponseData.attributeTypeMap;
    }
}
exports.OAuthScopesRestrictionResponseData = OAuthScopesRestrictionResponseData;
/**
 * @ignore
 */
OAuthScopesRestrictionResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "OAuthScopesRestrictionResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
        format: "uuid",
    },
    type: {
        baseName: "type",
        type: "OAuthScopesRestrictionType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
