// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OAuthClientRegistrationResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OAuthClientRegistrationResponse = void 0;
/**
 * Response payload for a successful OAuth2 dynamic client registration as defined by RFC 7591.
 */
class OAuthClientRegistrationResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OAuthClientRegistrationResponse.attributeTypeMap;
    }
}
exports.OAuthClientRegistrationResponse = OAuthClientRegistrationResponse;
/**
 * @ignore
 */
OAuthClientRegistrationResponse.attributeTypeMap = {
    clientId: {
        baseName: "client_id",
        type: "string",
        required: true,
        format: "uuid",
    },
    clientName: {
        baseName: "client_name",
        type: "string",
        required: true,
    },
    grantTypes: {
        baseName: "grant_types",
        type: "Array<OAuthClientRegistrationGrantType>",
        required: true,
    },
    redirectUris: {
        baseName: "redirect_uris",
        type: "Array<string>",
        required: true,
    },
    responseTypes: {
        baseName: "response_types",
        type: "Array<OAuthClientRegistrationResponseType>",
        required: true,
    },
    tokenEndpointAuthMethod: {
        baseName: "token_endpoint_auth_method",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
