// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OAuthClientRegistrationError.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OAuthClientRegistrationError = void 0;
/**
 * Error payload returned by OAuth2 dynamic client registration as defined by RFC 7591.
 */
class OAuthClientRegistrationError {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OAuthClientRegistrationError.attributeTypeMap;
    }
}
exports.OAuthClientRegistrationError = OAuthClientRegistrationError;
/**
 * @ignore
 */
OAuthClientRegistrationError.attributeTypeMap = {
    error: {
        baseName: "error",
        type: "string",
        required: true,
    },
    errorDescription: {
        baseName: "error_description",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
