// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OAuthClientRegistrationRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OAuthClientRegistrationRequest = void 0;
/**
 * Request payload for OAuth2 dynamic client registration as defined by RFC 7591.
 */
class OAuthClientRegistrationRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OAuthClientRegistrationRequest.attributeTypeMap;
    }
}
exports.OAuthClientRegistrationRequest = OAuthClientRegistrationRequest;
/**
 * @ignore
 */
OAuthClientRegistrationRequest.attributeTypeMap = {
    clientName: {
        baseName: "client_name",
        type: "string",
        required: true,
    },
    clientUri: {
        baseName: "client_uri",
        type: "string",
    },
    grantTypes: {
        baseName: "grant_types",
        type: "Array<OAuthClientRegistrationGrantType>",
    },
    jwksUri: {
        baseName: "jwks_uri",
        type: "string",
    },
    logoUri: {
        baseName: "logo_uri",
        type: "string",
    },
    policyUri: {
        baseName: "policy_uri",
        type: "string",
    },
    redirectUris: {
        baseName: "redirect_uris",
        type: "Array<string>",
        required: true,
    },
    responseTypes: {
        baseName: "response_types",
        type: "Array<OAuthClientRegistrationResponseType>",
    },
    scope: {
        baseName: "scope",
        type: "string",
    },
    tokenEndpointAuthMethod: {
        baseName: "token_endpoint_auth_method",
        type: "string",
    },
    tosUri: {
        baseName: "tos_uri",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
