// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OAuth2WellKnownSitesAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OAuth2WellKnownSitesAttributes = void 0;
/**
 * Attributes containing the list of public OAuth2 sites.
 */
class OAuth2WellKnownSitesAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OAuth2WellKnownSitesAttributes.attributeTypeMap;
    }
}
exports.OAuth2WellKnownSitesAttributes = OAuth2WellKnownSitesAttributes;
/**
 * @ignore
 */
OAuth2WellKnownSitesAttributes.attributeTypeMap = {
    sites: {
        baseName: "sites",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
