// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineTagCardinalityLimitProcessorPerMetricLimit.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineTagCardinalityLimitProcessorPerMetricLimit = void 0;
/**
 * A cardinality override applied to a specific metric.
 */
class ObservabilityPipelineTagCardinalityLimitProcessorPerMetricLimit {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineTagCardinalityLimitProcessorPerMetricLimit.attributeTypeMap;
    }
}
exports.ObservabilityPipelineTagCardinalityLimitProcessorPerMetricLimit = ObservabilityPipelineTagCardinalityLimitProcessorPerMetricLimit;
/**
 * @ignore
 */
ObservabilityPipelineTagCardinalityLimitProcessorPerMetricLimit.attributeTypeMap = {
    limitExceededAction: {
        baseName: "limit_exceeded_action",
        type: "ObservabilityPipelineTagCardinalityLimitProcessorAction",
    },
    metricName: {
        baseName: "metric_name",
        type: "string",
        required: true,
    },
    mode: {
        baseName: "mode",
        type: "ObservabilityPipelineTagCardinalityLimitProcessorPerMetricMode",
        required: true,
    },
    perTagLimits: {
        baseName: "per_tag_limits",
        type: "Array<ObservabilityPipelineTagCardinalityLimitProcessorPerTagLimit>",
    },
    valueLimit: {
        baseName: "value_limit",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
