// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineWebsocketSource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineWebsocketSource = void 0;
/**
 * The `websocket` source ingests logs from a WebSocket server using the `ws://` or `wss://` protocol.
 *
 * **Supported pipeline types:** logs.
 */
class ObservabilityPipelineWebsocketSource {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineWebsocketSource.attributeTypeMap;
    }
}
exports.ObservabilityPipelineWebsocketSource = ObservabilityPipelineWebsocketSource;
/**
 * @ignore
 */
ObservabilityPipelineWebsocketSource.attributeTypeMap = {
    authStrategy: {
        baseName: "auth_strategy",
        type: "ObservabilityPipelineWebsocketSourceAuthStrategy",
        required: true,
    },
    customKey: {
        baseName: "custom_key",
        type: "string",
    },
    decoding: {
        baseName: "decoding",
        type: "ObservabilityPipelineDecoding",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    passwordKey: {
        baseName: "password_key",
        type: "string",
    },
    tls: {
        baseName: "tls",
        type: "ObservabilityPipelineWebsocketSourceTls",
    },
    tokenKey: {
        baseName: "token_key",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "ObservabilityPipelineWebsocketSourceType",
        required: true,
    },
    uriKey: {
        baseName: "uri_key",
        type: "string",
    },
    usernameKey: {
        baseName: "username_key",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
