// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineTagCardinalityLimitProcessor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineTagCardinalityLimitProcessor = void 0;
/**
 * The `tag_cardinality_limit` processor caps the number of distinct tag value combinations on metrics, dropping tags or events once the limit is exceeded.
 *
 * **Supported pipeline types:** metrics
 */
class ObservabilityPipelineTagCardinalityLimitProcessor {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineTagCardinalityLimitProcessor.attributeTypeMap;
    }
}
exports.ObservabilityPipelineTagCardinalityLimitProcessor = ObservabilityPipelineTagCardinalityLimitProcessor;
/**
 * @ignore
 */
ObservabilityPipelineTagCardinalityLimitProcessor.attributeTypeMap = {
    displayName: {
        baseName: "display_name",
        type: "string",
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    include: {
        baseName: "include",
        type: "string",
        required: true,
    },
    limitExceededAction: {
        baseName: "limit_exceeded_action",
        type: "ObservabilityPipelineTagCardinalityLimitProcessorAction",
        required: true,
    },
    perMetricLimits: {
        baseName: "per_metric_limits",
        type: "Array<ObservabilityPipelineTagCardinalityLimitProcessorPerMetricLimit>",
    },
    type: {
        baseName: "type",
        type: "ObservabilityPipelineTagCardinalityLimitProcessorType",
        required: true,
    },
    valueLimit: {
        baseName: "value_limit",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
