// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineMtlsServerTls.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineMtlsServerTls = void 0;
/**
 * Configuration for enabling TLS encryption between the pipeline component and external connecting clients.
 */
class ObservabilityPipelineMtlsServerTls {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineMtlsServerTls.attributeTypeMap;
    }
}
exports.ObservabilityPipelineMtlsServerTls = ObservabilityPipelineMtlsServerTls;
/**
 * @ignore
 */
ObservabilityPipelineMtlsServerTls.attributeTypeMap = {
    caFile: {
        baseName: "ca_file",
        type: "string",
    },
    crtFile: {
        baseName: "crt_file",
        type: "string",
        required: true,
    },
    keyFile: {
        baseName: "key_file",
        type: "string",
    },
    keyPassKey: {
        baseName: "key_pass_key",
        type: "string",
    },
    verifyCertificate: {
        baseName: "verify_certificate",
        type: "boolean",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
