// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineSplunkHecMetricsDestination.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineSplunkHecMetricsDestination = void 0;
/**
 * The `splunk_hec_metrics` destination forwards metrics to Splunk using the HTTP Event Collector (HEC).
 *
 * **Supported pipeline types:** metrics
 */
class ObservabilityPipelineSplunkHecMetricsDestination {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineSplunkHecMetricsDestination.attributeTypeMap;
    }
}
exports.ObservabilityPipelineSplunkHecMetricsDestination = ObservabilityPipelineSplunkHecMetricsDestination;
/**
 * @ignore
 */
ObservabilityPipelineSplunkHecMetricsDestination.attributeTypeMap = {
    buffer: {
        baseName: "buffer",
        type: "ObservabilityPipelineBufferOptions",
    },
    compression: {
        baseName: "compression",
        type: "ObservabilityPipelineSplunkHecMetricsDestinationCompression",
    },
    defaultNamespace: {
        baseName: "default_namespace",
        type: "string",
    },
    endpointUrlKey: {
        baseName: "endpoint_url_key",
        type: "string",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    index: {
        baseName: "index",
        type: "string",
    },
    inputs: {
        baseName: "inputs",
        type: "Array<string>",
        required: true,
    },
    source: {
        baseName: "source",
        type: "string",
    },
    sourcetype: {
        baseName: "sourcetype",
        type: "string",
    },
    tls: {
        baseName: "tls",
        type: "ObservabilityPipelineTls",
    },
    tokenKey: {
        baseName: "token_key",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "ObservabilityPipelineSplunkHecMetricsDestinationType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
