// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineAggregateProcessor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineAggregateProcessor = void 0;
/**
 * The `aggregate` processor combines metrics that share the same name and tags into a single metric over a configurable interval.
 *
 * **Supported pipeline types:** metrics
 */
class ObservabilityPipelineAggregateProcessor {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineAggregateProcessor.attributeTypeMap;
    }
}
exports.ObservabilityPipelineAggregateProcessor = ObservabilityPipelineAggregateProcessor;
/**
 * @ignore
 */
ObservabilityPipelineAggregateProcessor.attributeTypeMap = {
    displayName: {
        baseName: "display_name",
        type: "string",
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    include: {
        baseName: "include",
        type: "string",
        required: true,
    },
    intervalSecs: {
        baseName: "interval_secs",
        type: "number",
        required: true,
        format: "int64",
    },
    mode: {
        baseName: "mode",
        type: "ObservabilityPipelineAggregateProcessorMode",
        required: true,
    },
    type: {
        baseName: "type",
        type: "ObservabilityPipelineAggregateProcessorType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
