// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineHttpServerSourceValidToken.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineHttpServerSourceValidToken = void 0;
/**
 * An accepted token used to authenticate incoming HTTP server requests.
 */
class ObservabilityPipelineHttpServerSourceValidToken {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineHttpServerSourceValidToken.attributeTypeMap;
    }
}
exports.ObservabilityPipelineHttpServerSourceValidToken = ObservabilityPipelineHttpServerSourceValidToken;
/**
 * @ignore
 */
ObservabilityPipelineHttpServerSourceValidToken.attributeTypeMap = {
    enabled: {
        baseName: "enabled",
        type: "boolean",
    },
    fieldToAdd: {
        baseName: "field_to_add",
        type: "ObservabilityPipelineSourceValidTokenFieldToAdd",
    },
    pathToToken: {
        baseName: "path_to_token",
        type: "ObservabilityPipelineHttpServerSourceValidTokenPathToToken",
    },
    tokenKey: {
        baseName: "token_key",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
