// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineClickhouseDestinationBatchEncoding.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineClickhouseDestinationBatchEncoding = void 0;
/**
 * Batch encoding configuration for the ClickHouse destination.
 * Required when `format` is `arrow_stream`. The `codec` field must be set to `arrow_stream`.
 */
class ObservabilityPipelineClickhouseDestinationBatchEncoding {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineClickhouseDestinationBatchEncoding.attributeTypeMap;
    }
}
exports.ObservabilityPipelineClickhouseDestinationBatchEncoding = ObservabilityPipelineClickhouseDestinationBatchEncoding;
/**
 * @ignore
 */
ObservabilityPipelineClickhouseDestinationBatchEncoding.attributeTypeMap = {
    allowNullableFields: {
        baseName: "allow_nullable_fields",
        type: "boolean",
    },
    codec: {
        baseName: "codec",
        type: "ObservabilityPipelineClickhouseDestinationBatchEncodingCodec",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
