// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineTagCardinalityLimitProcessorPerTagLimit.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineTagCardinalityLimitProcessorPerTagLimit = void 0;
/**
 * A cardinality override for a specific tag key within a per-metric limit.
 */
class ObservabilityPipelineTagCardinalityLimitProcessorPerTagLimit {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineTagCardinalityLimitProcessorPerTagLimit.attributeTypeMap;
    }
}
exports.ObservabilityPipelineTagCardinalityLimitProcessorPerTagLimit = ObservabilityPipelineTagCardinalityLimitProcessorPerTagLimit;
/**
 * @ignore
 */
ObservabilityPipelineTagCardinalityLimitProcessorPerTagLimit.attributeTypeMap = {
    mode: {
        baseName: "mode",
        type: "ObservabilityPipelineTagCardinalityLimitProcessorPerTagMode",
        required: true,
    },
    tagKey: {
        baseName: "tag_key",
        type: "string",
        required: true,
    },
    valueLimit: {
        baseName: "value_limit",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
