// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineRenameMetricTagsProcessor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineRenameMetricTagsProcessor = void 0;
/**
 * The `rename_metric_tags` processor changes the keys of tags on metrics.
 *
 * **Supported pipeline types:** metrics
 */
class ObservabilityPipelineRenameMetricTagsProcessor {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineRenameMetricTagsProcessor.attributeTypeMap;
    }
}
exports.ObservabilityPipelineRenameMetricTagsProcessor = ObservabilityPipelineRenameMetricTagsProcessor;
/**
 * @ignore
 */
ObservabilityPipelineRenameMetricTagsProcessor.attributeTypeMap = {
    displayName: {
        baseName: "display_name",
        type: "string",
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    include: {
        baseName: "include",
        type: "string",
        required: true,
    },
    tags: {
        baseName: "tags",
        type: "Array<ObservabilityPipelineRenameMetricTagsProcessorTag>",
        required: true,
    },
    type: {
        baseName: "type",
        type: "ObservabilityPipelineRenameMetricTagsProcessorType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
