// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineClickhouseDestinationAuth.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineClickhouseDestinationAuth = void 0;
/**
 * HTTP Basic Authentication credentials for the ClickHouse destination.
 * When `strategy` is `basic`, provide `username_key` and `password_key` that reference environment variables or secrets containing the credentials.
 */
class ObservabilityPipelineClickhouseDestinationAuth {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineClickhouseDestinationAuth.attributeTypeMap;
    }
}
exports.ObservabilityPipelineClickhouseDestinationAuth = ObservabilityPipelineClickhouseDestinationAuth;
/**
 * @ignore
 */
ObservabilityPipelineClickhouseDestinationAuth.attributeTypeMap = {
    passwordKey: {
        baseName: "password_key",
        type: "string",
    },
    strategy: {
        baseName: "strategy",
        type: "ObservabilityPipelineClickhouseDestinationAuthStrategy",
        required: true,
    },
    usernameKey: {
        baseName: "username_key",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
