// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineSplunkHecSourceValidToken.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineSplunkHecSourceValidToken = void 0;
/**
 * An accepted HEC token used to authenticate incoming Splunk HEC requests.
 */
class ObservabilityPipelineSplunkHecSourceValidToken {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineSplunkHecSourceValidToken.attributeTypeMap;
    }
}
exports.ObservabilityPipelineSplunkHecSourceValidToken = ObservabilityPipelineSplunkHecSourceValidToken;
/**
 * @ignore
 */
ObservabilityPipelineSplunkHecSourceValidToken.attributeTypeMap = {
    enabled: {
        baseName: "enabled",
        type: "boolean",
    },
    fieldToAdd: {
        baseName: "field_to_add",
        type: "ObservabilityPipelineSourceValidTokenFieldToAdd",
    },
    tokenKey: {
        baseName: "token_key",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
