// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineDatabricksZerobusDestination.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineDatabricksZerobusDestination = void 0;
/**
 * The `databricks_zerobus` destination sends logs to Databricks using the Zerobus ingestion API, streaming data directly into your Databricks Lakehouse.
 *
 * **Supported pipeline types:** Logs, rehydration
 */
class ObservabilityPipelineDatabricksZerobusDestination {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineDatabricksZerobusDestination.attributeTypeMap;
    }
}
exports.ObservabilityPipelineDatabricksZerobusDestination = ObservabilityPipelineDatabricksZerobusDestination;
/**
 * @ignore
 */
ObservabilityPipelineDatabricksZerobusDestination.attributeTypeMap = {
    auth: {
        baseName: "auth",
        type: "ObservabilityPipelineDatabricksZerobusDestinationAuth",
        required: true,
    },
    buffer: {
        baseName: "buffer",
        type: "ObservabilityPipelineBufferOptions",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    ingestionEndpointKey: {
        baseName: "ingestion_endpoint_key",
        type: "string",
    },
    inputs: {
        baseName: "inputs",
        type: "Array<string>",
        required: true,
    },
    tableName: {
        baseName: "table_name",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "ObservabilityPipelineDatabricksZerobusDestinationType",
        required: true,
    },
    unityCatalogEndpointKey: {
        baseName: "unity_catalog_endpoint_key",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
