// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineClickhouseDestinationCompressionObject.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineClickhouseDestinationCompressionObject = void 0;
/**
 * Structured compression configuration for the ClickHouse destination.
 * Use `algorithm` to specify the compression type and `level` (optional, gzip only) to control compression strength.
 */
class ObservabilityPipelineClickhouseDestinationCompressionObject {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineClickhouseDestinationCompressionObject.attributeTypeMap;
    }
}
exports.ObservabilityPipelineClickhouseDestinationCompressionObject = ObservabilityPipelineClickhouseDestinationCompressionObject;
/**
 * @ignore
 */
ObservabilityPipelineClickhouseDestinationCompressionObject.attributeTypeMap = {
    algorithm: {
        baseName: "algorithm",
        type: "ObservabilityPipelineClickhouseDestinationCompressionAlgorithm",
        required: true,
    },
    level: {
        baseName: "level",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
