// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineRenameMetricTagsProcessorTag.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineRenameMetricTagsProcessorTag = void 0;
/**
 * Defines how to rename a tag on metric events.
 */
class ObservabilityPipelineRenameMetricTagsProcessorTag {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineRenameMetricTagsProcessorTag.attributeTypeMap;
    }
}
exports.ObservabilityPipelineRenameMetricTagsProcessorTag = ObservabilityPipelineRenameMetricTagsProcessorTag;
/**
 * @ignore
 */
ObservabilityPipelineRenameMetricTagsProcessorTag.attributeTypeMap = {
    renameTo: {
        baseName: "rename_to",
        type: "string",
        required: true,
    },
    tag: {
        baseName: "tag",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
