// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineHttpServerSourceValidTokenPathToTokenHeader.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineHttpServerSourceValidTokenPathToTokenHeader = void 0;
/**
 * Extract the token from a specific HTTP request header.
 */
class ObservabilityPipelineHttpServerSourceValidTokenPathToTokenHeader {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineHttpServerSourceValidTokenPathToTokenHeader.attributeTypeMap;
    }
}
exports.ObservabilityPipelineHttpServerSourceValidTokenPathToTokenHeader = ObservabilityPipelineHttpServerSourceValidTokenPathToTokenHeader;
/**
 * @ignore
 */
ObservabilityPipelineHttpServerSourceValidTokenPathToTokenHeader.attributeTypeMap = {
    header: {
        baseName: "header",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
