// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineClickhouseDestinationBatch.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineClickhouseDestinationBatch = void 0;
/**
 * Batching configuration for ClickHouse inserts.
 */
class ObservabilityPipelineClickhouseDestinationBatch {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineClickhouseDestinationBatch.attributeTypeMap;
    }
}
exports.ObservabilityPipelineClickhouseDestinationBatch = ObservabilityPipelineClickhouseDestinationBatch;
/**
 * @ignore
 */
ObservabilityPipelineClickhouseDestinationBatch.attributeTypeMap = {
    maxEvents: {
        baseName: "max_events",
        type: "number",
        format: "int64",
    },
    timeoutSecs: {
        baseName: "timeout_secs",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
