// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineAddMetricTagsProcessor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineAddMetricTagsProcessor = void 0;
/**
 * The `add_metric_tags` processor adds static tags to metrics.
 *
 * **Supported pipeline types:** metrics
 */
class ObservabilityPipelineAddMetricTagsProcessor {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineAddMetricTagsProcessor.attributeTypeMap;
    }
}
exports.ObservabilityPipelineAddMetricTagsProcessor = ObservabilityPipelineAddMetricTagsProcessor;
/**
 * @ignore
 */
ObservabilityPipelineAddMetricTagsProcessor.attributeTypeMap = {
    displayName: {
        baseName: "display_name",
        type: "string",
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    include: {
        baseName: "include",
        type: "string",
        required: true,
    },
    tags: {
        baseName: "tags",
        type: "Array<ObservabilityPipelineFieldValue>",
        required: true,
    },
    type: {
        baseName: "type",
        type: "ObservabilityPipelineAddMetricTagsProcessorType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
