// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineGenerateMetricsV2Processor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineGenerateMetricsV2Processor = void 0;
/**
 * The `generate_metrics` processor creates custom metrics from logs.
 * Metrics can be counters, gauges, or distributions and optionally grouped by log fields.
 * The generated metrics must be routed to a metrics destination using the input `<processor-id>.metrics`.
 *
 * **Supported pipeline types:** logs
 */
class ObservabilityPipelineGenerateMetricsV2Processor {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineGenerateMetricsV2Processor.attributeTypeMap;
    }
}
exports.ObservabilityPipelineGenerateMetricsV2Processor = ObservabilityPipelineGenerateMetricsV2Processor;
/**
 * @ignore
 */
ObservabilityPipelineGenerateMetricsV2Processor.attributeTypeMap = {
    displayName: {
        baseName: "display_name",
        type: "string",
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    include: {
        baseName: "include",
        type: "string",
    },
    metrics: {
        baseName: "metrics",
        type: "Array<ObservabilityPipelineGeneratedMetric>",
    },
    type: {
        baseName: "type",
        type: "ObservabilityPipelineGenerateMetricsV2ProcessorType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
