// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineClickhouseDestination.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineClickhouseDestination = void 0;
/**
 * The `clickhouse` destination sends log events to a ClickHouse database table over HTTP.
 *
 * **Supported pipeline types:** logs.
 */
class ObservabilityPipelineClickhouseDestination {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineClickhouseDestination.attributeTypeMap;
    }
}
exports.ObservabilityPipelineClickhouseDestination = ObservabilityPipelineClickhouseDestination;
/**
 * @ignore
 */
ObservabilityPipelineClickhouseDestination.attributeTypeMap = {
    auth: {
        baseName: "auth",
        type: "ObservabilityPipelineClickhouseDestinationAuth",
    },
    batch: {
        baseName: "batch",
        type: "ObservabilityPipelineClickhouseDestinationBatch",
    },
    batchEncoding: {
        baseName: "batch_encoding",
        type: "ObservabilityPipelineClickhouseDestinationBatchEncoding",
    },
    buffer: {
        baseName: "buffer",
        type: "ObservabilityPipelineBufferOptions",
    },
    compression: {
        baseName: "compression",
        type: "ObservabilityPipelineClickhouseDestinationCompression",
    },
    database: {
        baseName: "database",
        type: "string",
    },
    dateTimeBestEffort: {
        baseName: "date_time_best_effort",
        type: "boolean",
    },
    endpointUrlKey: {
        baseName: "endpoint_url_key",
        type: "string",
    },
    format: {
        baseName: "format",
        type: "ObservabilityPipelineClickhouseDestinationFormat",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    inputs: {
        baseName: "inputs",
        type: "Array<string>",
        required: true,
    },
    skipUnknownFields: {
        baseName: "skip_unknown_fields",
        type: "boolean",
    },
    table: {
        baseName: "table",
        type: "string",
        required: true,
    },
    tls: {
        baseName: "tls",
        type: "ObservabilityPipelineTls",
    },
    type: {
        baseName: "type",
        type: "ObservabilityPipelineClickhouseDestinationType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
