// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineWebsocketSourceTlsWithClientCert.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineWebsocketSourceTlsWithClientCert = void 0;
/**
 * TLS configuration that enables encryption and presents a client certificate for mutual TLS authentication.
 */
class ObservabilityPipelineWebsocketSourceTlsWithClientCert {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineWebsocketSourceTlsWithClientCert.attributeTypeMap;
    }
}
exports.ObservabilityPipelineWebsocketSourceTlsWithClientCert = ObservabilityPipelineWebsocketSourceTlsWithClientCert;
/**
 * @ignore
 */
ObservabilityPipelineWebsocketSourceTlsWithClientCert.attributeTypeMap = {
    caFile: {
        baseName: "ca_file",
        type: "string",
    },
    crtFile: {
        baseName: "crt_file",
        type: "string",
        required: true,
    },
    keyFile: {
        baseName: "key_file",
        type: "string",
    },
    keyPassKey: {
        baseName: "key_pass_key",
        type: "string",
    },
    mode: {
        baseName: "mode",
        type: "ObservabilityPipelineWebsocketSourceTlsWithClientCertMode",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
