// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ObservabilityPipelineSourceValidTokenFieldToAdd.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObservabilityPipelineSourceValidTokenFieldToAdd = void 0;
/**
 * An optional metadata field that is attached to every event authenticated by the
 * associated token. Both `key` and `value` must match `^[A-Za-z0-9_]+$`.
 */
class ObservabilityPipelineSourceValidTokenFieldToAdd {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ObservabilityPipelineSourceValidTokenFieldToAdd.attributeTypeMap;
    }
}
exports.ObservabilityPipelineSourceValidTokenFieldToAdd = ObservabilityPipelineSourceValidTokenFieldToAdd;
/**
 * @ignore
 */
ObservabilityPipelineSourceValidTokenFieldToAdd.attributeTypeMap = {
    key: {
        baseName: "key",
        type: "string",
        required: true,
    },
    value: {
        baseName: "value",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
