// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OrgSAMLPreferencesAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrgSAMLPreferencesAttributes = void 0;
/**
 * Attributes for updating an organization's SAML preferences.
 */
class OrgSAMLPreferencesAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OrgSAMLPreferencesAttributes.attributeTypeMap;
    }
}
exports.OrgSAMLPreferencesAttributes = OrgSAMLPreferencesAttributes;
/**
 * @ignore
 */
OrgSAMLPreferencesAttributes.attributeTypeMap = {
    defaultRoleUuids: {
        baseName: "default_role_uuids",
        type: "[string]",
        required: true,
    },
    jitDomains: {
        baseName: "jit_domains",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
