// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OwnershipHistoryItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OwnershipHistoryItem = void 0;
/**
 * A single ownership inference history entry.
 */
class OwnershipHistoryItem {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OwnershipHistoryItem.attributeTypeMap;
    }
}
exports.OwnershipHistoryItem = OwnershipHistoryItem;
/**
 * @ignore
 */
OwnershipHistoryItem.attributeTypeMap = {
    checksum: {
        baseName: "checksum",
        type: "string",
        required: true,
    },
    confidence: {
        baseName: "confidence",
        type: "string",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    evidenceVersions: {
        baseName: "evidence_versions",
        type: "Array<{ [key: string]: any; }>",
        required: true,
    },
    explanation: {
        baseName: "explanation",
        type: "string",
        required: true,
    },
    failedAt: {
        baseName: "failed_at",
        type: "Date",
        format: "date-time",
    },
    failureReason: {
        baseName: "failure_reason",
        type: "string",
    },
    id: {
        baseName: "id",
        type: "number",
        required: true,
        format: "int64",
    },
    ownerType: {
        baseName: "owner_type",
        type: "OwnershipOwnerType",
        required: true,
    },
    primaryContactRef: {
        baseName: "primary_contact_ref",
        type: "string",
    },
    resourceId: {
        baseName: "resource_id",
        type: "string",
        required: true,
    },
    retrySchedule: {
        baseName: "retry_schedule",
        type: "Date",
        format: "date-time",
    },
    sources: {
        baseName: "sources",
        type: "Array<{ [key: string]: any; }>",
        required: true,
    },
    status: {
        baseName: "status",
        type: "OwnershipInferenceStatus",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
