// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OwnershipFeedbackRequestAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OwnershipFeedbackRequestAttributes = void 0;
/**
 * The attributes of an ownership feedback request.
 */
class OwnershipFeedbackRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OwnershipFeedbackRequestAttributes.attributeTypeMap;
    }
}
exports.OwnershipFeedbackRequestAttributes = OwnershipFeedbackRequestAttributes;
/**
 * @ignore
 */
OwnershipFeedbackRequestAttributes.attributeTypeMap = {
    action: {
        baseName: "action",
        type: "OwnershipFeedbackAction",
        required: true,
    },
    actorHandle: {
        baseName: "actor_handle",
        type: "string",
        required: true,
    },
    actorType: {
        baseName: "actor_type",
        type: "string",
        required: true,
    },
    correctedOwnerHandle: {
        baseName: "corrected_owner_handle",
        type: "string",
    },
    correctedOwnerType: {
        baseName: "corrected_owner_type",
        type: "string",
    },
    inferenceChecksum: {
        baseName: "inference_checksum",
        type: "string",
        required: true,
    },
    reason: {
        baseName: "reason",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
