// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OwnershipFeedbackResultAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OwnershipFeedbackResultAttributes = void 0;
/**
 * The attributes of an ownership feedback result.
 */
class OwnershipFeedbackResultAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OwnershipFeedbackResultAttributes.attributeTypeMap;
    }
}
exports.OwnershipFeedbackResultAttributes = OwnershipFeedbackResultAttributes;
/**
 * @ignore
 */
OwnershipFeedbackResultAttributes.attributeTypeMap = {
    action: {
        baseName: "action",
        type: "OwnershipFeedbackAction",
        required: true,
    },
    checksum: {
        baseName: "checksum",
        type: "string",
        required: true,
    },
    newStatus: {
        baseName: "new_status",
        type: "OwnershipInferenceStatus",
        required: true,
    },
    ownerType: {
        baseName: "owner_type",
        type: "OwnershipOwnerType",
        required: true,
    },
    previousStatus: {
        baseName: "previous_status",
        type: "OwnershipInferenceStatus",
        required: true,
    },
    primaryContactRef: {
        baseName: "primary_contact_ref",
        type: "string",
    },
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
