// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OwnershipEvidenceAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OwnershipEvidenceAttributes = void 0;
/**
 * The attributes of an ownership evidence response.
 */
class OwnershipEvidenceAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OwnershipEvidenceAttributes.attributeTypeMap;
    }
}
exports.OwnershipEvidenceAttributes = OwnershipEvidenceAttributes;
/**
 * @ignore
 */
OwnershipEvidenceAttributes.attributeTypeMap = {
    evidenceVersions: {
        baseName: "evidence_versions",
        type: "Array<{ [key: string]: any; }>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
