// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OwnershipInferenceItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OwnershipInferenceItem = void 0;
/**
 * A single ownership inference, scoped to a specific owner type.
 */
class OwnershipInferenceItem {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OwnershipInferenceItem.attributeTypeMap;
    }
}
exports.OwnershipInferenceItem = OwnershipInferenceItem;
/**
 * @ignore
 */
OwnershipInferenceItem.attributeTypeMap = {
    checksum: {
        baseName: "checksum",
        type: "string",
        required: true,
    },
    confidence: {
        baseName: "confidence",
        type: "string",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    evidenceVersions: {
        baseName: "evidence_versions",
        type: "Array<{ [key: string]: any; }>",
        required: true,
    },
    explanation: {
        baseName: "explanation",
        type: "string",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    ownerType: {
        baseName: "owner_type",
        type: "OwnershipOwnerType",
        required: true,
    },
    primaryContactRef: {
        baseName: "primary_contact_ref",
        type: "string",
    },
    sources: {
        baseName: "sources",
        type: "Array<{ [key: string]: any; }>",
        required: true,
    },
    status: {
        baseName: "status",
        type: "OwnershipInferenceStatus",
        required: true,
    },
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
