// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/OwnershipHistoryPagination.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OwnershipHistoryPagination = void 0;
/**
 * Cursor-based pagination metadata for the history response.
 */
class OwnershipHistoryPagination {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return OwnershipHistoryPagination.attributeTypeMap;
    }
}
exports.OwnershipHistoryPagination = OwnershipHistoryPagination;
/**
 * @ignore
 */
OwnershipHistoryPagination.attributeTypeMap = {
    hasMore: {
        baseName: "has_more",
        type: "boolean",
        required: true,
    },
    nextCursor: {
        baseName: "next_cursor",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
