// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/PageAnnotationsAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PageAnnotationsAttributes = void 0;
/**
 * Attributes of the annotations on a page.
 */
class PageAnnotationsAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return PageAnnotationsAttributes.attributeTypeMap;
    }
}
exports.PageAnnotationsAttributes = PageAnnotationsAttributes;
/**
 * @ignore
 */
PageAnnotationsAttributes.attributeTypeMap = {
    annotations: {
        baseName: "annotations",
        type: "{ [key: string]: AnnotationInPage; }",
        required: true,
    },
    globalAnnotations: {
        baseName: "global_annotations",
        type: "Array<string>",
        required: true,
    },
    widgetMapping: {
        baseName: "widget_mapping",
        type: "{ [key: string]: Array<string>; }",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
