// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ReactNativeSourcemapAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReactNativeSourcemapAttributes = void 0;
/**
 * Attributes of a React Native source map.
 */
class ReactNativeSourcemapAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ReactNativeSourcemapAttributes.attributeTypeMap;
    }
}
exports.ReactNativeSourcemapAttributes = ReactNativeSourcemapAttributes;
/**
 * @ignore
 */
ReactNativeSourcemapAttributes.attributeTypeMap = {
    buildNumber: {
        baseName: "build_number",
        type: "string",
    },
    bundleName: {
        baseName: "bundle_name",
        type: "string",
    },
    bundleVersion: {
        baseName: "bundle_version",
        type: "string",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    debugId: {
        baseName: "debug_id",
        type: "string",
    },
    mapkind: {
        baseName: "mapkind",
        type: "string",
        required: true,
    },
    platform: {
        baseName: "platform",
        type: "string",
    },
    service: {
        baseName: "service",
        type: "string",
    },
    size: {
        baseName: "size",
        type: "number",
        required: true,
        format: "int64",
    },
    version: {
        baseName: "version",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
