// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RecommendationsFilterRequestSortItems.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecommendationsFilterRequestSortItems = void 0;
/**
 * A single sort clause applied to the cost recommendations result set.
 */
class RecommendationsFilterRequestSortItems {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RecommendationsFilterRequestSortItems.attributeTypeMap;
    }
}
exports.RecommendationsFilterRequestSortItems = RecommendationsFilterRequestSortItems;
/**
 * @ignore
 */
RecommendationsFilterRequestSortItems.attributeTypeMap = {
    expression: {
        baseName: "expression",
        type: "string",
    },
    order: {
        baseName: "order",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
