// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RecommendationsPageMetaPage.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecommendationsPageMetaPage = void 0;
/**
 * Pagination metadata for a page of cost recommendations.
 */
class RecommendationsPageMetaPage {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RecommendationsPageMetaPage.attributeTypeMap;
    }
}
exports.RecommendationsPageMetaPage = RecommendationsPageMetaPage;
/**
 * @ignore
 */
RecommendationsPageMetaPage.attributeTypeMap = {
    filter: {
        baseName: "filter",
        type: "string",
    },
    nextPageToken: {
        baseName: "next_page_token",
        type: "string",
    },
    pageSize: {
        baseName: "page_size",
        type: "number",
        format: "int32",
    },
    pageToken: {
        baseName: "page_token",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
