// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ReportScheduleListResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportScheduleListResponse = void 0;
/**
 * Response containing a list of report schedules.
 */
class ReportScheduleListResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ReportScheduleListResponse.attributeTypeMap;
    }
}
exports.ReportScheduleListResponse = ReportScheduleListResponse;
/**
 * @ignore
 */
ReportScheduleListResponse.attributeTypeMap = {
    data: {
        baseName: "data",
        type: "Array<ReportScheduleListResponseData>",
        required: true,
    },
    included: {
        baseName: "included",
        type: "Array<ReportScheduleIncludedResource>",
    },
    links: {
        baseName: "links",
        type: "ReportScheduleListResponseLinks",
    },
    meta: {
        baseName: "meta",
        type: "ReportScheduleListResponseMeta",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
