// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ReportSchedulePatchRequestAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportSchedulePatchRequestAttributes = void 0;
/**
 * The updated configuration of the report schedule. These values replace the existing
 * ones; the targeted resource (`resource_id` and `resource_type`) cannot be changed.
 */
class ReportSchedulePatchRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ReportSchedulePatchRequestAttributes.attributeTypeMap;
    }
}
exports.ReportSchedulePatchRequestAttributes = ReportSchedulePatchRequestAttributes;
/**
 * @ignore
 */
ReportSchedulePatchRequestAttributes.attributeTypeMap = {
    deliveryFormat: {
        baseName: "delivery_format",
        type: "ReportScheduleDeliveryFormat",
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    recipients: {
        baseName: "recipients",
        type: "Array<string>",
        required: true,
    },
    rrule: {
        baseName: "rrule",
        type: "string",
        required: true,
    },
    tabId: {
        baseName: "tab_id",
        type: "string",
        format: "uuid",
    },
    templateVariables: {
        baseName: "template_variables",
        type: "Array<ReportScheduleTemplateVariable>",
        required: true,
    },
    timeframe: {
        baseName: "timeframe",
        type: "string",
        required: true,
    },
    timezone: {
        baseName: "timezone",
        type: "string",
        required: true,
    },
    title: {
        baseName: "title",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
