// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ReportScheduleResponseAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportScheduleResponseAttributes = void 0;
/**
 * The configuration and derived state of a report schedule.
 */
class ReportScheduleResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ReportScheduleResponseAttributes.attributeTypeMap;
    }
}
exports.ReportScheduleResponseAttributes = ReportScheduleResponseAttributes;
/**
 * @ignore
 */
ReportScheduleResponseAttributes.attributeTypeMap = {
    deliveryFormat: {
        baseName: "delivery_format",
        type: "ReportScheduleResponseAttributesDeliveryFormat",
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    nextRecurrence: {
        baseName: "next_recurrence",
        type: "number",
        required: true,
        format: "int64",
    },
    recipients: {
        baseName: "recipients",
        type: "Array<string>",
        required: true,
    },
    resourceId: {
        baseName: "resource_id",
        type: "string",
        required: true,
    },
    resourceType: {
        baseName: "resource_type",
        type: "ReportScheduleResourceType",
        required: true,
    },
    rrule: {
        baseName: "rrule",
        type: "string",
        required: true,
    },
    status: {
        baseName: "status",
        type: "ReportScheduleStatus",
        required: true,
    },
    tabId: {
        baseName: "tab_id",
        type: "string",
        required: true,
    },
    templateVariables: {
        baseName: "template_variables",
        type: "Array<ReportScheduleTemplateVariable>",
        required: true,
    },
    timeframe: {
        baseName: "timeframe",
        type: "string",
        required: true,
    },
    timezone: {
        baseName: "timezone",
        type: "string",
        required: true,
    },
    title: {
        baseName: "title",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
