// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ReportScheduleTemplateVariable.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportScheduleTemplateVariable = void 0;
/**
 * A dashboard template variable applied when rendering the report.
 */
class ReportScheduleTemplateVariable {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ReportScheduleTemplateVariable.attributeTypeMap;
    }
}
exports.ReportScheduleTemplateVariable = ReportScheduleTemplateVariable;
/**
 * @ignore
 */
ReportScheduleTemplateVariable.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    values: {
        baseName: "values",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
