// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ReportScheduleCreateRequestAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportScheduleCreateRequestAttributes = void 0;
/**
 * The configuration of the report schedule to create.
 */
class ReportScheduleCreateRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ReportScheduleCreateRequestAttributes.attributeTypeMap;
    }
}
exports.ReportScheduleCreateRequestAttributes = ReportScheduleCreateRequestAttributes;
/**
 * @ignore
 */
ReportScheduleCreateRequestAttributes.attributeTypeMap = {
    deliveryFormat: {
        baseName: "delivery_format",
        type: "ReportScheduleDeliveryFormat",
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    recipients: {
        baseName: "recipients",
        type: "Array<string>",
        required: true,
    },
    resourceId: {
        baseName: "resource_id",
        type: "string",
        required: true,
    },
    resourceType: {
        baseName: "resource_type",
        type: "ReportScheduleResourceType",
        required: true,
    },
    rrule: {
        baseName: "rrule",
        type: "string",
        required: true,
    },
    tabId: {
        baseName: "tab_id",
        type: "string",
        format: "uuid",
    },
    templateVariables: {
        baseName: "template_variables",
        type: "Array<ReportScheduleTemplateVariable>",
        required: true,
    },
    timeframe: {
        baseName: "timeframe",
        type: "string",
        required: true,
    },
    timezone: {
        baseName: "timezone",
        type: "string",
        required: true,
    },
    title: {
        baseName: "title",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
