// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ReportScheduleResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportScheduleResponseData = void 0;
/**
 * The JSON:API data object representing a report schedule.
 */
class ReportScheduleResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ReportScheduleResponseData.attributeTypeMap;
    }
}
exports.ReportScheduleResponseData = ReportScheduleResponseData;
/**
 * @ignore
 */
ReportScheduleResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "ReportScheduleResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    relationships: {
        baseName: "relationships",
        type: "ReportScheduleResponseRelationships",
        required: true,
    },
    type: {
        baseName: "type",
        type: "ReportScheduleType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
