// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ReportScheduleListResponsePagination.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportScheduleListResponsePagination = void 0;
/**
 * Offset and limit pagination metadata for a report schedule list response.
 */
class ReportScheduleListResponsePagination {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ReportScheduleListResponsePagination.attributeTypeMap;
    }
}
exports.ReportScheduleListResponsePagination = ReportScheduleListResponsePagination;
/**
 * @ignore
 */
ReportScheduleListResponsePagination.attributeTypeMap = {
    firstOffset: {
        baseName: "first_offset",
        type: "number",
        format: "int64",
    },
    lastOffset: {
        baseName: "last_offset",
        type: "number",
        format: "int64",
    },
    limit: {
        baseName: "limit",
        type: "number",
        format: "int64",
    },
    nextOffset: {
        baseName: "next_offset",
        type: "number",
        format: "int64",
    },
    offset: {
        baseName: "offset",
        type: "number",
        format: "int64",
    },
    prevOffset: {
        baseName: "prev_offset",
        type: "number",
        format: "int64",
    },
    total: {
        baseName: "total",
        type: "number",
        format: "int64",
    },
    type: {
        baseName: "type",
        type: "ReportScheduleListResponsePaginationType",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
