// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ReportScheduleListResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportScheduleListResponseData = void 0;
/**
 * The JSON:API data object representing a report schedule in a list response.
 */
class ReportScheduleListResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ReportScheduleListResponseData.attributeTypeMap;
    }
}
exports.ReportScheduleListResponseData = ReportScheduleListResponseData;
/**
 * @ignore
 */
ReportScheduleListResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "ReportScheduleListResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    relationships: {
        baseName: "relationships",
        type: "ReportScheduleListResponseRelationships",
        required: true,
    },
    type: {
        baseName: "type",
        type: "ReportScheduleType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
