// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RoutingRuleEscalationPolicyActionSupportHours.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoutingRuleEscalationPolicyActionSupportHours = void 0;
/**
 * Support hours during which the escalation policy will be executed. Outside of these hours, the escalation policy will be on hold and triggered once the next support hours window starts. This is mutually exclusive with the top-level `time_restriction` field on the routing rule.
 */
class RoutingRuleEscalationPolicyActionSupportHours {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RoutingRuleEscalationPolicyActionSupportHours.attributeTypeMap;
    }
}
exports.RoutingRuleEscalationPolicyActionSupportHours = RoutingRuleEscalationPolicyActionSupportHours;
/**
 * @ignore
 */
RoutingRuleEscalationPolicyActionSupportHours.attributeTypeMap = {
    restrictions: {
        baseName: "restrictions",
        type: "Array<TimeRestriction>",
    },
    timeZone: {
        baseName: "time_zone",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
