// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RoutingRuleEscalationPolicyAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoutingRuleEscalationPolicyAction = void 0;
/**
 * Triggers an escalation policy.
 */
class RoutingRuleEscalationPolicyAction {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RoutingRuleEscalationPolicyAction.attributeTypeMap;
    }
}
exports.RoutingRuleEscalationPolicyAction = RoutingRuleEscalationPolicyAction;
/**
 * @ignore
 */
RoutingRuleEscalationPolicyAction.attributeTypeMap = {
    ackTimeoutMinutes: {
        baseName: "ack_timeout_minutes",
        type: "number",
        format: "int64",
    },
    policyId: {
        baseName: "policy_id",
        type: "string",
        required: true,
    },
    supportHours: {
        baseName: "support_hours",
        type: "RoutingRuleEscalationPolicyActionSupportHours",
    },
    type: {
        baseName: "type",
        type: "RoutingRuleEscalationPolicyActionType",
        required: true,
    },
    urgency: {
        baseName: "urgency",
        type: "Urgency",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
