// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RumPermanentRetentionFilterEditability.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RumPermanentRetentionFilterEditability = void 0;
/**
 * Indicates which cross-product fields of a permanent RUM retention filter can be updated.
 */
class RumPermanentRetentionFilterEditability {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RumPermanentRetentionFilterEditability.attributeTypeMap;
    }
}
exports.RumPermanentRetentionFilterEditability = RumPermanentRetentionFilterEditability;
/**
 * @ignore
 */
RumPermanentRetentionFilterEditability.attributeTypeMap = {
    traceEditable: {
        baseName: "trace_editable",
        type: "boolean",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
