// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RumPermanentRetentionFilterAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RumPermanentRetentionFilterAttributes = void 0;
/**
 * The attributes of a permanent RUM retention filter.
 */
class RumPermanentRetentionFilterAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RumPermanentRetentionFilterAttributes.attributeTypeMap;
    }
}
exports.RumPermanentRetentionFilterAttributes = RumPermanentRetentionFilterAttributes;
/**
 * @ignore
 */
RumPermanentRetentionFilterAttributes.attributeTypeMap = {
    crossProductSampling: {
        baseName: "cross_product_sampling",
        type: "RumCrossProductSampling",
    },
    description: {
        baseName: "description",
        type: "string",
    },
    editability: {
        baseName: "editability",
        type: "RumPermanentRetentionFilterEditability",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
