// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RumPermanentRetentionFilterUpdateData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RumPermanentRetentionFilterUpdateData = void 0;
/**
 * The new permanent RUM retention filter configuration to update.
 */
class RumPermanentRetentionFilterUpdateData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RumPermanentRetentionFilterUpdateData.attributeTypeMap;
    }
}
exports.RumPermanentRetentionFilterUpdateData = RumPermanentRetentionFilterUpdateData;
/**
 * @ignore
 */
RumPermanentRetentionFilterUpdateData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "RumPermanentRetentionFilterUpdateAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "RumPermanentRetentionFilterID",
        required: true,
    },
    type: {
        baseName: "type",
        type: "RumPermanentRetentionFilterType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
