// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RumRateLimitConfigAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RumRateLimitConfigAttributes = void 0;
/**
 * The RUM rate limit configuration properties.
 */
class RumRateLimitConfigAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RumRateLimitConfigAttributes.attributeTypeMap;
    }
}
exports.RumRateLimitConfigAttributes = RumRateLimitConfigAttributes;
/**
 * @ignore
 */
RumRateLimitConfigAttributes.attributeTypeMap = {
    adaptive: {
        baseName: "adaptive",
        type: "RumRateLimitAdaptiveConfig",
    },
    custom: {
        baseName: "custom",
        type: "RumRateLimitCustomConfig",
    },
    mode: {
        baseName: "mode",
        type: "RumRateLimitMode",
        required: true,
    },
    orgId: {
        baseName: "org_id",
        type: "number",
        required: true,
        format: "int64",
    },
    updatedAt: {
        baseName: "updated_at",
        type: "string",
    },
    updatedBy: {
        baseName: "updated_by",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
