// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RumRateLimitAdaptiveConfig.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RumRateLimitAdaptiveConfig = void 0;
/**
 * The configuration used when `mode` is `adaptive`.
 */
class RumRateLimitAdaptiveConfig {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RumRateLimitAdaptiveConfig.attributeTypeMap;
    }
}
exports.RumRateLimitAdaptiveConfig = RumRateLimitAdaptiveConfig;
/**
 * @ignore
 */
RumRateLimitAdaptiveConfig.attributeTypeMap = {
    maxRetentionRate: {
        baseName: "max_retention_rate",
        type: "number",
        required: true,
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
