// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RumRateLimitCustomConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RumRateLimitCustomConfig = void 0;
/**
 * The configuration used when `mode` is `custom`.
 */
class RumRateLimitCustomConfig {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RumRateLimitCustomConfig.attributeTypeMap;
    }
}
exports.RumRateLimitCustomConfig = RumRateLimitCustomConfig;
/**
 * @ignore
 */
RumRateLimitCustomConfig.attributeTypeMap = {
    dailyResetTime: {
        baseName: "daily_reset_time",
        type: "string",
        required: true,
    },
    dailyResetTimezone: {
        baseName: "daily_reset_timezone",
        type: "string",
        required: true,
    },
    quotaReachedAction: {
        baseName: "quota_reached_action",
        type: "RumRateLimitQuotaReachedAction",
        required: true,
    },
    sessionLimit: {
        baseName: "session_limit",
        type: "number",
        required: true,
        format: "int64",
    },
    windowType: {
        baseName: "window_type",
        type: "RumRateLimitWindowType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
