// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RumSdkConfigDynamicOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RumSdkConfigDynamicOption = void 0;
/**
 * A dynamic configuration option that extracts a value at runtime using a specified strategy.
 */
class RumSdkConfigDynamicOption {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RumSdkConfigDynamicOption.attributeTypeMap;
    }
}
exports.RumSdkConfigDynamicOption = RumSdkConfigDynamicOption;
/**
 * @ignore
 */
RumSdkConfigDynamicOption.attributeTypeMap = {
    attribute: {
        baseName: "attribute",
        type: "string",
    },
    extractor: {
        baseName: "extractor",
        type: "RumSdkConfigSerializedRegex",
    },
    key: {
        baseName: "key",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    path: {
        baseName: "path",
        type: "string",
    },
    rcSerializedType: {
        baseName: "rc_serialized_type",
        type: "RumSdkConfigDynamicOptionSerializedType",
        required: true,
    },
    selector: {
        baseName: "selector",
        type: "string",
    },
    strategy: {
        baseName: "strategy",
        type: "RumSdkConfigDynamicOptionStrategy",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
