// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RumSdkConfigMeta.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RumSdkConfigMeta = void 0;
/**
 * Metadata associated with a RUM SDK configuration.
 */
class RumSdkConfigMeta {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RumSdkConfigMeta.attributeTypeMap;
    }
}
exports.RumSdkConfigMeta = RumSdkConfigMeta;
/**
 * @ignore
 */
RumSdkConfigMeta.attributeTypeMap = {
    updatedAt: {
        baseName: "updated_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    updatedBy: {
        baseName: "updated_by",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
