// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/RumSdkConfigRumUpdateAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RumSdkConfigRumUpdateAttributes = void 0;
/**
 * The RUM SDK settings to apply when updating a configuration.
 */
class RumSdkConfigRumUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return RumSdkConfigRumUpdateAttributes.attributeTypeMap;
    }
}
exports.RumSdkConfigRumUpdateAttributes = RumSdkConfigRumUpdateAttributes;
/**
 * @ignore
 */
RumSdkConfigRumUpdateAttributes.attributeTypeMap = {
    allowedTracingUrls: {
        baseName: "allowed_tracing_urls",
        type: "Array<RumSdkConfigTracingUrlConfig>",
    },
    allowedTrackingOrigins: {
        baseName: "allowed_tracking_origins",
        type: "Array<RumSdkConfigMatchOption>",
    },
    context: {
        baseName: "context",
        type: "Array<RumSdkConfigDynamicOptionPair>",
    },
    defaultPrivacyLevel: {
        baseName: "default_privacy_level",
        type: "string",
        required: true,
    },
    enablePrivacyForActionName: {
        baseName: "enable_privacy_for_action_name",
        type: "boolean",
        required: true,
    },
    env: {
        baseName: "env",
        type: "string",
    },
    service: {
        baseName: "service",
        type: "string",
    },
    sessionReplaySampleRate: {
        baseName: "session_replay_sample_rate",
        type: "number",
        required: true,
        format: "int64",
    },
    sessionSampleRate: {
        baseName: "session_sample_rate",
        type: "number",
        required: true,
        format: "int64",
    },
    traceSampleRate: {
        baseName: "trace_sample_rate",
        type: "number",
        format: "int64",
    },
    trackSessionAcrossSubdomains: {
        baseName: "track_session_across_subdomains",
        type: "boolean",
    },
    user: {
        baseName: "user",
        type: "Array<RumSdkConfigDynamicOptionPair>",
    },
    version: {
        baseName: "version",
        type: "RumSdkConfigDynamicOption",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
