// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SalesforceIncidentsOrganizationResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SalesforceIncidentsOrganizationResponseData = void 0;
/**
 * Salesforce organization data from a response.
 */
class SalesforceIncidentsOrganizationResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SalesforceIncidentsOrganizationResponseData.attributeTypeMap;
    }
}
exports.SalesforceIncidentsOrganizationResponseData = SalesforceIncidentsOrganizationResponseData;
/**
 * @ignore
 */
SalesforceIncidentsOrganizationResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SalesforceIncidentsOrganizationResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "SalesforceIncidentsOrganizationType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
