// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SalesforceIncidentsOrganizationResponseAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SalesforceIncidentsOrganizationResponseAttributes = void 0;
/**
 * Attributes of a Salesforce organization connected to the Datadog Salesforce integration.
 */
class SalesforceIncidentsOrganizationResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SalesforceIncidentsOrganizationResponseAttributes.attributeTypeMap;
    }
}
exports.SalesforceIncidentsOrganizationResponseAttributes = SalesforceIncidentsOrganizationResponseAttributes;
/**
 * @ignore
 */
SalesforceIncidentsOrganizationResponseAttributes.attributeTypeMap = {
    instanceUrl: {
        baseName: "instance_url",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    sfdcOrgId: {
        baseName: "sfdc_org_id",
        type: "string",
    },
    sfdcOrgType: {
        baseName: "sfdc_org_type",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
