// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SalesforceIncidentsTemplateCreateAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SalesforceIncidentsTemplateCreateAttributes = void 0;
/**
 * Salesforce incident template attributes for a create request.
 */
class SalesforceIncidentsTemplateCreateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SalesforceIncidentsTemplateCreateAttributes.attributeTypeMap;
    }
}
exports.SalesforceIncidentsTemplateCreateAttributes = SalesforceIncidentsTemplateCreateAttributes;
/**
 * @ignore
 */
SalesforceIncidentsTemplateCreateAttributes.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    ownerId: {
        baseName: "owner_id",
        type: "string",
        required: true,
    },
    priority: {
        baseName: "priority",
        type: "SalesforceIncidentsTemplatePriority",
        required: true,
    },
    salesforceOrgId: {
        baseName: "salesforce_org_id",
        type: "string",
        required: true,
        format: "uuid",
    },
    subject: {
        baseName: "subject",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
