// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SalesforceIncidentsTemplateResponseAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SalesforceIncidentsTemplateResponseAttributes = void 0;
/**
 * Salesforce incident template attributes returned by the API.
 */
class SalesforceIncidentsTemplateResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SalesforceIncidentsTemplateResponseAttributes.attributeTypeMap;
    }
}
exports.SalesforceIncidentsTemplateResponseAttributes = SalesforceIncidentsTemplateResponseAttributes;
/**
 * @ignore
 */
SalesforceIncidentsTemplateResponseAttributes.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    ownerId: {
        baseName: "owner_id",
        type: "string",
    },
    priority: {
        baseName: "priority",
        type: "SalesforceIncidentsTemplatePriority",
    },
    salesforceOrgId: {
        baseName: "salesforce_org_id",
        type: "string",
        format: "uuid",
    },
    subject: {
        baseName: "subject",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
