// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SalesforceIncidentsTemplateUpdateData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SalesforceIncidentsTemplateUpdateData = void 0;
/**
 * Salesforce incident template data for an update request.
 */
class SalesforceIncidentsTemplateUpdateData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SalesforceIncidentsTemplateUpdateData.attributeTypeMap;
    }
}
exports.SalesforceIncidentsTemplateUpdateData = SalesforceIncidentsTemplateUpdateData;
/**
 * @ignore
 */
SalesforceIncidentsTemplateUpdateData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SalesforceIncidentsTemplateUpdateAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "SalesforceIncidentsTemplateType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
