// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SalesforceIncidentsTemplateResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SalesforceIncidentsTemplateResponseData = void 0;
/**
 * Salesforce incident template data from a response.
 */
class SalesforceIncidentsTemplateResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SalesforceIncidentsTemplateResponseData.attributeTypeMap;
    }
}
exports.SalesforceIncidentsTemplateResponseData = SalesforceIncidentsTemplateResponseData;
/**
 * @ignore
 */
SalesforceIncidentsTemplateResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SalesforceIncidentsTemplateResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "SalesforceIncidentsTemplateType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
