// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SAMLConfigurationUpdateData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SAMLConfigurationUpdateData = void 0;
/**
 * Data for updating a SAML configuration.
 */
class SAMLConfigurationUpdateData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SAMLConfigurationUpdateData.attributeTypeMap;
    }
}
exports.SAMLConfigurationUpdateData = SAMLConfigurationUpdateData;
/**
 * @ignore
 */
SAMLConfigurationUpdateData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SAMLConfigurationUpdateAttributes",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    relationships: {
        baseName: "relationships",
        type: "SAMLConfigurationRelationships",
    },
    type: {
        baseName: "type",
        type: "SAMLConfigurationsType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
