// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SAMLConfigurationAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SAMLConfigurationAttributes = void 0;
/**
 * Attributes of a SAML configuration.
 */
class SAMLConfigurationAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SAMLConfigurationAttributes.attributeTypeMap;
    }
}
exports.SAMLConfigurationAttributes = SAMLConfigurationAttributes;
/**
 * @ignore
 */
SAMLConfigurationAttributes.attributeTypeMap = {
    assertionConsumerService: {
        baseName: "assertion_consumer_service",
        type: "Array<string>",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        format: "date-time",
    },
    entityId: {
        baseName: "entity_id",
        type: "string",
    },
    expiresAt: {
        baseName: "expires_at",
        type: "Date",
        format: "date-time",
    },
    idpInitiated: {
        baseName: "idp_initiated",
        type: "boolean",
    },
    jitDomains: {
        baseName: "jit_domains",
        type: "Array<string>",
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        format: "date-time",
    },
    ssoUrl: {
        baseName: "sso_url",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
