// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SAMLConfigurationUpdateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SAMLConfigurationUpdateAttributes = void 0;
/**
 * Attributes for updating a SAML configuration.
 */
class SAMLConfigurationUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SAMLConfigurationUpdateAttributes.attributeTypeMap;
    }
}
exports.SAMLConfigurationUpdateAttributes = SAMLConfigurationUpdateAttributes;
/**
 * @ignore
 */
SAMLConfigurationUpdateAttributes.attributeTypeMap = {
    idpInitiated: {
        baseName: "idp_initiated",
        type: "boolean",
    },
    jitDomains: {
        baseName: "jit_domains",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
