// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SampleLogGenerationBulkSubscriptionItemMeta.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SampleLogGenerationBulkSubscriptionItemMeta = void 0;
/**
 * Per-item status returned for a bulk subscription request.
 */
class SampleLogGenerationBulkSubscriptionItemMeta {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SampleLogGenerationBulkSubscriptionItemMeta.attributeTypeMap;
    }
}
exports.SampleLogGenerationBulkSubscriptionItemMeta = SampleLogGenerationBulkSubscriptionItemMeta;
/**
 * @ignore
 */
SampleLogGenerationBulkSubscriptionItemMeta.attributeTypeMap = {
    error: {
        baseName: "error",
        type: "string",
    },
    status: {
        baseName: "status",
        type: "number",
        required: true,
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
