// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SampleLogGenerationSubscriptionAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SampleLogGenerationSubscriptionAttributes = void 0;
/**
 * The attributes describing a sample log generation subscription.
 */
class SampleLogGenerationSubscriptionAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SampleLogGenerationSubscriptionAttributes.attributeTypeMap;
    }
}
exports.SampleLogGenerationSubscriptionAttributes = SampleLogGenerationSubscriptionAttributes;
/**
 * @ignore
 */
SampleLogGenerationSubscriptionAttributes.attributeTypeMap = {
    contentPackId: {
        baseName: "content_pack_id",
        type: "string",
        required: true,
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    expiresAt: {
        baseName: "expires_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    isActive: {
        baseName: "is_active",
        type: "boolean",
        required: true,
    },
    status: {
        baseName: "status",
        type: "SampleLogGenerationSubscriptionStatus",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
