// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SampleLogGenerationBulkSubscriptionAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SampleLogGenerationBulkSubscriptionAttributes = void 0;
/**
 * The attributes for creating sample log generation subscriptions for multiple content packs.
 */
class SampleLogGenerationBulkSubscriptionAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SampleLogGenerationBulkSubscriptionAttributes.attributeTypeMap;
    }
}
exports.SampleLogGenerationBulkSubscriptionAttributes = SampleLogGenerationBulkSubscriptionAttributes;
/**
 * @ignore
 */
SampleLogGenerationBulkSubscriptionAttributes.attributeTypeMap = {
    contentPackIds: {
        baseName: "content_pack_ids",
        type: "Array<string>",
        required: true,
    },
    duration: {
        baseName: "duration",
        type: "SampleLogGenerationDuration",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
