// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SampleLogGenerationSubscriptionData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SampleLogGenerationSubscriptionData = void 0;
/**
 * A sample log generation subscription.
 */
class SampleLogGenerationSubscriptionData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SampleLogGenerationSubscriptionData.attributeTypeMap;
    }
}
exports.SampleLogGenerationSubscriptionData = SampleLogGenerationSubscriptionData;
/**
 * @ignore
 */
SampleLogGenerationSubscriptionData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SampleLogGenerationSubscriptionAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "SampleLogGenerationSubscriptionResourceType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
