// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SampleLogGenerationSubscriptionsResponseMeta.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SampleLogGenerationSubscriptionsResponseMeta = void 0;
/**
 * Metadata returned alongside a list of sample log generation subscriptions.
 */
class SampleLogGenerationSubscriptionsResponseMeta {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SampleLogGenerationSubscriptionsResponseMeta.attributeTypeMap;
    }
}
exports.SampleLogGenerationSubscriptionsResponseMeta = SampleLogGenerationSubscriptionsResponseMeta;
/**
 * @ignore
 */
SampleLogGenerationSubscriptionsResponseMeta.attributeTypeMap = {
    totalSubscriptions: {
        baseName: "total_subscriptions",
        type: "number",
        required: true,
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
