// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SampleLogGenerationBulkSubscriptionResultItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SampleLogGenerationBulkSubscriptionResultItem = void 0;
/**
 * A single result entry returned by the bulk subscription endpoint.
 */
class SampleLogGenerationBulkSubscriptionResultItem {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SampleLogGenerationBulkSubscriptionResultItem.attributeTypeMap;
    }
}
exports.SampleLogGenerationBulkSubscriptionResultItem = SampleLogGenerationBulkSubscriptionResultItem;
/**
 * @ignore
 */
SampleLogGenerationBulkSubscriptionResultItem.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SampleLogGenerationSubscriptionAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    meta: {
        baseName: "meta",
        type: "SampleLogGenerationBulkSubscriptionItemMeta",
        required: true,
    },
    type: {
        baseName: "type",
        type: "SampleLogGenerationSubscriptionResourceType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
