// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SastRulesetDataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SastRulesetDataAttributes = void 0;
/**
 * The attributes of a SAST ruleset, including its name, description, and rules.
 */
class SastRulesetDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SastRulesetDataAttributes.attributeTypeMap;
    }
}
exports.SastRulesetDataAttributes = SastRulesetDataAttributes;
/**
 * @ignore
 */
SastRulesetDataAttributes.attributeTypeMap = {
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    rules: {
        baseName: "rules",
        type: "Array<GetMultipleRulesetsResponseDataAttributesRulesetsItemsRulesItems>",
        required: true,
    },
    shortDescription: {
        baseName: "short_description",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
