// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ScorecardScoreAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScorecardScoreAttributes = void 0;
/**
 * Attributes of a scorecard score.
 */
class ScorecardScoreAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ScorecardScoreAttributes.attributeTypeMap;
    }
}
exports.ScorecardScoreAttributes = ScorecardScoreAttributes;
/**
 * @ignore
 */
ScorecardScoreAttributes.attributeTypeMap = {
    aggregation: {
        baseName: "aggregation",
        type: "ScorecardScoresAggregation",
    },
    denominator: {
        baseName: "denominator",
        type: "number",
        format: "int64",
    },
    level: {
        baseName: "level",
        type: "number",
        format: "int64",
    },
    numerator: {
        baseName: "numerator",
        type: "number",
        format: "int64",
    },
    score: {
        baseName: "score",
        type: "number",
        format: "double",
    },
    totalEntities: {
        baseName: "total_entities",
        type: "number",
        format: "int64",
    },
    totalFail: {
        baseName: "total_fail",
        type: "number",
        format: "int64",
    },
    totalNoData: {
        baseName: "total_no_data",
        type: "number",
        format: "int64",
    },
    totalPass: {
        baseName: "total_pass",
        type: "number",
        format: "int64",
    },
    totalSkip: {
        baseName: "total_skip",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
