// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ScorecardScoreRelationships.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScorecardScoreRelationships = void 0;
/**
 * Relationships for a scorecard score, depending on the aggregation type.
 */
class ScorecardScoreRelationships {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ScorecardScoreRelationships.attributeTypeMap;
    }
}
exports.ScorecardScoreRelationships = ScorecardScoreRelationships;
/**
 * @ignore
 */
ScorecardScoreRelationships.attributeTypeMap = {
    entity: {
        baseName: "entity",
        type: "ScorecardScoreRelationshipItem",
    },
    rule: {
        baseName: "rule",
        type: "ScorecardScoreRelationshipItem",
    },
    scorecard: {
        baseName: "scorecard",
        type: "ScorecardScoreRelationshipItem",
    },
    service: {
        baseName: "service",
        type: "ScorecardScoreRelationshipItem",
    },
    team: {
        baseName: "team",
        type: "ScorecardScoreRelationshipItem",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
