// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ScorecardScoreData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScorecardScoreData = void 0;
/**
 * A scorecard score object for a single entity, rule, scorecard, service, or team.
 */
class ScorecardScoreData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ScorecardScoreData.attributeTypeMap;
    }
}
exports.ScorecardScoreData = ScorecardScoreData;
/**
 * @ignore
 */
ScorecardScoreData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "ScorecardScoreAttributes",
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    relationships: {
        baseName: "relationships",
        type: "ScorecardScoreRelationships",
    },
    type: {
        baseName: "type",
        type: "ScorecardScoreDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
