// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringRuleConvertBulkData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringRuleConvertBulkData = void 0;
/**
 * Data for bulk converting security monitoring rules to Terraform.
 */
class SecurityMonitoringRuleConvertBulkData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringRuleConvertBulkData.attributeTypeMap;
    }
}
exports.SecurityMonitoringRuleConvertBulkData = SecurityMonitoringRuleConvertBulkData;
/**
 * @ignore
 */
SecurityMonitoringRuleConvertBulkData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SecurityMonitoringRuleConvertBulkAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
    },
    type: {
        baseName: "type",
        type: "SecurityMonitoringRuleConvertBulkDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
