// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringIntegrationCredentialsValidateData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringIntegrationCredentialsValidateData = void 0;
/**
 * The credentials to validate.
 */
class SecurityMonitoringIntegrationCredentialsValidateData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringIntegrationCredentialsValidateData.attributeTypeMap;
    }
}
exports.SecurityMonitoringIntegrationCredentialsValidateData = SecurityMonitoringIntegrationCredentialsValidateData;
/**
 * @ignore
 */
SecurityMonitoringIntegrationCredentialsValidateData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SecurityMonitoringIntegrationCredentialsValidateAttributes",
        required: true,
    },
    type: {
        baseName: "type",
        type: "SecurityMonitoringIntegrationConfigResourceType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
