// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityFilterVersionEntry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityFilterVersionEntry = void 0;
/**
 * A single security filter as it existed at a given configuration version.
 */
class SecurityFilterVersionEntry {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityFilterVersionEntry.attributeTypeMap;
    }
}
exports.SecurityFilterVersionEntry = SecurityFilterVersionEntry;
/**
 * @ignore
 */
SecurityFilterVersionEntry.attributeTypeMap = {
    exclusionFilters: {
        baseName: "exclusion_filters",
        type: "Array<SecurityFilterExclusionFilterResponse>",
        required: true,
    },
    filteredDataType: {
        baseName: "filtered_data_type",
        type: "SecurityFilterFilteredDataType",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    isBuiltin: {
        baseName: "is_builtin",
        type: "boolean",
        required: true,
    },
    isEnabled: {
        baseName: "is_enabled",
        type: "boolean",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    query: {
        baseName: "query",
        type: "string",
        required: true,
    },
    version: {
        baseName: "version",
        type: "number",
        required: true,
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
