// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringIntegrationConfigUpdateData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringIntegrationConfigUpdateData = void 0;
/**
 * The entity context sync configuration fields to update.
 */
class SecurityMonitoringIntegrationConfigUpdateData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringIntegrationConfigUpdateData.attributeTypeMap;
    }
}
exports.SecurityMonitoringIntegrationConfigUpdateData = SecurityMonitoringIntegrationConfigUpdateData;
/**
 * @ignore
 */
SecurityMonitoringIntegrationConfigUpdateData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SecurityMonitoringIntegrationConfigUpdateAttributes",
        required: true,
    },
    type: {
        baseName: "type",
        type: "SecurityMonitoringIntegrationConfigResourceType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
