// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringDatasetDependentsAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringDatasetDependentsAttributes = void 0;
/**
 * The attributes of a dataset dependents entry.
 */
class SecurityMonitoringDatasetDependentsAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringDatasetDependentsAttributes.attributeTypeMap;
    }
}
exports.SecurityMonitoringDatasetDependentsAttributes = SecurityMonitoringDatasetDependentsAttributes;
/**
 * @ignore
 */
SecurityMonitoringDatasetDependentsAttributes.attributeTypeMap = {
    count: {
        baseName: "count",
        type: "number",
        required: true,
        format: "int64",
    },
    datasetId: {
        baseName: "datasetId",
        type: "string",
        required: true,
    },
    ids: {
        baseName: "ids",
        type: "Array<string>",
        required: true,
    },
    resourceType: {
        baseName: "resource_type",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
