// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringDatasetVersionHistoryAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringDatasetVersionHistoryAttributes = void 0;
/**
 * The attributes of a dataset version history response.
 */
class SecurityMonitoringDatasetVersionHistoryAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringDatasetVersionHistoryAttributes.attributeTypeMap;
    }
}
exports.SecurityMonitoringDatasetVersionHistoryAttributes = SecurityMonitoringDatasetVersionHistoryAttributes;
/**
 * @ignore
 */
SecurityMonitoringDatasetVersionHistoryAttributes.attributeTypeMap = {
    count: {
        baseName: "count",
        type: "number",
        required: true,
        format: "int64",
    },
    data: {
        baseName: "data",
        type: "{ [key: string]: SecurityMonitoringDatasetVersionEntry; }",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
