// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringDatasetVersionHistoryData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringDatasetVersionHistoryData = void 0;
/**
 * The data wrapper of a dataset version history response.
 */
class SecurityMonitoringDatasetVersionHistoryData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringDatasetVersionHistoryData.attributeTypeMap;
    }
}
exports.SecurityMonitoringDatasetVersionHistoryData = SecurityMonitoringDatasetVersionHistoryData;
/**
 * @ignore
 */
SecurityMonitoringDatasetVersionHistoryData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SecurityMonitoringDatasetVersionHistoryAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "SecurityMonitoringDatasetVersionHistoryType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
