// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringDatasetDependentsData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringDatasetDependentsData = void 0;
/**
 * A single entry describing the dependents of one dataset.
 */
class SecurityMonitoringDatasetDependentsData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringDatasetDependentsData.attributeTypeMap;
    }
}
exports.SecurityMonitoringDatasetDependentsData = SecurityMonitoringDatasetDependentsData;
/**
 * @ignore
 */
SecurityMonitoringDatasetDependentsData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SecurityMonitoringDatasetDependentsAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "SecurityMonitoringDatasetDependentsType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
