// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringIntegrationConfigUpdateAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringIntegrationConfigUpdateAttributes = void 0;
/**
 * Fields to update on the entity context sync configuration. All fields are optional.
 */
class SecurityMonitoringIntegrationConfigUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringIntegrationConfigUpdateAttributes.attributeTypeMap;
    }
}
exports.SecurityMonitoringIntegrationConfigUpdateAttributes = SecurityMonitoringIntegrationConfigUpdateAttributes;
/**
 * @ignore
 */
SecurityMonitoringIntegrationConfigUpdateAttributes.attributeTypeMap = {
    domain: {
        baseName: "domain",
        type: "string",
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
    },
    integrationType: {
        baseName: "integration_type",
        type: "SecurityMonitoringIntegrationType",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    secrets: {
        baseName: "secrets",
        type: "{ [key: string]: any; }",
    },
    settings: {
        baseName: "settings",
        type: "{ [key: string]: any; }",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
