// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityFindingType.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WORKLOAD_ACTIVITY = exports.STATIC_CODE_VULNERABILITY = exports.SECRET = exports.RUNTIME_CODE_VULNERABILITY = exports.MISCONFIGURATION = exports.LIBRARY_VULNERABILITY = exports.IDENTITY_RISK = exports.IAC_MISCONFIGURATION = exports.HOST_AND_CONTAINER_VULNERABILITY = exports.ATTACK_PATH = exports.API_SECURITY = void 0;
exports.API_SECURITY = "api_security";
exports.ATTACK_PATH = "attack_path";
exports.HOST_AND_CONTAINER_VULNERABILITY = "host_and_container_vulnerability";
exports.IAC_MISCONFIGURATION = "iac_misconfiguration";
exports.IDENTITY_RISK = "identity_risk";
exports.LIBRARY_VULNERABILITY = "library_vulnerability";
exports.MISCONFIGURATION = "misconfiguration";
exports.RUNTIME_CODE_VULNERABILITY = "runtime_code_vulnerability";
exports.SECRET = "secret";
exports.STATIC_CODE_VULNERABILITY = "static_code_vulnerability";
exports.WORKLOAD_ACTIVITY = "workload_activity";
