// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringIntegrationConfigData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringIntegrationConfigData = void 0;
/**
 * An entity context sync configuration.
 */
class SecurityMonitoringIntegrationConfigData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringIntegrationConfigData.attributeTypeMap;
    }
}
exports.SecurityMonitoringIntegrationConfigData = SecurityMonitoringIntegrationConfigData;
/**
 * @ignore
 */
SecurityMonitoringIntegrationConfigData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SecurityMonitoringIntegrationConfigAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "SecurityMonitoringIntegrationConfigResourceType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
