// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringIntegrationConfigCreateData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringIntegrationConfigCreateData = void 0;
/**
 * The entity context sync configuration to create.
 */
class SecurityMonitoringIntegrationConfigCreateData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringIntegrationConfigCreateData.attributeTypeMap;
    }
}
exports.SecurityMonitoringIntegrationConfigCreateData = SecurityMonitoringIntegrationConfigCreateData;
/**
 * @ignore
 */
SecurityMonitoringIntegrationConfigCreateData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SecurityMonitoringIntegrationConfigCreateAttributes",
        required: true,
    },
    type: {
        baseName: "type",
        type: "SecurityMonitoringIntegrationConfigResourceType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
