// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringIntegrationConfigAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringIntegrationConfigAttributes = void 0;
/**
 * The attributes of an entity context sync configuration as returned by the API.
 */
class SecurityMonitoringIntegrationConfigAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringIntegrationConfigAttributes.attributeTypeMap;
    }
}
exports.SecurityMonitoringIntegrationConfigAttributes = SecurityMonitoringIntegrationConfigAttributes;
/**
 * @ignore
 */
SecurityMonitoringIntegrationConfigAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        format: "date-time",
    },
    domain: {
        baseName: "domain",
        type: "string",
        required: true,
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
        required: true,
    },
    integrationType: {
        baseName: "integration_type",
        type: "SecurityMonitoringIntegrationType",
        required: true,
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        format: "date-time",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    settings: {
        baseName: "settings",
        type: "{ [key: string]: any; }",
    },
    state: {
        baseName: "state",
        type: "SecurityMonitoringIntegrationConfigState",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
