// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringDatasetsListMeta.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringDatasetsListMeta = void 0;
/**
 * Metadata returned with a list of datasets.
 */
class SecurityMonitoringDatasetsListMeta {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringDatasetsListMeta.attributeTypeMap;
    }
}
exports.SecurityMonitoringDatasetsListMeta = SecurityMonitoringDatasetsListMeta;
/**
 * @ignore
 */
SecurityMonitoringDatasetsListMeta.attributeTypeMap = {
    totalCount: {
        baseName: "totalCount",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
