// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringDatasetDefinition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringDatasetDefinition = void 0;
/**
 * The definition of the dataset. The shape depends on the value of `data_source`.
 * Use `reference_table` or `managed_resource` for a referential dataset, or one of the
 * event platform sources (for example `logs`, `audit`, `events`, `spans`, `rum`) for
 * an event platform dataset.
 */
class SecurityMonitoringDatasetDefinition {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringDatasetDefinition.attributeTypeMap;
    }
}
exports.SecurityMonitoringDatasetDefinition = SecurityMonitoringDatasetDefinition;
/**
 * @ignore
 */
SecurityMonitoringDatasetDefinition.attributeTypeMap = {
    columns: {
        baseName: "columns",
        type: "Array<SecurityMonitoringDatasetColumn>",
    },
    dataSource: {
        baseName: "data_source",
        type: "string",
        required: true,
    },
    indexes: {
        baseName: "indexes",
        type: "Array<string>",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    queryFilter: {
        baseName: "query_filter",
        type: "string",
    },
    search: {
        baseName: "search",
        type: "SecurityMonitoringDatasetSearch",
    },
    storage: {
        baseName: "storage",
        type: "string",
    },
    tableName: {
        baseName: "table_name",
        type: "string",
    },
    timeWindow: {
        baseName: "time_window",
        type: "SecurityMonitoringDatasetTimeWindow",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
