// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityAutomationRulesLinks.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityAutomationRulesLinks = void 0;
/**
 * Pagination links for the list of automation rules.
 */
class SecurityAutomationRulesLinks {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityAutomationRulesLinks.attributeTypeMap;
    }
}
exports.SecurityAutomationRulesLinks = SecurityAutomationRulesLinks;
/**
 * @ignore
 */
SecurityAutomationRulesLinks.attributeTypeMap = {
    first: {
        baseName: "first",
        type: "string",
        required: true,
    },
    last: {
        baseName: "last",
        type: "string",
        required: true,
    },
    next: {
        baseName: "next",
        type: "string",
    },
    prev: {
        baseName: "prev",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
