// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringIntegrationConfigCreateAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringIntegrationConfigCreateAttributes = void 0;
/**
 * The attributes of the entity context sync configuration to create.
 */
class SecurityMonitoringIntegrationConfigCreateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringIntegrationConfigCreateAttributes.attributeTypeMap;
    }
}
exports.SecurityMonitoringIntegrationConfigCreateAttributes = SecurityMonitoringIntegrationConfigCreateAttributes;
/**
 * @ignore
 */
SecurityMonitoringIntegrationConfigCreateAttributes.attributeTypeMap = {
    domain: {
        baseName: "domain",
        type: "string",
        required: true,
    },
    integrationType: {
        baseName: "integration_type",
        type: "SecurityMonitoringIntegrationType",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    secrets: {
        baseName: "secrets",
        type: "{ [key: string]: any; }",
        required: true,
    },
    settings: {
        baseName: "settings",
        type: "{ [key: string]: any; }",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
