// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringDatasetVersionFieldChange.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringDatasetVersionFieldChange = void 0;
/**
 * A single field change between two versions of a dataset.
 */
class SecurityMonitoringDatasetVersionFieldChange {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringDatasetVersionFieldChange.attributeTypeMap;
    }
}
exports.SecurityMonitoringDatasetVersionFieldChange = SecurityMonitoringDatasetVersionFieldChange;
/**
 * @ignore
 */
SecurityMonitoringDatasetVersionFieldChange.attributeTypeMap = {
    current: {
        baseName: "current",
        type: "any",
        required: true,
    },
    field: {
        baseName: "field",
        type: "string",
        required: true,
    },
    previous: {
        baseName: "previous",
        type: "any",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
