// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringDatasetDependenciesRequestAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringDatasetDependenciesRequestAttributes = void 0;
/**
 * The attributes of a dataset dependencies request.
 */
class SecurityMonitoringDatasetDependenciesRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringDatasetDependenciesRequestAttributes.attributeTypeMap;
    }
}
exports.SecurityMonitoringDatasetDependenciesRequestAttributes = SecurityMonitoringDatasetDependenciesRequestAttributes;
/**
 * @ignore
 */
SecurityMonitoringDatasetDependenciesRequestAttributes.attributeTypeMap = {
    datasetIds: {
        baseName: "datasetIds",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
