// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityAutomationRulesPageInfo.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityAutomationRulesPageInfo = void 0;
/**
 * Pagination information for the list of automation rules.
 */
class SecurityAutomationRulesPageInfo {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityAutomationRulesPageInfo.attributeTypeMap;
    }
}
exports.SecurityAutomationRulesPageInfo = SecurityAutomationRulesPageInfo;
/**
 * @ignore
 */
SecurityAutomationRulesPageInfo.attributeTypeMap = {
    totalFilteredCount: {
        baseName: "total_filtered_count",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
