// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringDatasetAttributesResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringDatasetAttributesResponse = void 0;
/**
 * The attributes of a Cloud SIEM dataset.
 */
class SecurityMonitoringDatasetAttributesResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringDatasetAttributesResponse.attributeTypeMap;
    }
}
exports.SecurityMonitoringDatasetAttributesResponse = SecurityMonitoringDatasetAttributesResponse;
/**
 * @ignore
 */
SecurityMonitoringDatasetAttributesResponse.attributeTypeMap = {
    createdAt: {
        baseName: "createdAt",
        type: "string",
        required: true,
    },
    createdByHandle: {
        baseName: "createdByHandle",
        type: "string",
        required: true,
    },
    createdByName: {
        baseName: "createdByName",
        type: "string",
        required: true,
    },
    definition: {
        baseName: "definition",
        type: "SecurityMonitoringDatasetDefinition",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    isDefault: {
        baseName: "isDefault",
        type: "boolean",
        required: true,
    },
    isDeprecated: {
        baseName: "isDeprecated",
        type: "boolean",
        required: true,
    },
    modifiedAt: {
        baseName: "modifiedAt",
        type: "string",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    updatedByHandle: {
        baseName: "updatedByHandle",
        type: "string",
        required: true,
    },
    updatedByName: {
        baseName: "updatedByName",
        type: "string",
        required: true,
    },
    version: {
        baseName: "version",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
