// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringIntegrationCredentialsValidateAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringIntegrationCredentialsValidateAttributes = void 0;
/**
 * The credentials to validate against the external entity source.
 */
class SecurityMonitoringIntegrationCredentialsValidateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringIntegrationCredentialsValidateAttributes.attributeTypeMap;
    }
}
exports.SecurityMonitoringIntegrationCredentialsValidateAttributes = SecurityMonitoringIntegrationCredentialsValidateAttributes;
/**
 * @ignore
 */
SecurityMonitoringIntegrationCredentialsValidateAttributes.attributeTypeMap = {
    domain: {
        baseName: "domain",
        type: "string",
        required: true,
    },
    integrationType: {
        baseName: "integration_type",
        type: "SecurityMonitoringIntegrationType",
        required: true,
    },
    secrets: {
        baseName: "secrets",
        type: "{ [key: string]: any; }",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
