// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityFilterVersionAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityFilterVersionAttributes = void 0;
/**
 * The attributes describing a single security filter configuration version.
 */
class SecurityFilterVersionAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityFilterVersionAttributes.attributeTypeMap;
    }
}
exports.SecurityFilterVersionAttributes = SecurityFilterVersionAttributes;
/**
 * @ignore
 */
SecurityFilterVersionAttributes.attributeTypeMap = {
    date: {
        baseName: "date",
        type: "number",
        required: true,
        format: "int64",
    },
    filters: {
        baseName: "filters",
        type: "Array<SecurityFilterVersionEntry>",
        required: true,
    },
    version: {
        baseName: "version",
        type: "number",
        required: true,
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
