// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringRuleConvertBulkAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringRuleConvertBulkAttributes = void 0;
/**
 * Attributes for bulk converting security monitoring rules to Terraform.
 */
class SecurityMonitoringRuleConvertBulkAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringRuleConvertBulkAttributes.attributeTypeMap;
    }
}
exports.SecurityMonitoringRuleConvertBulkAttributes = SecurityMonitoringRuleConvertBulkAttributes;
/**
 * @ignore
 */
SecurityMonitoringRuleConvertBulkAttributes.attributeTypeMap = {
    ruleIds: {
        baseName: "ruleIds",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
