// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SecurityMonitoringDatasetAttributesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityMonitoringDatasetAttributesRequest = void 0;
/**
 * The attributes of a dataset create or update request.
 */
class SecurityMonitoringDatasetAttributesRequest {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SecurityMonitoringDatasetAttributesRequest.attributeTypeMap;
    }
}
exports.SecurityMonitoringDatasetAttributesRequest = SecurityMonitoringDatasetAttributesRequest;
/**
 * @ignore
 */
SecurityMonitoringDatasetAttributesRequest.attributeTypeMap = {
    definition: {
        baseName: "definition",
        type: "SecurityMonitoringDatasetDefinition",
        required: true,
    },
    description: {
        baseName: "description",
        type: "string",
    },
    version: {
        baseName: "version",
        type: "number",
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
