// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ServiceRepositoryInfoRequestAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceRepositoryInfoRequestAttributes = void 0;
/**
 * Attributes for the service repository info request.
 */
class ServiceRepositoryInfoRequestAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ServiceRepositoryInfoRequestAttributes.attributeTypeMap;
    }
}
exports.ServiceRepositoryInfoRequestAttributes = ServiceRepositoryInfoRequestAttributes;
/**
 * @ignore
 */
ServiceRepositoryInfoRequestAttributes.attributeTypeMap = {
    service: {
        baseName: "service",
        type: "string",
        required: true,
    },
    version: {
        baseName: "version",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
