// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ServiceAccountAccessTokenUpdateData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceAccountAccessTokenUpdateData = void 0;
/**
 * Object used to update a service account access token.
 */
class ServiceAccountAccessTokenUpdateData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ServiceAccountAccessTokenUpdateData.attributeTypeMap;
    }
}
exports.ServiceAccountAccessTokenUpdateData = ServiceAccountAccessTokenUpdateData;
/**
 * @ignore
 */
ServiceAccountAccessTokenUpdateData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "ServiceAccountAccessTokenUpdateAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "ServiceAccessTokensType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
