// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ServiceAccountAccessTokenUpdateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceAccountAccessTokenUpdateAttributes = void 0;
/**
 * Attributes used to update a service account access token.
 */
class ServiceAccountAccessTokenUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ServiceAccountAccessTokenUpdateAttributes.attributeTypeMap;
    }
}
exports.ServiceAccountAccessTokenUpdateAttributes = ServiceAccountAccessTokenUpdateAttributes;
/**
 * @ignore
 */
ServiceAccountAccessTokenUpdateAttributes.attributeTypeMap = {
    name: {
        baseName: "name",
        type: "string",
    },
    scopes: {
        baseName: "scopes",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
