// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ServiceRepositoryInfoResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceRepositoryInfoResponseData = void 0;
/**
 * Data object for the service repository info response.
 */
class ServiceRepositoryInfoResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ServiceRepositoryInfoResponseData.attributeTypeMap;
    }
}
exports.ServiceRepositoryInfoResponseData = ServiceRepositoryInfoResponseData;
/**
 * @ignore
 */
ServiceRepositoryInfoResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "ServiceRepositoryInfoResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "ServiceRepositoryInfoDataType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
