// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/ServiceAccessTokenAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceAccessTokenAttributes = void 0;
/**
 * Attributes of an access token.
 */
class ServiceAccessTokenAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return ServiceAccessTokenAttributes.attributeTypeMap;
    }
}
exports.ServiceAccessTokenAttributes = ServiceAccessTokenAttributes;
/**
 * @ignore
 */
ServiceAccessTokenAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        format: "date-time",
    },
    expiresAt: {
        baseName: "expires_at",
        type: "Date",
        format: "date-time",
    },
    lastUsedAt: {
        baseName: "last_used_at",
        type: "Date",
        format: "date-time",
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        format: "date-time",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    publicPortion: {
        baseName: "public_portion",
        type: "string",
    },
    scopes: {
        baseName: "scopes",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
