// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SharedDashboardResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedDashboardResponse = void 0;
/**
 * A shared dashboard response resource.
 */
class SharedDashboardResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SharedDashboardResponse.attributeTypeMap;
    }
}
exports.SharedDashboardResponse = SharedDashboardResponse;
/**
 * @ignore
 */
SharedDashboardResponse.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "SharedDashboardResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    relationships: {
        baseName: "relationships",
        type: "SharedDashboardRelationships",
        required: true,
    },
    type: {
        baseName: "type",
        type: "SharedDashboardType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
