// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SharedDashboardResponseAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedDashboardResponseAttributes = void 0;
/**
 * Attributes of a shared dashboard response.
 */
class SharedDashboardResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SharedDashboardResponseAttributes.attributeTypeMap;
    }
}
exports.SharedDashboardResponseAttributes = SharedDashboardResponseAttributes;
/**
 * @ignore
 */
SharedDashboardResponseAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    embeddableDomains: {
        baseName: "embeddable_domains",
        type: "Array<string>",
        required: true,
    },
    expiration: {
        baseName: "expiration",
        type: "Date",
        required: true,
        format: "date-time",
    },
    globalTime: {
        baseName: "global_time",
        type: "{ [key: string]: any; }",
        required: true,
    },
    globalTimeSelectable: {
        baseName: "global_time_selectable",
        type: "boolean",
        required: true,
    },
    invitees: {
        baseName: "invitees",
        type: "Array<SharedDashboardInvitee>",
        required: true,
    },
    lastAccessed: {
        baseName: "last_accessed",
        type: "Date",
        required: true,
        format: "date-time",
    },
    selectableTemplateVars: {
        baseName: "selectable_template_vars",
        type: "Array<SharedDashboardSelectableTemplateVariable>",
        required: true,
    },
    shareType: {
        baseName: "share_type",
        type: "SharedDashboardShareType",
        required: true,
    },
    sharerDisabled: {
        baseName: "sharer_disabled",
        type: "boolean",
        required: true,
    },
    status: {
        baseName: "status",
        type: "SharedDashboardStatus",
        required: true,
    },
    title: {
        baseName: "title",
        type: "string",
        required: true,
    },
    token: {
        baseName: "token",
        type: "string",
        required: true,
    },
    url: {
        baseName: "url",
        type: "string",
        required: true,
    },
    viewingPreferences: {
        baseName: "viewing_preferences",
        type: "SharedDashboardViewingPreferences",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
