// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SharedDashboardSelectableTemplateVariable.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedDashboardSelectableTemplateVariable = void 0;
/**
 * A template variable that viewers can modify on the shared dashboard.
 */
class SharedDashboardSelectableTemplateVariable {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SharedDashboardSelectableTemplateVariable.attributeTypeMap;
    }
}
exports.SharedDashboardSelectableTemplateVariable = SharedDashboardSelectableTemplateVariable;
/**
 * @ignore
 */
SharedDashboardSelectableTemplateVariable.attributeTypeMap = {
    allowAnyValue: {
        baseName: "allow_any_value",
        type: "boolean",
        required: true,
    },
    defaultValues: {
        baseName: "default_values",
        type: "Array<string>",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    prefix: {
        baseName: "prefix",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "string",
        required: true,
    },
    visibleTags: {
        baseName: "visible_tags",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
