// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SharedDashboardInvitee.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedDashboardInvitee = void 0;
/**
 * Invitee that can access an invite-only shared dashboard.
 */
class SharedDashboardInvitee {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SharedDashboardInvitee.attributeTypeMap;
    }
}
exports.SharedDashboardInvitee = SharedDashboardInvitee;
/**
 * @ignore
 */
SharedDashboardInvitee.attributeTypeMap = {
    accessExpiration: {
        baseName: "access_expiration",
        type: "Date",
        required: true,
        format: "date-time",
    },
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    email: {
        baseName: "email",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
