// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SharedDashboardIncludedDashboardAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedDashboardIncludedDashboardAttributes = void 0;
/**
 * Attributes of the included dashboard.
 */
class SharedDashboardIncludedDashboardAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SharedDashboardIncludedDashboardAttributes.attributeTypeMap;
    }
}
exports.SharedDashboardIncludedDashboardAttributes = SharedDashboardIncludedDashboardAttributes;
/**
 * @ignore
 */
SharedDashboardIncludedDashboardAttributes.attributeTypeMap = {
    title: {
        baseName: "title",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
