// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SignalsProblemsDetections.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SignalsProblemsDetections = void 0;
/**
 * Grouped detection results by detection type.
 */
class SignalsProblemsDetections {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SignalsProblemsDetections.attributeTypeMap;
    }
}
exports.SignalsProblemsDetections = SignalsProblemsDetections;
/**
 * @ignore
 */
SignalsProblemsDetections.attributeTypeMap = {
    highFrozenFrameRates: {
        baseName: "high_frozen_frame_rates",
        type: "Array<AggregatedHighFrozenFrameRate>",
    },
    highScriptEvaluations: {
        baseName: "high_script_evaluations",
        type: "Array<AggregatedHighScriptEval>",
    },
    lowCacheHitRates: {
        baseName: "low_cache_hit_rates",
        type: "Array<AggregatedLowCacheHitRate>",
    },
    mobileScrollFrictions: {
        baseName: "mobile_scroll_frictions",
        type: "Array<AggregatedMobileScrollFriction>",
    },
    slowFcpHighBytes: {
        baseName: "slow_fcp_high_bytes",
        type: "Array<AggregatedSlowFCPHighBytes>",
    },
    slowInteractionLongTasks: {
        baseName: "slow_interaction_long_tasks",
        type: "Array<AggregatedSlowInteractionLongTask>",
    },
    uncompressedResources: {
        baseName: "uncompressed_resources",
        type: "Array<AggregatedUncompressedResource>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
