// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SignalsProblemsSampleMetadata.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SignalsProblemsSampleMetadata = void 0;
/**
 * Metadata about the sampling quality for a signals and problems query.
 */
class SignalsProblemsSampleMetadata {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SignalsProblemsSampleMetadata.attributeTypeMap;
    }
}
exports.SignalsProblemsSampleMetadata = SignalsProblemsSampleMetadata;
/**
 * @ignore
 */
SignalsProblemsSampleMetadata.attributeTypeMap = {
    failed: {
        baseName: "failed",
        type: "number",
        required: true,
        format: "int32",
    },
    requested: {
        baseName: "requested",
        type: "number",
        required: true,
        format: "int32",
    },
    sampledViewIds: {
        baseName: "sampled_view_ids",
        type: "Array<string>",
        required: true,
    },
    succeeded: {
        baseName: "succeeded",
        type: "number",
        required: true,
        format: "int32",
    },
    successRate: {
        baseName: "success_rate",
        type: "number",
        required: true,
        format: "double",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
