// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SourcemapsListMetaPage.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SourcemapsListMetaPage = void 0;
/**
 * Page information for the source maps list response.
 */
class SourcemapsListMetaPage {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SourcemapsListMetaPage.attributeTypeMap;
    }
}
exports.SourcemapsListMetaPage = SourcemapsListMetaPage;
/**
 * @ignore
 */
SourcemapsListMetaPage.attributeTypeMap = {
    hasMoreResults: {
        baseName: "has_more_results",
        type: "boolean",
        required: true,
    },
    totalFilteredCount: {
        baseName: "total_filtered_count",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
