// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SourcemapFileAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SourcemapFileAttributes = void 0;
/**
 * Attributes of a JavaScript source map file.
 */
class SourcemapFileAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SourcemapFileAttributes.attributeTypeMap;
    }
}
exports.SourcemapFileAttributes = SourcemapFileAttributes;
/**
 * @ignore
 */
SourcemapFileAttributes.attributeTypeMap = {
    file: {
        baseName: "file",
        type: "string",
        required: true,
    },
    mappings: {
        baseName: "mappings",
        type: "string",
        required: true,
    },
    minifiedLineLengths: {
        baseName: "minifiedLineLengths",
        type: "Array<number>",
        required: true,
    },
    names: {
        baseName: "names",
        type: "Array<any>",
        required: true,
    },
    sourceRoot: {
        baseName: "sourceRoot",
        type: "string",
        required: true,
    },
    sources: {
        baseName: "sources",
        type: "Array<string>",
        required: true,
    },
    sourcesContent: {
        baseName: "sourcesContent",
        type: "Array<string>",
        required: true,
    },
    version: {
        baseName: "version",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
