// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/StatuspageUrlSettingCreateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StatuspageUrlSettingCreateAttributes = void 0;
/**
 * The Statuspage URL setting attributes for a create request.
 */
class StatuspageUrlSettingCreateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return StatuspageUrlSettingCreateAttributes.attributeTypeMap;
    }
}
exports.StatuspageUrlSettingCreateAttributes = StatuspageUrlSettingCreateAttributes;
/**
 * @ignore
 */
StatuspageUrlSettingCreateAttributes.attributeTypeMap = {
    customTags: {
        baseName: "custom_tags",
        type: "string",
        required: true,
    },
    url: {
        baseName: "url",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
