// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/StatuspageUrlSettingResponseAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StatuspageUrlSettingResponseAttributes = void 0;
/**
 * The attributes from a Statuspage URL setting response.
 */
class StatuspageUrlSettingResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return StatuspageUrlSettingResponseAttributes.attributeTypeMap;
    }
}
exports.StatuspageUrlSettingResponseAttributes = StatuspageUrlSettingResponseAttributes;
/**
 * @ignore
 */
StatuspageUrlSettingResponseAttributes.attributeTypeMap = {
    customTags: {
        baseName: "custom_tags",
        type: "string",
    },
    url: {
        baseName: "url",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
