// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/StatuspageUrlSettingUpdateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StatuspageUrlSettingUpdateAttributes = void 0;
/**
 * The Statuspage URL setting attributes for an update request.
 */
class StatuspageUrlSettingUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return StatuspageUrlSettingUpdateAttributes.attributeTypeMap;
    }
}
exports.StatuspageUrlSettingUpdateAttributes = StatuspageUrlSettingUpdateAttributes;
/**
 * @ignore
 */
StatuspageUrlSettingUpdateAttributes.attributeTypeMap = {
    customTags: {
        baseName: "custom_tags",
        type: "string",
    },
    url: {
        baseName: "url",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
