// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/StegadographyWidgetAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StegadographyWidgetAttributes = void 0;
/**
 * Attributes of a watermarked widget recovered from an image.
 */
class StegadographyWidgetAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return StegadographyWidgetAttributes.attributeTypeMap;
    }
}
exports.StegadographyWidgetAttributes = StegadographyWidgetAttributes;
/**
 * @ignore
 */
StegadographyWidgetAttributes.attributeTypeMap = {
    locationx: {
        baseName: "locationx",
        type: "number",
        required: true,
        format: "int64",
    },
    locationy: {
        baseName: "locationy",
        type: "number",
        required: true,
        format: "int64",
    },
    rawData: {
        baseName: "rawData",
        type: "string",
        required: true,
    },
    watermark: {
        baseName: "watermark",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
