// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/SummarizedSpan.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SummarizedSpan = void 0;
/**
 * A node in the pruned trace tree.
 */
class SummarizedSpan {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return SummarizedSpan.attributeTypeMap;
    }
}
exports.SummarizedSpan = SummarizedSpan;
/**
 * @ignore
 */
SummarizedSpan.attributeTypeMap = {
    children: {
        baseName: "children",
        type: "Array<SummarizedSpan>",
        required: true,
    },
    durationSeconds: {
        baseName: "durationSeconds",
        type: "number",
        required: true,
        format: "double",
    },
    endTime: {
        baseName: "endTime",
        type: "Date",
        required: true,
        format: "date-time",
    },
    error: {
        baseName: "error",
        type: "APMSpanErrorFlag",
        required: true,
    },
    hiddenChildSpansCount: {
        baseName: "hidden_child_spans_count",
        type: "number",
        required: true,
        format: "int32",
    },
    meta: {
        baseName: "meta",
        type: "{ [key: string]: string; }",
        required: true,
    },
    metrics: {
        baseName: "metrics",
        type: "{ [key: string]: number; }",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    parentId: {
        baseName: "parentID",
        type: "number",
        required: true,
        format: "int64",
    },
    resource: {
        baseName: "resource",
        type: "string",
        required: true,
    },
    service: {
        baseName: "service",
        type: "string",
        required: true,
    },
    spanId: {
        baseName: "spanID",
        type: "number",
        required: true,
        format: "int64",
    },
    spanKind: {
        baseName: "span_kind",
        type: "string",
        required: true,
    },
    startTime: {
        baseName: "startTime",
        type: "Date",
        required: true,
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
