// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TagIndexingRuleExemptionAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagIndexingRuleExemptionAttributes = void 0;
/**
 * Attributes of a tag indexing rule exemption.
 */
class TagIndexingRuleExemptionAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TagIndexingRuleExemptionAttributes.attributeTypeMap;
    }
}
exports.TagIndexingRuleExemptionAttributes = TagIndexingRuleExemptionAttributes;
/**
 * @ignore
 */
TagIndexingRuleExemptionAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        format: "date-time",
    },
    createdByHandle: {
        baseName: "created_by_handle",
        type: "string",
    },
    kind: {
        baseName: "kind",
        type: "string",
    },
    reason: {
        baseName: "reason",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
