// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TagIndexingRuleDynamicTags.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagIndexingRuleDynamicTags = void 0;
/**
 * Configuration for including dynamically queried tags.
 */
class TagIndexingRuleDynamicTags {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TagIndexingRuleDynamicTags.attributeTypeMap;
    }
}
exports.TagIndexingRuleDynamicTags = TagIndexingRuleDynamicTags;
/**
 * @ignore
 */
TagIndexingRuleDynamicTags.attributeTypeMap = {
    queriedTagsWindowSeconds: {
        baseName: "queried_tags_window_seconds",
        type: "number",
        format: "int64",
    },
    relatedAssetTags: {
        baseName: "related_asset_tags",
        type: "boolean",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
