// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TagIndexingRuleOptionsData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagIndexingRuleOptionsData = void 0;
/**
 * Data payload for tag indexing rule options.
 */
class TagIndexingRuleOptionsData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TagIndexingRuleOptionsData.attributeTypeMap;
    }
}
exports.TagIndexingRuleOptionsData = TagIndexingRuleOptionsData;
/**
 * @ignore
 */
TagIndexingRuleOptionsData.attributeTypeMap = {
    dynamicTags: {
        baseName: "dynamic_tags",
        type: "TagIndexingRuleDynamicTags",
    },
    managePreexistingMetrics: {
        baseName: "manage_preexisting_metrics",
        type: "boolean",
    },
    metricMatch: {
        baseName: "metric_match",
        type: "TagIndexingRuleMetricMatch",
    },
    overridePreviousRules: {
        baseName: "override_previous_rules",
        type: "boolean",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
