// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TagIndexingRuleExemptionCreateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagIndexingRuleExemptionCreateAttributes = void 0;
/**
 * Attributes for creating a tag indexing rule exemption.
 */
class TagIndexingRuleExemptionCreateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TagIndexingRuleExemptionCreateAttributes.attributeTypeMap;
    }
}
exports.TagIndexingRuleExemptionCreateAttributes = TagIndexingRuleExemptionCreateAttributes;
/**
 * @ignore
 */
TagIndexingRuleExemptionCreateAttributes.attributeTypeMap = {
    reason: {
        baseName: "reason",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
