// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TagIndexingRuleCreateAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagIndexingRuleCreateAttributes = void 0;
/**
 * Attributes for creating a tag indexing rule.
 */
class TagIndexingRuleCreateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TagIndexingRuleCreateAttributes.attributeTypeMap;
    }
}
exports.TagIndexingRuleCreateAttributes = TagIndexingRuleCreateAttributes;
/**
 * @ignore
 */
TagIndexingRuleCreateAttributes.attributeTypeMap = {
    excludeTagsMode: {
        baseName: "exclude_tags_mode",
        type: "boolean",
    },
    ignoredMetricNameMatches: {
        baseName: "ignored_metric_name_matches",
        type: "Array<string>",
    },
    metricNameMatches: {
        baseName: "metric_name_matches",
        type: "Array<string>",
        required: true,
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    options: {
        baseName: "options",
        type: "TagIndexingRuleOptions",
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
