// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TagIndexingRuleMetricMatch.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagIndexingRuleMetricMatch = void 0;
/**
 * Criteria for matching metrics based on query state.
 */
class TagIndexingRuleMetricMatch {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TagIndexingRuleMetricMatch.attributeTypeMap;
    }
}
exports.TagIndexingRuleMetricMatch = TagIndexingRuleMetricMatch;
/**
 * @ignore
 */
TagIndexingRuleMetricMatch.attributeTypeMap = {
    isQueried: {
        baseName: "is_queried",
        type: "boolean",
    },
    notQueried: {
        baseName: "not_queried",
        type: "boolean",
    },
    notUsedInAssets: {
        baseName: "not_used_in_assets",
        type: "boolean",
    },
    queriedWindowSeconds: {
        baseName: "queried_window_seconds",
        type: "number",
        format: "int64",
    },
    usedInAssets: {
        baseName: "used_in_assets",
        type: "boolean",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
