// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TagIndexingRuleAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagIndexingRuleAttributes = void 0;
/**
 * Attributes of a tag indexing rule.
 */
class TagIndexingRuleAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TagIndexingRuleAttributes.attributeTypeMap;
    }
}
exports.TagIndexingRuleAttributes = TagIndexingRuleAttributes;
/**
 * @ignore
 */
TagIndexingRuleAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        format: "date-time",
    },
    createdByHandle: {
        baseName: "created_by_handle",
        type: "string",
    },
    excludeTagsMode: {
        baseName: "exclude_tags_mode",
        type: "boolean",
    },
    ignoredMetricNameMatches: {
        baseName: "ignored_metric_name_matches",
        type: "Array<string>",
    },
    metricNameMatches: {
        baseName: "metric_name_matches",
        type: "Array<string>",
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        format: "date-time",
    },
    modifiedByHandle: {
        baseName: "modified_by_handle",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    options: {
        baseName: "options",
        type: "TagIndexingRuleOptions",
    },
    ruleOrder: {
        baseName: "rule_order",
        type: "number",
        format: "int64",
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
