// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TagIndexingRuleUpdateAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagIndexingRuleUpdateAttributes = void 0;
/**
 * Attributes for updating a tag indexing rule. All fields are optional; omitted fields are unchanged.
 */
class TagIndexingRuleUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TagIndexingRuleUpdateAttributes.attributeTypeMap;
    }
}
exports.TagIndexingRuleUpdateAttributes = TagIndexingRuleUpdateAttributes;
/**
 * @ignore
 */
TagIndexingRuleUpdateAttributes.attributeTypeMap = {
    excludeTagsMode: {
        baseName: "exclude_tags_mode",
        type: "boolean",
    },
    ignoredMetricNameMatches: {
        baseName: "ignored_metric_name_matches",
        type: "Array<string>",
    },
    metricNameMatches: {
        baseName: "metric_name_matches",
        type: "Array<string>",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    options: {
        baseName: "options",
        type: "TagIndexingRuleOptions",
    },
    ruleOrder: {
        baseName: "rule_order",
        type: "number",
        format: "int64",
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
