// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TagPolicyAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagPolicyAttributes = void 0;
/**
 * The attributes of a tag policy resource.
 */
class TagPolicyAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TagPolicyAttributes.attributeTypeMap;
    }
}
exports.TagPolicyAttributes = TagPolicyAttributes;
/**
 * @ignore
 */
TagPolicyAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "created_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    createdBy: {
        baseName: "created_by",
        type: "string",
        required: true,
    },
    deletedAt: {
        baseName: "deleted_at",
        type: "Date",
        format: "date-time",
    },
    deletedBy: {
        baseName: "deleted_by",
        type: "string",
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
        required: true,
    },
    modifiedAt: {
        baseName: "modified_at",
        type: "Date",
        required: true,
        format: "date-time",
    },
    modifiedBy: {
        baseName: "modified_by",
        type: "string",
        required: true,
    },
    negated: {
        baseName: "negated",
        type: "boolean",
        required: true,
    },
    policyName: {
        baseName: "policy_name",
        type: "string",
        required: true,
    },
    policyType: {
        baseName: "policy_type",
        type: "TagPolicyType",
        required: true,
    },
    required: {
        baseName: "required",
        type: "boolean",
        required: true,
    },
    scope: {
        baseName: "scope",
        type: "string",
        required: true,
    },
    source: {
        baseName: "source",
        type: "TagPolicySource",
        required: true,
    },
    tagKey: {
        baseName: "tag_key",
        type: "string",
        required: true,
    },
    tagValuePatterns: {
        baseName: "tag_value_patterns",
        type: "Array<string>",
        required: true,
    },
    version: {
        baseName: "version",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
