// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TagPolicyUpdateAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagPolicyUpdateAttributes = void 0;
/**
 * Mutable attributes of a tag policy. Each field is optional; omitting a field leaves its
 * current value unchanged. The `source` of a policy cannot be changed.
 */
class TagPolicyUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TagPolicyUpdateAttributes.attributeTypeMap;
    }
}
exports.TagPolicyUpdateAttributes = TagPolicyUpdateAttributes;
/**
 * @ignore
 */
TagPolicyUpdateAttributes.attributeTypeMap = {
    enabled: {
        baseName: "enabled",
        type: "boolean",
    },
    negated: {
        baseName: "negated",
        type: "boolean",
    },
    policyName: {
        baseName: "policy_name",
        type: "string",
    },
    policyType: {
        baseName: "policy_type",
        type: "TagPolicyType",
    },
    required: {
        baseName: "required",
        type: "boolean",
    },
    scope: {
        baseName: "scope",
        type: "string",
    },
    tagKey: {
        baseName: "tag_key",
        type: "string",
    },
    tagValuePatterns: {
        baseName: "tag_value_patterns",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
