// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TagPolicyCreateAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagPolicyCreateAttributes = void 0;
/**
 * Attributes that can be supplied when creating a tag policy.
 */
class TagPolicyCreateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TagPolicyCreateAttributes.attributeTypeMap;
    }
}
exports.TagPolicyCreateAttributes = TagPolicyCreateAttributes;
/**
 * @ignore
 */
TagPolicyCreateAttributes.attributeTypeMap = {
    enabled: {
        baseName: "enabled",
        type: "boolean",
    },
    negated: {
        baseName: "negated",
        type: "boolean",
    },
    policyName: {
        baseName: "policy_name",
        type: "string",
        required: true,
    },
    policyType: {
        baseName: "policy_type",
        type: "TagPolicyCreateType",
        required: true,
    },
    required: {
        baseName: "required",
        type: "boolean",
    },
    scope: {
        baseName: "scope",
        type: "string",
        required: true,
    },
    source: {
        baseName: "source",
        type: "TagPolicySource",
        required: true,
    },
    tagKey: {
        baseName: "tag_key",
        type: "string",
        required: true,
    },
    tagValuePatterns: {
        baseName: "tag_value_patterns",
        type: "Array<string>",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
