// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TagPolicyScoreAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagPolicyScoreAttributes = void 0;
/**
 * Attributes of a tag policy compliance score.
 */
class TagPolicyScoreAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TagPolicyScoreAttributes.attributeTypeMap;
    }
}
exports.TagPolicyScoreAttributes = TagPolicyScoreAttributes;
/**
 * @ignore
 */
TagPolicyScoreAttributes.attributeTypeMap = {
    score: {
        baseName: "score",
        type: "number",
        required: true,
        format: "double",
    },
    tsEnd: {
        baseName: "ts_end",
        type: "number",
        required: true,
        format: "int64",
    },
    tsStart: {
        baseName: "ts_start",
        type: "number",
        required: true,
        format: "int64",
    },
    version: {
        baseName: "version",
        type: "number",
        required: true,
        format: "int64",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
