// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TicketCreationRuleActionResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TicketCreationRuleActionResponse = void 0;
/**
 * The action to take when the ticket creation rule matches a finding.
 */
class TicketCreationRuleActionResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TicketCreationRuleActionResponse.attributeTypeMap;
    }
}
exports.TicketCreationRuleActionResponse = TicketCreationRuleActionResponse;
/**
 * @ignore
 */
TicketCreationRuleActionResponse.attributeTypeMap = {
    assigneeId: {
        baseName: "assignee_id",
        type: "string",
        format: "uuid",
    },
    autoDisabledReason: {
        baseName: "auto_disabled_reason",
        type: "string",
    },
    fields: {
        baseName: "fields",
        type: "any",
    },
    maxTicketsPerDay: {
        baseName: "max_tickets_per_day",
        type: "number",
        required: true,
        format: "int64",
    },
    projectId: {
        baseName: "project_id",
        type: "string",
        required: true,
        format: "uuid",
    },
    target: {
        baseName: "target",
        type: "TicketCreationTarget",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
