// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TicketCreationRuleAttributesCreate.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TicketCreationRuleAttributesCreate = void 0;
/**
 * Attributes for creating or updating a ticket creation rule.
 */
class TicketCreationRuleAttributesCreate {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TicketCreationRuleAttributesCreate.attributeTypeMap;
    }
}
exports.TicketCreationRuleAttributesCreate = TicketCreationRuleAttributesCreate;
/**
 * @ignore
 */
TicketCreationRuleAttributesCreate.attributeTypeMap = {
    action: {
        baseName: "action",
        type: "TicketCreationRuleAction",
        required: true,
    },
    enabled: {
        baseName: "enabled",
        type: "boolean",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    rule: {
        baseName: "rule",
        type: "AutomationRuleScope",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
