// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TicketCreationRuleDataResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TicketCreationRuleDataResponse = void 0;
/**
 * The data object for a ticket creation rule returned by the API.
 */
class TicketCreationRuleDataResponse {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TicketCreationRuleDataResponse.attributeTypeMap;
    }
}
exports.TicketCreationRuleDataResponse = TicketCreationRuleDataResponse;
/**
 * @ignore
 */
TicketCreationRuleDataResponse.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "TicketCreationRuleAttributesResponse",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
        format: "uuid",
    },
    type: {
        baseName: "type",
        type: "TicketCreationRuleType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
