// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/TopLongTaskInvoker.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TopLongTaskInvoker = void 0;
/**
 * A top long task invoker within an invoker type.
 */
class TopLongTaskInvoker {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return TopLongTaskInvoker.attributeTypeMap;
    }
}
exports.TopLongTaskInvoker = TopLongTaskInvoker;
/**
 * @ignore
 */
TopLongTaskInvoker.attributeTypeMap = {
    criteriaViewOccurrences: {
        baseName: "criteria_view_occurrences",
        type: "number",
        format: "int32",
    },
    file: {
        baseName: "file",
        type: "string",
        required: true,
    },
    impactScore: {
        baseName: "impact_score",
        type: "number",
        format: "double",
    },
    invoker: {
        baseName: "invoker",
        type: "string",
        required: true,
    },
    statsPerView: {
        baseName: "stats_per_view",
        type: "LongTaskStatsPerView",
        required: true,
    },
    viewOccurrences: {
        baseName: "view_occurrences",
        type: "number",
        required: true,
        format: "int32",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
