// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/UpsertAndPublishFormVersionUpsertParams.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpsertAndPublishFormVersionUpsertParams = void 0;
/**
 * Concurrency control parameters for the upsert and publish operation.
 */
class UpsertAndPublishFormVersionUpsertParams {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return UpsertAndPublishFormVersionUpsertParams.attributeTypeMap;
    }
}
exports.UpsertAndPublishFormVersionUpsertParams = UpsertAndPublishFormVersionUpsertParams;
/**
 * @ignore
 */
UpsertAndPublishFormVersionUpsertParams.attributeTypeMap = {
    etag: {
        baseName: "etag",
        type: "string",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
