// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/UpsertFormVersionDataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpsertFormVersionDataAttributes = void 0;
/**
 * The attributes for creating or updating a form version.
 */
class UpsertFormVersionDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return UpsertFormVersionDataAttributes.attributeTypeMap;
    }
}
exports.UpsertFormVersionDataAttributes = UpsertFormVersionDataAttributes;
/**
 * @ignore
 */
UpsertFormVersionDataAttributes.attributeTypeMap = {
    dataDefinition: {
        baseName: "data_definition",
        type: "FormDataDefinition",
        required: true,
    },
    state: {
        baseName: "state",
        type: "FormVersionState",
        required: true,
    },
    uiDefinition: {
        baseName: "ui_definition",
        type: "FormUiDefinition",
        required: true,
    },
    upsertParams: {
        baseName: "upsert_params",
        type: "UpsertFormVersionUpsertParams",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
