// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/UpsertAndPublishFormVersionDataAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpsertAndPublishFormVersionDataAttributes = void 0;
/**
 * The attributes for upserting and publishing a form version.
 */
class UpsertAndPublishFormVersionDataAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return UpsertAndPublishFormVersionDataAttributes.attributeTypeMap;
    }
}
exports.UpsertAndPublishFormVersionDataAttributes = UpsertAndPublishFormVersionDataAttributes;
/**
 * @ignore
 */
UpsertAndPublishFormVersionDataAttributes.attributeTypeMap = {
    dataDefinition: {
        baseName: "data_definition",
        type: "FormDataDefinition",
        required: true,
    },
    uiDefinition: {
        baseName: "ui_definition",
        type: "FormUiDefinition",
        required: true,
    },
    upsertParams: {
        baseName: "upsert_params",
        type: "UpsertAndPublishFormVersionUpsertParams",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
