// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/UsageSummaryAvailableFieldsAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsageSummaryAvailableFieldsAttributes = void 0;
/**
 * The lists of field names returned by `GET /api/v1/usage/summary` at each
 * of its three response levels. Each list contains every key the data endpoint
 * emits—both typed fields declared in the OpenAPI spec and untyped keys
 * exposed through `additionalProperties`.
 */
class UsageSummaryAvailableFieldsAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return UsageSummaryAvailableFieldsAttributes.attributeTypeMap;
    }
}
exports.UsageSummaryAvailableFieldsAttributes = UsageSummaryAvailableFieldsAttributes;
/**
 * @ignore
 */
UsageSummaryAvailableFieldsAttributes.attributeTypeMap = {
    dateFields: {
        baseName: "date_fields",
        type: "Array<string>",
    },
    dateOrgFields: {
        baseName: "date_org_fields",
        type: "Array<string>",
    },
    responseFields: {
        baseName: "response_fields",
        type: "Array<string>",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
