// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/WebhooksOAuth2ClientCredentialsResponseAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhooksOAuth2ClientCredentialsResponseAttributes = void 0;
/**
 * OAuth2 client credentials attributes returned by the API. The `client_secret` is never echoed.
 */
class WebhooksOAuth2ClientCredentialsResponseAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return WebhooksOAuth2ClientCredentialsResponseAttributes.attributeTypeMap;
    }
}
exports.WebhooksOAuth2ClientCredentialsResponseAttributes = WebhooksOAuth2ClientCredentialsResponseAttributes;
/**
 * @ignore
 */
WebhooksOAuth2ClientCredentialsResponseAttributes.attributeTypeMap = {
    accessTokenUrl: {
        baseName: "access_token_url",
        type: "string",
    },
    audience: {
        baseName: "audience",
        type: "string",
    },
    clientId: {
        baseName: "client_id",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    protocol: {
        baseName: "protocol",
        type: "WebhooksAuthMethodProtocol",
    },
    scope: {
        baseName: "scope",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
