// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/WebhooksOAuth2ClientCredentialsResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhooksOAuth2ClientCredentialsResponseData = void 0;
/**
 * OAuth2 client credentials data from a response.
 */
class WebhooksOAuth2ClientCredentialsResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return WebhooksOAuth2ClientCredentialsResponseData.attributeTypeMap;
    }
}
exports.WebhooksOAuth2ClientCredentialsResponseData = WebhooksOAuth2ClientCredentialsResponseData;
/**
 * @ignore
 */
WebhooksOAuth2ClientCredentialsResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "WebhooksOAuth2ClientCredentialsResponseAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    type: {
        baseName: "type",
        type: "WebhooksOAuth2ClientCredentialsType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
