// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/WebhooksAuthMethodResponseData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhooksAuthMethodResponseData = void 0;
/**
 * Webhooks auth method data from a response.
 */
class WebhooksAuthMethodResponseData {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return WebhooksAuthMethodResponseData.attributeTypeMap;
    }
}
exports.WebhooksAuthMethodResponseData = WebhooksAuthMethodResponseData;
/**
 * @ignore
 */
WebhooksAuthMethodResponseData.attributeTypeMap = {
    attributes: {
        baseName: "attributes",
        type: "WebhooksAuthMethodAttributes",
        required: true,
    },
    id: {
        baseName: "id",
        type: "string",
        required: true,
    },
    relationships: {
        baseName: "relationships",
        type: "WebhooksAuthMethodRelationships",
    },
    type: {
        baseName: "type",
        type: "WebhooksAuthMethodType",
        required: true,
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
