// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/WebhooksOAuth2ClientCredentialsUpdateAttributes.js
"use strict";
/**
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache-2.0 License.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2020-Present Datadog, Inc.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhooksOAuth2ClientCredentialsUpdateAttributes = void 0;
/**
 * OAuth2 client credentials attributes for an update request.
 */
class WebhooksOAuth2ClientCredentialsUpdateAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return WebhooksOAuth2ClientCredentialsUpdateAttributes.attributeTypeMap;
    }
}
exports.WebhooksOAuth2ClientCredentialsUpdateAttributes = WebhooksOAuth2ClientCredentialsUpdateAttributes;
/**
 * @ignore
 */
WebhooksOAuth2ClientCredentialsUpdateAttributes.attributeTypeMap = {
    accessTokenUrl: {
        baseName: "access_token_url",
        type: "string",
    },
    audience: {
        baseName: "audience",
        type: "string",
    },
    clientId: {
        baseName: "client_id",
        type: "string",
    },
    clientSecret: {
        baseName: "client_secret",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
    },
    scope: {
        baseName: "scope",
        type: "string",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
