// @datadog/datadog-api-client/dist/packages/datadog-api-client-v2/models/WorkflowListItemAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkflowListItemAttributes = void 0;
/**
 * Attributes of a workflow returned in a list response.
 */
class WorkflowListItemAttributes {
    constructor() { }
    /**
     * @ignore
     */
    static getAttributeTypeMap() {
        return WorkflowListItemAttributes.attributeTypeMap;
    }
}
exports.WorkflowListItemAttributes = WorkflowListItemAttributes;
/**
 * @ignore
 */
WorkflowListItemAttributes.attributeTypeMap = {
    createdAt: {
        baseName: "createdAt",
        type: "Date",
        format: "date-time",
    },
    description: {
        baseName: "description",
        type: "string",
    },
    name: {
        baseName: "name",
        type: "string",
        required: true,
    },
    published: {
        baseName: "published",
        type: "boolean",
    },
    spec: {
        baseName: "spec",
        type: "Spec",
    },
    tags: {
        baseName: "tags",
        type: "Array<string>",
    },
    updatedAt: {
        baseName: "updatedAt",
        type: "Date",
        format: "date-time",
    },
    additionalProperties: {
        baseName: "additionalProperties",
        type: "{ [key: string]: any; }",
    },
};
